{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Estimate domains of translated protein product of isoforms\n",
    "\n",
    "This notebook requires that the `hmmscan` compute job in [0.2.11] be complete."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/matplotlib/__init__.py:810: UserWarning: Found matplotlib configuration in ~/.matplotlib/. To conform with the XDG base directory standard, this configuration location has been deprecated on Linux, and the new location is now /home/obotvinnik/.config/matplotlib/. Please move your configuration there to ensure that matplotlib will continue to find it in the future.\n",
      "  _get_xdg_config_dir())\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import pandas as pd\n",
    "import gffutils\n",
    "import pybedtools\n",
    "import re\n",
    "import numpy as np\n",
    "\n",
    "v19db_filename = '/projects/ps-yeolab/genomes/hg19/gencode/v19/gencode.v19.annotation.gtf.db'\n",
    "v19db = gffutils.FeatureDB(v19db_filename)\n",
    "\n",
    "folder = '/projects/ps-yeolab/obotvinnik/singlecell_pnms'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# splicing_feature_data = pd.read_csv('{}/splicing_feature_data.csv'.format(folder), index_col=0)\n",
    "# print splicing_feature_data.shape\n",
    "# splicing_feature_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "folder = '/projects/ps-yeolab/obotvinnik/singlecell_pnms'\n",
    "csv_folder = '{}/csvs_for_paper/'.format(folder)\n",
    "\n",
    "\n",
    "splicing_feature_folder = '{}/splicing_feature_data'.format(csv_folder)\n",
    "alternative_feature_folder = '{}/alternative'.format(splicing_feature_folder)\n",
    "constitutive_feature_folder = '{}/constitutive'.format(splicing_feature_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations.fa\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations.fa_iupred_out\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations.fa_iupred_tmp\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations_hmmscan_pfamA.txt\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations_hmmscan_pfamA.txt.err\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations_hmmscan_pfamA.txt.out\r\n",
      "/projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/protein_translations.csv\r\n"
     ]
    }
   ],
   "source": [
    "! ls $alternative_feature_folder/*translation*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "! ln -s $alternative_feature_folder/protein_translations.csv $alternative_feature_folder/isoform_translations.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isoform1_translation</th>\n",
       "      <th>isoform2_translation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-</th>\n",
       "      <td>NaN</td>\n",
       "      <td>HSASSLRPADLLALILLVQDLYPSESTAEDDIQPSPRRARSSQNIP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</th>\n",
       "      <td>NaN</td>\n",
       "      <td>ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</th>\n",
       "      <td>NaN</td>\n",
       "      <td>LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</th>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAK...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                               isoform1_translation  \\\n",
       "exon:chr10:100190328-100190427:-@exon:chr10:100...                                              NaN   \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101...                                              NaN   \n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101...                                              NaN   \n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101...                                              NaN   \n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101...  FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV   \n",
       "\n",
       "                                                                                 isoform2_translation  \n",
       "exon:chr10:100190328-100190427:-@exon:chr10:100...  HSASSLRPADLLALILLVQDLYPSESTAEDDIQPSPRRARSSQNIP...  \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101...  NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSA...  \n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101...  ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDF...  \n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101...  LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKM...  \n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101...  FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAK...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isoform_translations = pd.read_csv('{}/isoform_translations.csv'.format(alternative_feature_folder), index_col=0)\n",
    "isoform_translations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "prefix = 'isoform_translations'\n",
    "translated_fasta = '{}/{}.fa'.format(alternative_feature_folder, prefix)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read HMMScan output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.options.display.max_columns = 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- 1 obotvinnik yeo-group 28M Apr 11 20:34 /projects/ps-yeolab/obotvinnik/singlecell_pnms/csvs_for_paper//splicing_feature_data/alternative/isoform_translations_hmmscan_pfamA.txt\n",
      "(74296, 23)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/workspace-git/poshsplice/poshsplice/hmmscan.py:35: FutureWarning: convert_objects is deprecated.  Use the data-type specific converters pd.to_datetime, pd.to_timedelta and pd.to_numeric.\n",
      "  df = df.convert_objects(convert_numeric=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_name</th>\n",
       "      <th>target_accession</th>\n",
       "      <th>target_length</th>\n",
       "      <th>query_name</th>\n",
       "      <th>query_accession</th>\n",
       "      <th>query_length</th>\n",
       "      <th>sequence_e_value</th>\n",
       "      <th>sequence_score</th>\n",
       "      <th>sequence_bias</th>\n",
       "      <th>domain_number</th>\n",
       "      <th>domain_total</th>\n",
       "      <th>domain_conditional_e_value</th>\n",
       "      <th>domain_independent_e_value</th>\n",
       "      <th>domain_score</th>\n",
       "      <th>domain_bias</th>\n",
       "      <th>target_start</th>\n",
       "      <th>target_stop</th>\n",
       "      <th>query_start</th>\n",
       "      <th>query_stop</th>\n",
       "      <th>query_domain_envelope_start</th>\n",
       "      <th>query_domain_envelope_stop</th>\n",
       "      <th>mean_posterior_probability</th>\n",
       "      <th>target_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>363</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>140</td>\n",
       "      <td>1.600000e-37</td>\n",
       "      <td>129.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.200000e-41</td>\n",
       "      <td>1.700000e-37</td>\n",
       "      <td>129.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>142</td>\n",
       "      <td>270</td>\n",
       "      <td>2</td>\n",
       "      <td>138</td>\n",
       "      <td>1</td>\n",
       "      <td>140</td>\n",
       "      <td>0.96</td>\n",
       "      <td>Aminotransferase class I and II</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>436</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>1.800000e-27</td>\n",
       "      <td>95.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>2.100000e-27</td>\n",
       "      <td>95.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>104</td>\n",
       "      <td>75</td>\n",
       "      <td>179</td>\n",
       "      <td>72</td>\n",
       "      <td>179</td>\n",
       "      <td>0.90</td>\n",
       "      <td>GDA1/CD39 (nucleoside phosphatase) family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OppC_N</td>\n",
       "      <td>PF12911.2</td>\n",
       "      <td>56</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>8.500000e-02</td>\n",
       "      <td>12.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.200000e-05</td>\n",
       "      <td>1.600000e-01</td>\n",
       "      <td>11.3</td>\n",
       "      <td>0.2</td>\n",
       "      <td>12</td>\n",
       "      <td>37</td>\n",
       "      <td>21</td>\n",
       "      <td>46</td>\n",
       "      <td>17</td>\n",
       "      <td>56</td>\n",
       "      <td>0.92</td>\n",
       "      <td>N-terminal TM domain of oligopeptide transport...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>302</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>117</td>\n",
       "      <td>8.100000e-23</td>\n",
       "      <td>80.7</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.300000e-27</td>\n",
       "      <td>9.300000e-23</td>\n",
       "      <td>80.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>165</td>\n",
       "      <td>270</td>\n",
       "      <td>7</td>\n",
       "      <td>117</td>\n",
       "      <td>2</td>\n",
       "      <td>117</td>\n",
       "      <td>0.85</td>\n",
       "      <td>Cytochrome oxidase assembly protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>88</td>\n",
       "      <td>8.500000e-24</td>\n",
       "      <td>83.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.400000e-28</td>\n",
       "      <td>9.500000e-24</td>\n",
       "      <td>83.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>123</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>80</td>\n",
       "      <td>0.99</td>\n",
       "      <td>CutC family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     target_name target_accession  target_length  \\\n",
       "0  Aminotran_1_2       PF00155.16            363   \n",
       "1      GDA1_CD39       PF01150.12            436   \n",
       "2         OppC_N        PF12911.2             56   \n",
       "3     COX15-CtaA       PF02628.10            302   \n",
       "4           CutC        PF03932.9            202   \n",
       "\n",
       "                                          query_name query_accession  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:10...               -   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:10...               -   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:10...               -   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:10...               -   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:10...               -   \n",
       "\n",
       "   query_length  sequence_e_value  sequence_score  sequence_bias  \\\n",
       "0           140      1.600000e-37           129.3            0.0   \n",
       "1           179      1.800000e-27            95.9            0.0   \n",
       "2           179      8.500000e-02            12.2            0.2   \n",
       "3           117      8.100000e-23            80.7            0.2   \n",
       "4            88      8.500000e-24            83.9            0.0   \n",
       "\n",
       "   domain_number  domain_total  domain_conditional_e_value  \\\n",
       "0              1             1                1.200000e-41   \n",
       "1              1             1                2.800000e-31   \n",
       "2              1             1                2.200000e-05   \n",
       "3              1             1                6.300000e-27   \n",
       "4              1             1                6.400000e-28   \n",
       "\n",
       "   domain_independent_e_value  domain_score  domain_bias  target_start  \\\n",
       "0                1.700000e-37         129.2          0.0           142   \n",
       "1                2.100000e-27          95.7          0.0             4   \n",
       "2                1.600000e-01          11.3          0.2            12   \n",
       "3                9.300000e-23          80.5          0.2           165   \n",
       "4                9.500000e-24          83.7          0.0           123   \n",
       "\n",
       "   target_stop  query_start  query_stop  query_domain_envelope_start  \\\n",
       "0          270            2         138                            1   \n",
       "1          104           75         179                           72   \n",
       "2           37           21          46                           17   \n",
       "3          270            7         117                            2   \n",
       "4          201            1          79                            1   \n",
       "\n",
       "   query_domain_envelope_stop  mean_posterior_probability  \\\n",
       "0                         140                        0.96   \n",
       "1                         179                        0.90   \n",
       "2                          56                        0.92   \n",
       "3                         117                        0.85   \n",
       "4                          80                        0.99   \n",
       "\n",
       "                                  target_description  \n",
       "0                    Aminotransferase class I and II  \n",
       "1          GDA1/CD39 (nucleoside phosphatase) family  \n",
       "2  N-terminal TM domain of oligopeptide transport...  \n",
       "3                Cytochrome oxidase assembly protein  \n",
       "4                                        CutC family  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from poshsplice.hmmscan import read_hmmscan\n",
    "\n",
    "hmmscan_out = '{}/{}_hmmscan_pfamA.txt'.format(alternative_feature_folder, prefix)\n",
    "! ls -lha $hmmscan_out\n",
    "hmmscan = read_hmmscan(hmmscan_out)\n",
    "print hmmscan.shape\n",
    "hmmscan.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split query name (event name + cds + isoform information) into separate columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 'exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-|CDS:chr10:101165513-101165617:-:0@CDS:chr10:101163481-101163631:-:0@CDS:chr10:101163226-101163391:-:2|isoform2',\n",
       "       'exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+|CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-101439632:+:2|isoform2',\n",
       "       'exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+|CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-101439632:+:2|isoform2',\n",
       "       'exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-|CDS:chr10:101480744-101480825:-:0@CDS:chr10:101478103-101478257:-:2@CDS:chr10:101476105-101476218:-:0|isoform2',\n",
       "       'exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+|CDS:chr10:101507014-101507147:+:2@CDS:chr10:101510126-101510153:+:0@CDS:chr10:101514286-101514391:+:2|isoform2'], dtype=object)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan.query_name.head().values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   0  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:10...   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:10...   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:10...   \n",
       "\n",
       "                                                   1         2  \n",
       "0  CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...  isoform2  \n",
       "1  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "2  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "3  CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...  isoform2  \n",
       "4  CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...  isoform2  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_ids = hmmscan.query_name.str.split('|').apply(pd.Series)\n",
    "split_ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>event_name</th>\n",
       "      <th>cds</th>\n",
       "      <th>isoform</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:10...   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:10...   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:10...   \n",
       "\n",
       "                                                 cds   isoform  \n",
       "0  CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...  isoform2  \n",
       "1  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "2  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "3  CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...  isoform2  \n",
       "4  CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...  isoform2  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_ids = split_ids.rename(columns={0: 'event_name', 1:'cds', 2: 'isoform'})\n",
    "split_ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(74296, 3)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_ids.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(74296, 23)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_name</th>\n",
       "      <th>target_accession</th>\n",
       "      <th>target_length</th>\n",
       "      <th>query_name</th>\n",
       "      <th>query_accession</th>\n",
       "      <th>query_length</th>\n",
       "      <th>sequence_e_value</th>\n",
       "      <th>sequence_score</th>\n",
       "      <th>sequence_bias</th>\n",
       "      <th>domain_number</th>\n",
       "      <th>domain_total</th>\n",
       "      <th>domain_conditional_e_value</th>\n",
       "      <th>domain_independent_e_value</th>\n",
       "      <th>domain_score</th>\n",
       "      <th>domain_bias</th>\n",
       "      <th>target_start</th>\n",
       "      <th>target_stop</th>\n",
       "      <th>query_start</th>\n",
       "      <th>query_stop</th>\n",
       "      <th>query_domain_envelope_start</th>\n",
       "      <th>query_domain_envelope_stop</th>\n",
       "      <th>mean_posterior_probability</th>\n",
       "      <th>target_description</th>\n",
       "      <th>event_name</th>\n",
       "      <th>cds</th>\n",
       "      <th>isoform</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>363</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>140</td>\n",
       "      <td>1.600000e-37</td>\n",
       "      <td>129.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.200000e-41</td>\n",
       "      <td>1.700000e-37</td>\n",
       "      <td>129.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>142</td>\n",
       "      <td>270</td>\n",
       "      <td>2</td>\n",
       "      <td>138</td>\n",
       "      <td>1</td>\n",
       "      <td>140</td>\n",
       "      <td>0.96</td>\n",
       "      <td>Aminotransferase class I and II</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>436</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>1.800000e-27</td>\n",
       "      <td>95.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>2.100000e-27</td>\n",
       "      <td>95.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>104</td>\n",
       "      <td>75</td>\n",
       "      <td>179</td>\n",
       "      <td>72</td>\n",
       "      <td>179</td>\n",
       "      <td>0.90</td>\n",
       "      <td>GDA1/CD39 (nucleoside phosphatase) family</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OppC_N</td>\n",
       "      <td>PF12911.2</td>\n",
       "      <td>56</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>8.500000e-02</td>\n",
       "      <td>12.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.200000e-05</td>\n",
       "      <td>1.600000e-01</td>\n",
       "      <td>11.3</td>\n",
       "      <td>0.2</td>\n",
       "      <td>12</td>\n",
       "      <td>37</td>\n",
       "      <td>21</td>\n",
       "      <td>46</td>\n",
       "      <td>17</td>\n",
       "      <td>56</td>\n",
       "      <td>0.92</td>\n",
       "      <td>N-terminal TM domain of oligopeptide transport...</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>302</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>117</td>\n",
       "      <td>8.100000e-23</td>\n",
       "      <td>80.7</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.300000e-27</td>\n",
       "      <td>9.300000e-23</td>\n",
       "      <td>80.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>165</td>\n",
       "      <td>270</td>\n",
       "      <td>7</td>\n",
       "      <td>117</td>\n",
       "      <td>2</td>\n",
       "      <td>117</td>\n",
       "      <td>0.85</td>\n",
       "      <td>Cytochrome oxidase assembly protein</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>-</td>\n",
       "      <td>88</td>\n",
       "      <td>8.500000e-24</td>\n",
       "      <td>83.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.400000e-28</td>\n",
       "      <td>9.500000e-24</td>\n",
       "      <td>83.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>123</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>80</td>\n",
       "      <td>0.99</td>\n",
       "      <td>CutC family</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:10...</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...</td>\n",
       "      <td>isoform2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     target_name target_accession  target_length  \\\n",
       "0  Aminotran_1_2       PF00155.16            363   \n",
       "1      GDA1_CD39       PF01150.12            436   \n",
       "2         OppC_N        PF12911.2             56   \n",
       "3     COX15-CtaA       PF02628.10            302   \n",
       "4           CutC        PF03932.9            202   \n",
       "\n",
       "                                          query_name query_accession  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:10...               -   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:10...               -   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:10...               -   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:10...               -   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:10...               -   \n",
       "\n",
       "   query_length  sequence_e_value  sequence_score  sequence_bias  \\\n",
       "0           140      1.600000e-37           129.3            0.0   \n",
       "1           179      1.800000e-27            95.9            0.0   \n",
       "2           179      8.500000e-02            12.2            0.2   \n",
       "3           117      8.100000e-23            80.7            0.2   \n",
       "4            88      8.500000e-24            83.9            0.0   \n",
       "\n",
       "   domain_number  domain_total  domain_conditional_e_value  \\\n",
       "0              1             1                1.200000e-41   \n",
       "1              1             1                2.800000e-31   \n",
       "2              1             1                2.200000e-05   \n",
       "3              1             1                6.300000e-27   \n",
       "4              1             1                6.400000e-28   \n",
       "\n",
       "   domain_independent_e_value  domain_score  domain_bias  target_start  \\\n",
       "0                1.700000e-37         129.2          0.0           142   \n",
       "1                2.100000e-27          95.7          0.0             4   \n",
       "2                1.600000e-01          11.3          0.2            12   \n",
       "3                9.300000e-23          80.5          0.2           165   \n",
       "4                9.500000e-24          83.7          0.0           123   \n",
       "\n",
       "   target_stop  query_start  query_stop  query_domain_envelope_start  \\\n",
       "0          270            2         138                            1   \n",
       "1          104           75         179                           72   \n",
       "2           37           21          46                           17   \n",
       "3          270            7         117                            2   \n",
       "4          201            1          79                            1   \n",
       "\n",
       "   query_domain_envelope_stop  mean_posterior_probability  \\\n",
       "0                         140                        0.96   \n",
       "1                         179                        0.90   \n",
       "2                          56                        0.92   \n",
       "3                         117                        0.85   \n",
       "4                          80                        0.99   \n",
       "\n",
       "                                  target_description  \\\n",
       "0                    Aminotransferase class I and II   \n",
       "1          GDA1/CD39 (nucleoside phosphatase) family   \n",
       "2  N-terminal TM domain of oligopeptide transport...   \n",
       "3                Cytochrome oxidase assembly protein   \n",
       "4                                        CutC family   \n",
       "\n",
       "                                          event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:10...   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:10...   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:10...   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:10...   \n",
       "\n",
       "                                                 cds   isoform  \n",
       "0  CDS:chr10:101165513-101165617:-:0@CDS:chr10:10...  isoform2  \n",
       "1  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "2  CDS:chr10:101421203-101421385:+:1@CDS:chr10:10...  isoform2  \n",
       "3  CDS:chr10:101480744-101480825:-:0@CDS:chr10:10...  isoform2  \n",
       "4  CDS:chr10:101507014-101507147:+:2@CDS:chr10:10...  isoform2  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan_ids = pd.concat([hmmscan, split_ids], axis=1)\n",
    "hmmscan_ids.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.options.display.max_colwidth = 100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "e_value_cutoff = 1e-5\n",
    "hmmscan_filtered = hmmscan_ids.loc[hmmscan_ids.domain_independent_e_value < e_value_cutoff]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add protein translations\n",
    "\n",
    "To check if the isoforms went from no annotated reading frame to either a translation OR domain."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>isoform</th>\n",
       "      <th>event_name</th>\n",
       "      <th>translation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>isoform1</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>isoform1</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>isoform1</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>isoform1</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>isoform1</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    isoform  \\\n",
       "0  isoform1   \n",
       "1  isoform1   \n",
       "2  isoform1   \n",
       "3  isoform1   \n",
       "4  isoform1   \n",
       "\n",
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-   \n",
       "1  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "2  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "3  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "                                       translation  \n",
       "0                                              NaN  \n",
       "1                                              NaN  \n",
       "2                                              NaN  \n",
       "3                                              NaN  \n",
       "4  FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "translations_tidy = isoform_translations.unstack().reset_index()\n",
    "translations_tidy = translations_tidy.rename(columns={'level_0': 'isoform', 'level_1': 'event_name', 0: 'translation'})\n",
    "translations_tidy['isoform'] = translations_tidy['isoform'].str.split('_').str[0]\n",
    "translations_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15746, 26)\n",
      "(48730, 27)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_name</th>\n",
       "      <th>target_accession</th>\n",
       "      <th>target_length</th>\n",
       "      <th>query_name</th>\n",
       "      <th>query_accession</th>\n",
       "      <th>query_length</th>\n",
       "      <th>sequence_e_value</th>\n",
       "      <th>sequence_score</th>\n",
       "      <th>sequence_bias</th>\n",
       "      <th>domain_number</th>\n",
       "      <th>domain_total</th>\n",
       "      <th>domain_conditional_e_value</th>\n",
       "      <th>domain_independent_e_value</th>\n",
       "      <th>domain_score</th>\n",
       "      <th>domain_bias</th>\n",
       "      <th>target_start</th>\n",
       "      <th>target_stop</th>\n",
       "      <th>query_start</th>\n",
       "      <th>query_stop</th>\n",
       "      <th>query_domain_envelope_start</th>\n",
       "      <th>query_domain_envelope_stop</th>\n",
       "      <th>mean_posterior_probability</th>\n",
       "      <th>target_description</th>\n",
       "      <th>event_name</th>\n",
       "      <th>cds</th>\n",
       "      <th>isoform</th>\n",
       "      <th>translation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>363</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391...</td>\n",
       "      <td>-</td>\n",
       "      <td>140</td>\n",
       "      <td>1.600000e-37</td>\n",
       "      <td>129.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.200000e-41</td>\n",
       "      <td>1.700000e-37</td>\n",
       "      <td>129.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>142</td>\n",
       "      <td>270</td>\n",
       "      <td>2</td>\n",
       "      <td>138</td>\n",
       "      <td>1</td>\n",
       "      <td>140</td>\n",
       "      <td>0.96</td>\n",
       "      <td>Aminotransferase class I and II</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>CDS:chr10:101165513-101165617:-:0@CDS:chr10:101163481-101163631:-:0@CDS:chr10:101163226-10116339...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSAYQGFASGNLERDAWAIRYFVSEGFEFFCAQSFSKNFGLYNERVGNLTVVG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>436</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>1.800000e-27</td>\n",
       "      <td>95.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>2.100000e-27</td>\n",
       "      <td>95.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4</td>\n",
       "      <td>104</td>\n",
       "      <td>75</td>\n",
       "      <td>179</td>\n",
       "      <td>72</td>\n",
       "      <td>179</td>\n",
       "      <td>0.90</td>\n",
       "      <td>GDA1/CD39 (nucleoside phosphatase) family</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-10143963...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDFRHWSASLPRDRQYERYLARVGELEATDTEDPNLNYGLVVDCGSSGSRIFV...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>302</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218...</td>\n",
       "      <td>-</td>\n",
       "      <td>117</td>\n",
       "      <td>8.100000e-23</td>\n",
       "      <td>80.7</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.300000e-27</td>\n",
       "      <td>9.300000e-23</td>\n",
       "      <td>80.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>165</td>\n",
       "      <td>270</td>\n",
       "      <td>7</td>\n",
       "      <td>117</td>\n",
       "      <td>2</td>\n",
       "      <td>117</td>\n",
       "      <td>0.85</td>\n",
       "      <td>Cytochrome oxidase assembly protein</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>CDS:chr10:101480744-101480825:-:0@CDS:chr10:101478103-101478257:-:2@CDS:chr10:101476105-10147621...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKMGESWIPEDLFTFSPILRNVFENPTMVQFDHRILGITSVTAITVLYFLSRR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...</td>\n",
       "      <td>-</td>\n",
       "      <td>88</td>\n",
       "      <td>8.500000e-24</td>\n",
       "      <td>83.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.400000e-28</td>\n",
       "      <td>9.500000e-24</td>\n",
       "      <td>83.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>123</td>\n",
       "      <td>201</td>\n",
       "      <td>1</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>80</td>\n",
       "      <td>0.99</td>\n",
       "      <td>CutC family</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:101510126-101510153:+:0@CDS:chr10:101514286-10151439...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAKGRIVVMPGGGITDRNLQRILEGSGATEFHCSARSTRDSGMKF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...</td>\n",
       "      <td>-</td>\n",
       "      <td>47</td>\n",
       "      <td>1.700000e-11</td>\n",
       "      <td>43.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.300000e-15</td>\n",
       "      <td>1.900000e-11</td>\n",
       "      <td>43.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>123</td>\n",
       "      <td>166</td>\n",
       "      <td>1</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>0.98</td>\n",
       "      <td>CutC family</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:101514286-101514294:+:0</td>\n",
       "      <td>isoform1</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     target_name target_accession  target_length  \\\n",
       "0  Aminotran_1_2       PF00155.16            363   \n",
       "1      GDA1_CD39       PF01150.12            436   \n",
       "2     COX15-CtaA       PF02628.10            302   \n",
       "3           CutC        PF03932.9            202   \n",
       "4           CutC        PF03932.9            202   \n",
       "\n",
       "                                                                                            query_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391...   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632...   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218...   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...   \n",
       "\n",
       "  query_accession  query_length  sequence_e_value  sequence_score  \\\n",
       "0               -           140      1.600000e-37           129.3   \n",
       "1               -           179      1.800000e-27            95.9   \n",
       "2               -           117      8.100000e-23            80.7   \n",
       "3               -            88      8.500000e-24            83.9   \n",
       "4               -            47      1.700000e-11            43.7   \n",
       "\n",
       "   sequence_bias  domain_number  domain_total  domain_conditional_e_value  \\\n",
       "0            0.0              1             1                1.200000e-41   \n",
       "1            0.0              1             1                2.800000e-31   \n",
       "2            0.2              1             1                6.300000e-27   \n",
       "3            0.0              1             1                6.400000e-28   \n",
       "4            0.0              1             1                1.300000e-15   \n",
       "\n",
       "   domain_independent_e_value  domain_score  domain_bias  target_start  \\\n",
       "0                1.700000e-37         129.2          0.0           142   \n",
       "1                2.100000e-27          95.7          0.0             4   \n",
       "2                9.300000e-23          80.5          0.2           165   \n",
       "3                9.500000e-24          83.7          0.0           123   \n",
       "4                1.900000e-11          43.6          0.0           123   \n",
       "\n",
       "   target_stop  query_start  query_stop  query_domain_envelope_start  \\\n",
       "0          270            2         138                            1   \n",
       "1          104           75         179                           72   \n",
       "2          270            7         117                            2   \n",
       "3          201            1          79                            1   \n",
       "4          166            1          44                            1   \n",
       "\n",
       "   query_domain_envelope_stop  mean_posterior_probability  \\\n",
       "0                         140                        0.96   \n",
       "1                         179                        0.90   \n",
       "2                         117                        0.85   \n",
       "3                          80                        0.99   \n",
       "4                          46                        0.98   \n",
       "\n",
       "                          target_description  \\\n",
       "0            Aminotransferase class I and II   \n",
       "1  GDA1/CD39 (nucleoside phosphatase) family   \n",
       "2        Cytochrome oxidase assembly protein   \n",
       "3                                CutC family   \n",
       "4                                CutC family   \n",
       "\n",
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "                                                                                                   cds  \\\n",
       "0  CDS:chr10:101165513-101165617:-:0@CDS:chr10:101163481-101163631:-:0@CDS:chr10:101163226-10116339...   \n",
       "1  CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-10143963...   \n",
       "2  CDS:chr10:101480744-101480825:-:0@CDS:chr10:101478103-101478257:-:2@CDS:chr10:101476105-10147621...   \n",
       "3  CDS:chr10:101507014-101507147:+:2@CDS:chr10:101510126-101510153:+:0@CDS:chr10:101514286-10151439...   \n",
       "4                                  CDS:chr10:101507014-101507147:+:2@CDS:chr10:101514286-101514294:+:0   \n",
       "\n",
       "    isoform  \\\n",
       "0  isoform2   \n",
       "1  isoform2   \n",
       "2  isoform2   \n",
       "3  isoform2   \n",
       "4  isoform1   \n",
       "\n",
       "                                                                                           translation  \n",
       "0  NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSAYQGFASGNLERDAWAIRYFVSEGFEFFCAQSFSKNFGLYNERVGNLTVVG...  \n",
       "1  ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDFRHWSASLPRDRQYERYLARVGELEATDTEDPNLNYGLVVDCGSSGSRIFV...  \n",
       "2  LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKMGESWIPEDLFTFSPILRNVFENPTMVQFDHRILGITSVTAITVLYFLSRR...  \n",
       "3             FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAKGRIVVMPGGGITDRNLQRILEGSGATEFHCSARSTRDSGMKF  \n",
       "4                                                      FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merge_cols = ['isoform', 'event_name']\n",
    "print hmmscan_filtered.shape\n",
    "hmmscan_filtered = hmmscan_filtered.merge(translations_tidy, how='outer', left_on=merge_cols, right_on=merge_cols)\n",
    "print hmmscan_filtered.shape\n",
    "hmmscan_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(48730, 27)\n",
      "(30395, 27)\n"
     ]
    }
   ],
   "source": [
    "print hmmscan_filtered.shape\n",
    "hmmscan_filtered = hmmscan_filtered.dropna(subset=['translation'], axis=0)\n",
    "print hmmscan_filtered.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Annotate with Pfam-A clans and Gene ontology\n",
    "\n",
    "First add a column that has the PFam accession ID (\"`target_accession`\") without the period/decimal and everything after it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "hmmscan_filtered['pfam_accession_no_version'] = hmmscan_filtered.target_accession.str.split('.').str[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pfam-A clans"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(14831, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>clan_accession</th>\n",
       "      <th>clan_id</th>\n",
       "      <th>pfamA_id</th>\n",
       "      <th>pfamA_description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pfamA_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PF00389</th>\n",
       "      <td>CL0325</td>\n",
       "      <td>Form_Glyc_dh</td>\n",
       "      <td>2-Hacid_dh</td>\n",
       "      <td>D-isomer specific 2-hydroxyacid dehydrogenase, catalytic domain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF00198</th>\n",
       "      <td>CL0149</td>\n",
       "      <td>CoA-acyltrans</td>\n",
       "      <td>2-oxoacid_dh</td>\n",
       "      <td>2-oxoacid dehydrogenases acyltransferase (catalytic domain)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF04029</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2-ph_phosp</td>\n",
       "      <td>2-phosphosulpholactate phosphatase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF03171</th>\n",
       "      <td>CL0029</td>\n",
       "      <td>Cupin</td>\n",
       "      <td>2OG-FeII_Oxy</td>\n",
       "      <td>2OG-Fe(II) oxygenase superfamily</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF01073</th>\n",
       "      <td>CL0063</td>\n",
       "      <td>NADP_Rossmann</td>\n",
       "      <td>3Beta_HSD</td>\n",
       "      <td>3-beta hydroxysteroid dehydrogenase/isomerase family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                clan_accession        clan_id      pfamA_id  \\\n",
       "pfamA_accession                                               \n",
       "PF00389                 CL0325   Form_Glyc_dh    2-Hacid_dh   \n",
       "PF00198                 CL0149  CoA-acyltrans  2-oxoacid_dh   \n",
       "PF04029                    NaN            NaN    2-ph_phosp   \n",
       "PF03171                 CL0029          Cupin  2OG-FeII_Oxy   \n",
       "PF01073                 CL0063  NADP_Rossmann     3Beta_HSD   \n",
       "\n",
       "                                                               pfamA_description  \n",
       "pfamA_accession                                                                   \n",
       "PF00389          D-isomer specific 2-hydroxyacid dehydrogenase, catalytic domain  \n",
       "PF00198              2-oxoacid dehydrogenases acyltransferase (catalytic domain)  \n",
       "PF04029                                       2-phosphosulpholactate phosphatase  \n",
       "PF03171                                         2OG-Fe(II) oxygenase superfamily  \n",
       "PF01073                     3-beta hydroxysteroid dehydrogenase/isomerase family  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "column_names = ['pfamA_accession', 'clan_accession', 'clan_id', 'pfamA_id', 'pfamA_description']\n",
    "pfam_clans = pd.read_table('/projects/ps-yeolab/genomes/pfam/release_27/ftp.sanger.ac.uk/pub/databases/Pfam/releases/Pfam27.0/Pfam-A.clans.tsv',\n",
    "                           names=column_names, index_col=0,\n",
    "                           header=None)\n",
    "pfam_clans = pfam_clans.replace('\\\\N', np.nan)\n",
    "print pfam_clans.shape\n",
    "pfam_clans.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Join to `hmmscan_filtered`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target_name</th>\n",
       "      <th>target_accession</th>\n",
       "      <th>target_length</th>\n",
       "      <th>query_name</th>\n",
       "      <th>query_accession</th>\n",
       "      <th>query_length</th>\n",
       "      <th>sequence_e_value</th>\n",
       "      <th>sequence_score</th>\n",
       "      <th>sequence_bias</th>\n",
       "      <th>domain_number</th>\n",
       "      <th>domain_total</th>\n",
       "      <th>domain_conditional_e_value</th>\n",
       "      <th>domain_independent_e_value</th>\n",
       "      <th>domain_score</th>\n",
       "      <th>domain_bias</th>\n",
       "      <th>...</th>\n",
       "      <th>query_start</th>\n",
       "      <th>query_stop</th>\n",
       "      <th>query_domain_envelope_start</th>\n",
       "      <th>query_domain_envelope_stop</th>\n",
       "      <th>mean_posterior_probability</th>\n",
       "      <th>target_description</th>\n",
       "      <th>event_name</th>\n",
       "      <th>cds</th>\n",
       "      <th>isoform</th>\n",
       "      <th>translation</th>\n",
       "      <th>pfam_accession_no_version</th>\n",
       "      <th>clan_accession</th>\n",
       "      <th>clan_id</th>\n",
       "      <th>pfamA_id</th>\n",
       "      <th>pfamA_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>363</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391...</td>\n",
       "      <td>-</td>\n",
       "      <td>140</td>\n",
       "      <td>1.600000e-37</td>\n",
       "      <td>129.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.200000e-41</td>\n",
       "      <td>1.700000e-37</td>\n",
       "      <td>129.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>138</td>\n",
       "      <td>1</td>\n",
       "      <td>140</td>\n",
       "      <td>0.96</td>\n",
       "      <td>Aminotransferase class I and II</td>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>CDS:chr10:101165513-101165617:-:0@CDS:chr10:101163481-101163631:-:0@CDS:chr10:101163226-10116339...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSAYQGFASGNLERDAWAIRYFVSEGFEFFCAQSFSKNFGLYNERVGNLTVVG...</td>\n",
       "      <td>PF00155</td>\n",
       "      <td>CL0061</td>\n",
       "      <td>PLP_aminotran</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>Aminotransferase class I and II</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>436</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632...</td>\n",
       "      <td>-</td>\n",
       "      <td>179</td>\n",
       "      <td>1.800000e-27</td>\n",
       "      <td>95.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2.800000e-31</td>\n",
       "      <td>2.100000e-27</td>\n",
       "      <td>95.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>75</td>\n",
       "      <td>179</td>\n",
       "      <td>72</td>\n",
       "      <td>179</td>\n",
       "      <td>0.90</td>\n",
       "      <td>GDA1/CD39 (nucleoside phosphatase) family</td>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-10143963...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDFRHWSASLPRDRQYERYLARVGELEATDTEDPNLNYGLVVDCGSSGSRIFV...</td>\n",
       "      <td>PF01150</td>\n",
       "      <td>CL0108</td>\n",
       "      <td>Actin_ATPase</td>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>GDA1/CD39 (nucleoside phosphatase) family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>302</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218...</td>\n",
       "      <td>-</td>\n",
       "      <td>117</td>\n",
       "      <td>8.100000e-23</td>\n",
       "      <td>80.7</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.300000e-27</td>\n",
       "      <td>9.300000e-23</td>\n",
       "      <td>80.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>117</td>\n",
       "      <td>2</td>\n",
       "      <td>117</td>\n",
       "      <td>0.85</td>\n",
       "      <td>Cytochrome oxidase assembly protein</td>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>CDS:chr10:101480744-101480825:-:0@CDS:chr10:101478103-101478257:-:2@CDS:chr10:101476105-10147621...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKMGESWIPEDLFTFSPILRNVFENPTMVQFDHRILGITSVTAITVLYFLSRR...</td>\n",
       "      <td>PF02628</td>\n",
       "      <td>CL0328</td>\n",
       "      <td>2heme_cytochrom</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>Cytochrome oxidase assembly protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...</td>\n",
       "      <td>-</td>\n",
       "      <td>88</td>\n",
       "      <td>8.500000e-24</td>\n",
       "      <td>83.9</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>6.400000e-28</td>\n",
       "      <td>9.500000e-24</td>\n",
       "      <td>83.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>79</td>\n",
       "      <td>1</td>\n",
       "      <td>80</td>\n",
       "      <td>0.99</td>\n",
       "      <td>CutC family</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:101510126-101510153:+:0@CDS:chr10:101514286-10151439...</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAKGRIVVMPGGGITDRNLQRILEGSGATEFHCSARSTRDSGMKF</td>\n",
       "      <td>PF03932</td>\n",
       "      <td>CL0036</td>\n",
       "      <td>TIM_barrel</td>\n",
       "      <td>CutC</td>\n",
       "      <td>CutC family</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>202</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...</td>\n",
       "      <td>-</td>\n",
       "      <td>47</td>\n",
       "      <td>1.700000e-11</td>\n",
       "      <td>43.7</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.300000e-15</td>\n",
       "      <td>1.900000e-11</td>\n",
       "      <td>43.6</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>44</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>0.98</td>\n",
       "      <td>CutC family</td>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>CDS:chr10:101507014-101507147:+:2@CDS:chr10:101514286-101514294:+:0</td>\n",
       "      <td>isoform1</td>\n",
       "      <td>FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV</td>\n",
       "      <td>PF03932</td>\n",
       "      <td>CL0036</td>\n",
       "      <td>TIM_barrel</td>\n",
       "      <td>CutC</td>\n",
       "      <td>CutC family</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     target_name target_accession  target_length  \\\n",
       "0  Aminotran_1_2       PF00155.16            363   \n",
       "1      GDA1_CD39       PF01150.12            436   \n",
       "2     COX15-CtaA       PF02628.10            302   \n",
       "3           CutC        PF03932.9            202   \n",
       "4           CutC        PF03932.9            202   \n",
       "\n",
       "                                                                                            query_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391...   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632...   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218...   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391...   \n",
       "\n",
       "  query_accession  query_length  sequence_e_value  sequence_score  \\\n",
       "0               -           140      1.600000e-37           129.3   \n",
       "1               -           179      1.800000e-27            95.9   \n",
       "2               -           117      8.100000e-23            80.7   \n",
       "3               -            88      8.500000e-24            83.9   \n",
       "4               -            47      1.700000e-11            43.7   \n",
       "\n",
       "   sequence_bias  domain_number  domain_total  domain_conditional_e_value  \\\n",
       "0            0.0              1             1                1.200000e-41   \n",
       "1            0.0              1             1                2.800000e-31   \n",
       "2            0.2              1             1                6.300000e-27   \n",
       "3            0.0              1             1                6.400000e-28   \n",
       "4            0.0              1             1                1.300000e-15   \n",
       "\n",
       "   domain_independent_e_value  domain_score  domain_bias  \\\n",
       "0                1.700000e-37         129.2          0.0   \n",
       "1                2.100000e-27          95.7          0.0   \n",
       "2                9.300000e-23          80.5          0.2   \n",
       "3                9.500000e-24          83.7          0.0   \n",
       "4                1.900000e-11          43.6          0.0   \n",
       "\n",
       "                     ...                      query_start  query_stop  \\\n",
       "0                    ...                                2         138   \n",
       "1                    ...                               75         179   \n",
       "2                    ...                                7         117   \n",
       "3                    ...                                1          79   \n",
       "4                    ...                                1          44   \n",
       "\n",
       "   query_domain_envelope_start  query_domain_envelope_stop  \\\n",
       "0                            1                         140   \n",
       "1                           72                         179   \n",
       "2                            2                         117   \n",
       "3                            1                          80   \n",
       "4                            1                          46   \n",
       "\n",
       "   mean_posterior_probability                         target_description  \\\n",
       "0                        0.96            Aminotransferase class I and II   \n",
       "1                        0.90  GDA1/CD39 (nucleoside phosphatase) family   \n",
       "2                        0.85        Cytochrome oxidase assembly protein   \n",
       "3                        0.99                                CutC family   \n",
       "4                        0.98                                CutC family   \n",
       "\n",
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "                                                                                                   cds  \\\n",
       "0  CDS:chr10:101165513-101165617:-:0@CDS:chr10:101163481-101163631:-:0@CDS:chr10:101163226-10116339...   \n",
       "1  CDS:chr10:101421203-101421385:+:1@CDS:chr10:101439018-101439223:+:1@CDS:chr10:101439482-10143963...   \n",
       "2  CDS:chr10:101480744-101480825:-:0@CDS:chr10:101478103-101478257:-:2@CDS:chr10:101476105-10147621...   \n",
       "3  CDS:chr10:101507014-101507147:+:2@CDS:chr10:101510126-101510153:+:0@CDS:chr10:101514286-10151439...   \n",
       "4                                  CDS:chr10:101507014-101507147:+:2@CDS:chr10:101514286-101514294:+:0   \n",
       "\n",
       "    isoform  \\\n",
       "0  isoform2   \n",
       "1  isoform2   \n",
       "2  isoform2   \n",
       "3  isoform2   \n",
       "4  isoform1   \n",
       "\n",
       "                                                                                           translation  \\\n",
       "0  NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSAYQGFASGNLERDAWAIRYFVSEGFEFFCAQSFSKNFGLYNERVGNLTVVG...   \n",
       "1  ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDFRHWSASLPRDRQYERYLARVGELEATDTEDPNLNYGLVVDCGSSGSRIFV...   \n",
       "2  LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKMGESWIPEDLFTFSPILRNVFENPTMVQFDHRILGITSVTAITVLYFLSRR...   \n",
       "3             FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAKGRIVVMPGGGITDRNLQRILEGSGATEFHCSARSTRDSGMKF   \n",
       "4                                                      FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV   \n",
       "\n",
       "  pfam_accession_no_version clan_accession          clan_id       pfamA_id  \\\n",
       "0                   PF00155         CL0061    PLP_aminotran  Aminotran_1_2   \n",
       "1                   PF01150         CL0108     Actin_ATPase      GDA1_CD39   \n",
       "2                   PF02628         CL0328  2heme_cytochrom     COX15-CtaA   \n",
       "3                   PF03932         CL0036       TIM_barrel           CutC   \n",
       "4                   PF03932         CL0036       TIM_barrel           CutC   \n",
       "\n",
       "                           pfamA_description  \n",
       "0            Aminotransferase class I and II  \n",
       "1  GDA1/CD39 (nucleoside phosphatase) family  \n",
       "2        Cytochrome oxidase assembly protein  \n",
       "3                                CutC family  \n",
       "4                                CutC family  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan_clans = hmmscan_filtered.join(pfam_clans, on='pfam_accession_no_version')\n",
    "hmmscan_clans.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(30395, 32)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan_clans.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get the net effect of domain changes from isoform1 to isoform2\n",
    "\n",
    "If the Pfam Clan of the domain switches from isoform1 to isoform2 --> Call this \"switching domains\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "hmmscan_clans_isoform = hmmscan_clans.set_index('isoform').sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<pandas.core.groupby.DataFrameGroupBy object at 0x2b87f54203d0>"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hmmscan_clans_isoform.groupby('event_name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2         NaN   \n",
      "\n",
      "                                                                                        translation  \\\n",
      "isoform                                                                                               \n",
      "isoform2  HSASSLRPADLLALILLVQDLYPSESTAEDDIQPSPRRARSSQNIPVQQAWSPHSTGPTGGSSAETETDSFSLPEEYFTPAPSPGDQSS   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform2     NaN  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "            target_name  \\\n",
      "isoform                   \n",
      "isoform2  Aminotran_1_2   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  NAPEFSIVVLHACAHNPTGIDPTPEQWKQIASVMKHRFLFPFFDSAYQGFASGNLERDAWAIRYFVSEGFEFFCAQSFSKNFGLYNERVGNLTVVG...   \n",
      "\n",
      "                clan_id  \n",
      "isoform                  \n",
      "isoform2  PLP_aminotran  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2   GDA1_CD39   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  ISFSYLCPASWYFTVPTVSPFLRQRVAFLGLFFISCLLLLMLIIDFRHWSASLPRDRQYERYLARVGELEATDTEDPNLNYGLVVDCGSSGSRIFV...   \n",
      "\n",
      "               clan_id  \n",
      "isoform                 \n",
      "isoform2  Actin_ATPase  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2  COX15-CtaA   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  LPETHQLLQLRRFAHGTAGLVFLTALSGAFVAGLDAGLVYNSFPKMGESWIPEDLFTFSPILRNVFENPTMVQFDHRILGITSVTAITVLYFLSRR...   \n",
      "\n",
      "                  clan_id  \n",
      "isoform                    \n",
      "isoform2  2heme_cytochrom  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform1        CutC   \n",
      "isoform2        CutC   \n",
      "\n",
      "                                                                                       translation  \\\n",
      "isoform                                                                                              \n",
      "isoform1                                           FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQEVV   \n",
      "isoform2  FDMVHDPMAALETLLTLGFERVLTSGCDSSALEGLPLIKRLIEQAKGRIVVMPGGGITDRNLQRILEGSGATEFHCSARSTRDSGMKF   \n",
      "\n",
      "             clan_id  \n",
      "isoform               \n",
      "isoform1  TIM_barrel  \n",
      "isoform2  TIM_barrel  \n",
      "\t---> Same status\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2         NaN   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  LSLLRYNANLTKMKNTLISASQQLKAKLEFFHKSIQLDLERYSEQMTYGISSEKMLKAWKEMEEKAIHYAEVGVIGYLEDQIMSLHAEIMELQKSP...   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform2     NaN  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2         NaN   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  IVHILNMTSAKIISFLLPPDESLHSLQSRIERETGINTGSQELLSETGISLDPRKPASQCVLDGVRGCDSYMVYLFDKSKTVYEGPFASRSLSDCV...   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform2     NaN  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2         NaN   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  LVVEPMENWLQLMLNWDPQQRGGPVDLTLKQPRCFVLMDHILNLKIVHILNMTSAKIISFLLPPDESLHSLQSRIERETGINTGSQELLSETGISL...   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform2     NaN  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform1    CwfJ_C_1   \n",
      "isoform2    CwfJ_C_1   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform1  CYLALAKGGLSDDHVLILPIGHYQSVVELSAEVVEEVEKYKATLRRFFKSRGKWCVVFERNYKSHHLQLQIAQPGAAYFYVELDTGEKLFHRIKKN...   \n",
      "isoform2  CYLALAKGGLSDDHVLILPIGHYQSVVELSAEVVEEVEKYKATLRRFFKSRGKWCVVFERNYKSHHLQLQVIPVPISCSTTDDIKDAFITQAQEQQ...   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform1     HIT  \n",
      "isoform2     HIT  \n",
      "\t---> Same status\n",
      "\n",
      "         target_name  \\\n",
      "isoform                \n",
      "isoform2    CwfJ_C_1   \n",
      "\n",
      "                                                                                                  translation  \\\n",
      "isoform                                                                                                         \n",
      "isoform2  QPPGPCWFCLASPEVEKHLVVNIGTHCYLALAKGGLSDDHVLILPIGHYQSVVELSAEVVEEVEKYKATLRRFFKSRGKWCVVFERNYKSHHLQLQ...   \n",
      "\n",
      "         clan_id  \n",
      "isoform           \n",
      "isoform2     HIT  \n",
      "\t---> No annotated reading frame --> Protein +/- domain\n"
     ]
    }
   ],
   "source": [
    "def get_col_set(df, isoform, col):\n",
    "    try:\n",
    "        return set(df.loc[isoform][col].values)\n",
    "    except AttributeError:\n",
    "        # Only one item --> str or float has no attribute \"values\"\n",
    "        return set([df.loc[isoform][col]])\n",
    "    \n",
    "def get_domain_effects(df):\n",
    "    \n",
    "    if 'isoform1' not in df.index:\n",
    "        return 'No annotated reading frame --> Protein +/- domain'\n",
    "    elif 'isoform2' not in df.index:\n",
    "        return 'Protein +/- domain --> No annotated reading frame'\n",
    "    \n",
    "    \n",
    "    df_clans = df.dropna(subset=['clan_id'], axis=0)\n",
    "    try:\n",
    "        isoform1 = get_col_set(df_clans, 'isoform1', 'clan_id')\n",
    "        isoform2 = get_col_set(df_clans, 'isoform2', 'clan_id')\n",
    "        \n",
    "        # Get domains unique to either isoform\n",
    "        unique_domains = isoform1.symmetric_difference(isoform2)\n",
    "        if len(unique_domains) > 0:\n",
    "            return \"Domain switch\"\n",
    "        else: \n",
    "            return \"Same status\"\n",
    "    except KeyError:\n",
    "        # Both isoforms don't have clans, but both are translated\n",
    "        \n",
    "        df_domains = df.dropna(subset=['target_name'], axis=0)\n",
    "        if len(df_domains) == 0:\n",
    "            return 'Same status'\n",
    "        elif 'isoform1' not in df_domains.index:\n",
    "            return 'No domain --> Domain'\n",
    "        elif 'isoform2' not in df_domains.index:\n",
    "            return 'Domain --> No domain'\n",
    "        \n",
    "        isoform1 = get_col_set(df_domains, 'isoform1', 'target_name')\n",
    "        isoform2 = get_col_set(df_domains, 'isoform2', 'target_name')\n",
    "        unique_domains = isoform1.symmetric_difference(isoform2)\n",
    "        if len(unique_domains) > 0:\n",
    "            return \"Domain switch\"\n",
    "        else: \n",
    "            return \"Same status\"\n",
    "\n",
    "        \n",
    "for i, (name, df) in zip(range(10), hmmscan_clans_isoform.groupby('event_name')):\n",
    "    print '\\n', df[['target_name', 'translation', 'clan_id']]\n",
    "    print '\\t--->', get_domain_effects(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 12.4 s, sys: 1e+03 s, total: 12.4 s\n",
      "Wall time: 12.4 s\n"
     ]
    }
   ],
   "source": [
    "%time domain_effects = hmmscan_clans_isoform.groupby('event_name').apply(get_domain_effects)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name\n",
       "exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+                                          Same status\n",
       "dtype: object"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domain_effects.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "domain_effects.name = 'isoform_domain_effects'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name\n",
       "exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-    No annotated reading frame --> Protein +/- domain\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+                                          Same status\n",
       "Name: isoform_domain_effects, dtype: object"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domain_effects.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "domain_effects.to_csv('{}/isoform_domain_effects.csv'.format(alternative_feature_folder), header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "isoform_domain_effects\n",
       "Domain --> No domain                                   125\n",
       "Domain switch                                           40\n",
       "No annotated reading frame --> Protein +/- domain    14132\n",
       "No domain --> Domain                                   654\n",
       "Protein +/- domain --> No annotated reading frame     4203\n",
       "Same status                                           2998\n",
       "dtype: int64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domain_effects.groupby(domain_effects).size()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get event to Pfam Accession and Name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name                                                                                          isoform \n",
       "exon:chr10:100190328-100190427:-@exon:chr10:100189548-100189646:-@exon:chr10:100189330-100189399:-  isoform2           nan[1]\n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-  isoform2    PF00155.16[1]\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+  isoform2    PF01150.12[1]\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-  isoform2    PF02628.10[1]\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+  isoform1     PF03932.9[1]\n",
       "dtype: object"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from collections import Counter\n",
    "\n",
    "grouped = hmmscan_filtered.groupby(['event_name', 'isoform'])\n",
    "event_to_pfam_accession = grouped.apply(lambda x: ','.join('{}[{}]'.format(k, v) for k, v in Counter(x['target_accession']).items()))\n",
    "event_to_pfam_accession.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>event_name</th>\n",
       "      <th>isoform</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>n_domains</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform1</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "    isoform pfam_accession  n_domains  \n",
       "0  isoform2     PF00155.16          1  \n",
       "1  isoform2     PF01150.12          1  \n",
       "2  isoform2     PF02628.10          1  \n",
       "3  isoform1      PF03932.9          1  \n",
       "4  isoform2      PF03932.9          1  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam_accession_tidy = hmmscan_filtered.groupby(['event_name', 'isoform']).apply(\n",
    "    lambda x: pd.Series(Counter(x['target_accession'])))\n",
    "pfam_accession_tidy = pfam_accession_tidy.dropna()\n",
    "pfam_accession_tidy = pfam_accession_tidy.reset_index()\n",
    "pfam_accession_tidy = pfam_accession_tidy.rename(columns={'level_2': 'pfam_accession', 0: 'n_domains'})\n",
    "pfam_accession_tidy.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(30395,)\n",
      "(2824,)\n"
     ]
    }
   ],
   "source": [
    "pfam_accession_to_name = pd.Series(hmmscan_filtered.target_name.values, index=hmmscan_filtered.target_accession.values,\n",
    "                                  name='pfam_name')\n",
    "print pfam_accession_to_name.shape\n",
    "pfam_accession_to_name = pfam_accession_to_name.drop_duplicates()\n",
    "pfam_accession_to_name.head()\n",
    "print pfam_accession_to_name.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pfam_tidy.shape (14016, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>event_name</th>\n",
       "      <th>isoform</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>n_domains</th>\n",
       "      <th>pfam_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>1</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>1</td>\n",
       "      <td>GDA1_CD39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>1</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform1</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "      <td>CutC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "      <td>CutC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "    isoform pfam_accession  n_domains      pfam_name  \n",
       "0  isoform2     PF00155.16          1  Aminotran_1_2  \n",
       "1  isoform2     PF01150.12          1      GDA1_CD39  \n",
       "2  isoform2     PF02628.10          1     COX15-CtaA  \n",
       "3  isoform1      PF03932.9          1           CutC  \n",
       "4  isoform2      PF03932.9          1           CutC  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam_tidy = pfam_accession_tidy.join(pfam_accession_to_name, on='pfam_accession')\n",
    "print 'pfam_tidy.shape', pfam_tidy.shape\n",
    "pfam_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>event_name</th>\n",
       "      <th>isoform</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>n_domains</th>\n",
       "      <th>pfam_name</th>\n",
       "      <th>pfam_accession_no_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>1</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>1</td>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>1</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform1</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF03932.9</td>\n",
       "      <td>1</td>\n",
       "      <td>CutC</td>\n",
       "      <td>PF03932</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "3  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "4  exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+   \n",
       "\n",
       "    isoform pfam_accession  n_domains      pfam_name pfam_accession_no_version  \n",
       "0  isoform2     PF00155.16          1  Aminotran_1_2                   PF00155  \n",
       "1  isoform2     PF01150.12          1      GDA1_CD39                   PF01150  \n",
       "2  isoform2     PF02628.10          1     COX15-CtaA                   PF02628  \n",
       "3  isoform1      PF03932.9          1           CutC                   PF03932  \n",
       "4  isoform2      PF03932.9          1           CutC                   PF03932  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam_tidy['pfam_accession_no_version'] = pfam_tidy.pfam_accession.str.split('.').str[0]\n",
    "pfam_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name                                                                                          isoform \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-  isoform2    Aminotran_1_2[1]\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+  isoform2        GDA1_CD39[1]\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-  isoform2       COX15-CtaA[1]\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+  isoform1             CutC[1]\n",
       "                                                                                                    isoform2             CutC[1]\n",
       "Name: pfam_name, dtype: object"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "event_to_isoform_pfam_name = pfam_tidy.groupby(['event_name', 'isoform']).apply(\n",
    "    lambda x: ','.join('{}[{}]'.format(a, int(b)) for a, b in zip(x.pfam_name, x.n_domains)))\n",
    "event_to_isoform_pfam_name.name = 'pfam_name'\n",
    "event_to_isoform_pfam_name.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name                                                                                          isoform \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-  isoform2    PF00155.16[1]\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+  isoform2    PF01150.12[1]\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-  isoform2    PF02628.10[1]\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+  isoform1     PF03932.9[1]\n",
       "                                                                                                    isoform2     PF03932.9[1]\n",
       "Name: pfam_accession, dtype: object"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "event_to_isoform_pfam_accession = pfam_tidy.groupby(['event_name', 'isoform']).apply(\n",
    "    lambda x: ','.join('{}[{}]'.format(a, int(b)) for a, b in zip(x.pfam_accession, x.n_domains)))\n",
    "event_to_isoform_pfam_accession.name = 'pfam_accession'\n",
    "event_to_isoform_pfam_accession.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add GO annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2016-04-14 09:49:11--  http://www.geneontology.org/external2go/pfam2go\n",
      "Resolving www.geneontology.org... 52.27.86.54\n",
      "Connecting to www.geneontology.org|52.27.86.54|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 713246 (697K) [text/plain]\n",
      "Saving to: `pfam2go.1'\n",
      "\n",
      "100%[======================================>] 713,246     2.72M/s   in 0.3s    \n",
      "\n",
      "2016-04-14 09:49:11 (2.72 MB/s) - `pfam2go.1' saved [713246/713246]\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/ipykernel/__main__.py:3: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators; you can avoid this warning by specifying engine='python'.\n",
      "  app.launch_new_instance()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pfam</th>\n",
       "      <th>go</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>pfam_id</th>\n",
       "      <th>go_name</th>\n",
       "      <th>go_id</th>\n",
       "      <th>go_name_only</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pfam_accession_only</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PF00001</th>\n",
       "      <td>Pfam:PF00001 7tm_1</td>\n",
       "      <td>GO:G-protein coupled receptor activity ; GO:0004930</td>\n",
       "      <td>Pfam:PF00001</td>\n",
       "      <td>7tm_1</td>\n",
       "      <td>GO:G-protein coupled receptor activity</td>\n",
       "      <td>GO:0004930</td>\n",
       "      <td>G-protein coupled receptor activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF00001</th>\n",
       "      <td>Pfam:PF00001 7tm_1</td>\n",
       "      <td>GO:G-protein coupled receptor signaling pathway ; GO:0007186</td>\n",
       "      <td>Pfam:PF00001</td>\n",
       "      <td>7tm_1</td>\n",
       "      <td>GO:G-protein coupled receptor signaling pathway</td>\n",
       "      <td>GO:0007186</td>\n",
       "      <td>G-protein coupled receptor signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF00001</th>\n",
       "      <td>Pfam:PF00001 7tm_1</td>\n",
       "      <td>GO:integral component of membrane ; GO:0016021</td>\n",
       "      <td>Pfam:PF00001</td>\n",
       "      <td>7tm_1</td>\n",
       "      <td>GO:integral component of membrane</td>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>integral component of membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF00002</th>\n",
       "      <td>Pfam:PF00002 7tm_2</td>\n",
       "      <td>GO:G-protein coupled receptor activity ; GO:0004930</td>\n",
       "      <td>Pfam:PF00002</td>\n",
       "      <td>7tm_2</td>\n",
       "      <td>GO:G-protein coupled receptor activity</td>\n",
       "      <td>GO:0004930</td>\n",
       "      <td>G-protein coupled receptor activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PF00002</th>\n",
       "      <td>Pfam:PF00002 7tm_2</td>\n",
       "      <td>GO:G-protein coupled receptor signaling pathway ; GO:0007186</td>\n",
       "      <td>Pfam:PF00002</td>\n",
       "      <td>7tm_2</td>\n",
       "      <td>GO:G-protein coupled receptor signaling pathway</td>\n",
       "      <td>GO:0007186</td>\n",
       "      <td>G-protein coupled receptor signaling pathway</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pfam  \\\n",
       "pfam_accession_only                       \n",
       "PF00001              Pfam:PF00001 7tm_1   \n",
       "PF00001              Pfam:PF00001 7tm_1   \n",
       "PF00001              Pfam:PF00001 7tm_1   \n",
       "PF00002              Pfam:PF00002 7tm_2   \n",
       "PF00002              Pfam:PF00002 7tm_2   \n",
       "\n",
       "                                                                               go  \\\n",
       "pfam_accession_only                                                                 \n",
       "PF00001                       GO:G-protein coupled receptor activity ; GO:0004930   \n",
       "PF00001              GO:G-protein coupled receptor signaling pathway ; GO:0007186   \n",
       "PF00001                            GO:integral component of membrane ; GO:0016021   \n",
       "PF00002                       GO:G-protein coupled receptor activity ; GO:0004930   \n",
       "PF00002              GO:G-protein coupled receptor signaling pathway ; GO:0007186   \n",
       "\n",
       "                    pfam_accession pfam_id  \\\n",
       "pfam_accession_only                          \n",
       "PF00001               Pfam:PF00001   7tm_1   \n",
       "PF00001               Pfam:PF00001   7tm_1   \n",
       "PF00001               Pfam:PF00001   7tm_1   \n",
       "PF00002               Pfam:PF00002   7tm_2   \n",
       "PF00002               Pfam:PF00002   7tm_2   \n",
       "\n",
       "                                                             go_name  \\\n",
       "pfam_accession_only                                                    \n",
       "PF00001                       GO:G-protein coupled receptor activity   \n",
       "PF00001              GO:G-protein coupled receptor signaling pathway   \n",
       "PF00001                            GO:integral component of membrane   \n",
       "PF00002                       GO:G-protein coupled receptor activity   \n",
       "PF00002              GO:G-protein coupled receptor signaling pathway   \n",
       "\n",
       "                          go_id                                  go_name_only  \n",
       "pfam_accession_only                                                            \n",
       "PF00001              GO:0004930           G-protein coupled receptor activity  \n",
       "PF00001              GO:0007186  G-protein coupled receptor signaling pathway  \n",
       "PF00001              GO:0016021                integral component of membrane  \n",
       "PF00002              GO:0004930           G-protein coupled receptor activity  \n",
       "PF00002              GO:0007186  G-protein coupled receptor signaling pathway  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "! wget http://www.geneontology.org/external2go/pfam2go \n",
    "\n",
    "pfam_to_go = pd.read_table('pfam2go', sep=' > ', skiprows=6, header=None, names=['pfam', 'go'])\n",
    "# pfam_to_go.head()\n",
    "pfam_to_go[['pfam_accession', 'pfam_id']] = pfam_to_go.pfam.str.split(' ').apply(pd.Series)\n",
    "# pfam_to_go.head()\n",
    "pfam_to_go[['go_name', 'go_id']] = pfam_to_go.go.str.split(' ; ').apply(pd.Series)\n",
    "# pfam_to_go.head()\n",
    "pfam_to_go['pfam_accession_only'] = pfam_to_go.pfam_accession.str.split(':').str[1]\n",
    "# pfam_to_go.head()\n",
    "pfam_to_go = pfam_to_go.set_index('pfam_accession_only')\n",
    "# pfam_to_go.head()\n",
    "pfam_to_go['go_name_only'] = pfam_to_go.go_name.str.split(':').str[1]\n",
    "pfam_to_go.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>event_name</th>\n",
       "      <th>isoform</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>n_domains</th>\n",
       "      <th>pfam_name</th>\n",
       "      <th>pfam_accession_no_version</th>\n",
       "      <th>go</th>\n",
       "      <th>go_id</th>\n",
       "      <th>go_name</th>\n",
       "      <th>go_name_only</th>\n",
       "      <th>pfam</th>\n",
       "      <th>pfam_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>1</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155</td>\n",
       "      <td>GO:pyridoxal phosphate binding ; GO:0030170</td>\n",
       "      <td>GO:0030170</td>\n",
       "      <td>GO:pyridoxal phosphate binding</td>\n",
       "      <td>pyridoxal phosphate binding</td>\n",
       "      <td>Pfam:PF00155 Aminotran_1_2</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF00155.16</td>\n",
       "      <td>1</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "      <td>PF00155</td>\n",
       "      <td>GO:biosynthetic process ; GO:0009058</td>\n",
       "      <td>GO:0009058</td>\n",
       "      <td>GO:biosynthetic process</td>\n",
       "      <td>biosynthetic process</td>\n",
       "      <td>Pfam:PF00155 Aminotran_1_2</td>\n",
       "      <td>Aminotran_1_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF01150.12</td>\n",
       "      <td>1</td>\n",
       "      <td>GDA1_CD39</td>\n",
       "      <td>PF01150</td>\n",
       "      <td>GO:hydrolase activity ; GO:0016787</td>\n",
       "      <td>GO:0016787</td>\n",
       "      <td>GO:hydrolase activity</td>\n",
       "      <td>hydrolase activity</td>\n",
       "      <td>Pfam:PF01150 GDA1_CD39</td>\n",
       "      <td>GDA1_CD39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>1</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628</td>\n",
       "      <td>GO:oxidoreductase activity, acting on the CH-CH group of donors ; GO:0016627</td>\n",
       "      <td>GO:0016627</td>\n",
       "      <td>GO:oxidoreductase activity, acting on the CH-CH group of donors</td>\n",
       "      <td>oxidoreductase activity, acting on the CH-CH group of donors</td>\n",
       "      <td>Pfam:PF02628 COX15-CtaA</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-</td>\n",
       "      <td>isoform2</td>\n",
       "      <td>PF02628.10</td>\n",
       "      <td>1</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "      <td>PF02628</td>\n",
       "      <td>GO:heme a biosynthetic process ; GO:0006784</td>\n",
       "      <td>GO:0006784</td>\n",
       "      <td>GO:heme a biosynthetic process</td>\n",
       "      <td>heme a biosynthetic process</td>\n",
       "      <td>Pfam:PF02628 COX15-CtaA</td>\n",
       "      <td>COX15-CtaA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                           event_name  \\\n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "0  exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-   \n",
       "1  exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "2  exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-   \n",
       "\n",
       "    isoform pfam_accession  n_domains      pfam_name  \\\n",
       "0  isoform2     PF00155.16          1  Aminotran_1_2   \n",
       "0  isoform2     PF00155.16          1  Aminotran_1_2   \n",
       "1  isoform2     PF01150.12          1      GDA1_CD39   \n",
       "2  isoform2     PF02628.10          1     COX15-CtaA   \n",
       "2  isoform2     PF02628.10          1     COX15-CtaA   \n",
       "\n",
       "  pfam_accession_no_version  \\\n",
       "0                   PF00155   \n",
       "0                   PF00155   \n",
       "1                   PF01150   \n",
       "2                   PF02628   \n",
       "2                   PF02628   \n",
       "\n",
       "                                                                             go  \\\n",
       "0                                   GO:pyridoxal phosphate binding ; GO:0030170   \n",
       "0                                          GO:biosynthetic process ; GO:0009058   \n",
       "1                                            GO:hydrolase activity ; GO:0016787   \n",
       "2  GO:oxidoreductase activity, acting on the CH-CH group of donors ; GO:0016627   \n",
       "2                                   GO:heme a biosynthetic process ; GO:0006784   \n",
       "\n",
       "        go_id  \\\n",
       "0  GO:0030170   \n",
       "0  GO:0009058   \n",
       "1  GO:0016787   \n",
       "2  GO:0016627   \n",
       "2  GO:0006784   \n",
       "\n",
       "                                                           go_name  \\\n",
       "0                                   GO:pyridoxal phosphate binding   \n",
       "0                                          GO:biosynthetic process   \n",
       "1                                            GO:hydrolase activity   \n",
       "2  GO:oxidoreductase activity, acting on the CH-CH group of donors   \n",
       "2                                   GO:heme a biosynthetic process   \n",
       "\n",
       "                                                   go_name_only  \\\n",
       "0                                   pyridoxal phosphate binding   \n",
       "0                                          biosynthetic process   \n",
       "1                                            hydrolase activity   \n",
       "2  oxidoreductase activity, acting on the CH-CH group of donors   \n",
       "2                                   heme a biosynthetic process   \n",
       "\n",
       "                         pfam        pfam_id  \n",
       "0  Pfam:PF00155 Aminotran_1_2  Aminotran_1_2  \n",
       "0  Pfam:PF00155 Aminotran_1_2  Aminotran_1_2  \n",
       "1      Pfam:PF01150 GDA1_CD39      GDA1_CD39  \n",
       "2     Pfam:PF02628 COX15-CtaA     COX15-CtaA  \n",
       "2     Pfam:PF02628 COX15-CtaA     COX15-CtaA  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "join_cols = pfam_to_go.columns.difference(pfam_tidy.columns)\n",
    "pfam_tidy_go = pfam_tidy.join(pfam_to_go[join_cols], on='pfam_accession_no_version')\n",
    "pfam_tidy_go.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name                                                                                          isoform \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-  isoform2                                GO:0030170[1],GO:0009058[1]\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+  isoform2                                              GO:0016787[1]\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-  isoform2    GO:0016627[1],GO:0006784[1],GO:0055114[1],GO:0016021[1]\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+  isoform1                                GO:0005507[1],GO:0055070[1]\n",
       "                                                                                                    isoform2                                GO:0005507[1],GO:0055070[1]\n",
       "Name: pfam_go_id, dtype: object"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "event_to_isoform_pfam_go_id = pfam_tidy_go.groupby(['event_name', 'isoform']).apply(\n",
    "    lambda x: ','.join('{}[{}]'.format(a, int(b)) for a, b in zip(x.go_id, x.n_domains)))\n",
    "event_to_isoform_pfam_go_id.name = 'pfam_go_id'\n",
    "event_to_isoform_pfam_go_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "event_name                                                                                          isoform \n",
       "exon:chr10:101165513-101165617:-@exon:chr10:101163481-101163631:-@exon:chr10:101163226-101163391:-  isoform2                                             (pyridoxal phosphate binding)[1],(biosynthetic process)[1]\n",
       "exon:chr10:101421203-101421385:+@exon:chr10:101439018-101439223:+@exon:chr10:101439482-101439632:+  isoform2                                                                                (hydrolase activity)[1]\n",
       "exon:chr10:101480744-101480825:-@exon:chr10:101478103-101478257:-@exon:chr10:101476105-101476218:-  isoform2    (oxidoreductase activity, acting on the CH-CH group of donors)[1],(heme a biosynthetic process)[...\n",
       "exon:chr10:101507014-101507147:+@exon:chr10:101510126-101510153:+@exon:chr10:101514286-101514391:+  isoform1                                                    (copper ion binding)[1],(copper ion homeostasis)[1]\n",
       "                                                                                                    isoform2                                                    (copper ion binding)[1],(copper ion homeostasis)[1]\n",
       "Name: pfam_go_name, dtype: object"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "event_to_isoform_pfam_go_name = pfam_tidy_go.groupby(['event_name', 'isoform']).apply(\n",
    "    lambda x: ','.join('({})[{}]'.format(a, int(b)) for a, b in zip(x.go_name_only, x.n_domains)))\n",
    "event_to_isoform_pfam_go_name.name = 'pfam_go_name'\n",
    "event_to_isoform_pfam_go_name.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "event_to_isoform_series = {'pfam_go_name': event_to_isoform_pfam_go_name, \n",
    "                           'pfam_go_id': event_to_isoform_pfam_go_id,\n",
    "                           'pfam_name': event_to_isoform_pfam_name, \n",
    "                           'pfam_accession': event_to_isoform_pfam_accession}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save the data!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for name, series in event_to_isoform_series.items():\n",
    "    df = series.unstack()\n",
    "    df.columns = df.columns.map(lambda x: x + '_' + name)\n",
    "    df.to_csv('{}/isoform_domain_{}.csv'.format(alternative_feature_folder, name))\n",
    "#     print df.head()\n",
    "#     splicing_feature_data = splicing_feature_data.join(df)\n",
    "# splicing_feature_data.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
