{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>junctions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100188908-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100188913-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100189330-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:100193697-100193848:-</td>\n",
       "      <td>exon:chr10:100190888-100191048:-</td>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>(chr10:100191049-100193696:-, chr10:100190428-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:100195392-100195529:-</td>\n",
       "      <td>exon:chr10:100195029-100195171:-</td>\n",
       "      <td>exon:chr10:100193697-100193848:-</td>\n",
       "      <td>(chr10:100195172-100195391:-, chr10:100193849-...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              exon1                             exon2  \\\n",
       "0  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "1  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "2  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "3  exon:chr10:100193697-100193848:-  exon:chr10:100190888-100191048:-   \n",
       "4  exon:chr10:100195392-100195529:-  exon:chr10:100195029-100195171:-   \n",
       "\n",
       "                              exon3  \\\n",
       "0  exon:chr10:100188908-100189399:-   \n",
       "1  exon:chr10:100188913-100189399:-   \n",
       "2  exon:chr10:100189330-100189399:-   \n",
       "3  exon:chr10:100190328-100190427:-   \n",
       "4  exon:chr10:100193697-100193848:-   \n",
       "\n",
       "                                           junctions  \n",
       "0  (chr10:100189647-100190327:-, chr10:100189400-...  \n",
       "1  (chr10:100189647-100190327:-, chr10:100189400-...  \n",
       "2  (chr10:100189647-100190327:-, chr10:100189400-...  \n",
       "3  (chr10:100191049-100193696:-, chr10:100190428-...  \n",
       "4  (chr10:100195172-100195391:-, chr10:100193849-...  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import gffutils\n",
    "import pybedtools\n",
    "import re\n",
    "import numpy as np\n",
    "\n",
    "v19db_filename = '/projects/ps-yeolab/genomes/hg19/gencode/v19/gencode.v19.annotation.gtf.db'\n",
    "v19db = gffutils.FeatureDB(v19db_filename)\n",
    "\n",
    "folder = '/projects/ps-yeolab/obotvinnik/singlecell_pnms'\n",
    "\n",
    "exons_to_junctions = pd.read_csv('{}/exons_to_junctions_se.csv'.format(folder), index_col=[0, 1, 2], squeeze=True, header=None)\n",
    "exons_to_junctions = exons_to_junctions.reset_index()\n",
    "exons_to_junctions = exons_to_junctions.rename(columns={0: 'exon1', 1: 'exon2', 2: 'exon3', 3:'junctions'})\n",
    "exons_to_junctions['junctions'] = exons_to_junctions.junctions.map(eval)\n",
    "exons_to_junctions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>junctions</th>\n",
       "      <th>event_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100188908-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100188913-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>exon:chr10:100189548-100189646:-</td>\n",
       "      <td>exon:chr10:100189330-100189399:-</td>\n",
       "      <td>(chr10:100189647-100190327:-, chr10:100189400-...</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:100193697-100193848:-</td>\n",
       "      <td>exon:chr10:100190888-100191048:-</td>\n",
       "      <td>exon:chr10:100190328-100190427:-</td>\n",
       "      <td>(chr10:100191049-100193696:-, chr10:100190428-...</td>\n",
       "      <td>exon:chr10:100193697-100193848:-@exon:chr10:10...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:100195392-100195529:-</td>\n",
       "      <td>exon:chr10:100195029-100195171:-</td>\n",
       "      <td>exon:chr10:100193697-100193848:-</td>\n",
       "      <td>(chr10:100195172-100195391:-, chr10:100193849-...</td>\n",
       "      <td>exon:chr10:100195392-100195529:-@exon:chr10:10...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              exon1                             exon2  \\\n",
       "0  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "1  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "2  exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-   \n",
       "3  exon:chr10:100193697-100193848:-  exon:chr10:100190888-100191048:-   \n",
       "4  exon:chr10:100195392-100195529:-  exon:chr10:100195029-100195171:-   \n",
       "\n",
       "                              exon3  \\\n",
       "0  exon:chr10:100188908-100189399:-   \n",
       "1  exon:chr10:100188913-100189399:-   \n",
       "2  exon:chr10:100189330-100189399:-   \n",
       "3  exon:chr10:100190328-100190427:-   \n",
       "4  exon:chr10:100193697-100193848:-   \n",
       "\n",
       "                                           junctions  \\\n",
       "0  (chr10:100189647-100190327:-, chr10:100189400-...   \n",
       "1  (chr10:100189647-100190327:-, chr10:100189400-...   \n",
       "2  (chr10:100189647-100190327:-, chr10:100189400-...   \n",
       "3  (chr10:100191049-100193696:-, chr10:100190428-...   \n",
       "4  (chr10:100195172-100195391:-, chr10:100193849-...   \n",
       "\n",
       "                                            event_id  \n",
       "0  exon:chr10:100190328-100190427:-@exon:chr10:10...  \n",
       "1  exon:chr10:100190328-100190427:-@exon:chr10:10...  \n",
       "2  exon:chr10:100190328-100190427:-@exon:chr10:10...  \n",
       "3  exon:chr10:100193697-100193848:-@exon:chr10:10...  \n",
       "4  exon:chr10:100195392-100195529:-@exon:chr10:10...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exons_to_junctions['event_id'] = exons_to_junctions['exon1'] + '@' \\\n",
    "    + exons_to_junctions['exon2'] + '@' + \\\n",
    "    exons_to_junctions['exon3']\n",
    "exons_to_junctions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(706732, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_location</th>\n",
       "      <th>exon</th>\n",
       "      <th>direction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100008749-100010821:-</td>\n",
       "      <td>exon:chr10:100010822-100010933:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10:100010934-100011322:-</td>\n",
       "      <td>exon:chr10:100011323-100011459:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100143626-100144703:-</td>\n",
       "      <td>exon:chr10:100144704-100144824:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100144825-100146957:-</td>\n",
       "      <td>exon:chr10:100146958-100147064:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100147065-100155147:-</td>\n",
       "      <td>exon:chr10:100155148-100155209:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             junction_location                              exon direction\n",
       "0  chr10:100008749-100010821:-  exon:chr10:100010822-100010933:-  upstream\n",
       "1  chr10:100010934-100011322:-  exon:chr10:100011323-100011459:-  upstream\n",
       "2  chr10:100143626-100144703:-  exon:chr10:100144704-100144824:-  upstream\n",
       "3  chr10:100144825-100146957:-  exon:chr10:100146958-100147064:-  upstream\n",
       "4  chr10:100147065-100155147:-  exon:chr10:100155148-100155209:-  upstream"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_junction_direction = pd.read_csv('{}/junction_exons.csv'.format(folder), index_col=0)\n",
    "print exon_junction_direction.shape\n",
    "exon_junction_direction.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write alternative exons to bed file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(58842, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189548</td>\n",
       "      <td>100189646</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189548</td>\n",
       "      <td>100189646</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189548</td>\n",
       "      <td>100189646</td>\n",
       "      <td>exon:chr10:100190328-100190427:-@exon:chr10:10...</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100190888</td>\n",
       "      <td>100191048</td>\n",
       "      <td>exon:chr10:100193697-100193848:-@exon:chr10:10...</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100195029</td>\n",
       "      <td>100195171</td>\n",
       "      <td>exon:chr10:100195392-100195529:-@exon:chr10:10...</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom      start       stop  \\\n",
       "0  chr10  100189548  100189646   \n",
       "1  chr10  100189548  100189646   \n",
       "2  chr10  100189548  100189646   \n",
       "3  chr10  100190888  100191048   \n",
       "4  chr10  100195029  100195171   \n",
       "\n",
       "                                                name  score strand  \n",
       "0  exon:chr10:100190328-100190427:-@exon:chr10:10...   1000      -  \n",
       "1  exon:chr10:100190328-100190427:-@exon:chr10:10...   1000      -  \n",
       "2  exon:chr10:100190328-100190427:-@exon:chr10:10...   1000      -  \n",
       "3  exon:chr10:100193697-100193848:-@exon:chr10:10...   1000      -  \n",
       "4  exon:chr10:100195392-100195529:-@exon:chr10:10...   1000      -  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from poshsplice.region import Region\n",
    "\n",
    "exons_to_junctions['exon2_region'] = exons_to_junctions.exon2.map(Region)\n",
    "\n",
    "exon2_bed = pd.DataFrame.from_records(exons_to_junctions.exon2_region.map(\n",
    "    lambda x: pd.Series(dict(chrom=x.chrom, start=x._start, stop=x._stop, strand=x.strand, score=1000))))\n",
    "\n",
    "exon2_bed['name'] = exons_to_junctions.event_id\n",
    "exon2_bed = exon2_bed.reindex(columns=['chrom', 'start', 'stop', 'name', 'score', 'strand'])\n",
    "\n",
    "print exon2_bed.shape\n",
    "exon2_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "exon2_bed.to_csv('{}/skipped_exon_exon2.bed'.format(folder), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get internal, constitutive exons\n",
    "\n",
    "### Internal: have both downstream and upstream junctions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "exon                              direction \n",
       "exon:chr10:100003848-100004106:+  downstream    1\n",
       "exon:chr10:100003848-100004321:+  downstream    1\n",
       "exon:chr10:100003848-100004651:+  downstream    1\n",
       "exon:chr10:100003848-100004654:+  downstream    1\n",
       "exon:chr10:100007447-100008748:-  downstream    1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_direction = exon_junction_direction.groupby(['exon', 'direction']).size()\n",
    "exon_direction.head()\n",
    "# constitutive = constitutive[constitutive == 2]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Group on the number of \"direction\" items they have. Must have two to be internal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "exon\n",
       "exon:chr10:100010822-100010933:-    2\n",
       "exon:chr10:100144704-100144824:-    2\n",
       "exon:chr10:100146958-100147064:-    2\n",
       "exon:chr10:100155148-100155209:-    2\n",
       "exon:chr10:100157102-100157255:-    2\n",
       "dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "internal_exons = exon_direction.groupby(level=0).size()\n",
    "internal_exons = internal_exons[internal_exons == 2]\n",
    "internal_exons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction_location</th>\n",
       "      <th>exon</th>\n",
       "      <th>direction</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10:100008749-100010821:-</td>\n",
       "      <td>exon:chr10:100010822-100010933:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10:100143626-100144703:-</td>\n",
       "      <td>exon:chr10:100144704-100144824:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10:100144825-100146957:-</td>\n",
       "      <td>exon:chr10:100146958-100147064:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10:100147065-100155147:-</td>\n",
       "      <td>exon:chr10:100155148-100155209:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10:100155210-100157101:-</td>\n",
       "      <td>exon:chr10:100157102-100157255:-</td>\n",
       "      <td>upstream</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             junction_location                              exon direction\n",
       "0  chr10:100008749-100010821:-  exon:chr10:100010822-100010933:-  upstream\n",
       "2  chr10:100143626-100144703:-  exon:chr10:100144704-100144824:-  upstream\n",
       "3  chr10:100144825-100146957:-  exon:chr10:100146958-100147064:-  upstream\n",
       "4  chr10:100147065-100155147:-  exon:chr10:100155148-100155209:-  upstream\n",
       "6  chr10:100155210-100157101:-  exon:chr10:100157102-100157255:-  upstream"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_juction_direction_internal = exon_junction_direction.loc[exon_junction_direction.exon.isin(internal_exons.index)]\n",
    "exon_juction_direction_internal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(168130,)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "exon                              direction \n",
       "exon:chr10:100010822-100010933:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100144704-100144824:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100146958-100147064:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100155148-100155209:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100157102-100157255:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100159859-100160014:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100167339-100167412:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100167661-100167754:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100170696-100170715:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100177932-100178014:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100179802-100179915:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100182126-100182270:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100183360-100183425:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100183510-100183644:-  downstream    1\n",
       "                                  upstream      1\n",
       "exon:chr10:100184070-100184131:-  downstream    1\n",
       "                                  upstream      1\n",
       "                                               ..\n",
       "exon:chrY:6846254-6846284:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6863845-6863939:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6893076-6893183:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6911021-6911166:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6938237-6938369:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6938761-6938902:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6939601-6939664:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6939774-6939871:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6942601-6942661:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6948773-6948894:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6953939-6954013:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6954331-6954458:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:6955308-6955473:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:7235397-7235474:+       downstream    1\n",
       "                                  upstream      1\n",
       "exon:chrY:7239784-7239930:+       downstream    1\n",
       "                                  upstream      1\n",
       "dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "single_upstream_downstream = exon_juction_direction_internal.groupby(['exon', 'direction']).size()\n",
    "single_upstream_downstream = single_upstream_downstream[single_upstream_downstream == 1]\n",
    "single_upstream_downstream = single_upstream_downstream.groupby(level=0).filter(lambda x: len(x) == 2)\n",
    "print single_upstream_downstream.shape\n",
    "single_upstream_downstream"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon</th>\n",
       "      <th>direction</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>168125</th>\n",
       "      <td>exon:chrY:6955308-6955473:+</td>\n",
       "      <td>upstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168126</th>\n",
       "      <td>exon:chrY:7235397-7235474:+</td>\n",
       "      <td>downstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168127</th>\n",
       "      <td>exon:chrY:7235397-7235474:+</td>\n",
       "      <td>upstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168128</th>\n",
       "      <td>exon:chrY:7239784-7239930:+</td>\n",
       "      <td>downstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168129</th>\n",
       "      <td>exon:chrY:7239784-7239930:+</td>\n",
       "      <td>upstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               exon   direction  0\n",
       "168125  exon:chrY:6955308-6955473:+    upstream  1\n",
       "168126  exon:chrY:7235397-7235474:+  downstream  1\n",
       "168127  exon:chrY:7235397-7235474:+    upstream  1\n",
       "168128  exon:chrY:7239784-7239930:+  downstream  1\n",
       "168129  exon:chrY:7239784-7239930:+    upstream  1"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "single_upstream_downstream = single_upstream_downstream.reset_index()\n",
    "single_upstream_downstream.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(84065,)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "single_upstream_downstream.exon.unique().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove all exons that appear as exon2 in alternative annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(74549,)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon</th>\n",
       "      <th>direction</th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>exon:chr10:100010822-100010933:-</td>\n",
       "      <td>downstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>exon:chr10:100010822-100010933:-</td>\n",
       "      <td>upstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>exon:chr10:100144704-100144824:-</td>\n",
       "      <td>downstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>exon:chr10:100144704-100144824:-</td>\n",
       "      <td>upstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>exon:chr10:100146958-100147064:-</td>\n",
       "      <td>downstream</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               exon   direction  0\n",
       "0  exon:chr10:100010822-100010933:-  downstream  1\n",
       "1  exon:chr10:100010822-100010933:-    upstream  1\n",
       "2  exon:chr10:100144704-100144824:-  downstream  1\n",
       "3  exon:chr10:100144704-100144824:-    upstream  1\n",
       "4  exon:chr10:100146958-100147064:-  downstream  1"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "constitutive = single_upstream_downstream.loc[~single_upstream_downstream.exon.isin(exons_to_junctions.exon2)]\n",
    "print constitutive.exon.unique().shape\n",
    "constitutive.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    exon:chr10:100010822-100010933:-\n",
       "1    exon:chr10:100144704-100144824:-\n",
       "2    exon:chr10:100146958-100147064:-\n",
       "3    exon:chr10:100155148-100155209:-\n",
       "4    exon:chr10:100157102-100157255:-\n",
       "dtype: object"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from poshsplice.region import Region\n",
    "constitutive_regions = pd.Series(constitutive.exon.unique()).map(Region)\n",
    "constitutive_regions.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that the junctions for these exons are seen in 20+ single cells per celltype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(214, 573064)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10:100008749-100010821:-|5p</th>\n",
       "      <th>chr10:100010934-100011322:-|5p</th>\n",
       "      <th>chr10:100143626-100144703:-|5p</th>\n",
       "      <th>chr10:100144825-100146957:-|5p</th>\n",
       "      <th>chr10:100147065-100155147:-|5p</th>\n",
       "      <th>chr10:100150512-100150766:-|5p</th>\n",
       "      <th>chr10:100155210-100157101:-|5p</th>\n",
       "      <th>chr10:100157256-100159858:-|5p</th>\n",
       "      <th>chr10:100160015-100167338:-|5p</th>\n",
       "      <th>chr10:100167413-100167660:-|5p</th>\n",
       "      <th>...</th>\n",
       "      <th>chrY:7194211-7209155:+|3p</th>\n",
       "      <th>chrY:7194211-7224175:+|3p</th>\n",
       "      <th>chrY:7198575-7201095:+|3p</th>\n",
       "      <th>chrY:7208064-7209155:+|3p</th>\n",
       "      <th>chrY:7209276-7224175:+|3p</th>\n",
       "      <th>chrY:7223339-7223437:+|3p</th>\n",
       "      <th>chrY:7224272-7235396:+|3p</th>\n",
       "      <th>chrY:7225609-7226951:+|3p</th>\n",
       "      <th>chrY:7235475-7239783:+|3p</th>\n",
       "      <th>chrY:7239931-7243752:+|3p</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 573064 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr10:100008749-100010821:-|5p  chr10:100010934-100011322:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                NaN                             NaN   \n",
       "\n",
       "           chr10:100143626-100144703:-|5p  chr10:100144825-100146957:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                NaN                             NaN   \n",
       "\n",
       "           chr10:100147065-100155147:-|5p  chr10:100150512-100150766:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                NaN                             NaN   \n",
       "\n",
       "           chr10:100155210-100157101:-|5p  chr10:100157256-100159858:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                NaN                             NaN   \n",
       "\n",
       "           chr10:100160015-100167338:-|5p  chr10:100167413-100167660:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                NaN                             NaN   \n",
       "\n",
       "                     ...              chrY:7194211-7209155:+|3p  \\\n",
       "sample_id            ...                                          \n",
       "CVN_01               ...                                    NaN   \n",
       "CVN_02               ...                                      0   \n",
       "CVN_03               ...                                    NaN   \n",
       "CVN_04               ...                                    NaN   \n",
       "CVN_05               ...                                    NaN   \n",
       "\n",
       "           chrY:7194211-7224175:+|3p  chrY:7198575-7201095:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                           NaN                        NaN   \n",
       "CVN_02                             0                        NaN   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                           NaN                        NaN   \n",
       "CVN_05                           NaN                        NaN   \n",
       "\n",
       "           chrY:7208064-7209155:+|3p  chrY:7209276-7224175:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                           NaN                        NaN   \n",
       "CVN_02                             1                          1   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                           NaN                        NaN   \n",
       "CVN_05                           NaN                        NaN   \n",
       "\n",
       "           chrY:7223339-7223437:+|3p  chrY:7224272-7235396:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                           NaN                        NaN   \n",
       "CVN_02                           NaN                          1   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                           NaN                        NaN   \n",
       "CVN_05                           NaN                        NaN   \n",
       "\n",
       "           chrY:7225609-7226951:+|3p  chrY:7235475-7239783:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                           NaN                        NaN   \n",
       "CVN_02                           NaN                          1   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                           NaN                        NaN   \n",
       "CVN_05                           NaN                        NaN   \n",
       "\n",
       "           chrY:7239931-7243752:+|3p  \n",
       "sample_id                             \n",
       "CVN_01                           NaN  \n",
       "CVN_02                             1  \n",
       "CVN_03                           NaN  \n",
       "CVN_04                           NaN  \n",
       "CVN_05                           NaN  \n",
       "\n",
       "[5 rows x 573064 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_folder = '/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper'\n",
    "psi3_psi5 = pd.read_hdf('{}/psi5_psi3_combined.hdf'.format(csv_folder), 'psi')\n",
    "# psi3_psi5 = pd.read_csv('{}/psi5_psi3_combined.csv'.format(csv_folder), index_col=0)\n",
    "print psi3_psi5.shape\n",
    "psi3_psi5.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "expression.csv              psi5_psi3_combined.hdf\r\n",
      "expression_stressed.csv     psi5_psi3_combined_metadata.csv\r\n",
      "mapping_stats.csv           psi5_psi3_combined_metadata.hdf\r\n",
      "mapping_stats_stressed.csv  psi5_stressed.csv\r\n",
      "metadata.csv                sj_raw.csv\r\n",
      "metadata_stressed.csv       skipped_exon_psi_from_junctions.csv\r\n",
      "psi3.csv                    splice_junction_metadata.csv\r\n",
      "psi3_stressed.csv           splice_junctions_psi.csv\r\n",
      "psi5.csv                    splicing.csv\r\n",
      "psi5_psi3_combined.csv      splicing_stressed.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls $csv_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intron_location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:120933-129054:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120933</td>\n",
       "      <td>129054</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:135767-137718:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135767</td>\n",
       "      <td>137718</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:326417-327587:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:334298-342391:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>334298</td>\n",
       "      <td>342391</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>RP4-669L17.10</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:566538-566579:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>566538</td>\n",
       "      <td>566579</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>RP5-857K21.4</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        chrom  intron_start  intron_stop strand intron_motif  \\\n",
       "intron_location                                                                \n",
       "chr1:120933-129054:-|5p  chr1        120933       129054      -        GT/AG   \n",
       "chr1:135767-137718:-|5p  chr1        135767       137718      -        GC/AG   \n",
       "chr1:326417-327587:+|5p  chr1        326417       327587      +        GC/AG   \n",
       "chr1:334298-342391:+|5p  chr1        334298       342391      +        GT/AG   \n",
       "chr1:566538-566579:-|5p  chr1        566538       566579      -        GC/AG   \n",
       "\n",
       "                        annotated                           gencode_id  \\\n",
       "intron_location                                                          \n",
       "chr1:120933-129054:-|5p      True                    ENSG00000238009.2   \n",
       "chr1:135767-137718:-|5p     False                    ENSG00000237683.5   \n",
       "chr1:326417-327587:+|5p     False  ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:334298-342391:+|5p      True                    ENSG00000237094.7   \n",
       "chr1:566538-566579:-|5p     False                    ENSG00000230021.3   \n",
       "\n",
       "                                              ensembl_id  \\\n",
       "intron_location                                            \n",
       "chr1:120933-129054:-|5p                  ENSG00000238009   \n",
       "chr1:135767-137718:-|5p                  ENSG00000237683   \n",
       "chr1:326417-327587:+|5p  ENSG00000237094,ENSG00000250575   \n",
       "chr1:334298-342391:+|5p                  ENSG00000237094   \n",
       "chr1:566538-566579:-|5p                  ENSG00000230021   \n",
       "\n",
       "                                                     gene_id  \\\n",
       "intron_location                                                \n",
       "chr1:120933-129054:-|5p                    ENSG00000238009.2   \n",
       "chr1:135767-137718:-|5p                    ENSG00000237683.5   \n",
       "chr1:326417-327587:+|5p  ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:334298-342391:+|5p                    ENSG00000237094.7   \n",
       "chr1:566538-566579:-|5p                    ENSG00000230021.3   \n",
       "\n",
       "                                          gene_name      ...       \\\n",
       "intron_location                                          ...        \n",
       "chr1:120933-129054:-|5p                RP11-34P13.7      ...        \n",
       "chr1:135767-137718:-|5p                  AL627309.1      ...        \n",
       "chr1:326417-327587:+|5p  RP4-669L17.10,RP4-669L17.8      ...        \n",
       "chr1:334298-342391:+|5p               RP4-669L17.10      ...        \n",
       "chr1:566538-566579:-|5p                RP5-857K21.4      ...        \n",
       "\n",
       "                        gerstberger2014_rbp_target_ncrna  \\\n",
       "intron_location                                            \n",
       "chr1:120933-129054:-|5p                            False   \n",
       "chr1:135767-137718:-|5p                            False   \n",
       "chr1:326417-327587:+|5p                            False   \n",
       "chr1:334298-342391:+|5p                            False   \n",
       "chr1:566538-566579:-|5p                            False   \n",
       "\n",
       "                        gerstberger2014_rbp_target_ribosome  \\\n",
       "intron_location                                               \n",
       "chr1:120933-129054:-|5p                               False   \n",
       "chr1:135767-137718:-|5p                               False   \n",
       "chr1:326417-327587:+|5p                               False   \n",
       "chr1:334298-342391:+|5p                               False   \n",
       "chr1:566538-566579:-|5p                               False   \n",
       "\n",
       "                        gerstberger2014_rbp_target_trna gerstberger2014_tf  \\\n",
       "intron_location                                                              \n",
       "chr1:120933-129054:-|5p                           False              False   \n",
       "chr1:135767-137718:-|5p                           False              False   \n",
       "chr1:326417-327587:+|5p                           False              False   \n",
       "chr1:334298-342391:+|5p                           False              False   \n",
       "chr1:566538-566579:-|5p                           False              False   \n",
       "\n",
       "                           rbp ribosomal ribosomal_subunit synapse  \\\n",
       "intron_location                                                      \n",
       "chr1:120933-129054:-|5p  False     False             False   False   \n",
       "chr1:135767-137718:-|5p  False     False             False   False   \n",
       "chr1:326417-327587:+|5p  False     False             False   False   \n",
       "chr1:334298-342391:+|5p  False     False             False   False   \n",
       "chr1:566538-566579:-|5p  False     False             False   False   \n",
       "\n",
       "                        transcription_factor intron_length  \n",
       "intron_location                                             \n",
       "chr1:120933-129054:-|5p                False          8121  \n",
       "chr1:135767-137718:-|5p                False          1951  \n",
       "chr1:326417-327587:+|5p                False          1170  \n",
       "chr1:334298-342391:+|5p                False          8093  \n",
       "chr1:566538-566579:-|5p                False            41  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata = pd.read_hdf('{}/psi5_psi3_combined_metadata.hdf'.format(csv_folder), 'metadata')\n",
    "sj_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata['exon_stop'] = sj_metadata['intron_start'] - 1\n",
    "sj_metadata['exon_start'] = sj_metadata['intron_stop'] + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>single</th>\n",
       "      <th>pooled</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>neuron</th>\n",
       "      <th>progenitor</th>\n",
       "      <th>stressed</th>\n",
       "      <th>craig_venter</th>\n",
       "      <th>differentiation_batch</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       single pooled phenotype neuron progenitor stressed craig_venter  \\\n",
       "CVN_01   True  False       NPC  False       True    False         True   \n",
       "CVN_02   True  False       NPC  False       True    False         True   \n",
       "CVN_03   True  False       NPC  False       True    False         True   \n",
       "CVN_04   True  False       NPC  False       True    False         True   \n",
       "CVN_05   True  False       NPC  False       True    False         True   \n",
       "\n",
       "       differentiation_batch split  \n",
       "CVN_01                   NaN   NaN  \n",
       "CVN_02                   NaN   NaN  \n",
       "CVN_03                   NaN   NaN  \n",
       "CVN_04                   NaN   NaN  \n",
       "CVN_05                   NaN   NaN  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata = pd.read_csv('{}/metadata.csv'.format(csv_folder), index_col=0)\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "phenotype = metadata.phenotype[metadata.single]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(214, 573064)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi3_psi5.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10:101157444-101162336:-|5p</th>\n",
       "      <th>chr10:101162480-101163225:-|5p</th>\n",
       "      <th>chr10:101163392-101163480:-|5p</th>\n",
       "      <th>chr10:101163392-101165512:-|5p</th>\n",
       "      <th>chr10:101163632-101165512:-|5p</th>\n",
       "      <th>chr10:101166007-101190204:-|5p</th>\n",
       "      <th>chr10:101437763-101476104:-|5p</th>\n",
       "      <th>chr10:101456299-101476104:-|5p</th>\n",
       "      <th>chr10:101473237-101476104:-|5p</th>\n",
       "      <th>chr10:101474476-101476104:-|5p</th>\n",
       "      <th>...</th>\n",
       "      <th>chrY:2713785-2734833:+|3p</th>\n",
       "      <th>chrY:2714508-2722640:+|3p</th>\n",
       "      <th>chrY:2721733-2722640:+|3p</th>\n",
       "      <th>chrY:2722746-2734833:+|3p</th>\n",
       "      <th>chrY:2722813-2733128:+|3p</th>\n",
       "      <th>chrY:2722813-2734833:+|3p</th>\n",
       "      <th>chrY:2722813-2789783:+|3p</th>\n",
       "      <th>chrY:2722813-2796904:+|3p</th>\n",
       "      <th>chrY:2733287-2734833:+|3p</th>\n",
       "      <th>chrY:2733362-2734833:+|3p</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 61766 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr10:101157444-101162336:-|5p  chr10:101162480-101163225:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                  1                               1   \n",
       "CVN_02                                  1                               1   \n",
       "CVN_03                                  1                               1   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               1   \n",
       "\n",
       "           chr10:101163392-101163480:-|5p  chr10:101163392-101165512:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                  1                               0   \n",
       "CVN_02                                  1                               0   \n",
       "CVN_03                                  1                               0   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               0   \n",
       "\n",
       "           chr10:101163632-101165512:-|5p  chr10:101166007-101190204:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                  1                               0   \n",
       "CVN_02                                  1                               0   \n",
       "CVN_03                                  1                               0   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               0   \n",
       "\n",
       "           chr10:101437763-101476104:-|5p  chr10:101456299-101476104:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  0                               1   \n",
       "\n",
       "           chr10:101473237-101476104:-|5p  chr10:101474476-101476104:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  0                               1   \n",
       "\n",
       "                     ...              chrY:2713785-2734833:+|3p  \\\n",
       "sample_id            ...                                          \n",
       "CVN_01               ...                                      0   \n",
       "CVN_02               ...                                    NaN   \n",
       "CVN_03               ...                                    NaN   \n",
       "CVN_04               ...                                      0   \n",
       "CVN_05               ...                                      0   \n",
       "\n",
       "           chrY:2714508-2722640:+|3p  chrY:2721733-2722640:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                             0                          0   \n",
       "CVN_02                           NaN                        NaN   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                             0                          0   \n",
       "CVN_05                             0                          0   \n",
       "\n",
       "           chrY:2722746-2734833:+|3p  chrY:2722813-2733128:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                             0                          1   \n",
       "CVN_02                           NaN                        NaN   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                             0                          1   \n",
       "CVN_05                             0                          1   \n",
       "\n",
       "           chrY:2722813-2734833:+|3p  chrY:2722813-2789783:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                             0                          0   \n",
       "CVN_02                           NaN                        NaN   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                             0                          0   \n",
       "CVN_05                             0                          0   \n",
       "\n",
       "           chrY:2722813-2796904:+|3p  chrY:2733287-2734833:+|3p  \\\n",
       "sample_id                                                         \n",
       "CVN_01                             0                          1   \n",
       "CVN_02                           NaN                        NaN   \n",
       "CVN_03                           NaN                        NaN   \n",
       "CVN_04                             0                          1   \n",
       "CVN_05                             0                          1   \n",
       "\n",
       "           chrY:2733362-2734833:+|3p  \n",
       "sample_id                             \n",
       "CVN_01                             0  \n",
       "CVN_02                           NaN  \n",
       "CVN_03                           NaN  \n",
       "CVN_04                             0  \n",
       "CVN_05                             0  \n",
       "\n",
       "[5 rows x 61766 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi20_all = psi3_psi5.loc[:, (psi3_psi5.groupby(phenotype).count() >= 20).all()]\n",
    "psi20_all.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get Junctions whose psi score is equal to 1 in all celltypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10:101157444-101162336:-|5p</th>\n",
       "      <th>chr10:101162480-101163225:-|5p</th>\n",
       "      <th>chr10:101163392-101163480:-|5p</th>\n",
       "      <th>chr10:101163392-101165512:-|5p</th>\n",
       "      <th>chr10:101163632-101165512:-|5p</th>\n",
       "      <th>chr10:101166007-101190204:-|5p</th>\n",
       "      <th>chr10:101437763-101476104:-|5p</th>\n",
       "      <th>chr10:101456299-101476104:-|5p</th>\n",
       "      <th>chr10:101473237-101476104:-|5p</th>\n",
       "      <th>chr10:101474476-101476104:-|5p</th>\n",
       "      <th>...</th>\n",
       "      <th>chrY:2713785-2734833:+|3p</th>\n",
       "      <th>chrY:2714508-2722640:+|3p</th>\n",
       "      <th>chrY:2721733-2722640:+|3p</th>\n",
       "      <th>chrY:2722746-2734833:+|3p</th>\n",
       "      <th>chrY:2722813-2733128:+|3p</th>\n",
       "      <th>chrY:2722813-2734833:+|3p</th>\n",
       "      <th>chrY:2722813-2789783:+|3p</th>\n",
       "      <th>chrY:2722813-2796904:+|3p</th>\n",
       "      <th>chrY:2733287-2734833:+|3p</th>\n",
       "      <th>chrY:2733362-2734833:+|3p</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phenotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MN</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPC</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>iPSC</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows Ã— 61766 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr10:101157444-101162336:-|5p chr10:101162480-101163225:-|5p  \\\n",
       "phenotype                                                                 \n",
       "MN                                  True                           True   \n",
       "NPC                                 True                           True   \n",
       "iPSC                                True                           True   \n",
       "\n",
       "          chr10:101163392-101163480:-|5p chr10:101163392-101165512:-|5p  \\\n",
       "phenotype                                                                 \n",
       "MN                                 False                          False   \n",
       "NPC                                 True                          False   \n",
       "iPSC                               False                          False   \n",
       "\n",
       "          chr10:101163632-101165512:-|5p chr10:101166007-101190204:-|5p  \\\n",
       "phenotype                                                                 \n",
       "MN                                  True                          False   \n",
       "NPC                                 True                          False   \n",
       "iPSC                                True                          False   \n",
       "\n",
       "          chr10:101437763-101476104:-|5p chr10:101456299-101476104:-|5p  \\\n",
       "phenotype                                                                 \n",
       "MN                                 False                          False   \n",
       "NPC                                False                          False   \n",
       "iPSC                               False                          False   \n",
       "\n",
       "          chr10:101473237-101476104:-|5p chr10:101474476-101476104:-|5p  \\\n",
       "phenotype                                                                 \n",
       "MN                                 False                          False   \n",
       "NPC                                False                          False   \n",
       "iPSC                               False                          False   \n",
       "\n",
       "                     ...            chrY:2713785-2734833:+|3p  \\\n",
       "phenotype            ...                                        \n",
       "MN                   ...                                False   \n",
       "NPC                  ...                                False   \n",
       "iPSC                 ...                                False   \n",
       "\n",
       "          chrY:2714508-2722640:+|3p chrY:2721733-2722640:+|3p  \\\n",
       "phenotype                                                       \n",
       "MN                            False                     False   \n",
       "NPC                           False                     False   \n",
       "iPSC                          False                     False   \n",
       "\n",
       "          chrY:2722746-2734833:+|3p chrY:2722813-2733128:+|3p  \\\n",
       "phenotype                                                       \n",
       "MN                            False                      True   \n",
       "NPC                           False                      True   \n",
       "iPSC                          False                     False   \n",
       "\n",
       "          chrY:2722813-2734833:+|3p chrY:2722813-2789783:+|3p  \\\n",
       "phenotype                                                       \n",
       "MN                            False                     False   \n",
       "NPC                           False                     False   \n",
       "iPSC                          False                     False   \n",
       "\n",
       "          chrY:2722813-2796904:+|3p chrY:2733287-2734833:+|3p  \\\n",
       "phenotype                                                       \n",
       "MN                            False                     False   \n",
       "NPC                           False                      True   \n",
       "iPSC                          False                     False   \n",
       "\n",
       "          chrY:2733362-2734833:+|3p  \n",
       "phenotype                            \n",
       "MN                            False  \n",
       "NPC                           False  \n",
       "iPSC                          False  \n",
       "\n",
       "[3 rows x 61766 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi_is_1 = psi20_all.groupby(phenotype).apply(lambda x: (x == 1)[x.notnull()].all(axis=0))\n",
    "psi_is_1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr10:101157444-101162336:-|5p</th>\n",
       "      <th>chr10:101162480-101163225:-|5p</th>\n",
       "      <th>chr10:101163632-101165512:-|5p</th>\n",
       "      <th>chr10:101480826-101483712:-|5p</th>\n",
       "      <th>chr10:101483881-101486724:-|5p</th>\n",
       "      <th>chr10:101487321-101489309:-|5p</th>\n",
       "      <th>chr10:101489492-101491716:-|5p</th>\n",
       "      <th>chr10:102035251-102039880:-|5p</th>\n",
       "      <th>chr10:102107289-102107820:+|5p</th>\n",
       "      <th>chr10:102286312-102286731:-|5p</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX:77378872-77380370:+|3p</th>\n",
       "      <th>chrX:77380923-77381286:+|3p</th>\n",
       "      <th>chrX:80532669-80533829:+|3p</th>\n",
       "      <th>chrX:80533911-80552693:+|3p</th>\n",
       "      <th>chrX:99887566-99888401:-|3p</th>\n",
       "      <th>chrX:99890250-99890554:-|3p</th>\n",
       "      <th>chrY:1455653-1456171:-|3p</th>\n",
       "      <th>chrY:1456313-1458133:-|3p</th>\n",
       "      <th>chrY:14774638-14776570:+|3p</th>\n",
       "      <th>chrY:2710284-2712117:+|3p</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 9313 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr10:101157444-101162336:-|5p  chr10:101162480-101163225:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                  1                               1   \n",
       "CVN_02                                  1                               1   \n",
       "CVN_03                                  1                               1   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               1   \n",
       "\n",
       "           chr10:101163632-101165512:-|5p  chr10:101480826-101483712:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                  1                             NaN   \n",
       "CVN_02                                  1                             NaN   \n",
       "CVN_03                                  1                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               1   \n",
       "\n",
       "           chr10:101483881-101486724:-|5p  chr10:101487321-101489309:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                             NaN   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                             NaN   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                               1   \n",
       "\n",
       "           chr10:101489492-101491716:-|5p  chr10:102035251-102039880:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                               1   \n",
       "CVN_02                                NaN                             NaN   \n",
       "CVN_03                                NaN                               1   \n",
       "CVN_04                                NaN                             NaN   \n",
       "CVN_05                                  1                             NaN   \n",
       "\n",
       "           chr10:102107289-102107820:+|5p  chr10:102286312-102286731:-|5p  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                NaN                               1   \n",
       "CVN_02                                  1                               1   \n",
       "CVN_03                                  1                               1   \n",
       "CVN_04                                NaN                               1   \n",
       "CVN_05                                NaN                               1   \n",
       "\n",
       "                     ...              chrX:77378872-77380370:+|3p  \\\n",
       "sample_id            ...                                            \n",
       "CVN_01               ...                                        1   \n",
       "CVN_02               ...                                        1   \n",
       "CVN_03               ...                                      NaN   \n",
       "CVN_04               ...                                        1   \n",
       "CVN_05               ...                                        1   \n",
       "\n",
       "           chrX:77380923-77381286:+|3p  chrX:80532669-80533829:+|3p  \\\n",
       "sample_id                                                             \n",
       "CVN_01                               1                            1   \n",
       "CVN_02                               1                            1   \n",
       "CVN_03                             NaN                            1   \n",
       "CVN_04                               1                            1   \n",
       "CVN_05                               1                            1   \n",
       "\n",
       "           chrX:80533911-80552693:+|3p  chrX:99887566-99888401:-|3p  \\\n",
       "sample_id                                                             \n",
       "CVN_01                               1                            1   \n",
       "CVN_02                               1                            1   \n",
       "CVN_03                               1                          NaN   \n",
       "CVN_04                               1                            1   \n",
       "CVN_05                               1                            1   \n",
       "\n",
       "           chrX:99890250-99890554:-|3p  chrY:1455653-1456171:-|3p  \\\n",
       "sample_id                                                           \n",
       "CVN_01                               1                          1   \n",
       "CVN_02                               1                          1   \n",
       "CVN_03                             NaN                          1   \n",
       "CVN_04                               1                          1   \n",
       "CVN_05                               1                          1   \n",
       "\n",
       "           chrY:1456313-1458133:-|3p  chrY:14774638-14776570:+|3p  \\\n",
       "sample_id                                                           \n",
       "CVN_01                             1                            1   \n",
       "CVN_02                             1                          NaN   \n",
       "CVN_03                             1                            1   \n",
       "CVN_04                             1                            1   \n",
       "CVN_05                             1                            1   \n",
       "\n",
       "           chrY:2710284-2712117:+|3p  \n",
       "sample_id                             \n",
       "CVN_01                             1  \n",
       "CVN_02                           NaN  \n",
       "CVN_03                           NaN  \n",
       "CVN_04                             1  \n",
       "CVN_05                             1  \n",
       "\n",
       "[5 rows x 9313 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi20_is_1_all = psi20_all.loc[:, psi_is_1.all()]\n",
    "psi20_is_1_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9313, 43)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "      <th>exon_stop</th>\n",
       "      <th>exon_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr10:101157444-101162336:-|5p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>ENSG00000120053</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>GOT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>4892</td>\n",
       "      <td>101157443</td>\n",
       "      <td>101162337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:101162480-101163225:-|5p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162480</td>\n",
       "      <td>101163225</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>ENSG00000120053</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>GOT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>745</td>\n",
       "      <td>101162479</td>\n",
       "      <td>101163226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:101163632-101165512:-|5p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101163632</td>\n",
       "      <td>101165512</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>ENSG00000120053</td>\n",
       "      <td>ENSG00000120053.9</td>\n",
       "      <td>GOT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1880</td>\n",
       "      <td>101163631</td>\n",
       "      <td>101165513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:101480826-101483712:-|5p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101480826</td>\n",
       "      <td>101483712</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000014919.8</td>\n",
       "      <td>ENSG00000014919</td>\n",
       "      <td>ENSG00000014919.8</td>\n",
       "      <td>COX15</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2886</td>\n",
       "      <td>101480825</td>\n",
       "      <td>101483713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:101483881-101486724:-|5p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101483881</td>\n",
       "      <td>101486724</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000014919.8</td>\n",
       "      <td>ENSG00000014919</td>\n",
       "      <td>ENSG00000014919.8</td>\n",
       "      <td>COX15</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2843</td>\n",
       "      <td>101483880</td>\n",
       "      <td>101486725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                chrom  intron_start  intron_stop strand  \\\n",
       "chr10:101157444-101162336:-|5p  chr10     101157444    101162336      -   \n",
       "chr10:101162480-101163225:-|5p  chr10     101162480    101163225      -   \n",
       "chr10:101163632-101165512:-|5p  chr10     101163632    101165512      -   \n",
       "chr10:101480826-101483712:-|5p  chr10     101480826    101483712      -   \n",
       "chr10:101483881-101486724:-|5p  chr10     101483881    101486724      -   \n",
       "\n",
       "                               intron_motif annotated         gencode_id  \\\n",
       "chr10:101157444-101162336:-|5p        GT/AG      True  ENSG00000120053.9   \n",
       "chr10:101162480-101163225:-|5p        GT/AG      True  ENSG00000120053.9   \n",
       "chr10:101163632-101165512:-|5p        GT/AG      True  ENSG00000120053.9   \n",
       "chr10:101480826-101483712:-|5p        GT/AG      True  ENSG00000014919.8   \n",
       "chr10:101483881-101486724:-|5p        GT/AG      True  ENSG00000014919.8   \n",
       "\n",
       "                                     ensembl_id            gene_id gene_name  \\\n",
       "chr10:101157444-101162336:-|5p  ENSG00000120053  ENSG00000120053.9      GOT1   \n",
       "chr10:101162480-101163225:-|5p  ENSG00000120053  ENSG00000120053.9      GOT1   \n",
       "chr10:101163632-101165512:-|5p  ENSG00000120053  ENSG00000120053.9      GOT1   \n",
       "chr10:101480826-101483712:-|5p  ENSG00000014919  ENSG00000014919.8     COX15   \n",
       "chr10:101483881-101486724:-|5p  ENSG00000014919  ENSG00000014919.8     COX15   \n",
       "\n",
       "                                  ...     gerstberger2014_rbp_target_trna  \\\n",
       "chr10:101157444-101162336:-|5p    ...                               False   \n",
       "chr10:101162480-101163225:-|5p    ...                               False   \n",
       "chr10:101163632-101165512:-|5p    ...                               False   \n",
       "chr10:101480826-101483712:-|5p    ...                               False   \n",
       "chr10:101483881-101486724:-|5p    ...                               False   \n",
       "\n",
       "                               gerstberger2014_tf    rbp ribosomal  \\\n",
       "chr10:101157444-101162336:-|5p              False  False     False   \n",
       "chr10:101162480-101163225:-|5p              False  False     False   \n",
       "chr10:101163632-101165512:-|5p              False  False     False   \n",
       "chr10:101480826-101483712:-|5p              False  False     False   \n",
       "chr10:101483881-101486724:-|5p              False  False     False   \n",
       "\n",
       "                               ribosomal_subunit synapse transcription_factor  \\\n",
       "chr10:101157444-101162336:-|5p             False   False                False   \n",
       "chr10:101162480-101163225:-|5p             False   False                False   \n",
       "chr10:101163632-101165512:-|5p             False   False                False   \n",
       "chr10:101480826-101483712:-|5p             False   False                False   \n",
       "chr10:101483881-101486724:-|5p             False   False                False   \n",
       "\n",
       "                               intron_length  exon_stop exon_start  \n",
       "chr10:101157444-101162336:-|5p          4892  101157443  101162337  \n",
       "chr10:101162480-101163225:-|5p           745  101162479  101163226  \n",
       "chr10:101163632-101165512:-|5p          1880  101163631  101165513  \n",
       "chr10:101480826-101483712:-|5p          2886  101480825  101483713  \n",
       "chr10:101483881-101486724:-|5p          2843  101483880  101486725  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj20_is_1_all = sj_metadata.loc[psi20_is_1_all.columns, :]\n",
    "print sj20_is_1_all.shape\n",
    "sj20_is_1_all.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100010822</td>\n",
       "      <td>100010933</td>\n",
       "      <td>exon:chr10:100010822-100010933:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100144704</td>\n",
       "      <td>100144824</td>\n",
       "      <td>exon:chr10:100144704-100144824:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100146958</td>\n",
       "      <td>100147064</td>\n",
       "      <td>exon:chr10:100146958-100147064:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100155148</td>\n",
       "      <td>100155209</td>\n",
       "      <td>exon:chr10:100155148-100155209:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100157102</td>\n",
       "      <td>100157255</td>\n",
       "      <td>exon:chr10:100157102-100157255:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom      start       stop                              name  score strand\n",
       "0  chr10  100010822  100010933  exon:chr10:100010822-100010933:-   1000      -\n",
       "1  chr10  100144704  100144824  exon:chr10:100144704-100144824:-   1000      -\n",
       "2  chr10  100146958  100147064  exon:chr10:100146958-100147064:-   1000      -\n",
       "3  chr10  100155148  100155209  exon:chr10:100155148-100155209:-   1000      -\n",
       "4  chr10  100157102  100157255  exon:chr10:100157102-100157255:-   1000      -"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "records = constitutive_regions.map(\n",
    "    lambda x: pd.Series(dict(chrom=x.chrom, start=x._start, stop=x._stop, strand=x.strand, score=1000, name=x.name)))\n",
    "constitutive_bed = pd.DataFrame.from_records(records)\n",
    "constitutive_bed = constitutive_bed.reindex(columns=['chrom', 'start', 'stop', 'name', 'score', 'strand'])\n",
    "constitutive_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "constitutive_exon_start.merge?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "constitutive_exon_stop (3501, 47)\n",
      "constitutive_exon_start (3420, 47)\n"
     ]
    }
   ],
   "source": [
    "constitutive_exon_stop = constitutive_bed.merge(sj20_is_1_all, \n",
    "                                                right_on=['chrom', 'exon_stop', 'strand'], \n",
    "                                                left_on=['chrom', 'stop', 'strand'])\n",
    "print 'constitutive_exon_stop', constitutive_exon_stop.shape\n",
    "constitutive_exon_start = constitutive_bed.merge(sj20_is_1_all, \n",
    "                                                right_on=['chrom', 'exon_start', 'strand'], \n",
    "                                                left_on=['chrom', 'start', 'strand'])\n",
    "print 'constitutive_exon_start', constitutive_exon_start.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "      <th>exon_stop</th>\n",
       "      <th>exon_start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>4892</td>\n",
       "      <td>101157443</td>\n",
       "      <td>101162337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>4892</td>\n",
       "      <td>101157443</td>\n",
       "      <td>101162337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101483713</td>\n",
       "      <td>101483880</td>\n",
       "      <td>exon:chr10:101483713-101483880:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101480826</td>\n",
       "      <td>101483712</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2886</td>\n",
       "      <td>101480825</td>\n",
       "      <td>101483713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101483713</td>\n",
       "      <td>101483880</td>\n",
       "      <td>exon:chr10:101483713-101483880:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101480826</td>\n",
       "      <td>101483712</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2886</td>\n",
       "      <td>101480825</td>\n",
       "      <td>101483713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101486725</td>\n",
       "      <td>101486911</td>\n",
       "      <td>exon:chr10:101486725-101486911:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101483881</td>\n",
       "      <td>101486724</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2843</td>\n",
       "      <td>101483880</td>\n",
       "      <td>101486725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom      start       stop                              name  score  \\\n",
       "0  chr10  101162337  101162479  exon:chr10:101162337-101162479:-   1000   \n",
       "1  chr10  101162337  101162479  exon:chr10:101162337-101162479:-   1000   \n",
       "2  chr10  101483713  101483880  exon:chr10:101483713-101483880:-   1000   \n",
       "3  chr10  101483713  101483880  exon:chr10:101483713-101483880:-   1000   \n",
       "4  chr10  101486725  101486911  exon:chr10:101486725-101486911:-   1000   \n",
       "\n",
       "  strand  intron_start  intron_stop intron_motif annotated    ...      \\\n",
       "0      -     101157444    101162336        GT/AG      True    ...       \n",
       "1      -     101157444    101162336        GT/AG      True    ...       \n",
       "2      -     101480826    101483712        GT/AG      True    ...       \n",
       "3      -     101480826    101483712        GT/AG      True    ...       \n",
       "4      -     101483881    101486724        GT/AG      True    ...       \n",
       "\n",
       "  gerstberger2014_rbp_target_trna gerstberger2014_tf    rbp ribosomal  \\\n",
       "0                           False              False  False     False   \n",
       "1                           False              False  False     False   \n",
       "2                           False              False  False     False   \n",
       "3                           False              False  False     False   \n",
       "4                           False              False  False     False   \n",
       "\n",
       "  ribosomal_subunit synapse transcription_factor intron_length  exon_stop  \\\n",
       "0             False   False                False          4892  101157443   \n",
       "1             False   False                False          4892  101157443   \n",
       "2             False   False                False          2886  101480825   \n",
       "3             False   False                False          2886  101480825   \n",
       "4             False   False                False          2843  101483880   \n",
       "\n",
       "  exon_start  \n",
       "0  101162337  \n",
       "1  101162337  \n",
       "2  101483713  \n",
       "3  101483713  \n",
       "4  101486725  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "constitutive_exon_start.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(5229, 93)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom_x</th>\n",
       "      <th>start_x</th>\n",
       "      <th>stop_x</th>\n",
       "      <th>name</th>\n",
       "      <th>score_x</th>\n",
       "      <th>strand_x</th>\n",
       "      <th>intron_start_x</th>\n",
       "      <th>intron_stop_x</th>\n",
       "      <th>intron_motif_x</th>\n",
       "      <th>annotated_x</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_trna_y</th>\n",
       "      <th>gerstberger2014_tf_y</th>\n",
       "      <th>rbp_y</th>\n",
       "      <th>ribosomal_y</th>\n",
       "      <th>ribosomal_subunit_y</th>\n",
       "      <th>synapse_y</th>\n",
       "      <th>transcription_factor_y</th>\n",
       "      <th>intron_length_y</th>\n",
       "      <th>exon_stop_y</th>\n",
       "      <th>exon_start_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>745</td>\n",
       "      <td>101162479</td>\n",
       "      <td>101163226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>745</td>\n",
       "      <td>101162479</td>\n",
       "      <td>101163226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>745</td>\n",
       "      <td>101162479</td>\n",
       "      <td>101163226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101157444</td>\n",
       "      <td>101162336</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>745</td>\n",
       "      <td>101162479</td>\n",
       "      <td>101163226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101483713</td>\n",
       "      <td>101483880</td>\n",
       "      <td>exon:chr10:101483713-101483880:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "      <td>101480826</td>\n",
       "      <td>101483712</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2843</td>\n",
       "      <td>101483880</td>\n",
       "      <td>101486725</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom_x    start_x     stop_x                              name  score_x  \\\n",
       "0   chr10  101162337  101162479  exon:chr10:101162337-101162479:-     1000   \n",
       "1   chr10  101162337  101162479  exon:chr10:101162337-101162479:-     1000   \n",
       "2   chr10  101162337  101162479  exon:chr10:101162337-101162479:-     1000   \n",
       "3   chr10  101162337  101162479  exon:chr10:101162337-101162479:-     1000   \n",
       "4   chr10  101483713  101483880  exon:chr10:101483713-101483880:-     1000   \n",
       "\n",
       "  strand_x  intron_start_x  intron_stop_x intron_motif_x annotated_x  \\\n",
       "0        -       101157444      101162336          GT/AG        True   \n",
       "1        -       101157444      101162336          GT/AG        True   \n",
       "2        -       101157444      101162336          GT/AG        True   \n",
       "3        -       101157444      101162336          GT/AG        True   \n",
       "4        -       101480826      101483712          GT/AG        True   \n",
       "\n",
       "      ...      gerstberger2014_rbp_target_trna_y gerstberger2014_tf_y  rbp_y  \\\n",
       "0     ...                                  False                False  False   \n",
       "1     ...                                  False                False  False   \n",
       "2     ...                                  False                False  False   \n",
       "3     ...                                  False                False  False   \n",
       "4     ...                                  False                False  False   \n",
       "\n",
       "  ribosomal_y ribosomal_subunit_y synapse_y transcription_factor_y  \\\n",
       "0       False               False     False                  False   \n",
       "1       False               False     False                  False   \n",
       "2       False               False     False                  False   \n",
       "3       False               False     False                  False   \n",
       "4       False               False     False                  False   \n",
       "\n",
       "  intron_length_y exon_stop_y exon_start_y  \n",
       "0             745   101162479    101163226  \n",
       "1             745   101162479    101163226  \n",
       "2             745   101162479    101163226  \n",
       "3             745   101162479    101163226  \n",
       "4            2843   101483880    101486725  \n",
       "\n",
       "[5 rows x 93 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "constitutive_exons = constitutive_exon_start.merge(constitutive_exon_stop, left_on='name', right_on='name')\n",
    "print constitutive_exons.shape\n",
    "constitutive_exons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1449, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101162337</td>\n",
       "      <td>101162479</td>\n",
       "      <td>exon:chr10:101162337-101162479:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101483713</td>\n",
       "      <td>101483880</td>\n",
       "      <td>exon:chr10:101483713-101483880:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>101489310</td>\n",
       "      <td>101489491</td>\n",
       "      <td>exon:chr10:101489310-101489491:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr10</td>\n",
       "      <td>104183414</td>\n",
       "      <td>104183474</td>\n",
       "      <td>exon:chr10:104183414-104183474:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr10</td>\n",
       "      <td>104245366</td>\n",
       "      <td>104245490</td>\n",
       "      <td>exon:chr10:104245366-104245490:-</td>\n",
       "      <td>1000</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom      start       stop                              name  score  \\\n",
       "0   chr10  101162337  101162479  exon:chr10:101162337-101162479:-   1000   \n",
       "4   chr10  101483713  101483880  exon:chr10:101483713-101483880:-   1000   \n",
       "8   chr10  101489310  101489491  exon:chr10:101489310-101489491:-   1000   \n",
       "12  chr10  104183414  104183474  exon:chr10:104183414-104183474:-   1000   \n",
       "16  chr10  104245366  104245490  exon:chr10:104245366-104245490:-   1000   \n",
       "\n",
       "   strand  \n",
       "0       -  \n",
       "4       -  \n",
       "8       -  \n",
       "12      -  \n",
       "16      -  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed = constitutive_exons[['chrom_x', 'start_x', 'stop_x', 'name', 'score_x', 'strand_x']]\n",
    "bed = bed.drop_duplicates()\n",
    "print bed.shape\n",
    "bed.columns = bed.columns.map(lambda x: x.split('_')[0])\n",
    "bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "bed.to_csv('{}/constitutive_exons.bed'.format(folder), index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Submit compute job to calculate exon conservation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "folder = '/projects/ps-yeolab/obotvinnik/singlecell_pnms'\n",
    "\n",
    "alt_exons_bedfile = '{}/skipped_exon_exon2.bed'.format(folder)\n",
    "constitutive_bedfile = '{}/constitutive_exons.bed'.format(folder)\n",
    "\n",
    "bedfiles = alt_exons_bedfile, constitutive_bedfile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1449"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import glob\n",
    "\n",
    "from gscripts.qtools import Submitter\n",
    "\n",
    "commands = []\n",
    "\n",
    "bw = '/projects/ps-yeolab/genomes/hg19/hg19_phastcons_placental_mammal.bw'\n",
    "\n",
    "for bedfile in bedfiles:\n",
    "    basename = os.path.basename(bedfile)\n",
    "    \n",
    "    prefix = basename.split('.bed')[0]\n",
    "    \n",
    "    prefix += '_phastcons_placental_mammal'\n",
    "    bedout = '{}/{}'.format(folder, prefix + '.bed')\n",
    "    outtab = '{}/{}'.format(folder, prefix + '.txt')\n",
    "    command = 'bigWigAverageOverBed {} {} {} -bedOut={}'.format(bw, bedfile, outtab, bedout)\n",
    "    print command\n",
    "    commands.append(command)\n",
    "\n",
    "jobname = 'exonbody_conservation'\n",
    "Submitter(commands, jobname, array=True, walltime='2:00:00', write_and_submit=True,\n",
    "          err_filename='{}/{}.err'.format(folder, jobname),\n",
    "          out_filename='{}/{}.out'.format(folder, jobname))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
