{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/matplotlib/__init__.py:810: UserWarning: Found matplotlib configuration in ~/.matplotlib/. To conform with the XDG base directory standard, this configuration location has been deprecated on Linux, and the new location is now /home/obotvinnik/.config/matplotlib/. Please move your configuration there to ensure that matplotlib will continue to find it in the future.\n",
      "  _get_xdg_config_dir())\n",
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/flotilla/__init__.py:55: DtypeWarning: Columns (19) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  load_species_data=load_species_data)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2016-05-13 10:39:25\tReading datapackage from /projects/ps-yeolab/obotvinnik/flotilla_projects/singlecell_pnm_figure1_post_kmeans_outliers/datapackage.json\n",
      "2016-05-13 10:39:25\tParsing datapackage to create a Study object\n",
      "2016-05-13 10:39:58\tInitializing Study\n",
      "2016-05-13 10:39:58\tInitializing Predictor configuration manager for Study\n",
      "2016-05-13 10:39:58\tPredictor ExtraTreesClassifier is of type <class 'sklearn.ensemble.forest.ExtraTreesClassifier'>\n",
      "2016-05-13 10:39:58\tAdded ExtraTreesClassifier to default predictors\n",
      "2016-05-13 10:39:58\tPredictor ExtraTreesRegressor is of type <class 'sklearn.ensemble.forest.ExtraTreesRegressor'>\n",
      "2016-05-13 10:39:58\tAdded ExtraTreesRegressor to default predictors\n",
      "2016-05-13 10:39:58\tPredictor GradientBoostingClassifier is of type <class 'sklearn.ensemble.gradient_boosting.GradientBoostingClassifier'>\n",
      "2016-05-13 10:39:58\tAdded GradientBoostingClassifier to default predictors\n",
      "2016-05-13 10:39:58\tPredictor GradientBoostingRegressor is of type <class 'sklearn.ensemble.gradient_boosting.GradientBoostingRegressor'>\n",
      "2016-05-13 10:39:58\tAdded GradientBoostingRegressor to default predictors\n",
      "2016-05-13 10:40:01\tBuilding Gene Ontology database...\n",
      "2016-05-13 10:40:08\t\tDone.\n",
      "2016-05-13 10:40:09\tLoading metadata\n",
      "2016-05-13 10:40:09\tLoading expression data\n",
      "2016-05-13 10:40:09\tInitializing expression\n",
      "2016-05-13 10:40:09\tDone initializing expression\n",
      "2016-05-13 10:40:10\tLoading splicing data\n",
      "2016-05-13 10:40:10\tInitializing splicing\n",
      "2016-05-13 10:40:11\tDone initializing splicing\n",
      "2016-05-13 10:40:11\tSuccessfully initialized a Study object!\n"
     ]
    }
   ],
   "source": [
    "import six\n",
    "\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib_venn\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "sns.set(style='ticks', context='paper')\n",
    "\n",
    "%matplotlib inline\n",
    "import flotilla\n",
    "\n",
    "folder = 'figures'\n",
    "\n",
    "flotilla_dir = '/projects/ps-yeolab/obotvinnik/flotilla_projects/'\n",
    "study = flotilla.embark('singlecell_pnm_figure1_post_kmeans_outliers', flotilla_dir=flotilla_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study.splicing.minimum_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `figures': File exists\r\n"
     ]
    }
   ],
   "source": [
    "mkdir $folder"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Expression Quality Control\n",
    "### Number of detected genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "detected_genes.mean() 8.02368224299\n",
      "detected_genes.groupby(study.sample_id_to_phenotype).mean() phenotype\n",
      "MN      7.667781\n",
      "NPC     7.594408\n",
      "iPSC    8.925308\n",
      "dtype: float64\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAK8AAACGCAYAAACrMdqLAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJztnXd0VNXah58zfSa9AwkJNST0zkWq9K6AEBX0IoggKlc+\nRVAQlAtKEbkWUBHExgUvoGALKoqC0pEWAiQYEtJJL9Nnzvn+iARiEgZkMkl0nrVcy5zZc/Yv5Ddn\ndnn3+wqSJEm4cVMPkdW2ADdu/ixu87qpt7jN66be4javm3qL27xu6i1u87qpt9QJ86akpDB27FgA\nNm7cyIoVK5g7dy6pqam1rMxNXabWzZubm8v27dvR6XTk5eVx/Phx7HY7/v7+NGzYsLbluanD1Lp5\nAwMDeeqpp9DpdKSnpyMIAs899xyRkZHs2rXrlu5ls9lIS0vDZrPVkFo3dQlFbQu4iiRJBAUF4eXl\nBYC/vz95eXnVtv/kk0/45JNPKlyzWCwkJiby/fffExYWVqN63dQ+dca8giDQsGFDQkJCWLJkCXq9\nnsWLF1fbPiYmhpiYmArX0tLSGDhwYE1LdVNHEP5KsQ1Xzet+8v49qPUxrxs3fxa3ed3UW9zmdVNv\nqTMTtrpCfHw8mzdvRqVS8cADD9CsWbPaluSmGtzmvY5PP/2UyZMnY7FYkMlkrFy5ktjYWPr27Vvb\n0txUgXvY8DuSJDFr1iyMRiN2ux2r1YrBYOCxxx7j7NmzJCQk1LZEN3/Abd7fMZlM5OTkVLp+9uxZ\nevToQceOHenatesNN07cuBa3eX9Ho9EQHh5e6bokSej1eoxGI6dPn2bWrFm1oM5NVbjN+zuCIPD+\n++/j4eGBTqfDw8OjUhur1crXX39dC+rcVIXDCVtiYiIGgwEoC1e877776NmzZ40Lqw369etHUlIS\nu3btQi6X8/jjj2M0Giu08fX1rSV1biohOSAmJka6dOmSNH36dCklJUWaNGmSo7fUGqmpqVJkZKSU\nmprqlPs9//zzkk6nkwAJkHQ6nbRx40an3NvN7ePwySuTyfD09ESn0xEeHk5paWnNf6LqCC+++CJN\nmjThrbfeQqVS8dRTTzFu3LjaluXmdxwG5mzcuJEff/yR+fPn89VXXxESEsI///lPV+m7JdyBOX8v\nHE7Ypk2bxpNPPklycjLjx4/nvvvuc4UuN9eh1+uZMWMGvr6+NGzYkNWrV+PgmfO3wKF5X3rpJXbv\n3s2mTZtITEzk6aefdoUuN9cxadIkPvjgA4qKisjKymLRokWsXbu2tmVVQG8xYrCYXNqnQ/OeO3eO\nBQsWoNVqGTZsWJUL+W5qjsLCQmJjYzGbzeXXDAYDa9asqUVV17CJdnL1BSzb+xYv//QOhcZiREl0\nSd8OJ2xarZYDBw4giiKnT5/G09PTFbrc/I7dbq/yusVicbGSqrHarfxz23zO5yQBkJCTzMZ7luGp\n0tV43w7N+/LLL/POO++g0+n4/PPPWbZsWY2LcnMNDw8PgoODSUtLK7+m1WqZNm1arWmKiYkhKSkJ\nuVzOLwd+4be8y+WvJealIBOhW7du1b6/WbNmlc4f/hkcmtfT05M777yTO+64AygbRgQHB992x25u\njgkTJpCbm1vh2oABA1iwYEEtKaKC8fQWA5M6jub9Xz8D4MFOd3H42FGOHj1a4zocmnf69Ok0a9aM\nwMDA8mv9+vWrUVFuykhPT2fPnj2YTBUnQsXFxSiVylpSVREPlY6n+07lgU5jQIJffznGY/Mf5e23\n36ZPnz4IglBjfTs0r0Kh4IUXXqgxAXWZ48ePA9ClS5da6b+oqAi5XF7pel2LbNMqNYR6hTBgwABO\nnjxJaWkpI0aMYMyYMWzevLnGDOzQvF5eXixZsoTmzZuXX5s0aVKNiKlrrF+/HoB33nkHm2gr+yNI\nEnKZa2L4o6Ki8PX1Ra/Xl1/TarVMnjzZJf3fCjt27ODEiRPlWvV6PZ9//jkHDx4sH3I6G4dLZf37\n96ddu3bodLry//4OHD9+nF9//ZW8vDysNitJJQmczDtCmj4Fu1T1CoCzkclkxMbGEh4ejk6nQ61W\nM2bMmDq51n7w4MEKHzIAs9nMwoULWb9+fY2EFTg076BBg8jKyiI+Ph4fH58a+Qq9PtHe0qVLWb58\nOXPmzKkww3Y1V5+6o8eMJsuQTr45B6toIdOYhsGmd/Bu59GuXTuSk5M5efIkqampbN26tc6Md68n\nMDAQmayinWw2G3v37mXOnDm0bduWwsJCp/bp8Ptv3rx5jBs3jk2bNjF+/HjmzZvHli1bKrR56qmn\nKuUHkyQJQRB47bXXbnj/6xPtGY1G+vTpQ79+/fj22285cOAAEydO/BO/lvNQq9VIQsVFd1c9ea8i\nCAItW7Z0aZ+3Ql5eHqtWrUIUq96cMBgMZGdns27dOp577jmn9evQvKWlpQwaNIgPP/yQqKgoFIrK\nb3nooYdYsGABa9asQa1W35KAq4n2pk+fjlarpV+/fly+fJnY2NgbrilXl6vMWTzyyCPMnDmT2K9j\nGTd+LIXWAgy2UvxUAXgrvZ3Wz1+B7du3V7uZchWTycSxY8ec2q9D80ZERLBmzRoKCgrYsGFDldFa\nbdu25YknniAnJ4cePXr8KSFXA0327NnDoUOHWL58+Q0/CDWdq6xLly507twZAKVcRRu/jggISJKI\nTKi8AvB3xmQyVfvUvYpWp6V///5O7femcpXt2bOHS5cuERERweDBg2tk6WP69Ok8//zzxMTE0KdP\nHwCGDRvGgAEDbvoezg6JrO2lsuqw2WxkZGQQEhJyy990NUFqaiqtWrWqcOpEoVAgChKiJCHIBPwb\nBRL/6xmC/YKc1q9D865cuRK4NoZVKBQEBwdz1113lacjrSv8HeJ5d+zYwcMPP4zZbEahUPDqq6/y\n8MMP17YsYmNjmTZtWnng1tatW3ly/yqMSQUoArQ0aB/B0dnbUcqdN9l0uNqQmpqKp6cnvXr1wtfX\nl/j4eEpLS/m///s/p4lwc3OkpKTwwAMPUFhYiNFopKSkhNmzZ3Py5Mnalsbw4cNJT08nLy+PDh06\nMHT0cEb1H45Pn3A8WgcxsvWdmG1Wp/bpcMxbWFhYfty7d+/eHDhwgJkzZ3L//fdX2X7z5s2MGDEC\nPz8/pwp1A1988UWlIHSz2cyOHTvo2LFjLam6hiAIeHt7IwgCnioda0bN50DKCZRyJd1C26JROneI\n49C8/v7+vP7667Ru3Zr4+Hj8/Pw4dOgQKpWqUttz586xbds28vPzeeKJJ5wq1E3ZbucfV3uUSiXe\n3nVz9UOjUDOg+T9q7P4Ohw2vvvoqbdu2JTk5maioKF555RUCAwN58803K7XdtGkTq1ev5siRI5WC\nSeorkiRhE23YRJvLgqyrY+zYsahUqgqbASqVqta3i/UWAzbRjt5icLjq4EwcmlculzNgwAAefvhh\nhgwZglwup0WLFpWC0jMyMtDr9TRv3pyxY8eybdu2GhNdU4iSnVJrCWa7GbtoR5IkzKKZxOJ4zhed\nQW8rRXTxBsX1eHt7c/jwYUaNGkVISAiDBw/m559/rtWqSUaribcPf0KXN8YxZduzFJlLbhjL60yq\nHTYYDIZbimNISEhgxowZAIwePZp333339tW5ELtk50JhHMXWsi3M5t5R+KkCuFh0jlJbMQAJhXF0\nCqy5r8GboUWLFrdcJakmuVKax7pD/wXg14x4Xv35fSbcH+PgXc6h2ifv1Uj9GxU1uZ7+/fvTvn17\noGwcVl9yesXExDBx4kRKSospthaikqmRCTKyDRkICNikazNkm2TDbrPRr18/unXrVmmTpCYx2UyI\nkojJVreGY+LvE8h724/gu2nvMafXFLp3705CTjL7Lh2j1GwgR5+P0ep83dWu87788sscPXqU5ORk\nmjVrVj7LFQSB7du3O12IM7iddV5RsmO2m5HLFMiQUWwpRKPQYrDp+a34AhIijT2a0EAb6rKQyKtY\n7Bb+m7CVC4UJtPFvTUyLCU5dL70dik2lHEk7TTP/xqz8aQMKmYK5fafhodLy4vdrAYmZPe7DW+1B\nY1/nDm+q/Ss8++yzAGzZsuVvkatBAtL0yeSZc1DKlLT27UieKQetQke3oDso++y6Lpb3ek7knCIu\n/ywAJ3NP0S6gLR0C27tcR1Wo5SruiOjM6PdnklyYDkBGyRXeHPM8/x48m7WH/kuroKZ8+OtOpnYd\n79S+Hf4lWrZsyb333ovNZkOSJF544QXatWtXoc2AAQPw9vausIxzdUeurj6l/4jFbibPXLY7ZBWt\npOmTCdVFYBUtiJKITJAhStd+L1cQExND79696Xh3pwrXTSYTK1euvOGk2FmHHG+E3mIkz1BIqHcI\nGSVXyq+nFWXho/HkYt5lHuoyjnxDIf2aOn8S59C8q1evZt26dfj7+5Ofn8+MGTMq/aM9+uijHD58\nGJVKRe/evenXr1+VKULrMnKh4j+FQqYEAXRKD6x2K/FFJ9EpPGnpHY1CcM1X9lXzWewWEkoSuVCQ\nQGv/aDo37MTiHxa55JDjjVDJlfhqvNBbDMzocS9vHPgImSDj0R73UWQq5dDlUzT1D6Nvk27olFqn\n9+/QvFeLWEPZhkVVZ6omTJjAhAkTMBqN7N+/nxUrVmA2m+natSsTJkxwumhnI0oiAtDUK5JsQzoa\nhY7GHk2xihYsdgsauYZgTUPSDZdJ06fQ2KMpcpnrIstkgoyYFhPwVHlitJlQypV14hybTbRhtJlY\n8eMGpnYbz/g2g9Eo1KiVKi7lpTGm9QD2XzrG5aIMfDVeqBUqlHLnDbsc3mn06NFMnjyZqKgoLly4\nwMiRI6ttq9VqadasGSkpKRw6dIhDhw7VG/MmFp/DTx1AS5/WWERzWRCSTEGRuYBCSx4NdKGIkoiI\nHQnX5Qmz2C2sP7uR5JJkAGJaTKBTUO1vBQPk6gsI9gxg17nv2XXue1oGROCp0vHhxBWE+TZg8Map\nFJlKkAkyPpiwnM6NWrvWvA888AAjR44kLS2N0NBQAgICKrU5cOAAe/fu5cKFC0RGRjJw4ECmTp1a\n5VO6riIiklL6Gyn8hgwZnQN7km3MJE2fDMAVYxbt/LsgSnZsohWFiyZuRpup3LgAB7MO0zagrUv6\ndoSX2pPMkhx6hnfk4OWTJOalcHfrQRisJk5mnKPIVAKUPRy+vrCPtiEtnRrfcFN/AX9///KhQ1VM\nnTqV8PBwoqKiuHLlClu2bGHr1q0ADo8B1QVkgoxmXpEkFsVjk2w09miKXbJTZCkAoKEujFBdeHlb\no82IRuH8MVxV6JRaPJUelFrLzs019gxDLsgYPXo0NrHs6JVVLFuLVslULh3OqOQK8o1FrB+7hO1x\n36CSqxgZ1Y/LhZl0Cm2NSq7EYi/T1r+Z88e9TimcnZ5etkQiCEKlqKfQ0NDbvf1NU9U676jRo8jO\nyr7h+wRBYN/P+1AoFAiCQJ7pCr6qAPLMVyi05NPEswXnC09jFs34qvxp6dOaeyfeS3Jy8g3vG9Ig\nhC+/+PK2fiebaKPYUsKP6T/ir/bnHw16YLSbSClIITooCrPNzP8ubsdkNzGyyQjCPEJdugZstVsx\n2Swk5qXwzYX9fHF+L8Nb9mFOnynk6AvYdmY3d0R0onOj1uhULjbvzdSkOHjwIK+//jparZb58+cT\nGRnpVJE3S1Xm7datG+t2On76d2jQlVJrMRISGYbLtPCOBkBAINOYRrYxo7xta98OZJdkUWjKv+E9\nZ939L6etCFjtVmSCjMulqaw78zYSEt4qb57p9BSrTrxKkaUInULHom4LnDKkGTpiGPk5Nzcp9PPz\n45X/rMYvOACL3YpOruH11f/BarXSoVNHUlMus3v37kr1ParDPyiAb77e7bCdw9/y+eefZ/ny5bz0\n0kssXLiQ5557rpJ533jjDdauXYter2fZsmW8/fbbNyWyrqBWaJALcrxUPsgQkAkyFIICQZAhSRLK\nPyyNyQW5SwN0bKINi2jBKlo5nH2kfMJYbCnmcmkqD7aaxE8Z+4kvOOe0PvNz8rAOv7kjO1eAB7fO\nY3q/e/lnr/EcunySWc/8i1JjKc9/9ioXTcnQ3xO4uQyj+bE3l0bXKTUpFApF+bj46lO6LjHr7n/d\n8PXHn3icgPGBNNA2Itd0BaVMiUk0cr7wDCqZmtZ+HTGJRkqtpQRrGpJ/pZAHx011SfifKImY7Wa+\nufwd0X5RNPduxrErZWfrFIKCEG0wAgJ9GvViZMRw7KLdZZNJgCAvf+7tMZpmwRH0aNqBj099zr0d\nRqJVaChUqtg4dQXfxO3jpS+cnwzb4W85cOBA5syZw/z581m5cmV5cpD6hKNhQ2OfJlgkM1bRhk2y\noZXpyjPjGO0GEoriiPRpi4CA1W7lgiyONz91nNzZ0YfmZrCKVt4//yGXipP5regiszuUBfmnlabR\nJbgzGfpMQnTBrD3zNjPbPIK3ynWB6X46bz585FW+SfyFzxL2oFArmNBuONtO76aZfxhyQU7LwCZ0\niGhNn8ju7E844tT+qzXvxYsXgbKMkH379kUQhGor4WRnZ3PPPfcgSRKlpaXl/19ftofzjblE+kcj\nCDJCtI3Q20pQya6dFCm1liBDQJQkfss/Xz67dwUKQUGmPguAbGMOdtGOn8YPURI5mn2cjkEdUMvL\nlp9O5Z0mzDMUJa6ZsN3dZSj7U47z773rAPj6wj6+f/h92oS04P3jn9E1rB2dQ9tw8PJJujZt5zrz\nbty4ESgzsUwmIzo6msTERKDsnNr1LF68uMaSqbkCq92KJECJpRBPpReeCm88Pb3xUwditBoI0AQh\nASJ2ooPakVSQSKGpwCXabJKNXg3v4Pu0HwBQK9RY7BY6B3eiQ2B7jl45TrhnYwBa+0Uj4Jq4C4DW\njVpyMOtU+c8SEgm5yZzNTuS7iwf47uIBAnV+dGzYik+Sb2/VpSqqNe/LL78MlK3hbty4sXwZ7KGH\nHqrU9u2333aaebOzs3nzzTfLT2rMmzfPKfe9Ef7aQATAW+VLXMGvRPq0wS7ZKLEU4asOQCbI+DX3\nECJ2gjQhNPFtzsks52Z/qQ61XM3AsDvpGtwFSRKx2C1kGbL56PxmRElkeJNh5JhymNFmOqEejVAr\nXJPHQSbI6NykLWFBDfnfmd0YrSZCvUPoGd6RLqFtuJh3mdiEfZzLuUifiM58deoHp2twOOY1GAwc\nP36cyMhI4uLiqkypVFBQwE8//VTl+281EXVSUhKHDh2ibdu2REVF3dJ7qyKkQYjDsefChQsZOjIY\nhUyJ6vev4LiCEwBkGtOI8mlPgCaQHFM2OaZsIjybs3TWCjIyMm50W0IahNzw9RGjRpCTfeOZtSAI\nLHt5GX379cUsWsgz5tExsAPdQ7ohSiIqmRKr3UqQNgi73c6ECRMcrj8DBIUE8fWXf76OctOgxhSb\nSkkuyODwrE9IyE0mKqg5aUWZNPAKYtHAx9iffIx724/kld3vUmp2/kTeoXlXrVrFBx98wDvvvEPj\nxo2rrEKj1+s5c+ZMle+/VfM2aNCATZs2ERYWxrRp05gyZcptZYW52U2CDP1l/NSB+KsCKbZUzGZY\nbC1EKy+LkpMLcmSCzClHcXKycxj+UvWxIgCTO02ikXdDkoov8V3qHnzVvoxpOpqLhYnIBQWbE7fQ\nUNeADgHtifaPZv5rz7Lr7OcO+4597qvb0l5sLCHAw5fOqmgu5FwiOrg5aw9+zFuHtxLm04DYh97l\nwMytnE47T+zpH2+rr+q4qWpArVq1omnTpkBZ6qc/JpcODQ3l8ccfd4qgzZs3M358WdCyp6cnNput\nSvM6O9GerzqApJIEWvm0RZTspOiTECU7AgKBmmCKLQX4qvwJ84goizRzwfawTJAR7huOiMgH5z/C\nJtm4XJqKXCanT8PeFFuKsYk2MvVZ3B95L15KL0Qv1xzG7Na0A5cLM5nw339hsVtp7NOAnQ+s5dOz\n32G2WUgrzMJsNxMR0KjGNDg076OPPsqQIUMq1KT4I8OGDXOaoHHjxvHaa68RGhpKhw4dqo0Ldnai\nPbVMQxPPFmQZ0gnWNqRTQHcKzfl4KX2QC3I8lT54qXwpMOXS0KPxn+rjVtHINcgFGQarHpt0LYVs\nqaUUP7UvQdpAJkfeT4R3BBmlGfj6+WK1u2YlpFuzDmw9/VV57EJqURaHUk8xLLIPM3vcy68Z8UT4\nNsLPw4ddT77Lv3e+zrHkqr+d/ywOzevt7c306dNv2MaZaf5bt25dKzt0cpkcD5knOoUHoiRyqSQR\nuUzGFVMWwdqG+Kr8Mdj0NPRo7LIZvSAI2CU7KrmKHiHdOZx9BI1cw5DwweSb8/FW+tAhsD0mu4kW\nPs0RJZHUQtck5M4szCY6+FqpB5kgo21IS0I8Atge9y1j2wxi49HtfHF+L91C27Hq3ueY8u7TpOSl\nO02DQ/MqlUpmzJhBs2bNyo+/PPPMM04TUNe4WndCECDbmImvyh8kEAAvpTcywWGqC6ehUqiQkCgx\nlzC6yUhGNhmOQlCQVpJOsGcIn1/6kuNXfiXUsxEPRU9BLVORmHfRJdr2xP/Cx4/8B6vdxsnMc4xr\nMxgvlQfeAZ54qHUkF6Sz6finAMQm7KN9g0hGdRzI2u8/dJoGh+adOnVqhZ9vdH7LZrMRHx9fodSo\nqxJQOBO5TEETz5b4qQLwVHojISEXFNglu9PNe6OJU2RkJCXt7sfPww+T3YRWrkWURAK1AeQYczh6\npWy5LrU0jX3p+xgQdicNc0NYu/ANp2qsUltIU46mnsZoNTGkZW9OZ16gTUhLiowlnM1KxF/nU6G9\n0WbGV+Hco2EOzavRaIiNjUUURSRJIjc3t1pDPvHEE/j4+JQn2RMEoV6aF8qGEb5qf+ySSFLxeQot\n+eXhkHInJpe+0WqDgIBcKeNsXjxtA9ogl8k5k3eGInMxzX2aV2grE2RIQJfeXRyuYMDtrzYUGooJ\n0Pnyyv73APh6ynrUchUzPlvMB/csx1vryZCWvfg28RfahUTyz85jmfXBwtvq8484NO8LL7zAk08+\nyY4dO+jatesNC2fr9XreeustpwqsDeyiHbPdhFquIc9UFtMLUGjJJ9+UQ5C2gUt0SEh8fu5L7usQ\nQ5G1mFO5p2ju3YwmXk3wUHrQr1FfjmQfpbFXY+4M60+OIRcvpfNqQytvEN11QrYHr0EzWTX8GX74\n7RANvQLJKs3lrugBfJ90iC/P7WXJ4NmsGfksdknEarWRuPGAUzeuHZrXz8+Pvn378tVXX/HAAw8Q\nGxtbqc3VDQqdTsf69etp1apV+Wv1sVqmRTRxuuAYXQN7VRomCC4c8wK0CW6N3mZg9Yk12H9funu8\n/SxO554h0rcFQ8OHYBUt7E37kTvD7iSrOMtpfTsKifw1K57ggECmd5uAQq4kxDOAjo2i2RH3LcfS\n4xjx/iN4KLWsH7eETg1b4xcTxZVixzHCN/rQXI/Dv0RYWBg7d+7Ew8ODFStWVKq1BXDmzBnOnDlD\n27ZtsVgs5T9Xt3FR1ykw5xGoCcEqWgnQBBOgDkIuyAlQB+OvrnyGrybx0/pysfBieQUiCYm4vLNI\nArwb/x4LDj2P2W7mbP45fs74mTxDroM7Og+DxYjZZmXc5idIyEkmKT+VTo2iuafd0PIVGW+NJ+0a\ntCI2YT93dx7i1P4dPnlffPFFioqKGDFiBPv27WPKlCmV2lzdoHjttdf417+ubcVejY+ob/irg9DZ\njaTqk7CJNpp4NqeZdyunF1MJCglyOPZ8LPZRQvxDUF5SYhWtCAi09W/N/sxfKrWVJLiw7wKxaxyP\nZ4NCbr82xK4T3/HetFW8MWYhOqWGl358h7NZiWya8BKxD71LckEaPSM6cSwtjpTCdLqHO/fgqEPz\nHj58GKvVis1mY8OGDdx///2MGTOmQpudO3fy0UcfcenSJfbv319+jq1JkyZOFVsTxMTEkJSUVP6z\nIAisXLWSf9zRoyxnmbWQ0wVlwd9tPDox7aFJXLp0Cbj9rDQ3E1tQbCkhx5DDvM5PczrvDFG+rfBU\nepJjyEUtVzMgtD96qx5vlTf9wvqgvl/NpPtdU143IesSBy/+SrvQVvhoPGkREM7+5GPc9dFj9G3a\nlXV3vcCZrPMEevjxcLcJpOc7b0gDN3GGbeLEibz11lvMnTuXtWvXMn36dD7++OMq237zzTcMHTrU\nqQJvBWcXVLGKFhKL4jHZTTTShROkCXZ5rrI8Yx4IAnbRjk2y4q/2J6n4Ek29m6KWqzDbzKgUKoxW\nI2q52mmHL7t163ZTx4AUMjlLxz7N4HZ9KDIVs+HodlKLMpnefSKNvRvw7tFtKOVKpnUbT7G+lFH/\nmerwnsrYnJs6+3dTx4Dy8vIIDg5GkiSKi4srtVm8eDEvvvgi69evr5SXtz4Eo1eHQlCWnaAQBJBw\n6bHyq3gqPTmcfYTPf4+HDfcMZ0r0gxRbivFSev4esPM96fp0JraYQOegTihcqLNjeGvuaNUFs83M\nvNjVLBk8G5kg4KHS8eC2eZzKPA9AXHYCywY/6dS+HZp38uTJvPnmm8ydO5f33nuPRx99tFKbF198\nEYB169YREnLjMMD6hCAIKK7mMHNdjHcFRCSO55wo/9loN6CRayixFLM18X/4qn2Y3mYqb51Zz9n8\ns7QLaOsU8/oHBdzUQci5W6ex57dDtA5uxsrhc1HJFTy660U23fMSZ7ISytudzDiHp1J3UysJ/kE3\nNyl2aF4vLy/GjRtHUlJSpeyQf2TNmjXk5uYSHR1Nv3796NSpU73KmlMXkQkCzbybkq5Pp31AO+5p\nPo6jV46ikCmZ2OIe1sdv5OeMA9zR8B9o5FpUMucMG27m6LkoiVjtNkLtYXx78RcSZSkMatETo9VE\nRvEVRkX15/NzZUHo97QdiiAXnJoc0KF5ry53SZJUfgyourXb5cuXA2Vhk8uWLSM1NdXp9WZdSV2o\ngKmWqxkeMYwgbSDtA9uzIX4Tl0svA5Aa0oMREcM4lXuaHiHd8FB6uHRoY7XbSC3KZM3PH3A07Qx5\nhkJGR9/JqhHPMOfLl1k/7t9M6jgKlVxFy8AmaF1dyuqPcbpVHQO6yuzZsykoKCA6OpqZM2fStWvX\n21dYS9hdFVzLAAAJEElEQVRFO0HBgZw9G1/bUhAEiPRpiUqmLDcuwPmCCwwOH4S/xh8/jevr3uUZ\nCgnzbsDyYU8hSiIpBRks/PY/hHqHsOW+VxElkfYNolApauZAqEPzXn/YMi8v74ZZTwYNGsTx48e5\ncuVK+ddDba4+/Fnskp00fTJ6z1I63tGeUn0Jnh61U6rWZDOz69Ln+Kl96dXwDpp6N+FScTIAbfyj\nUcmU6OSuyZv2RwI9fEnITeGhbc9SaCphcsfRrLt7MYXGsrgHu2TnZMY5mviH4aXWOT1XmUPz6nS6\n8kgyPz8/HnzwwWrbjhkzhubNm3PkyBF++OEHUlJS6p15RUkk25BOprEsLrbYWki0R4da06OUKTid\nd4Z7mo/j2JXjTIl6kNN5cajlKtr4t0YtV2OymbDb7GgUGpdqM1hMvLp/E/nGIgA+PLGLWT3vR7BI\nWEUboz94lKT8VGSCjA8nrqBHWPsKNeRuF4d36tWrF3FxcXz55ZccOXLjc/f33XcfX375Ja1atWLj\nxo31rpwVlI3tTfaKlWvM9tqrwGMVrbT1b4NCUNDaL5rYlN3kGHPI0GcCkG/KZ/tvn7IzaRel1lKX\nFjoUBAFv9bVAIIVMjkah5vTJU+QbiujYMAqlTIEoiWw+8QV6683lKrtZHD55n332WaZMmcJjjz3G\n6dOnmTNnDps2baqy7ZYtW5wqrjaQy+Q01IWRZ87BLtnwUHjh71H9EaiaRqPQML75WOLyz3Io+zD9\nQvshSSIaNMiQsf7sBnJNZcEu+eYCpkZPcdkT2EfjxaKBj1Fi1pNRcoXHe5bt7BmDBd4+spUJ7YYx\nvFVfpn/6PB0bRaOWVy75ezs4NK/ZbKZPnz5A2SrD+vXrnSqgLqKWq+kc+A9K9MWo5Opa2Zy4ik20\nczznBDt+KzuVcDDrMIu6LWDunLm89vprFJivnXTOMea6XKu/zof/jH4OAQGFTMahy6eY/fVLAGw5\n+SXfTnuPdXctpm/Tbk6fuDkcNoSGhrJ69Wq+++47Vq9e7ZRt17qOTJAjF+RkpmZx/tx5l1X/+SMx\nMTEcOPQLFwuvHe2xiBaScpKQy+WcijvFHSHXMnb2bdCbvXv3urzAoUyQUWIuxWK3sT/lePl1uyQS\nl5VAm5CWKGvgQ+UwtkGSJPbs2UNKSgoREREMGjSo1v6YjnB2bENdQJREEgoT2Ri/CQkJX5Uv8zo/\nXR7DYLFbMNgMyAU5KrmqPG+ZKzFYjIiSyBV9ASabiXEfPYFVtOGn9Sb2oXf578kvmN59ouszo8fF\nxbF9+/YK59LqaqjjX9G8AGa7hWJLEZn6TFr5tUIhKGp1KHM9ZpsZELDabbxx8GP6NOlCU78w4rIT\n6RrWlq2nvmJgi55EBTVzet8Ox7yLFi1i3rx5N6xJURM8/fTTDBgwgBEjRri037qIWq4iSBtEkPb2\nY3Cdj4BNtLE7YR8bjm5jw9FtdGoUzcrhczGYjUzuNAZVDZUZcGjegIAAunXr5tT1OUe8//779a4I\n4d+J62OgY7/bTWLRZbJLrx3vOZFxjvjs39Bm2Fm/fn2FeGlwXnXOas07e/ZsBEEgJyeHu+66q7x4\ntiAINVrh54cffsDLy4uOHetGrTE3lbneeBa7lQCrLzHtR7A97hsuF2bSJrgFA1v8A2205k9nMLoZ\nqjXv9OnTXT5UAPjiiy/w8fEhKSkJpVJJr1698PHxqdTO2bnK3Pw5VHIlTf3DsIl2Yh8qywbpqdKh\nVjh3Tbcqqp2wPfjgg3z4ofOym9wqO3fuRKVS3dKY9686YXNTNdU+eUVRxGQyVaqrBmWZI2uau+++\nu8b7cFO/qda8Fy5c4JFHHik379XM6IIg1OoT2Y2bq1Rr3qioKLdJ3dRpql3/ioiIcKUON25umWrN\nu3TpUlfqcOPmlnFt4i03bpyI27xu6i1u87qpt7jN66be4javm3qL27xu6i2uTXlYw9jtZQmYs7Kc\nm0rTTe3ToEEDFIqKdv1LmfdqvQxn1oVzUzeoKtjK4TGg+oTJZCIuLo6goCCnJfibOXNmrRQ1dERd\n1QU1o+0v/+TVaDROz4+mUqnqZHhlXdUFrtPmnrC5qbe4zeum3uI2r5t6i9u8DnBl5plboa7qAtdp\n+0utNrj5e+F+8rqpt7jN66be4javm3qL27xu6i1/qR22W2XHjh3ltZI1Gg2lpaWMHz+e3r17s2TJ\nEmQyGUajkUGDBjF06FBOnjzJxx9/jEajoaCggNmzZ9OqVasa0/fZZ5+xbds2Pv74Y2QyGYsXL8Zi\nsWAwGCroHTBgAO+88w7JyclIkoRKpWLRokWVtlNrSuOqVav46aefUCqVpKenM2TIEHr27ElUVBRP\nP/00AA8//DAbNmxwat9/a/NmZmai1+tZvnw5AQEB5OTksHz5cpo3b47dbuff//43giDw2WefYbfb\nWbNmDZs2bUImk5Gfn8/Bgwdr1LwAgYGBvPLKKzzzzDPl1xYtWlRB79UkiFdTz/70009kZGQQHh5e\no9qu0r59e/bs2cPw4cP59NNP6dGjB4IgkJaWxo8//kj//v1rJKfz39q8UJY8e9myZajVagRBYNas\nWURERDBmzBhWrFiB2Wyme/fuFBQUEBISUm4Uf39/Ro4cWeP6hg0bxunTp9mzZ0/5taVLl6LRaMr1\nfv/997Rv37789eqKPNYUQ4YMYc+ePQwdOpTc3FxCQkLIzc1l/vz5LFy4kBYtWtRIv3978wqCwIIF\nCwgIuFbv9pdffsFqtTJ//nygrHDi4MGDyczMxGg0otVqyc/PZ+PGjcydO7fGNc6dO5eZM2dit9sJ\nCQlh4cKFFfSmp6dz4MABevYsS/G/c+dOGjVqRPfu3WtcG5QFRAUEBLB161buvPNOvvnmGwCUSiVL\nly5lwYIF2Gw2p/f7tzdvVXTo0IElS5bw1VdfIYoiXbp0QaFQMHv2bObMmYOfnx/FxcXMnj3bJXrk\ncjlLly5l3LhxNGjQoNLrffv25fjx48yZMwelUolWq2XUqFEu0XaV8ePHM3v2bHbv3l1uXigLZZwy\nZQpLlixxep/uHTY39Rb3UpmbeovbvG7qLW7zuqm3uM3rpt7iNq+beovbvG7qLW7zuqm3/D8Q08IN\nbv0jFQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ba6d40f52d0>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.set(context='paper', style='ticks')\n",
    "\n",
    "detected_genes = (study.expression.data_original > 1).sum(axis=1)\n",
    "detected_genes = detected_genes/1000\n",
    "\n",
    "six.print_('detected_genes.mean()', detected_genes.mean())\n",
    "six.print_('detected_genes.groupby(study.sample_id_to_phenotype).mean()', \n",
    "           detected_genes.groupby(study.sample_id_to_phenotype).mean())\n",
    "\n",
    "detected_genes = detected_genes.to_frame().join(study.metadata.data[['pooled', 'phenotype']])\n",
    "detected_genes = detected_genes.rename(columns={0: 'n_genes_detected'})\n",
    "\n",
    "detected_genes_pooled = detected_genes.ix[detected_genes.pooled]\n",
    "detected_genes_single = detected_genes.ix[~detected_genes.pooled]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2.5, 2))\n",
    "sns.boxplot(x='phenotype', y='n_genes_detected', order=study.phenotype_order, linewidth=1,\n",
    "               data=detected_genes_single, palette=study.phenotype_color_ordered)\n",
    "sns.stripplot(x='phenotype', y='n_genes_detected', order=study.phenotype_order, #size=2, \n",
    "              linewidth=1, edgecolor='white',\n",
    "               data=detected_genes_single, palette=study.phenotype_color_ordered, jitter=True)\n",
    "sns.stripplot(x='phenotype', y='n_genes_detected', order=study.phenotype_order,linewidth=1,\n",
    "              data=detected_genes_pooled, color='k', size=5, jitter=True)\n",
    "sns.despine()\n",
    "ax.set_xlabel('')\n",
    "\n",
    "ax.set_ylabel('Thousands of genes\\nwith TPM > 1')\n",
    "\n",
    "# Reduce number of ticks\n",
    "yloc = plt.MaxNLocator(5)\n",
    "ax.yaxis.set_major_locator(yloc)\n",
    "\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/detected_genes.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mapped reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Started job on</th>\n",
       "      <th>Started mapping on</th>\n",
       "      <th>Finished on</th>\n",
       "      <th>Mapping speed, Million of reads per hour</th>\n",
       "      <th>Number of input reads</th>\n",
       "      <th>Average input read length</th>\n",
       "      <th>UNIQUE READS</th>\n",
       "      <th>Uniquely mapped reads number</th>\n",
       "      <th>Uniquely mapped reads %</th>\n",
       "      <th>Average mapped length</th>\n",
       "      <th>...</th>\n",
       "      <th>% of reads mapped to too many loci</th>\n",
       "      <th>UNMAPPED READS</th>\n",
       "      <th>% of reads unmapped: too many mismatches</th>\n",
       "      <th>% of reads unmapped: too short</th>\n",
       "      <th>% of reads unmapped: other</th>\n",
       "      <th>% splices: Annotated (sjdb)</th>\n",
       "      <th>% splices: GT/AG</th>\n",
       "      <th>% splices: GC/AG</th>\n",
       "      <th>% splices: AT/AC</th>\n",
       "      <th>% splices: Non-canonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>Feb 21 15:54:22</td>\n",
       "      <td>Feb 21 15:55:44</td>\n",
       "      <td>Feb 21 15:57:19</td>\n",
       "      <td>459.82</td>\n",
       "      <td>12134131</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11128688</td>\n",
       "      <td>91.71</td>\n",
       "      <td>98.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>3.10</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.841858</td>\n",
       "      <td>99.126387</td>\n",
       "      <td>0.652647</td>\n",
       "      <td>0.134704</td>\n",
       "      <td>0.086261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>Feb 21 15:55:43</td>\n",
       "      <td>Feb 21 15:57:11</td>\n",
       "      <td>Feb 21 15:58:51</td>\n",
       "      <td>480.66</td>\n",
       "      <td>13351539</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12219037</td>\n",
       "      <td>91.52</td>\n",
       "      <td>98.13</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>3.96</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.614629</td>\n",
       "      <td>99.020137</td>\n",
       "      <td>0.801086</td>\n",
       "      <td>0.097722</td>\n",
       "      <td>0.081054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>Feb 21 16:51:58</td>\n",
       "      <td>Feb 21 16:53:51</td>\n",
       "      <td>Feb 21 16:55:31</td>\n",
       "      <td>484.15</td>\n",
       "      <td>13448587</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12210025</td>\n",
       "      <td>90.79</td>\n",
       "      <td>98.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4.99</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.596403</td>\n",
       "      <td>99.028733</td>\n",
       "      <td>0.796140</td>\n",
       "      <td>0.101854</td>\n",
       "      <td>0.073272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>Feb 21 16:45:53</td>\n",
       "      <td>Feb 21 16:47:23</td>\n",
       "      <td>Feb 21 16:48:59</td>\n",
       "      <td>474.71</td>\n",
       "      <td>12658833</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11805790</td>\n",
       "      <td>93.26</td>\n",
       "      <td>98.33</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2.89</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.831275</td>\n",
       "      <td>99.120483</td>\n",
       "      <td>0.711286</td>\n",
       "      <td>0.084708</td>\n",
       "      <td>0.083523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>Feb 21 16:50:26</td>\n",
       "      <td>Feb 21 16:52:19</td>\n",
       "      <td>Feb 21 16:54:01</td>\n",
       "      <td>486.68</td>\n",
       "      <td>13789345</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12894337</td>\n",
       "      <td>93.51</td>\n",
       "      <td>98.44</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2.49</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.605923</td>\n",
       "      <td>99.083942</td>\n",
       "      <td>0.761804</td>\n",
       "      <td>0.089186</td>\n",
       "      <td>0.065069</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Started job on Started mapping on      Finished on  \\\n",
       "CVN_01  Feb 21 15:54:22    Feb 21 15:55:44  Feb 21 15:57:19   \n",
       "CVN_02  Feb 21 15:55:43    Feb 21 15:57:11  Feb 21 15:58:51   \n",
       "CVN_03  Feb 21 16:51:58    Feb 21 16:53:51  Feb 21 16:55:31   \n",
       "CVN_04  Feb 21 16:45:53    Feb 21 16:47:23  Feb 21 16:48:59   \n",
       "CVN_05  Feb 21 16:50:26    Feb 21 16:52:19  Feb 21 16:54:01   \n",
       "\n",
       "        Mapping speed, Million of reads per hour  Number of input reads  \\\n",
       "CVN_01                                    459.82               12134131   \n",
       "CVN_02                                    480.66               13351539   \n",
       "CVN_03                                    484.15               13448587   \n",
       "CVN_04                                    474.71               12658833   \n",
       "CVN_05                                    486.68               13789345   \n",
       "\n",
       "        Average input read length  UNIQUE READS  Uniquely mapped reads number  \\\n",
       "CVN_01                         99           NaN                      11128688   \n",
       "CVN_02                         99           NaN                      12219037   \n",
       "CVN_03                         99           NaN                      12210025   \n",
       "CVN_04                         99           NaN                      11805790   \n",
       "CVN_05                         99           NaN                      12894337   \n",
       "\n",
       "        Uniquely mapped reads %  Average mapped length  \\\n",
       "CVN_01                    91.71                  98.27   \n",
       "CVN_02                    91.52                  98.13   \n",
       "CVN_03                    90.79                  98.07   \n",
       "CVN_04                    93.26                  98.33   \n",
       "CVN_05                    93.51                  98.44   \n",
       "\n",
       "                  ...             % of reads mapped to too many loci  \\\n",
       "CVN_01            ...                                           0.03   \n",
       "CVN_02            ...                                           0.04   \n",
       "CVN_03            ...                                           0.03   \n",
       "CVN_04            ...                                           0.04   \n",
       "CVN_05            ...                                           0.03   \n",
       "\n",
       "        UNMAPPED READS  % of reads unmapped: too many mismatches  \\\n",
       "CVN_01             NaN                                         0   \n",
       "CVN_02             NaN                                         0   \n",
       "CVN_03             NaN                                         0   \n",
       "CVN_04             NaN                                         0   \n",
       "CVN_05             NaN                                         0   \n",
       "\n",
       "        % of reads unmapped: too short  % of reads unmapped: other  \\\n",
       "CVN_01                            3.10                        0.02   \n",
       "CVN_02                            3.96                        0.02   \n",
       "CVN_03                            4.99                        0.02   \n",
       "CVN_04                            2.89                        0.02   \n",
       "CVN_05                            2.49                        0.02   \n",
       "\n",
       "        % splices: Annotated (sjdb)  % splices: GT/AG  % splices: GC/AG  \\\n",
       "CVN_01                    98.841858         99.126387          0.652647   \n",
       "CVN_02                    98.614629         99.020137          0.801086   \n",
       "CVN_03                    98.596403         99.028733          0.796140   \n",
       "CVN_04                    98.831275         99.120483          0.711286   \n",
       "CVN_05                    98.605923         99.083942          0.761804   \n",
       "\n",
       "        % splices: AT/AC  % splices: Non-canonical  \n",
       "CVN_01          0.134704                  0.086261  \n",
       "CVN_02          0.097722                  0.081054  \n",
       "CVN_03          0.101854                  0.073272  \n",
       "CVN_04          0.084708                  0.083523  \n",
       "CVN_05          0.089186                  0.065069  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "study.mapping_stats.data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "n_genes_detected    8.023682\n",
      "pooled              0.037383\n",
      "dtype: float64\n",
      "           n_genes_detected    pooled\n",
      "phenotype                            \n",
      "MN                 7.667781  0.041096\n",
      "NPC                7.594408  0.039474\n",
      "iPSC               8.925308  0.030769\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAK4AAACHCAYAAACPr2IQAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAHyxJREFUeJztnXlgFFW69n9V1Xs6a2dPCEgCJiIBcQAZvbKERQQ3mE9E\nHB3RQS4q3yCLCIqCyCJyQb0oKFxHBtwV9eJKVMRB/UDcgGFHlpCEELJ0Or1X1fdHQyQOsUNId9Kh\nfn+RorrO011Pv33qnPe8R1BVVUVDI8IQW1qAhkZT0IyrEZFoxtWISDTjakQkmnE1IpKINK7f76eo\nqAi/39/SUjRaiIg0bmlpKQUFBZSWlra0FI0WIiKNq6GhGVcjItGMqxGRaMbViEg042pEJJpxNSIS\nzbgaEYlmXI2IRDOuRkSiGVcjItGMqxGRaMbViEg042pEJLpQXvzw4cP87W9/Y926daxatYpjx47h\ncDh46KGH8Hq9LFy4kLi4OLKzsxkzZkwopWi0MUJm3PLyct566y0sFgter5etW7eyfPlytmzZwhtv\nvIHH4+H222+ne/fujBs3jltuuQVJkkIlR6ONEbKuQmJiIpMnT8ZisVBdXU1iYiIAKSkplJWVUV5e\nTlpaGgAxMTHU1NSESopGGySkXYXT2Gw2qqqqgEASeEpKCoqiUFJSQkpKCna7nZiYmLO+9vXXX+f1\n11+vd8zr9YZcs0brJizGFUWR3r17M3v2bOx2O7Nnz8btdjN//nzWrVvHoEGDEMWzB/9Ro0YxatSo\neseKioooKCgIh3SNVooQiZVsThv3s88+IzMz87yuJSsygiAAKiAgCtpASyQQlojbWlFUmQM1u6nw\nlGOSzFwS3x2DYGhpWRqN4IIOL05/LRWecgDcsovi2iPIitzCqiIHRVFw+70crDiKy+fG4w/fs0fQ\niLtv3z6cTicAq1atYvTo0fTp0yfkwsKBKNQffpMECUFoITERiEf2ccPqCRyoOEKsKZr1dywnPSY5\nLG0HjbiPPPIIsbGxLFu2jClTprBs2bJw6AoLRslEZlQHDKKRBGMi6ZZ2/2ZmjYbZfHgbByqOAFDt\nruHl79eFLeoGjbiiKGK1WrFYLGRlZeFwOMKhKyxIgkSaOZM0SyaqqiKJgY9DURVUVUEUpFMPbhpn\no0NcRr2/O9naI4nh+eIHNW5BQQGTJk1i+vTpPPnkk9x0003h0BU26j7oU/6UVZkqTwVV3pMkmlKw\n6mOQtCh8VtJjklk6fAZv7fiE3u3yGZ7bH12YjNvgcNj+/fvr/q2qar3Ik5OTE3plv0NzDoediaqq\nVHsr2V29ve5Yt4SemHWWZmujreGX/Th9bgySHhXQiSICAl7Zh8VgDlm7DUbcVatWAQEDi6JIXl4e\n+/btA2Dt2rUhE9SyqNh91fWO1PjsmnF/B52kI0ay4vF72XZsBxPfn4tX9vHIgAkMz+0XMvM2aNz5\n8+cDMHbsWFatWoUgCKiqyp133hkSIa0BQRBJNqVS6ixCQUEn6Egw2lpaVkQgqwqPFf43Ve5Azsnj\nnz/HTV0Ghay9oH1cp9PJtm3b6Ny5Mzt27GjzeQJ6yUCPxCtw+GqI1scgaDNpjUJVVUy6XydvDJKe\nwGxkaAh6VxYtWsTHH3/M5MmTKSwsZMmSJSET0xqQBAmdqCfOmIAk6rQp4EZilPQsunYalyRnc1F8\nJkuHz8CvKCFrL2jEFQSBDh060K5dO1RVZc2aNUyePDlkgjQiE52kIychi9dGL0EQBHSCiEEXuunz\noOHkgQceQJIkNm/eTGlpKQcPHgyZGI3Iodbrwu334PF7cfncQMC8UQYzFr0ppKaFRhg3Ojqa0aNH\nk5CQwPTp0+vyajUuXGq9Ltb88D7dnr6BXsv+D1uObscr+8KqIahxY2Nj+eKLLxAEgbVr11JZWRkO\nXRqtGJ0oseSff8evyDi8ThZ8+QI+ObzbGgQ17oIFC8jJyeHBBx/E5/OxePHicOjSaMUIgoBZb6r7\nO9YUjaqG7kHsbAR9OKuoqGDFihXY7Xb69u3b5OGwtWvXsmPHDnw+H99//z19+vRBlgNJ3KNHjyY/\nP79J19UIP7Ki8PyNjzHvi+VYjRYWXjMFnaij0lXN7rJf6J6ei17UoZNCl+4dNOLOmDGDMWPGUFVV\nxZVXXsncuXOb1NCYMWOYP38+aWlpLFu2jL1792KxWIiKiqJTp05NuqZGy2DWG7k8owuv3LKYFTfN\nISM2hSPVJfzx+dHc9sZUClb+JeR93qBfCY/HQ15eHgCpqamYTKYgr2iYgwcP4vf7ycvLY9q0afTs\n2ZONGzeyevVq7rnnnrO+Rlss2TrRSzr0pyKqy+fh1R/X15n1uOMk3xz5kYKc0OVtBzVu7969mTJl\nCqWlpcyePZtu3bo1ubFXXnmFsWPH4nA42LdvHz179iQuLg6fr+Fvp7ZYsnXj9nvRixI9Mrqw+of3\nABAFkbzk7JC2G9S4gwcPBgLRMisri0svvbTJjR04cID09PS66z3xxBM4HA6mTp3a5Gs2ldNJcVq+\nbdNwet38UnmUd3ZsoCCnDwXZfVg0dCrfHvmJm/OHEmeODmn7QVf53nrrrbzyyishFXGunG9ao6zK\nVLjL8Skeks3pp5bsaAY+F45VH6f/i7cjnxpNeHXUYi7LuCSQzqg3hfzzbFSSzY033khWVladmKef\nfjqkokKJrPg5WnuYUlcRAGXuUvITLkdASxY/F7Yd21lnWoDPDn5D9/Q8okKYg3smQY3bltaYQSBf\nqdr76ySKW3Yhn1qmo9F4+nbsSYzRit3jQC/qGNllMAadPmztBzVuRkZGsFMiCgEBmzGRImctAFE6\nq5YB1gRMOiNfjvsH/zy8jR7pXYg1WcPafoPGdTqdWCxtL/NfEiXSotoRbYjFK3tJMCUiXtjlJZqE\nUWfAqDNw7cV9W6T9Bu/YXXfdBcCjjz4aNjHhQhIkYg3xJJlTtAezCKXBiJufn8+IESM4dOgQO3fu\nrDd89NZbb4VNoIbG2WjQuA899BAAr776KqNHjw6bIA2NxhD04axTp07ccsst+P1+VFXlscceo2vX\nruHQpqHRIEGNu3jxYp577jkSEhKoqKjgnnvu4c033wyHNg2NBgn6OC3LMgkJCQAkJCRo+zRotAqC\nRtzrrruO2267jdzcXPbs2cOwYcPCoSusKKqMw1eDIIhYdFFayaUIIKhx//znPzNs2DCKiorIyMjA\nZmtbBTJkxc/Bmn2c9JQBkGrOoF1Uh7oCeBqtk0bdnYSEhLruQltDFKQ60wKUu4+TZe3Ygoo0GsMF\nP2WkoBCl+zUFz6qPRQnz+imNc+eCrkgOICKSF5dPsfMooiCSZs4MW43XtozdbgfAYDCc16qZhrig\nK5JDYCZQJ+rIjGpPuqWdZtpmwG63U1FRwZYtW9iwYUNIioGHrSJ5cXExEyZMIC8vj+TkZGJiYurt\n7RsfH9+k6zYVRVVQVAUBtBphzYjD4eCFF15g9erV9OnTh7lz5+J0OrFamzd7LOjdOl2R/K9//et5\nVSTfunUrSUlJAHTv3p3vvvuOWbNmMXLkSN54440mXbOpyKpMja+avdU7OeQ4gKwEilmcNrOiKtru\nO01k//79TJ06le3bt/PCCy/w0ksvoYSg+F3QiHvXXXcxaNAgvF4vN910U5MzqfLz87nyyitJTEzk\njjvuoF27dkBg5fCJEycafF1IVvmqKrurtqOigi+Qo5tl7YhHdrG7egd+xUdGVHvSzBnasNg58tt7\neeLECaKjm3/9WdC7cv/99+N2u+s2kYZfiz6fC7t27aJ79+4AGI1GysoCQ1ClpaUkJze8xVAoVvn6\nVX/AtKfwKh4UVeGw4yA+JfClKKo9RKq5bSXRh4OrrrqKq6++mk2bNtGhQwemTJlCVFRUs7cT1LjV\n1dWsXr36vBtq3749Tz75JDabjYKCArxeb729fcOJTtQTb7BR6T2JiBjYdQelXj9XQNDydJuA2Wzm\no48+wu12Y7FYQpYiENS4aWlprFq1iuzs7Lob2bfvuWe9d+nShaVLl567whAgCRI5sXnIih9JkKjw\nlGPVRdPemo2/xo9P9pIZ1T6Qg6x595yxWCwhXz0T1Ljt2rXD5XKxY8eOumNNMW5rQxIkJElCUWVO\nesqINyZS7DxKx+jO6AQd1d4qBM21rZagowr33nsvnTt3xmAw0LFjR+67775w6AoboiDRKbZL3d8/\nV3zH9ye/RUHWtkdtIrVeFz7Zj1f24fZ5QtJGo4rebd++nfbt27Nz505mzZoVEiHhJjDsJePy1yIQ\nKBvUITqH7gm96GHrQ6IxRVuy3gRqvS5Wf/8u+U9fz3+sGMOe8l9CMrQY1LhFRUVMnjyZIUOGMHXq\n1Hob90Uyiqrw48mt/FTxHT+c3IKsykiChElnxiAZtBm082DJP/+OV/ZRXlvJvC9W4DxVar85CWpc\nSZL46quvqK6u5quvvgrJvHNLcNJdhlcJ/Iz5FC8nXCUEqUal0QgkQaxXF9eiD41fgj6cLVq0iBUr\nVvCPf/yDrKwsFixYEBIhoWbUqFF1G68MHTqUCQ+Mr3+CT2T+/PmsW7euwWt07Njx3yZDNOqjqAqL\nr32QRZtWYbPEMXfw37Aamn+EocGid4WFhQwcOPCs25+OGTOm2YWcC82xl6+sypS7j1PhKSfeYCPJ\nnEpB/wI2btzYvGIvQDx+D4qqIisKZr0JSWz+PJAGI+7pLoHFYqkbv/3tZtSRjCRIJJlSsRmTEQUR\nURDJyclBVvwIgoiKiojYZt5vODHqjCFvo0HjHj58mMOHD4dcQEty2rCnWfr0Uvbbd1PpPYlFZ+WS\nuG7oBC1XoTXS4F05M9JeKNi9VVT6TgLg9DsocR4lI6q9lvLYCmnQuHv37q3bMf1M2rKZRUSMopFU\nSyYmyYyo1cxttTRo3P79+wNt26i/xSiYuTThcspcJZS5Skg2p9XLItNoPTRo3C1btnDfffcxceLE\nepFXEAR69uwZNoGhRFZkVFQUVUYSdXz7zbf06NeNo7W/AFDpPUl3W2+tCMo54PS6kVUZqyG0Xc0G\njXu60N2DDz5Y73hbicCyKlPhOcGBmj1AoJ7CFX+8AodcTXtrNkbJRLW3ErffiUlqG5MuocbhdbJ4\n0/9wvPYk9/W5jY7xmZj0oRlhaNC4Tz31VIMvakoieatDhWPOI6SaM0g0JaOoKhYxihgxhhJXESfd\nZSSb07DqQrt7TFvB4XEy+7P/5p2dGwD4+vAPfDshdJM1DRrX5/PV7UVWUFBQbwVEU/jhhx947bXX\nsFqtJCQkUFxc3OJbonaIykYUdRxxHEQUJDpGX0y5q5Si2kMAVHjK6ZHYR3tEawyCQK92+UiixPrd\nG6nx1FLjqcUUojHd3424siyzZcsWPv/8cyoqKsjLy2PgwIFNashut/Poo49isVi46667sNvtdO3a\nFVEUw74lqqLKeBQ3Zn0UfsWP0+/Er/pw+R3U+n9dxayi4pFd6MXwbcrRmjlz2vxMJk19gH5DBlBa\nU86gnD8y9g8jWfH/XsOkM/Jp4QZmPfwIshzIEGuuafOg+5yd5qeffuK5555j165dbNq0qckNLl++\nnNTUVDIyMuq2RN2zZ0+DW6KejfOZ8v1trbD21mxEQeKXmr2km7OIN9n4V+WPqKhYpCguTbhMS28M\ngsNTy5XLb8XhDRSOeea6mfS7qBd2j4Nj9jLy0y7GIDXvl7/BiOtyudi8eTNffPEFJSUl5Ofnc999\n9zW5qHNtbS3z5s3juuuu49JLL+X9999v1Jaozb7KVxDq1Qo77iomLy4fAQGz3oKExGW23rhkZ6B/\nqwra8p0g7Dy+v860AJ8d+JbcpI7c/saDWAxm3r/9OWhm4zYYcbt27Up6ejoDBgz4t6jWlCSbGTNm\ncOTIEdLT05Ekqa5AxOktUc+lqN7ZIu7w64ZzvPT4777ObDbzxcYv+Klia11KY7zBRnbMxXXnHKs9\nSo2vilhDPPFiEuPuHse+ffsapSslNYX1/7u+0e+jMdTU1KDT6epGc1pjWqnT6+I/Voyhyl0DwH9f\n/wg/luxi5da3yE7I4t3blzV7emODEXfOnDnN2tC8efOa9Xq/5XjpcZ579/d3vAxM3arkxnWlxHkU\nSdCRbslCVeGw4wAdorNJt7RDELJw+10csx9h0qLGL1WacOP/Pc93UR+Hw8HMmTNZtmwZSUlJfPrp\np+R1yQMBPLIHg2jAIBmatc2mIAoihXf/nQ92b+Sy9DzSopN5ctNK0mOSmX/NJHQhSMpv0LhNrVjT\nmlFUBaevFpfiJM6QgKwqOHzVxBriaR+djaKqSIKEikKt14HdU9Wieh0OB88++ywAx48f56GHHuLl\nNS/zzN5lOP1O8uJzuT33thZ9eHT5PDh9LmZ8soQO8emcqK3gP68YzadjV6Gc+jFv7v4tNLI+blvi\nl8oDdLLlghBYKBnYnE9AEgRkZECl2ltJvNmGQTKwv2JPi+j0yT5MJhOSJNU9kcfGxlLlqcLpD/Qn\nd1Xuptx1krSo1BbR6PDU4vJ5+Pv36yjc/3Xd8R7pXejbMbSzqxdc2pNHdrP35C4kTplWUFFR2Vn5\nE9+Xf0upoxidx0hldQUxpjikFkhrPF3PocpXxerVq8nNzWXIkCEsXboU0Ri4ZaIgIiAQpW/+KjGN\nwa/ISIKIR/YSb46t938JltgGXtV8XHARFyAnoTOl7mKKnUcA6GDNITOqPYpHZevXgWJ8qamprFy5\nkhRrKsU1RWHRdXqcdP3H64mOjUFAwGQy8eGHH1JWVsbChQuZ8/gcJnefRIolGY/sweP20PPKQHQL\n59Iij9+LX/Hz8rZ3Gd97FIcqj7Ht2A5uumQgHRPa4VfkkPRtT3PBGTfeZMOij2J39XbiDTa8ige/\n4sdmisYn+xkxYgQulwuA8ePH8+KqF8Om7bTpfLKPaq+dRGsi99xzD+Xl5UBg4eqTTz7Jl0c2sbFk\nEynmZMZ1uZutW7eGTeNpVFXl55I9tI9P5863ZnD/H29j7OUjiLfE4va5cXhqiTFFYw5RrsIF1VUQ\nEOgQn42iKnSz9SLFnE6n2EtIi8rE7q3G7XHXmRagrKwMXQtUa5REiVf3vY7P5+Pee++tOz72rrGU\n28vJjM5EURVKnKV8fORT3P7mX/4dDBWVH4p3MaLLYG7OH8qG/d9w0lWF1+9DL+mxWeIAFYenNiTt\nX1ARN8YUh1t2YZbM7Kz8EZdci0VnpXPMJcQa4qnyVDFx4kSeffZZrFYr8+bNw2g2QPMX1G4QWZHx\nKl4GZPRDb9YzZeoURo76Ex6vmw4XdWDtwVf4z67jMYgGvIoXXTPO6g259hoqTpwMep4oinxS+Ck3\ndimgcP9mLku/hBsuKUCQQZFlampqmPfdCjb+soVnrptJzf4TTJ82vVEaEpJsfPLhx0HPa1PGDTaO\nOmDAAB55/GFq/Q5cciASuP1OdKIeASjyH+ThOTOZPecxDAYjiqKw9L+e5u23324WfdcOv5YTxxuu\nBQzwz83/RBIlOsXlMH/bQiZ0Hc9u/R4q5Ere3PEOelEf6PtKJtpZMxmcNQiDaGhUjnRSShIfrv+w\nwf+vOHES39CkoNfpnHIRHsHHgZNH6ZaWx6BVd7Ls2kco++kwj89+nIyMDP7n5ZeQVZlZG57hzVuf\nbtR1ASo++v3P5zRtyrjBJiAMkhGz3oKiygiIqCgoKFR7K4k1xGMzJnHAuQuTaKa9KYdoYwx/njCG\ngj9f3aj2g31xgpkWQBYUdIJEmfMEA9MKULwq/TL68vLu1cQYYrjhouuo9tqZ1mMyHtmDJOjwyo2b\nAm9M+43BFh1PlN7C5RldUFSVL/66mmjBTGL3Afh8Pvbs2cOkiX9j5aqVrPnX+pBMmbcp4wYjSh+F\noigIgsCl8ZdR6irCLFmINcRT7iol0ZRCmqUdLmegO+GqdZFqTeO4oxhZbZ76V0Pn/f7OnEadAZ/s\nI5YYPl35Ce843mbag9OYcMl4dDodR2qOsvjHJfgUHyIij1/xGG6fJ+h1AT6a8UGzvId7+t3KzuP7\n2Vv+CxXOau78wwi8tZ56OSc1NTWcqK3g7l43U3SyuFnaPZM2Y9yU1JSgEW/YsGE8Muth9tr/RVZU\nR7Ks2bj9TkqdRaRY0jnqOESylM6Im0ZQWFhIQUEB7733Hu8897988sknjdLweySlJAU1z7iudxNl\niuLOv9zJe++9B8Ann3zCpm82UetzkhaViv/UnhWCIKAC7lp3o0yZlBL851of5Kc6NzeX7OQsdpbt\np9JtZ3DnK9lbfoiO0RlMmzaNxYsXY7PZmL9gPuYoC7Iis/mjL4Ne91xpdFpja+J80hp9ipdt5d8g\nCTo6WjsRZ0xAQQUVREFg9cv/YOzYsXXnv/jii9x9993N/RYaRFEV3C43nTp1org4EKkEQaDWWcu/\n7LvIt3Vlw9FCYg2xZFozEQSBJHMSZt35J7H07NkzaF/0mTGPYY4yc8urDwDQydaeV29ZjFlvorrG\nTozZiorKTyV7qPDaUVSF7qm5XLfkrkZp0H90olHDexfUcBgEKtgYRAOy6mdfzS4OOfbjld0IgkC1\ntxKjqf64Yyj2L/g9REFEUZR6X5abb76ZE44T5MbnoqgKfVKvwOFzUOuvJcmciDGMiTYdEjPZsC8w\nvTuyy2BeuWUxhQe+4aeS3URboxH1En5BITs5C5ffjaqqxEfFkhzTvHtAt5muQmNRVegSfxnFzqPo\nBT2plkz2Ve+ic9wl/FKzj+tvuJ477riD9evXc+2113LjjTeGXaPVamXatGmMHDkSp9NJ1/yu6I16\nVDUwPf3Uj/+F59QD2a2db6GbLb9ZHoASkmxBn+pLhh1lWG5fXv3pAx7qfw+jXpnEgYrADORD/cYx\nuNNV3P/+4xgkPY8MmMAHezaSFZdOn5qOfPjR7kZpaAwXXFcBAj/HPsWL3VtNsfMIsYZ4sqwd+b78\nG4ySmURSiLPE4/X6iI0J/bz7ac5cGqPT6Rg+fDg9evQgOyebDhd1QEHlqOMIK3aurHvNZUndSStN\n4cHJD4ZtyrfGU0ulqxqA/i/e8auW9DxmD7yf61dPACAvOZvnb3iUzw98yx2XN2+24QUTcc80xWtv\nvEZiho14YwIxhlhUVNweNxeZO3PcV4xdrCJKjmbhgoVs2LCh7hqhNsbvXVtWZBTFR1Z0FlZ9FA5f\nYBy6R+JlXNypc9imfVVVZcvR7fTJ6oYKZMSkcMweSOC/ukNP9p44VHdutbsGi8HM8Lz+za6jRSPu\n8ePHWbhwIXFxcWRnZzd6ZUVzlRmtcJ9AQCDelBjIw1XVumGvlpjqbQx+xY9P8bH95A4yrZnYTLaw\n9XFHjRpFcnIy9894gGq/A5POSJI1gXU7N9AuLpW+F/Viw97NzNywFJ0o8eTQqfyxfQ8OlB/mq7cL\nWbNmTfgXS4aCZ555hquvvpru3bszbtw4nn/++UZVjTl8+DCDBw9m7dq1pKY2LRdVkiTMFjMALqer\nLuc1EtDr9RiNRvx+P253+PMULDFRKKKKR/ETa4xCVlWMkh4VFVlV0Is6/IrMzyW7mbfxBV68aQ6i\nW23yWsHU1FR0uvqBpEXDSnl5OWlpaQDExMRQU1NDXFxcvXPOtliytjbwM9nSBaY1GkffeZ+f1+vP\n9svaosZNT0+ntLSUlJQU7HY7MTEx/3bO2bZEdbvd7Nixg6SkpGat6zV+/HiWL1/ebNdrbi5UfWf7\nVW1R4/7pT39i/vz5WK1WBg0ahNjIkusmk4k//OEPza7HYDA0uc8cDjR9v9Kixk1MTGTx4sUtKUEj\nQrngZs402gaacTUiEs24Z/Dbh8DWhqbvVyJyyldDQ4u4GhGJZlyNiEQzrkZEohlXIyJpnSlQIeDt\nt9/mq6++QlVVTCYTDoeDkSNHctVVVzFnzhxEUcTlcjFw4ECGDBnCjz/+yJo1azCZTFRWVjJx4kQu\nvvji4A2dB+vWrePNN99kzZo1iKLIo48+itfrxel01tM8YMAAVqxYwaFDh1BVFYPBwKxZs/4tESXU\nWhctWsSXX36JXq/n2LFjDB48mD59+pCbm8uUKVMAuPvuu1m5cmWQq507F4xxS0pKqK2tZcGCBdhs\nNk6cOMGCBQvIzs5GlmUef/xxBEFg3bp1yLLMkiVLeOmllxBFkYqKCr755puQGxcCs4lPPfUU06ZN\nqzs2a9aseppPT42f3v3oyy+/pLi4mKysrJDrO5P8/HwKCwsZOnQo77zzDr1790YQBIqKiti4cSP9\n+vUL2fZiF4xxIZAE/cQTT2A0GhEEgQkTJtC+fXuuv/56Fi5ciMfjoVevXlRWVpKSklJnkISEBIYN\nC778uzm45ppr+PnnnyksLKw7NnfuXEwmU53mzz77rN4uRX379g2Ltt8yePBgCgsLGTJkCOXl5aSk\npFBeXs706dN5+OGHycnJCVnbF5RxBUFg5syZ2Gy/rmvavHkzPp+P6dMDJYLuvPNOBg0aRElJCS6X\nC7PZTEVFBatWrWLq1Klh0Tl16lTGjx+PLMukpKTw8MMP19N87Ngxvv76a/r06QPAu+++S3p6Or16\n9QqLvtOYTCZsNhuvvfYa/fv3r1vCr9frmTt3LjNnzsTv94ek7QvKuGejW7duzJkzhw8++ABFUbj8\n8svR6XRMnDiRSZMmER8fj91uZ+LEiWHTJEkSc+fOZcSIEWdN6bv66qvZtm0bkyZNQq/XYzabGT58\neNj0ncnIkSOZOHEiH3/8cb3aE6mpqfzlL39p9i0ZTqPNnGlEJNpwmEZEohlXIyLRjKsRkWjG1YhI\nNONqRCSacTUiEs24GhHJ/wdX9BI0cuIG/wAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ba70b6f7f50>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# sns.set(context='paper', style='ticks')\n",
    "reads_col = 'Uniquely mapped reads number'\n",
    "mapped_reads = study.mapping_stats.data[reads_col]/1e6\n",
    "\n",
    "print detected_genes.mean()\n",
    "print detected_genes.groupby(study.sample_id_to_phenotype).mean()\n",
    "\n",
    "mapped_reads = mapped_reads.to_frame().join(study.metadata.data[['pooled', 'phenotype']])\n",
    "mapped_reads = mapped_reads.rename(columns={0: 'Millions of reads'})\n",
    "\n",
    "mapped_reads_pooled = mapped_reads.ix[mapped_reads.pooled]\n",
    "mapped_reads_single = mapped_reads.ix[~mapped_reads.pooled]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2.5, 2))\n",
    "sns.boxplot(x='phenotype', y=reads_col, order=study.phenotype_order, linewidth=1, \n",
    "               data=mapped_reads_single, palette=study.phenotype_color_ordered, fliersize=0)\n",
    "sns.stripplot(x='phenotype', y=reads_col, order=study.phenotype_order, linewidth=1,edgecolor='white', #size=2,\n",
    "               data=mapped_reads_single, palette=study.phenotype_color_ordered, jitter=True)\n",
    "sns.stripplot(x='phenotype', y=reads_col, order=study.phenotype_order, linewidth=1,edgecolor='white',\n",
    "              data=mapped_reads_pooled, color='k', size=5, jitter=True)\n",
    "sns.despine()\n",
    "ax.set_xlabel('')\n",
    "\n",
    "ax.set_ylabel('Millions of reads')\n",
    "\n",
    "ymin, ymax = ax.get_ylim()\n",
    "ax.set_ylim(0, ymax)\n",
    "\n",
    "# Reduce number of ticks\n",
    "yloc = plt.MaxNLocator(5)\n",
    "ax.yaxis.set_major_locator(yloc)\n",
    "\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/mapped_reads.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Number of detected genes, per transcript type"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Differentially expressed genes between outliers and non-outliers\n",
    "\n",
    "Focus on just NPC and MN because that's where the majority of the outliers are."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(143, 18594)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "phenotype\n",
       "MN     70\n",
       "NPC    73\n",
       "dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression = study.expression.singles.ix[study.metadata.data.phenotype.map(lambda x: x in ('MN', 'NPC'))].dropna()\n",
    "print expression.shape\n",
    "# Use only genes which have $\\log_2(TPM+1) > 1$ in at least 3 cells.\n",
    "expression = expression.ix[:, (expression > 1).sum() >= 3]\n",
    "expression.groupby(study.sample_id_to_phenotype).size()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "from scipy.stats import kruskal\n",
    "\n",
    "grouped = expression.groupby(study.sample_id_to_phenotype)\n",
    "\n",
    "def outliers_diff_expression(df1):\n",
    "    # Use only genes which have $\\log_2(TPM+1) > 1$ in at least 3 cells.\n",
    "    df1 = df1.ix[:, (df1 > 1).sum() >= 3].dropna()\n",
    "    return df1.groupby(level=0, axis=1).apply(lambda df2: pd.Series(kruskal(*[x for name, x in \n",
    "                                                                              df2.groupby(study.metadata.data.outlier)]), \n",
    "                                                                    index=['H_statistic', 'p_value']))\n",
    "\n",
    "kruskal_results = grouped.apply(outliers_diff_expression)\n",
    "kruskal_results.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "kruskal_results_tidy = kruskal_results.T\n",
    "kruskal_results_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "kruskal_results_tidy = kruskal_results_tidy.stack(level=0).reset_index()\n",
    "kruskal_results_tidy = kruskal_results_tidy.rename(columns={'level_0': 'gene_id'})\n",
    "kruskal_results_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "kruskal_results_tidy['bonferonni_p_value'] = kruskal_results_tidy.groupby(\n",
    "    'phenotype', as_index=False, group_keys=False).apply(lambda x: x.p_value * x.shape[0])\n",
    "kruskal_results_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fig, ax = plt.subplots(figsize=(2, 2))\n",
    "\n",
    "sns.distplot(kruskal_results_tidy.bonferonni_p_value, color='grey', ax=ax, kde=False)\n",
    "sns.despine(offset=2)\n",
    "ax.set(xlim=(0,10000))\n",
    "ax.locator_params(nbins=3)\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/outlier_neural_types_differential_expression_p_distribution.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "p = 0.001\n",
    "kruskal_results_tidy_significant = kruskal_results_tidy.ix[kruskal_results_tidy.bonferonni_p_value < p]\n",
    "print kruskal_results_tidy_significant.gene_id.unique().shape\n",
    "gene_ids = kruskal_results_tidy_significant.gene_id.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "go = study.go_enrichment(gene_ids, background=expression.columns, p_value_cutoff=p, domain='biological_process', \n",
    "                         min_background_size=10, min_feature_size=5)\n",
    "go = go.iloc[:10, :]\n",
    "sns.set(style='ticks', context='paper')\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(4, 2))\n",
    "\n",
    "y = np.arange(go.shape[0])\n",
    "width = -np.log10(go.bonferonni_corrected_p_value)\n",
    "ax.barh(y, width, color='grey')\n",
    "\n",
    "yticks = y + 0.4\n",
    "ax.set(yticklabels=go.go_name, yticks=yticks, xlabel='$\\log_{10}p$-value', ylim=(0, y.max()+1))\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/outlier_neural_types_go_enrichment.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Clustermap of DE genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# data = expression.T\n",
    "# data = (data - data.mean())/data.std()\n",
    "# data = data.ix[gene_ids, :]\n",
    "data = expression.ix[:, gene_ids].T\n",
    "\n",
    "\n",
    "# IN case this data is properly logged in the future...\n",
    "if study.expression.data.max().max() > 100:\n",
    "    data = np.log2(data+1)\n",
    "\n",
    "col_colors = [study.sample_id_to_color[data.columns], \n",
    "              study.metadata.data.outlier[data.columns].map(lambda x: 'lightgrey' if x else 'k')]\n",
    "yticklabels = study.expression.feature_data.gene_name[data.index]\n",
    "\n",
    "g = sns.clustermap(data=data, col_colors=col_colors, yticklabels=yticklabels, method='ward', figsize=(8, 8),\n",
    "                  cbar_kws=dict(label='$\\log_2(TPM+1)'))\n",
    "g.ax_heatmap.set(yticks=[], xticks=[])\n",
    "g.ax_col_colors.set(xticks=[], yticks=[])\n",
    "g.savefig('{}/outlier_neural_types_clustermap.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Colorbar legend"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# sns.set(style='ticks', context='talk')\n",
    "fig, axes = plt.subplots(nrows=2, figsize=(1.5, 1))\n",
    "\n",
    "phantom_bars = {'Celltype': {'NPC': study.phenotype_to_color['NPC'],\n",
    "                             'MN': study.phenotype_to_color['MN']},\n",
    "                '$k$-Means Outlier': {'Outlier': 'lightgrey', 'Not outlier': 'k'}}\n",
    "legends = []\n",
    "for (title, label_to_color), ax in zip(phantom_bars.items(), axes.flat):\n",
    "    for label, color in label_to_color.items():\n",
    "        ax.bar(0, 0, 0,  color=color, label=label)\n",
    "    legends.append(ax.legend(loc='center', title=title, ncol=2))\n",
    "    ax.axis('off')\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/outlier_neural_types_clustermap_color_labels.pdf'.format(folder), bbox_inches='tight', bbox_extra_artists=legends)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Splicing quality control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='paper')\n",
    "\n",
    "grouped = study.splicing.data_original.groupby(study.metadata.data.pooled)\n",
    "detected_events = grouped.apply(lambda x: x.groupby(study.sample_id_to_phenotype).apply(lambda y: y.count(axis=1)))\n",
    "detected_events = detected_events/1000\n",
    "detected_events = detected_events.reset_index()\n",
    "detected_events = detected_events.rename(columns={0: 'n_events', 'level_1': 'phenotype', 'level_2': 'sample_id'})\n",
    "detected_events.phenotype = pd.Categorical(detected_events.phenotype, categories=study.phenotype_order, ordered=True)\n",
    "detected_events.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "detected_events_singles = detected_events.ix[~detected_events.pooled]\n",
    "detected_events_pooled = detected_events.ix[detected_events.pooled]\n",
    "\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2.5, 2))\n",
    "sns.boxplot(x='phenotype', y='n_events', data=detected_events_singles, palette=study.phenotype_color_ordered, linewidth=1.5)\n",
    "sns.stripplot(x='phenotype', y='n_events', data=detected_events_singles, edgecolor='gray',\n",
    "              palette=study.phenotype_color_ordered, linewidth=1, #size=2, \n",
    "              jitter=True)\n",
    "sns.stripplot(x='phenotype', y='n_events', data=detected_events_pooled, color='k', size=5, linewidth=1, jitter=True)\n",
    "sns.despine()\n",
    "ax.set_xlabel('')\n",
    "ax.set_ylabel('AS events $\\times 10^3$')\n",
    "ymin, ymax = ax.get_ylim()\n",
    "ax.set_ylim(0, ymax)\n",
    "ax.locator_params('y', nbins=5)\n",
    "fig.tight_layout()\n",
    "fig.savefig(\"{}/A_splicing_events_detected.pdf\".format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import itertools\n",
    "\n",
    "sns.set(style='whitegrid')\n",
    "\n",
    "nrows = study.metadata.n_phenotypes\n",
    "\n",
    "largest_group = study.sample_id_to_phenotype.groupby(study.sample_id_to_phenotype).size().max()\n",
    "\n",
    "fig, axes = plt.subplots(nrows=nrows, figsize=(.5*largest_group, 6*study.metadata.n_phenotypes), sharey=True)\n",
    "paired = sns.color_palette('Paired', n_colors=4)\n",
    "\n",
    "splits = study.metadata.data.split.dropna()\n",
    "split_to_pair = dict((x, tuple(sorted([x, y]))) for x, y in splits.iteritems())\n",
    "unique_pairs = set([tuple(sorted([x, y])) for x, y in splits.iteritems()])\n",
    "pair_colors = sns.color_palette('husl', n_colors=len(unique_pairs))\n",
    "pair_to_colors = dict(zip(unique_pairs, pair_colors))\n",
    "\n",
    "\n",
    "total_width = 0.8\n",
    "# width = total_width/expression_binned.shape[1]\n",
    "\n",
    "\n",
    "for ax, phenotype in zip(axes, study.metadata.phenotype_order):\n",
    "    sample_ids = study.sample_id_to_phenotype.index[study.sample_id_to_phenotype == phenotype]\n",
    "    pooled = sample_ids[study.metadata.data.pooled[sample_ids]]\n",
    "    if phenotype == 'iPSC':\n",
    "        split = list(itertools.chain(*unique_pairs))\n",
    "    else:\n",
    "        split = []\n",
    "    singles = sample_ids.difference(pooled).difference(split)\n",
    "    \n",
    "    ordered_sample_ids = np.concatenate([pooled, split, singles])\n",
    "    \n",
    "    splicing = study.splicing.data_original.ix[ordered_sample_ids]\n",
    "    n_events = splicing.T.count()\n",
    "    print phenotype, 'n_events.mean(), n_events.std()', n_events.mean(), n_events.std()\n",
    "    \n",
    "    n_events = n_events.fillna(0)\n",
    "    left = np.arange(n_events.shape[0])\n",
    "    \n",
    "    color = study.phenotype_to_color[phenotype]\n",
    "    ax.bar(left, n_events, color=color, label='Number of splicing events detected' , width=total_width)\n",
    "#     ax.set_ylim(1, n_events.max().max()*1.1)\n",
    "    xticks = left+total_width/2\n",
    "    ax.set_xticks(xticks)\n",
    "    ax.set_xticklabels(ordered_sample_ids, rotation=90, visible=True, fontsize=24)\n",
    "    ax.set_xlim(0, xticks.max()+0.5)\n",
    "    \n",
    "    for label in ax.get_xticklabels():\n",
    "        text = label.get_text()\n",
    "        if label.get_text() in study.pooled:\n",
    "            label.set(backgroundcolor='lightgrey', weight='bold')\n",
    "        elif text in splits:\n",
    "            pair = split_to_pair[text]\n",
    "            color = pair_to_colors[pair]\n",
    "            label.set(backgroundcolor=color)\n",
    "    ax.xaxis.grid(False)\n",
    "\n",
    "    ax.set_title(phenotype, fontsize=32)\n",
    "    ax.set_ylabel('Alternatively spliced junctions', fontsize=24)\n",
    "    \n",
    "#     ax.set_yticks(np.arange(0, n_events.max()+1000, 2500))\n",
    "    ax.locator_params('y', nbins=5)\n",
    "    for label in ax.get_yticklabels():\n",
    "        label.set(size=24)\n",
    "#     if ax.is_first_row():\n",
    "#         ax.legend(loc=\"best\", fontsize=24)\n",
    "    \n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/quality_control_all_cells_splicing_barplots.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Number of cells per event"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='talk')\n",
    "\n",
    "for phenotype, df in study.splicing.data_original.groupby(study.sample_id_to_phenotype):    \n",
    "    counts = df.count()\n",
    "    fig, ax = plt.subplots(figsize=(3, 2))\n",
    "    color = study.phenotype_to_color[phenotype]\n",
    "    sns.distplot(counts, bins=np.arange(1, counts.max()+1), kde=False, color=color, \n",
    "                 hist_kws=dict(alpha=0.75, linewidth=0.1))\n",
    "\n",
    "    # Reduce nubmer of ticks\n",
    "    ymin, ymax = ax.get_ylim()\n",
    "    ax.vlines(10, ymin, ymax, linestyle='--')\n",
    "    \n",
    "    ax.set(ylabel='AS events', xlabel=\"Single cells\", title=phenotype)\n",
    "    ax.locator_params(nbins=4)\n",
    "    \n",
    "    fig.tight_layout()\n",
    "    sns.despine()\n",
    "    fig.savefig('{}/n_cells_per_event_{}.pdf'.format(folder, phenotype))\n",
    "\n",
    "    \n",
    "## All events\n",
    "phenotype = 'all'\n",
    "counts = study.splicing.data_original.count()\n",
    "fig, ax = plt.subplots(figsize=(3, 2))\n",
    "color = '#262626'\n",
    "sns.distplot(counts, bins=np.arange(1, counts.max()+1), kde=False, color=color, \n",
    "             hist_kws=dict(alpha=0.75, linewidth=0.1))\n",
    "ymin, ymax = ax.get_ylim()\n",
    "ax.set_title(phenotype)\n",
    "ax.set_xlabel('Cells')\n",
    "ax.set_ylabel('AS events')\n",
    "\n",
    "# Reduce number of tick\n",
    "ax.locator_params(nbins=4)\n",
    "\n",
    "ymin, ymax = ax.get_ylim()\n",
    "xmin, xmax = ax.get_xlim()\n",
    "x1 = [0, 0]\n",
    "x2 = [10,10]\n",
    "# x1 = 10,10]\n",
    "# x2 = [xmax, xmax]\n",
    "\n",
    "# ax.fill_betweenx([ymin, ymax], x1, x2, color='darkgrey', zorder=-1)\n",
    "ax.vlines(10, ymin, ymax, linestyle='--')\n",
    "\n",
    "fig.tight_layout()\n",
    "sns.despine()\n",
    "fig.savefig('{}/n_cells_per_event_{}.pdf'.format(folder, phenotype))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Diversity of splice junction features\n",
    "https://en.wikipedia.org/wiki/Diversity_index\n",
    "\n",
    "#### True diversity\n",
    "\n",
    "$^qD = \\frac{1}{M_{q-1}} = \\frac{1}{\\sqrt[q-1]{\\sum_{i=1}^{R} p_i p_i^{q-1}}} = \\left(\\sum_{i=1}^{R} p_i^q\\right)^{1/(1-q)}$\n",
    "\n",
    "* $M_{q-1}$: average proportional abundance of the types in the dataset\n",
    "* $R$: richness\n",
    "* $p_i$: Proportional abundance of the $i$th type\n",
    "\n",
    "Pick different values of $q$...\n",
    "\n",
    "With $q=1$, equation is undefined but limit approaches\n",
    "\n",
    "$\n",
    "^1D = \\frac{1}{\\prod_{i=1}^{R} p_i^{p_i}} = \\exp\\left(-\\sum_{i=1}^{R}p_i \\ln(p_i)\\right)\n",
    "$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "diversity_folder = '{}/diversity'.format(folder)\n",
    "! mkdir $diversity_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "splicing_original = study.splicing.data_original.copy()\n",
    "singles = splicing_original.loc[study.splicing.singles.index]\n",
    "print singles.shape\n",
    "singles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "counts = splicing_original.groupby(study.sample_id_to_phenotype).count()\n",
    "proportions = counts.divide(counts.sum(axis=1), axis='index')\n",
    "proportions.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Implementation of:\n",
    "\n",
    "$\n",
    "^1D = \\frac{1}{\\prod_{i=1}^{R} p_i^{p_i}} = \\exp\\left(-\\sum_{i=1}^{R}p_i \\ln(p_i)\\right)\n",
    "$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='talk')\n",
    "diversity = np.exp(- proportions.multiply(np.log(proportions)).sum(axis=1))\n",
    "diversity = diversity[study.phenotype_order]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(3, 2))\n",
    "left = np.arange(diversity.shape[0])\n",
    "color = [study.phenotype_to_color[p] for p in diversity.index]\n",
    "ax.bar(left, diversity, color=color, linewidth=1)\n",
    "\n",
    "xticks = left + 0.4\n",
    "ax.locator_params('y', nbins=4)\n",
    "ax.set(xticks=xticks, xticklabels=diversity.index, ylabel='Shannon Diversity Index')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/shannon_diversity_index.pdf'.format(diversity_folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Shannon diversity for Expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "expression_singles = study.expression.singles\n",
    "six.print_(expression_singles.shape)\n",
    "not_outliers = study.expression.singles.index.difference(study.expression.outliers.index)\n",
    "expression_singles = expression_singles.loc[not_outliers]\n",
    "six.print_(expression_singles.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "expression_sum = expression_singles.groupby(study.sample_id_to_phenotype).sum()\n",
    "expression_proportions = expression_sum.divide(expression_sum.sum(axis=1), axis=0)\n",
    "expression_proportions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='talk')\n",
    "expression_diversity = np.exp(- expression_proportions.multiply(np.log(expression_proportions)).sum(axis=1))\n",
    "expression_diversity = expression_diversity[study.phenotype_order]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(3, 2))\n",
    "left = np.arange(expression_diversity.shape[0])\n",
    "color = [study.phenotype_to_color[p] for p in diversity.index]\n",
    "ax.bar(left, expression_diversity, color=color, linewidth=1)\n",
    "\n",
    "xticks = left + 0.4\n",
    "ax.locator_params('y', nbins=4)\n",
    "ax.set(xticks=xticks, xticklabels=diversity.index, ylabel='Shannon Diversity Index')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/shannon_diversity_index_expression.pdf'.format(diversity_folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "expression_diversity_df = expression_diversity.reset_index()\n",
    "expression_diversity_df = expression_diversity_df.rename(columns={0: 'Shannon Diversity'})\n",
    "expression_diversity_df['data_type'] = 'Expression'\n",
    "expression_diversity_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "splicing_diversity_df = diversity.reset_index()\n",
    "splicing_diversity_df = splicing_diversity_df.rename(columns={0: 'Shannon Diversity'})\n",
    "splicing_diversity_df['data_type'] = 'Splicing'\n",
    "splicing_diversity_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "diversity_df = pd.concat([expression_diversity_df, splicing_diversity_df])\n",
    "diversity_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "fig, ax = plt.subplots(figsize=(3, 2))\n",
    "sns.barplot(x='data_type', y='Shannon Diversity', data=diversity_df, hue='phenotype', \n",
    "            palette=study.phenotype_color_ordered, hue_order=study.phenotype_order)\n",
    "ax.set(xlabel='', ylabel='Shannon Diversity')\n",
    "ax.locator_params(nbins=5, axis='y')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/shannon_diversity_expression_splicing.pdf'.format(diversity_folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Inverse Simpson index\n",
    "\n",
    "This is more intuitive because datasets with a larger number of species and fewer abundances per species will have larger diversity.\n",
    "\n",
    "$1/\\lambda = \\frac{1}{\\sum_{i=1}^R p_i^2} =\\, ^2D$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='talk')\n",
    "diversity = 1/np.square(proportions).sum(axis=1)\n",
    "diversity = diversity[study.phenotype_order]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(3, 2))\n",
    "left = np.arange(diversity.shape[0])\n",
    "color = [study.phenotype_to_color[p] for p in diversity.index]\n",
    "ax.bar(left, diversity, color=color, linewidth=1)\n",
    "\n",
    "xticks = left + 0.4\n",
    "ax.locator_params('y', nbins=4)\n",
    "ax.set(xticks=xticks, xticklabels=diversity.index, ylabel='Inverse Simpson Index')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/inverse_simpson_index.pdf'.format(diversity_folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Percentage of events detected in only one cell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "junction_counts = splicing_original.groupby(study.sample_id_to_phenotype).count().unstack()\n",
    "junction_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "junction_counts = junction_counts.reset_index()\n",
    "junction_counts = junction_counts.rename(columns={'level_0': 'event_name', 0: 'n_cells'})\n",
    "junction_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "print junction_counts.shape\n",
    "junction_counts = junction_counts.loc[junction_counts.n_cells > 0]\n",
    "print junction_counts.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "   \n",
    "def percent_junctions_at_quantiles(series, quantiles=[0, 0.25, 0.5, 1.]):\n",
    "    \"\"\"Get the percent of junctions observed in 25% of cells, 50% of cells, etc\n",
    "    \n",
    "    Note: 0% gets interpreted as how many junctions were observed in 1 cell, not 0 cells \n",
    "    (since 0 cells is boring)\n",
    "    \"\"\"\n",
    "    series.index = series.index.droplevel(0)\n",
    "    total = series.sum()\n",
    "    percentage = 100*series/total\n",
    "    \n",
    "    percentage_at_quantiles = pd.Series(index=quantiles)\n",
    "    for q in quantiles:\n",
    "        if q == 0:\n",
    "            percentage_at_quantiles.loc[0] = percentage[1]\n",
    "        else:\n",
    "            i = int(np.round(q * series.index.max()))\n",
    "#             print q, i, series[:i]/total\n",
    "            percentage_at_quantiles.loc[q] = percentage[i]\n",
    "    return percentage_at_quantiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "junctions_per_cell = junction_counts.groupby(['phenotype', 'n_cells']).size()\n",
    "percent_junctions_per_cell = junctions_per_cell/junctions_per_cell.groupby(level=0).sum()\n",
    "# percent_junctions_per_cell = percent_junctions_per_cell.unstack()\n",
    "# percent_junctions_per_cell.columns = percent_junctions_per_cell.columns + 1\n",
    "percent_junctions_per_cell.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "percentage_junctions_quantiles = junctions_per_cell.groupby(level=0).apply(percent_junctions_at_quantiles).reset_index()\n",
    "percentage_junctions_quantiles = percentage_junctions_quantiles.rename(\n",
    "    columns={'level_1': 'Quantity of cells', 0:'Percentage of junctions'})\n",
    "percentage_junctions_quantiles['Quantity of cells'] *= 100\n",
    "percentage_junctions_quantiles['Quantity of cells'] = percentage_junctions_quantiles['Quantity of cells'].replace(0, '1 cell')\n",
    "percentage_junctions_quantiles['Quantity of cells'] = percentage_junctions_quantiles['Quantity of cells'].map(\n",
    "    lambda x: x if isinstance(x, str) else '{:.0f}%'.format(x))\n",
    "percentage_junctions_quantiles.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "g = sns.factorplot(x='Quantity of cells', y='Percentage of junctions', data=percentage_junctions_quantiles, \n",
    "               hue='phenotype', kind='bar', hue_order=study.phenotype_order,\n",
    "              palette=study.phenotype_color_ordered)\n",
    "g.savefig('{}/quantity_of_cells_per_percentage_of_junctions_barplot.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "viz = study.plot_pca(show_vectors=False, scale_by_variance=False, markersize10)\n",
    "viz.fig_reduced.savefig('{}/pca_variant.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Only use splicing events detected in at least10 samples in at least 1 celltype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "splicing_filter_folder = '{}/splicing_filtering'.format(folder)\n",
    "! mkdir $splicing_filter_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "alternative = study.splicing.data_original.groupby(study.sample_id_to_phenotype).apply(lambda x: x.dropna(thresh=10, axis=1))\n",
    "print study.splicing.data_original.shape, alternative.shape\n",
    "# constitutive = study.supplemental.psi_constitutive.groupby(study.sample_id_to_phenotype).apply(lambda x: x.dropna(thresh10, axis=1))\n",
    "# print study.supplemental.psi_constitutive.shape, constitutive.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "study.splicing.data_original = alternative\n",
    "# study.supplemental.psi_constitutive = constitutive"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Distribution of gene expression for all genes, and genes for which we measured splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "study.splicing.feature_expression_id_col = 'ensembl_id'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# splicing_metadata = pd.read_csv('/projects/ps-yeolab/obotvinnik/singlecell_pnms/skipped_exon_metadata.csv', index_col=0)\n",
    "# study.splicing.feature_expression_id_col = 'ensembl_id'\n",
    "# study.splicing.feature_data = splicing_metadata\n",
    "# cols = study.expression.feature_data.columns.difference(study.splicing.feature_data.columns)\n",
    "# study.splicing.feature_data = study.splicing.feature_data.join(study.expression.feature_data[cols], on='one_ensembl_id')\n",
    "\n",
    "# study.splicing.feature_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='paper')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## Genes\n",
    "fig, ax = plt.subplots(figsize=(2, 1.5))\n",
    "\n",
    "sns.distplot(study.expression.data[study.expression.data >= 1].values.flat, kde=False,\n",
    "            bins=np.linspace(0, 16, 50), hist_kws=dict(range=(0, 16)),\n",
    "             ax=ax, color='grey')\n",
    "ax.set_ylabel('Genes')\n",
    "ax.set_xlabel('$\\log_2(TPM+1)$')\n",
    "ax.locator_params(axis='y', nbins=4)\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/distribution_of_expression_all_genes.pdf'.format(splicing_filter_folder))\n",
    "\n",
    "\n",
    "\n",
    "## Splicing events\n",
    "fig, ax = plt.subplots(figsize=(2, 1.5))\n",
    "no_duplicate_genes = study.tidy_splicing_with_expression.groupby(['sample_id', 'common_id']).expression.mean()\n",
    "sns.distplot(np.log2(no_duplicate_genes+1), kde=False, ax=ax)\n",
    "ax.locator_params(axis='y', nbins=4)\n",
    "ax.set_ylabel('Alt Spliced Junctions')\n",
    "ax.set_xlabel('$\\log_2(TPM+1)$')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Only use splicing events with mean gene expression TPM > 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "grouped = study.tidy_splicing_with_expression.groupby(['sample_id', 'event_name'], as_index=False)\n",
    "# Take the mean psi if there are multiple splicing events per gene, for each sample's splicing event\n",
    "summed_expression = grouped.agg({'expression':'mean', 'psi': 'mean'})\n",
    "splicing_tidy_filtered = summed_expression.ix[summed_expression.expression >= 1]\n",
    "splicing_filtered = splicing_tidy_filtered.pivot(index='sample_id', columns='event_name', values='psi')\n",
    "print splicing_filtered.shape\n",
    "splicing_filtered.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "splicing_filter_check = splicing_tidy_filtered.merge(study.tidy_splicing_with_expression, left_on=['sample_id', 'event_name'],\n",
    "                            right_on=['sample_id', 'event_name'])\n",
    "splicing_filter_check.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.set(style='ticks', context='paper')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "## Splicing events\n",
    "fig, ax = plt.subplots(figsize=(2, 1.5))\n",
    "# ind = study.tidy_splicing_with_expression['event_name'].isin(splicing_filtered.columns)\n",
    "# tidy_splicing_with_expression_filtered = study.tidy_splicing_with_expression.ix[ind]\n",
    "no_duplicate_genes = splicing_filter_check.groupby(['sample_id', 'common_id']).expression_x.sum()\n",
    "lower_quantile = np.log2(no_duplicate_genes.quantile(0.1)+1)\n",
    "upper_quantile = np.log2(no_duplicate_genes.quantile(0.9)+1)\n",
    "six.print_(lower_quantile, upper_quantile)\n",
    "\n",
    "sns.distplot(np.log2(no_duplicate_genes+1), kde=False, color='grey')\n",
    "ymin, ymax = ax.get_ylim()\n",
    "ax.vlines(lower_quantile, ymin, ymax, linestyle='--', color='Crimson')\n",
    "ax.vlines(upper_quantile, ymin, ymax, linestyle='--', color='Crimson')\n",
    "ax.locator_params(nbins=5)\n",
    "ax.set_ylabel('AS Events')\n",
    "ax.set_xlabel('$\\log_2(TPM+1)$')\n",
    "sns.despine()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/distribution_of_expression_of_splicing_events.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Assign this new splicing matrix to our `study`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "print study.splicing.data_original.shape\n",
    "study.splicing.data_original = splicing_filtered\n",
    "study.splicing.data = splicing_filtered\n",
    "print study.splicing.data_original.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "study.save('singlecell_pnm_figure1_supplementary_post_splicing_filtering', flotilla_dir=flotilla_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "study = flotilla.embark('singlecell_pnm_figure1_supplementary_post_splicing_filtering', flotilla_dir=flotilla_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "study.splicing.data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "study.splicing.data_original.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "study.splicing.feature_expression_id_col = 'ensembl_id'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Venn diagram of shared events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "events_per_phenotype = dict((name, df.dropna(thresh=10, axis=1).dropna(how='all',axis=0).columns)\n",
    "                            for name, df in study.splicing.data.groupby(study.sample_id_to_phenotype))\n",
    "events_per_phenotype_count = pd.Series(events_per_phenotype).map(len)\n",
    "\n",
    "phenotypes, events = zip(*events_per_phenotype.items())\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(2, 2))\n",
    "matplotlib_venn.venn3(events, set_labels=phenotypes, \n",
    "                      set_colors=[study.phenotype_to_color[p] for p in phenotypes])\n",
    "ax = plt.gca()\n",
    "# ax.set_title('Alternative splicing events\\10+ cells per celltype')\n",
    "fig = plt.gcf()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/shared_events_venn10_per_phenotype.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "events_per_phenotype = dict((name, df.dropna(how='all',axis=1).dropna(how='all', axis=0).columns)\n",
    "                            for name, df in study.splicing.data.groupby(study.sample_id_to_phenotype))\n",
    "events_per_phenotype_count = pd.Series(events_per_phenotype).map(len)\n",
    "\n",
    "phenotypes, events = zip(*events_per_phenotype.items())\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(4, 3))\n",
    "matplotlib_venn.venn3(events, set_labels=phenotypes, \n",
    "                      set_colors=[study.phenotype_to_color[p] for p in phenotypes])\n",
    "ax = plt.gca()\n",
    "ax.set_title('Shared splicing events\\n10+ cells total')\n",
    "fig = plt.gcf()\n",
    "fig.tight_layout()\n",
    "fig.savefig('{}/shared_events_venn10_total.pdf'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "event_to_genes =  study.tidy_splicing_with_expression.drop(['sample_id', 'psi', 'expression'], axis=1)\n",
    "event_to_genes = event_to_genes.drop_duplicates()\n",
    "event_to_genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "scatterplot_folder = '{}/scatterplots'.format(folder)\n",
    "! mkdir $scatterplot_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from scipy.stats import pearsonr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "comparisons = [('P1_05', 'P7_08'),\n",
    "               ('P1_05', 'P2_03'),\n",
    "               ('M2nd_33', 'M2nd_34'),\n",
    "               ('M2nd_33', 'P1_05')]\n",
    "for sample1, sample2 in comparisons:\n",
    "    study.plot_two_samples(sample1, sample2, 'splicing', size=2, color='#262626', stat_func=None, alpha=0.5, rasterized=True)\n",
    "    x = study.splicing.data.loc[sample1]\n",
    "    y = study.splicing.data.loc[sample2]\n",
    "    correlation = x.corr(y)\n",
    "    six.print_(sample1, sample2, correlation)\n",
    "    \n",
    "    fig = plt.gcf()\n",
    "    ax = fig.axes[0]\n",
    "    ax.set(xticks=[0, 0.5, 1], yticks=[0, 0.5, 1], xlabel='$\\Psi$', ylabel='$\\Psi$')\n",
    "    fig.savefig(\"{}/{}_vs_{}_splicing.pdf\".format(scatterplot_folder, sample1, sample2), dpi=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "comparisons = [('P1_05', 'P7_08'),\n",
    "               ('P1_05', 'P2_03'),\n",
    "               ('M2nd_33', 'M2nd_34'),\n",
    "               ('M2nd_33', 'P1_05')]\n",
    "for sample1, sample2 in comparisons:\n",
    "    x = np.log2(study.expression.data.loc[sample1]+1)\n",
    "    y = np.log2(study.expression.data.loc[sample2]+1)\n",
    "    correlation = x.corr(y)\n",
    "    six.print_(sample1, sample2, correlation)\n",
    "#     study.plot_two_samples(sample1, sample2, 'expression', size=2, color='#262626', stat_func=None, alpha=0.5, rasterized=True)\n",
    "    sns.jointplot(x, y, size=2, color='#262626', stat_func=None, alpha=0.1, rasterized=True)\n",
    "    fig = plt.gcf()\n",
    "\n",
    "    ax = fig.axes[0]\n",
    "    ax.set(#xticks=[0, 4, 8, 12], yticks=[0, 4, 8, 12], \n",
    "           xlabel='$\\log_2 (\\mathrm{TPM} + 1)$', ylabel='$\\log_2 (\\mathrm{TPM} + 1)$')\n",
    "    fig.savefig(\"{}/{}_vs_{}_expression.pdf\".format(scatterplot_folder, sample1, sample2), dpi=600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "\n",
    "seen = set([])\n",
    "tidy_splicing_with_expression = study.tidy_splicing_with_expression.copy()\n",
    "tidy_splicing_with_expression = tidy_splicing_with_expression.set_index('event_name').sort_index()\n",
    "dark2 = sns.color_palette('Dark2')\n",
    "\n",
    "splits = study.metadata.data.split.dropna()\n",
    "\n",
    "paired = sns.color_palette('Paired')\n",
    "\n",
    "sample_colors = {'P1_05': paired[1],\n",
    "                'P7_08': paired[0],\n",
    "                'P2_03': 'salmon',\n",
    "                'M2nd_33': '#262626',\n",
    "                'M2nd_34': 'lightgrey'}\n",
    "\n",
    "\n",
    "for sample1, sample2 in comparisons:\n",
    "    fig, ax = plt.subplots(figsize=(4,3))\n",
    "    \n",
    "    ind1 = tidy_splicing_with_expression.sample_id == sample1\n",
    "    ind2 = tidy_splicing_with_expression.sample_id == sample2\n",
    "    \n",
    "    psi1 = tidy_splicing_with_expression.ix[ind1, 'psi']\n",
    "    psi2 = tidy_splicing_with_expression.ix[ind2, 'psi']\n",
    "    delta_psi = (psi1 - psi2).dropna()\n",
    "\n",
    "    for sample in (sample1, sample2):\n",
    "        ind = tidy_splicing_with_expression.sample_id == sample\n",
    "        y = np.log2(tidy_splicing_with_expression.ix[ind, 'expression']+1)\n",
    "        x, y = delta_psi.align(y, 'inner')\n",
    "        color = sample_colors[sample]\n",
    "        ax.plot(x, y, 'o', label=sample, markerfacecolor=color, alpha=0.5, markeredgewidth=0.75, markeredgecolor='white',\n",
    "               rasterized=True)\n",
    "    legend = ax.legend(**{'bbox_to_anchor': (1, 0.5), 'loc': 'center left'})\n",
    "    ax.set_xlabel('$\\Delta \\Psi$')\n",
    "    ax.set_ylabel('$\\\\log_2{{TPM+1)}}$')\n",
    "    ax.set_xlim(-1, 1)\n",
    "    ymin, ymax = ax.get_ylim()\n",
    "    ax.set_ylim(0, ymax)\n",
    "    sns.despine()\n",
    "    fig.savefig('{}/split_delta_psi_vs_tpm_{}_{}.png'.format(folder, sample1, sample2), \n",
    "                format='png', dpi=300, **{'bbox_extra_artists': (legend,), 'bbox_inches': 'tight'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# sns.set(style='whitegrid')\n",
    "\n",
    "tidy_splicing_with_expression = study.tidy_splicing_with_expression\n",
    "tidy_splicing_with_expression = tidy_splicing_with_expression.set_index('event_name').sort_index()\n",
    "dark2 = sns.color_palette('Dark2')\n",
    "\n",
    "splits = study.metadata.data.split.dropna()\n",
    "\n",
    "paired = sns.color_palette('Paired')\n",
    "\n",
    "sample_colors = {'P1_05': paired[1],\n",
    "                'P7_08': paired[0],\n",
    "                'P2_03': 'salmon',\n",
    "                'M2nd_33': '#262626',\n",
    "                'M2nd_34': 'lightgrey'}\n",
    "\n",
    "# folder = '/home/obotvinnik/Dropbox/figures2/singlecell_pnm/figure2_splicing'\n",
    "\n",
    "for sample1, sample2 in comparisons:\n",
    "#     fig, ax = plt.subplots(figsize=(4,3))\n",
    "    \n",
    "    ind1 = tidy_splicing_with_expression.sample_id == sample1\n",
    "    ind2 = tidy_splicing_with_expression.sample_id == sample2\n",
    "    \n",
    "    psi1 = tidy_splicing_with_expression.ix[ind1, 'psi']\n",
    "    psi2 = tidy_splicing_with_expression.ix[ind2, 'psi']\n",
    "    delta_psi = (psi1 - psi2).dropna()\n",
    "\n",
    "    # Plot sample1 and set up the jointgrid\n",
    "    ind = tidy_splicing_with_expression.sample_id == sample1\n",
    "    y = np.log2(tidy_splicing_with_expression.ix[ind, 'expression']+1)\n",
    "    x, y = delta_psi.align(y, 'inner')\n",
    "    color = sample_colors[sample1]\n",
    "    g = sns.jointplot(x, y, color=color, joint_kws=dict(alpha=0.5, linewidth=1, edgecolor='white', label=sample1, \n",
    "                                                        rasterized=True, s=40), \n",
    "                      xlim=(-1, 1), ylim=(0, 14), size=4,\n",
    "                     stat_func=None)\n",
    "    \n",
    "    # Plot sample 2 using existing jointgrid\n",
    "    ind = tidy_splicing_with_expression.sample_id == sample2\n",
    "    y = np.log2(tidy_splicing_with_expression.ix[ind, 'expression']+1)\n",
    "    x, y = delta_psi.align(y, 'inner')\n",
    "    color = sample_colors[sample2] \n",
    "    g.x, g.y = x, y\n",
    "    g.plot_joint(plt.scatter, color=color, alpha=0.5, label=sample2, linewidth=1, edgecolor='white', rasterized=True,\n",
    "                s=40)\n",
    "    g.plot_marginals(sns.distplot, color=color)\n",
    "    g.ax_marg_x.set_xlabel('')\n",
    "    g.ax_marg_y.set_ylabel('')\n",
    "\n",
    "    g.ax_joint.legend(loc=\"upper left\", fontsize=10)\n",
    "#     legend = g.ax_joint.legend(**{'bbox_to_anchor': (1, 0.5), 'loc': 'center left'})\n",
    "    g.ax_joint.set_xlabel('$\\Delta \\Psi$')\n",
    "    g.ax_joint.set_ylabel('$\\\\log_2{{TPM+1)}}$')\n",
    "#     ax.set_xlim(-1, 1)\n",
    "#     ymin, ymax = ax.get_ylim()\n",
    "#     ax.set_ylim(0, ymax)\n",
    "    sns.despine()\n",
    "    g.savefig('{}/split_delta_psi_vs_tpm_{}_{}.png'.format(folder, sample1, sample2), dpi=300)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Venn diagrams of shared events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "threshold = 0.25\n",
    "\n",
    "for sample1, sample2 in comparisons:\n",
    "    color1 = sample_colors[sample1]\n",
    "    color2 = sample_colors[sample2]\n",
    "    \n",
    "    psi1_original = study.splicing.data_original.ix[sample1].dropna()\n",
    "    psi2_original = study.splicing.data_original.ix[sample2].dropna()\n",
    "    \n",
    "    union = psi1_original.index.union(psi2_original.index)\n",
    "    both = psi1_original.index.intersection(psi2_original.index)\n",
    "    percent_in_both = 100.*both.shape[0]/float(union.shape[0])\n",
    "    \n",
    "    fig, axes = plt.subplots(figsize=(4, 4), nrows=2)\n",
    "    ax = axes[0]\n",
    "    matplotlib_venn.venn2([psi1_original.index, psi2_original.index], set_labels=(sample1, sample2),\n",
    "                          set_colors=(color1, color2), ax=ax)\n",
    "    ax.set_title('{:.2f}% events detected in both'.format(percent_in_both))\n",
    "    \n",
    "    psi1, psi2 = psi1_original.align(psi2_original, 'inner')\n",
    "    diff = psi1-psi2\n",
    "    venn_data = {'11': diff[diff.abs() <= threshold].shape[0], \n",
    "                 '10': diff[diff > threshold].shape[0], \n",
    "                 '01': diff[diff < -threshold].shape[0]} \n",
    "    percent_in_both = 100.*float(venn_data['11'])/sum(venn_data.values())\n",
    "\n",
    "    ax = axes[1]\n",
    "    matplotlib_venn.venn2(venn_data, \n",
    "                          set_labels=('{:.1f}% events $\\Delta \\Psi\\leq {:.2f}$'.format(percent_in_both, threshold), ''),\n",
    "                          set_colors=(color1, color2), ax=ax)\n",
    "\n",
    "    # Add bracket line\n",
    "    connection = mpl.patches.ConnectionPatch(xyA=(0.5, 0.05), xyB=(0.5, 0.96), coordsA='axes fraction', coordsB='axes fraction',\n",
    "                                             axesA=axes[0], axesB=axes[1], arrowstyle='-[,widthB=3.0,lengthB=0.5', \n",
    "                                             edgecolor='darkgrey', linewidth=3, facecolor='darkgrey')\n",
    "    axes[1].add_artist(connection)\n",
    "    fig.tight_layout()\n",
    "    fig.savefig('{}/two_samples_{}_vs_{}_splicing_venn.pdf'.format(folder, sample1, sample2))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
