{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pe_expression.shape (243, 34608)\n",
      "pe_psi.shape (88098, 203)\n",
      "pe_redo_expression.shape (77, 34608)\n",
      "pe_redo_psi.shape (80259, 75)\n",
      "se_expression.shape (35, 34608)\n",
      "se_psi.shape (65950, 35)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "pe_expression = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_v6/combined_output/tpm_genes_with_stressed.csv', index_col=0)\n",
    "# pe_psi = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_v6/combined_output/psi_with_stressed.csv', index_col=[0, 1])\n",
    "\n",
    "pe_redo_expression = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_redo_v1/combined_output/tpm_genes.csv', index_col=0)\n",
    "# pe_redo_psi = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_redo_v1/combined_output/psi.csv', index_col=[0, 1])\n",
    "\n",
    "se_expression = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v6/combined_output/tpm_genes.csv', index_col=0)\n",
    "# se_psi = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v6/combined_output/psi.csv', index_col=[0, 1])\n",
    "\n",
    "\n",
    "print 'pe_expression.shape', pe_expression.shape\n",
    "print 'pe_psi.shape', pe_psi.shape\n",
    "print 'pe_redo_expression.shape', pe_redo_expression.shape\n",
    "print 'pe_redo_psi.shape', pe_redo_psi.shape\n",
    "print 'se_expression.shape', se_expression.shape\n",
    "print 'se_psi.shape', se_psi.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove samples that were redone"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "pe_expression = pe_expression.drop(pe_expression.index.intersection(pe_redo_expression.index))\n",
    "# pe_psi = pe_psi.drop(pe_psi.columns.intersection(pe_redo_psi.columns), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Started job on</th>\n",
       "      <th>Started mapping on</th>\n",
       "      <th>Finished on</th>\n",
       "      <th>Mapping speed, Million of reads per hour</th>\n",
       "      <th>Number of input reads</th>\n",
       "      <th>Average input read length</th>\n",
       "      <th>UNIQUE READS</th>\n",
       "      <th>Uniquely mapped reads number</th>\n",
       "      <th>Uniquely mapped reads %</th>\n",
       "      <th>Average mapped length</th>\n",
       "      <th>...</th>\n",
       "      <th>% of reads mapped to too many loci</th>\n",
       "      <th>UNMAPPED READS</th>\n",
       "      <th>% of reads unmapped: too many mismatches</th>\n",
       "      <th>% of reads unmapped: too short</th>\n",
       "      <th>% of reads unmapped: other</th>\n",
       "      <th>% splices: Annotated (sjdb)</th>\n",
       "      <th>% splices: GT/AG</th>\n",
       "      <th>% splices: GC/AG</th>\n",
       "      <th>% splices: AT/AC</th>\n",
       "      <th>% splices: Non-canonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01_R1</th>\n",
       "      <td>Feb 21 15:54:22</td>\n",
       "      <td>Feb 21 15:55:44</td>\n",
       "      <td>Feb 21 15:57:19</td>\n",
       "      <td>459.82</td>\n",
       "      <td>12134131</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11128688</td>\n",
       "      <td>91.71</td>\n",
       "      <td>98.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>3.10</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.841858</td>\n",
       "      <td>99.126387</td>\n",
       "      <td>0.652647</td>\n",
       "      <td>0.134704</td>\n",
       "      <td>0.086261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02_R1</th>\n",
       "      <td>Feb 21 15:55:43</td>\n",
       "      <td>Feb 21 15:57:11</td>\n",
       "      <td>Feb 21 15:58:51</td>\n",
       "      <td>480.66</td>\n",
       "      <td>13351539</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12219037</td>\n",
       "      <td>91.52</td>\n",
       "      <td>98.13</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>3.96</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.614629</td>\n",
       "      <td>99.020137</td>\n",
       "      <td>0.801086</td>\n",
       "      <td>0.097722</td>\n",
       "      <td>0.081054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03_R1</th>\n",
       "      <td>Feb 21 16:51:58</td>\n",
       "      <td>Feb 21 16:53:51</td>\n",
       "      <td>Feb 21 16:55:31</td>\n",
       "      <td>484.15</td>\n",
       "      <td>13448587</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12210025</td>\n",
       "      <td>90.79</td>\n",
       "      <td>98.07</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4.99</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.596403</td>\n",
       "      <td>99.028733</td>\n",
       "      <td>0.796140</td>\n",
       "      <td>0.101854</td>\n",
       "      <td>0.073272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04_R1</th>\n",
       "      <td>Feb 21 16:45:53</td>\n",
       "      <td>Feb 21 16:47:23</td>\n",
       "      <td>Feb 21 16:48:59</td>\n",
       "      <td>474.71</td>\n",
       "      <td>12658833</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11805790</td>\n",
       "      <td>93.26</td>\n",
       "      <td>98.33</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2.89</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.831275</td>\n",
       "      <td>99.120483</td>\n",
       "      <td>0.711286</td>\n",
       "      <td>0.084708</td>\n",
       "      <td>0.083523</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05_R1</th>\n",
       "      <td>Feb 21 16:50:26</td>\n",
       "      <td>Feb 21 16:52:19</td>\n",
       "      <td>Feb 21 16:54:01</td>\n",
       "      <td>486.68</td>\n",
       "      <td>13789345</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12894337</td>\n",
       "      <td>93.51</td>\n",
       "      <td>98.44</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2.49</td>\n",
       "      <td>0.02</td>\n",
       "      <td>98.605923</td>\n",
       "      <td>99.083942</td>\n",
       "      <td>0.761804</td>\n",
       "      <td>0.089186</td>\n",
       "      <td>0.065069</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Started job on Started mapping on      Finished on  \\\n",
       "CVN_01_R1  Feb 21 15:54:22    Feb 21 15:55:44  Feb 21 15:57:19   \n",
       "CVN_02_R1  Feb 21 15:55:43    Feb 21 15:57:11  Feb 21 15:58:51   \n",
       "CVN_03_R1  Feb 21 16:51:58    Feb 21 16:53:51  Feb 21 16:55:31   \n",
       "CVN_04_R1  Feb 21 16:45:53    Feb 21 16:47:23  Feb 21 16:48:59   \n",
       "CVN_05_R1  Feb 21 16:50:26    Feb 21 16:52:19  Feb 21 16:54:01   \n",
       "\n",
       "           Mapping speed, Million of reads per hour  Number of input reads  \\\n",
       "CVN_01_R1                                    459.82               12134131   \n",
       "CVN_02_R1                                    480.66               13351539   \n",
       "CVN_03_R1                                    484.15               13448587   \n",
       "CVN_04_R1                                    474.71               12658833   \n",
       "CVN_05_R1                                    486.68               13789345   \n",
       "\n",
       "           Average input read length  UNIQUE READS  \\\n",
       "CVN_01_R1                         99           NaN   \n",
       "CVN_02_R1                         99           NaN   \n",
       "CVN_03_R1                         99           NaN   \n",
       "CVN_04_R1                         99           NaN   \n",
       "CVN_05_R1                         99           NaN   \n",
       "\n",
       "           Uniquely mapped reads number  Uniquely mapped reads %  \\\n",
       "CVN_01_R1                      11128688                    91.71   \n",
       "CVN_02_R1                      12219037                    91.52   \n",
       "CVN_03_R1                      12210025                    90.79   \n",
       "CVN_04_R1                      11805790                    93.26   \n",
       "CVN_05_R1                      12894337                    93.51   \n",
       "\n",
       "           Average mapped length            ...             \\\n",
       "CVN_01_R1                  98.27            ...              \n",
       "CVN_02_R1                  98.13            ...              \n",
       "CVN_03_R1                  98.07            ...              \n",
       "CVN_04_R1                  98.33            ...              \n",
       "CVN_05_R1                  98.44            ...              \n",
       "\n",
       "           % of reads mapped to too many loci  UNMAPPED READS  \\\n",
       "CVN_01_R1                                0.03             NaN   \n",
       "CVN_02_R1                                0.04             NaN   \n",
       "CVN_03_R1                                0.03             NaN   \n",
       "CVN_04_R1                                0.04             NaN   \n",
       "CVN_05_R1                                0.03             NaN   \n",
       "\n",
       "           % of reads unmapped: too many mismatches  \\\n",
       "CVN_01_R1                                         0   \n",
       "CVN_02_R1                                         0   \n",
       "CVN_03_R1                                         0   \n",
       "CVN_04_R1                                         0   \n",
       "CVN_05_R1                                         0   \n",
       "\n",
       "           % of reads unmapped: too short  % of reads unmapped: other  \\\n",
       "CVN_01_R1                            3.10                        0.02   \n",
       "CVN_02_R1                            3.96                        0.02   \n",
       "CVN_03_R1                            4.99                        0.02   \n",
       "CVN_04_R1                            2.89                        0.02   \n",
       "CVN_05_R1                            2.49                        0.02   \n",
       "\n",
       "           % splices: Annotated (sjdb)  % splices: GT/AG  % splices: GC/AG  \\\n",
       "CVN_01_R1                    98.841858         99.126387          0.652647   \n",
       "CVN_02_R1                    98.614629         99.020137          0.801086   \n",
       "CVN_03_R1                    98.596403         99.028733          0.796140   \n",
       "CVN_04_R1                    98.831275         99.120483          0.711286   \n",
       "CVN_05_R1                    98.605923         99.083942          0.761804   \n",
       "\n",
       "           % splices: AT/AC  % splices: Non-canonical  \n",
       "CVN_01_R1          0.134704                  0.086261  \n",
       "CVN_02_R1          0.097722                  0.081054  \n",
       "CVN_03_R1          0.101854                  0.073272  \n",
       "CVN_04_R1          0.084708                  0.083523  \n",
       "CVN_05_R1          0.089186                  0.065069  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pe_mapping_stats = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_v6/combined_output/mapping_stats_with_stressed.csv', index_col=0)\n",
    "pe_mapping_stats.head()\n",
    "pe_redo_mapping_stats = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_redo_v1/combined_output/mapping_stats.csv', index_col=0)\n",
    "pe_redo_mapping_stats.head()\n",
    "pe_mapping_stats = pe_mapping_stats.drop(pe_mapping_stats.index.intersection(pe_redo_mapping_stats.index))\n",
    "se_mapping_stats = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v6/combined_output/mapping_stats.csv', index_col=0)\n",
    "se_mapping_stats.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "expression.shape (321, 34608)\n",
      "splicing.shape (295, 91229)\n",
      "mapping_stats.shape (319, 35)\n"
     ]
    }
   ],
   "source": [
    "expression = pd.concat([pe_expression, pe_redo_expression, se_expression])\n",
    "print 'expression.shape', expression.shape\n",
    "\n",
    "# # Concatenate psis on columns instead of rows since they're (n_features, n_samples),\n",
    "# # but then transform the matrix\n",
    "# splicing = pd.concat([pe_psi, pe_redo_psi, se_psi], axis=1).T\n",
    "# print 'splicing.shape', splicing.shape\n",
    "\n",
    "mapping_stats = pd.concat([pe_mapping_stats, pe_redo_mapping_stats, se_mapping_stats])\n",
    "print 'mapping_stats.shape', mapping_stats.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keep only RI and TandemUTR annotations \n",
    "\n",
    "Use splice junctions to calculate percent spliced in for all other event types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>event_name</th>\n",
       "      <th>chr10:101901555:101901876:-@chr10:101899839:101901554:-</th>\n",
       "      <th>chr10:101982148:101982972:-@chr10:101982044:101982147:-</th>\n",
       "      <th>chr10:102024660:102025207:-@chr10:102024445:102024659:-</th>\n",
       "      <th>chr10:102247855-102247784:-@chr10:102247552-102246403:-</th>\n",
       "      <th>chr10:102265958-102265797:-@chr10:102265252-102265118:-</th>\n",
       "      <th>chr10:102577366:102579341:+@chr10:102579342:102579688:+</th>\n",
       "      <th>chr10:102743062-102744114:+@chr10:102744247-102745373:+</th>\n",
       "      <th>chr10:102782142-102781966:-@chr10:102781702-102781555:-</th>\n",
       "      <th>chr10:102783825-102783685:-@chr10:102783367-102783193:-</th>\n",
       "      <th>chr10:103339574-103339430:-@chr10:103339310-103338639:-</th>\n",
       "      <th>...</th>\n",
       "      <th>chrY:15026796-15026894:+@chrY:15026979-15027139:+</th>\n",
       "      <th>chrY:15448215-15447772:-@chrY:15447680-15447443:-</th>\n",
       "      <th>chrY:15448215-15448116:-@chrY:15447680-15447443:-</th>\n",
       "      <th>chrY:15467278-15467255:-@chrY:15466972-15466883:-</th>\n",
       "      <th>chrY:165764-166059:+@chrY:169260-170022:+</th>\n",
       "      <th>chrY:20326874:20327267:-@chrY:20326696:20326873:-</th>\n",
       "      <th>chrY:21163620:21163904:+@chrY:21163905:21164427:+</th>\n",
       "      <th>chrY:21868749-21868680:-@chrY:21868526-21868327:-</th>\n",
       "      <th>chrY:9236076-9236162:+@chrY:9236424-9236561:+</th>\n",
       "      <th>chrY:9365535-9365621:+@chrY:9365883-9366020:+</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01_R1</th>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02_R1</th>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03_R1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.63</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04_R1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05_R1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.37</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 7700 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "event_name  chr10:101901555:101901876:-@chr10:101899839:101901554:-  \\\n",
       "CVN_01_R1                                                0.98         \n",
       "CVN_02_R1                                                0.97         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:101982148:101982972:-@chr10:101982044:101982147:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102024660:102025207:-@chr10:102024445:102024659:-  \\\n",
       "CVN_01_R1                                                0.95         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                0.63         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                0.11         \n",
       "\n",
       "event_name  chr10:102247855-102247784:-@chr10:102247552-102246403:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102265958-102265797:-@chr10:102265252-102265118:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102577366:102579341:+@chr10:102579342:102579688:+  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102743062-102744114:+@chr10:102744247-102745373:+  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102782142-102781966:-@chr10:102781702-102781555:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:102783825-102783685:-@chr10:102783367-102783193:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name  chr10:103339574-103339430:-@chr10:103339310-103338639:-  \\\n",
       "CVN_01_R1                                                 NaN         \n",
       "CVN_02_R1                                                 NaN         \n",
       "CVN_03_R1                                                 NaN         \n",
       "CVN_04_R1                                                 NaN         \n",
       "CVN_05_R1                                                 NaN         \n",
       "\n",
       "event_name                      ...                        \\\n",
       "CVN_01_R1                       ...                         \n",
       "CVN_02_R1                       ...                         \n",
       "CVN_03_R1                       ...                         \n",
       "CVN_04_R1                       ...                         \n",
       "CVN_05_R1                       ...                         \n",
       "\n",
       "event_name  chrY:15026796-15026894:+@chrY:15026979-15027139:+  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                0.05   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 NaN   \n",
       "\n",
       "event_name  chrY:15448215-15447772:-@chrY:15447680-15447443:-  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 0.9   \n",
       "\n",
       "event_name  chrY:15448215-15448116:-@chrY:15447680-15447443:-  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                0.37   \n",
       "\n",
       "event_name  chrY:15467278-15467255:-@chrY:15466972-15466883:-  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 NaN   \n",
       "\n",
       "event_name  chrY:165764-166059:+@chrY:169260-170022:+  \\\n",
       "CVN_01_R1                                        0.13   \n",
       "CVN_02_R1                                        1.00   \n",
       "CVN_03_R1                                         NaN   \n",
       "CVN_04_R1                                         NaN   \n",
       "CVN_05_R1                                         NaN   \n",
       "\n",
       "event_name  chrY:20326874:20327267:-@chrY:20326696:20326873:-  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 NaN   \n",
       "\n",
       "event_name  chrY:21163620:21163904:+@chrY:21163905:21164427:+  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                0.05   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 NaN   \n",
       "\n",
       "event_name  chrY:21868749-21868680:-@chrY:21868526-21868327:-  \\\n",
       "CVN_01_R1                                                 NaN   \n",
       "CVN_02_R1                                                 NaN   \n",
       "CVN_03_R1                                                 NaN   \n",
       "CVN_04_R1                                                 NaN   \n",
       "CVN_05_R1                                                 NaN   \n",
       "\n",
       "event_name  chrY:9236076-9236162:+@chrY:9236424-9236561:+  \\\n",
       "CVN_01_R1                                             NaN   \n",
       "CVN_02_R1                                             NaN   \n",
       "CVN_03_R1                                             NaN   \n",
       "CVN_04_R1                                             NaN   \n",
       "CVN_05_R1                                             NaN   \n",
       "\n",
       "event_name  chrY:9365535-9365621:+@chrY:9365883-9366020:+  \n",
       "CVN_01_R1                                             NaN  \n",
       "CVN_02_R1                                             NaN  \n",
       "CVN_03_R1                                             NaN  \n",
       "CVN_04_R1                                             NaN  \n",
       "CVN_05_R1                                             NaN  \n",
       "\n",
       "[5 rows x 7700 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splicing = splicing.drop(['ALE', 'AFE', 'A5SS', 'A3SS', 'SE', 'MXE'], level=1, axis=1).sort_index()\n",
    "splicing.columns = splicing.columns.droplevel(1)\n",
    "splicing.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove \"R1\" from the sample names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "splicing.index = splicing.index.map(lambda x: '_'.join(x.split('_')[:2]))\n",
    "expression.index = expression.index.map(lambda x: '_'.join(x.split('_')[:2]))\n",
    "mapping_stats.index = mapping_stats.index.map(lambda x: '_'.join(x.split('_')[:2]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Read metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>single</th>\n",
       "      <th>pooled</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>neuron</th>\n",
       "      <th>progenitor</th>\n",
       "      <th>stressed</th>\n",
       "      <th>craig_venter</th>\n",
       "      <th>differentiation_batch</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_31</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_32</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_33</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_34</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_35</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NPC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       single pooled phenotype neuron progenitor stressed craig_venter  \\\n",
       "CVN_31   True  False       NPC  False       True    False         True   \n",
       "CVN_32   True  False       NPC  False       True    False         True   \n",
       "CVN_33   True  False       NPC  False       True    False         True   \n",
       "CVN_34   True  False       NPC  False       True    False         True   \n",
       "CVN_35  False   True       NPC  False       True    False         True   \n",
       "\n",
       "       differentiation_batch split  \n",
       "CVN_31                   NaN   NaN  \n",
       "CVN_32                   NaN   NaN  \n",
       "CVN_33                   NaN   NaN  \n",
       "CVN_34                   NaN   NaN  \n",
       "CVN_35                   NaN   NaN  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names = 'single\tpooled\tphenotype\tneuron\tprogenitor\tstressed\tcraig_venter\tdifferentiation_batch\tsplit'.split()\n",
    "\n",
    "metadata = pd.read_excel('/home/obotvinnik/projects/singlecell_pnms/olgas_metadata_with_stressed.xlsx', \n",
    "                         skip_footer=1,\n",
    "                         index_col=0)\n",
    "metadata.single = metadata.single.astype(bool)\n",
    "metadata = metadata[names]\n",
    "metadata.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['P1_13',\n",
       " 'P1_14',\n",
       " u'P4_0203',\n",
       " u'P4_0405',\n",
       " u'P4_1314',\n",
       " u'N4_010203',\n",
       " u'N4_0708',\n",
       " u'N4_1112',\n",
       " u'N4_1314',\n",
       " u'N5_010203',\n",
       " u'N5_0405',\n",
       " u'N5_070809',\n",
       " u'N5_1112',\n",
       " u'N5_1314',\n",
       " u'M3_040506',\n",
       " u'M3_0809',\n",
       " u'M3_1011',\n",
       " u'M3_1213',\n",
       " u'M4_0102',\n",
       " u'M4_0304',\n",
       " u'M4_0708',\n",
       " u'M4_0910']"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_to_drop = ['P1_13', 'P1_14']\n",
    "samples_to_drop.extend(metadata.index[metadata.index.map(lambda x: len(x.split('_')[1]) > 2)])\n",
    "samples_to_drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[u'P4_0203',\n",
       " u'P4_0405',\n",
       " u'P4_1314',\n",
       " u'N4_010203',\n",
       " u'N4_0708',\n",
       " u'N4_1112',\n",
       " u'N4_1314',\n",
       " u'N5_010203',\n",
       " u'N5_0405',\n",
       " u'N5_070809',\n",
       " u'N5_1112',\n",
       " u'N5_1314',\n",
       " u'M3_040506',\n",
       " u'M3_0809',\n",
       " u'M3_1011',\n",
       " u'M3_1213',\n",
       " u'M4_0102',\n",
       " u'M4_0304',\n",
       " u'M4_0708',\n",
       " u'M4_0910']"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_samples = [x for x in metadata.index if len(x.split('_')[1]) > 2]\n",
    "combined_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove \"redundant\" samples, e.g. M4_02 for M4_0102"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['P4_02',\n",
       " 'P4_03',\n",
       " 'P4_04',\n",
       " 'P4_05',\n",
       " 'P4_13',\n",
       " 'P4_14',\n",
       " 'N4_01',\n",
       " 'N4_02',\n",
       " 'N4_03',\n",
       " 'N4_07',\n",
       " 'N4_08',\n",
       " 'N4_11',\n",
       " 'N4_12',\n",
       " 'N4_13',\n",
       " 'N4_14',\n",
       " 'N5_01',\n",
       " 'N5_02',\n",
       " 'N5_03',\n",
       " 'N5_04',\n",
       " 'N5_05',\n",
       " 'N5_07',\n",
       " 'N5_08',\n",
       " 'N5_09',\n",
       " 'N5_11',\n",
       " 'N5_12',\n",
       " 'N5_13',\n",
       " 'N5_14',\n",
       " 'M3_04',\n",
       " 'M3_05',\n",
       " 'M3_06',\n",
       " 'M3_08',\n",
       " 'M3_09',\n",
       " 'M3_10',\n",
       " 'M3_11',\n",
       " 'M3_12',\n",
       " 'M3_13',\n",
       " 'M4_01',\n",
       " 'M4_02',\n",
       " 'M4_03',\n",
       " 'M4_04',\n",
       " 'M4_07',\n",
       " 'M4_08',\n",
       " 'M4_09',\n",
       " 'M4_10']"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import itertools\n",
    "\n",
    "n = 2\n",
    "redundant_samples = list(itertools.chain(*[['{}_{}'.format(combined_sample.split('_')[0], numbers) \n",
    "                                       for numbers in [combined_sample.split('_')[1][i:i+n] \n",
    "                                                       for i in range(0, len(combined_sample.split('_')[1]), 2)]] \n",
    "                                      for combined_sample in combined_samples]))\n",
    "\n",
    "samples_to_drop.extend(redundant_samples)\n",
    "redundant_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove samples with fewer than 1 million reads mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "s = mapping_stats['Uniquely mapped reads number']\n",
    "samples_to_drop.extend(s[s < 1e6].index)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Don't use any sample that doesn't have splicing data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "samples_to_drop.extend(metadata.index.difference(splicing.index))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['P1_13',\n",
       " 'P1_14',\n",
       " u'P4_0203',\n",
       " u'P4_0405',\n",
       " u'P4_1314',\n",
       " u'N4_010203',\n",
       " u'N4_0708',\n",
       " u'N4_1112',\n",
       " u'N4_1314',\n",
       " u'N5_010203',\n",
       " u'N5_0405',\n",
       " u'N5_070809',\n",
       " u'N5_1112',\n",
       " u'N5_1314',\n",
       " u'M3_040506',\n",
       " u'M3_0809',\n",
       " u'M3_1011',\n",
       " u'M3_1213',\n",
       " u'M4_0102',\n",
       " u'M4_0304',\n",
       " u'M4_0708',\n",
       " u'M4_0910',\n",
       " 'P4_02',\n",
       " 'P4_03',\n",
       " 'P4_04',\n",
       " 'P4_05',\n",
       " 'P4_13',\n",
       " 'P4_14',\n",
       " 'N4_01',\n",
       " 'N4_02',\n",
       " 'N4_03',\n",
       " 'N4_07',\n",
       " 'N4_08',\n",
       " 'N4_11',\n",
       " 'N4_12',\n",
       " 'N4_13',\n",
       " 'N4_14',\n",
       " 'N5_01',\n",
       " 'N5_02',\n",
       " 'N5_03',\n",
       " 'N5_04',\n",
       " 'N5_05',\n",
       " 'N5_07',\n",
       " 'N5_08',\n",
       " 'N5_09',\n",
       " 'N5_11',\n",
       " 'N5_12',\n",
       " 'N5_13',\n",
       " 'N5_14',\n",
       " 'M3_04',\n",
       " 'M3_05',\n",
       " 'M3_06',\n",
       " 'M3_08',\n",
       " 'M3_09',\n",
       " 'M3_10',\n",
       " 'M3_11',\n",
       " 'M3_12',\n",
       " 'M3_13',\n",
       " 'M4_01',\n",
       " 'M4_02',\n",
       " 'M4_03',\n",
       " 'M4_04',\n",
       " 'M4_07',\n",
       " 'M4_08',\n",
       " 'M4_09',\n",
       " 'M4_10',\n",
       " 'M2nd_23',\n",
       " 'M2nd_30',\n",
       " 'M2nd_31',\n",
       " 'M2nd_32',\n",
       " 'M3_10',\n",
       " 'M3_12',\n",
       " 'M3_13',\n",
       " 'M4_07',\n",
       " 'MSA_25',\n",
       " 'N5_01',\n",
       " 'N5_03',\n",
       " 'N5_05',\n",
       " 'N5_08',\n",
       " 'N5_11',\n",
       " 'N5_12',\n",
       " 'N5_14',\n",
       " 'P8_07',\n",
       " 'P8_09',\n",
       " 'P8_12',\n",
       " 'N3_08',\n",
       " 'CVN_10',\n",
       " u'M2nd_31',\n",
       " u'M3_05',\n",
       " u'M3_06',\n",
       " u'M3_09',\n",
       " u'M3_11',\n",
       " u'M3_13',\n",
       " u'M4_02',\n",
       " u'M4_04',\n",
       " u'M4_08',\n",
       " u'M4_10',\n",
       " u'M6_01',\n",
       " u'MSA_25',\n",
       " u'N3_08',\n",
       " u'P1_13',\n",
       " u'P1_14',\n",
       " u'P4_03',\n",
       " u'P4_05',\n",
       " u'P4_14',\n",
       " u'P8_03',\n",
       " u'P8_04',\n",
       " u'P8_07',\n",
       " u'P8_08',\n",
       " u'P8_09',\n",
       " u'P8_12']"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples_to_drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'CVN_35' in metadata.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/kernel/__main__.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  app.launch_new_instance()\n"
     ]
    }
   ],
   "source": [
    "# Remove a split from the metadata. Was P8_03 but that sequencing sucked.\n",
    "metadata.split['P8_06'] = np.nan\n",
    "metadata.split['P8_03'] = np.nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "to_drop = metadata.index[metadata.index.isin(samples_to_drop)]\n",
    "metadata = metadata.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = splicing.index[splicing.index.isin(samples_to_drop)]\n",
    "splicing = splicing.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = expression.index[expression.index.isin(samples_to_drop)]\n",
    "expression = expression.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = mapping_stats.index[mapping_stats.index.isin(samples_to_drop)]\n",
    "mapping_stats = mapping_stats.drop(to_drop).sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(250, 9)\n",
      "(250, 34608)\n",
      "(250, 7700)\n",
      "(250, 35)\n"
     ]
    }
   ],
   "source": [
    "print metadata.shape\n",
    "print expression.shape\n",
    "print splicing.shape\n",
    "print mapping_stats.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reformat sample ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# import itertools\n",
    "# import numpy as np\n",
    "\n",
    "# new_sample_ids = list(itertools.chain(*itertools.chain(*[[\n",
    "#                     x1 + '_' + pd.Series(np.arange(df2.shape[0])+1).astype(str).map(lambda x: x.zfill(2))\n",
    "#                     if x2 else x1 + '_pooled_' + pd.Series(np.arange(df2.shape[0])+1).astype(str).map(lambda x: x.zfill(2))\n",
    "#                         for x2, df2 in df1.groupby(final_samples.single)]\n",
    "#  for x1, df1 in final_samples_bam_df.groupby(final_samples.phenotype)])))\n",
    "# new_sample_ids"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Separate stressed samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "stressed_samples = [x for x in splicing.index if 'MSA' in x]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>single</th>\n",
       "      <th>pooled</th>\n",
       "      <th>phenotype</th>\n",
       "      <th>neuron</th>\n",
       "      <th>progenitor</th>\n",
       "      <th>stressed</th>\n",
       "      <th>craig_venter</th>\n",
       "      <th>differentiation_batch</th>\n",
       "      <th>split</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MSA_01</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>sMN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_02</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>sMN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_03</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>sMN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_04</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>sMN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_05</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>sMN</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>batch2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       single pooled phenotype neuron progenitor stressed craig_venter  \\\n",
       "MSA_01   True  False       sMN   True      False     True         True   \n",
       "MSA_02   True  False       sMN   True      False     True         True   \n",
       "MSA_03   True  False       sMN   True      False     True         True   \n",
       "MSA_04   True  False       sMN   True      False     True         True   \n",
       "MSA_05   True  False       sMN   True      False     True         True   \n",
       "\n",
       "       differentiation_batch split  \n",
       "MSA_01                batch2   NaN  \n",
       "MSA_02                batch2   NaN  \n",
       "MSA_03                batch2   NaN  \n",
       "MSA_04                batch2   NaN  \n",
       "MSA_05                batch2   NaN  "
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_stressed = metadata.ix[stressed_samples]\n",
    "metadata_stressed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <th>ENSG00000000938</th>\n",
       "      <th>ENSG00000000971</th>\n",
       "      <th>ENSG00000001036</th>\n",
       "      <th>ENSG00000001084</th>\n",
       "      <th>ENSG00000001167</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSGR0000214717</th>\n",
       "      <th>ENSGR0000223511</th>\n",
       "      <th>ENSGR0000223571</th>\n",
       "      <th>ENSGR0000226179</th>\n",
       "      <th>ENSGR0000230542</th>\n",
       "      <th>ENSGR0000234622</th>\n",
       "      <th>ENSGR0000236017</th>\n",
       "      <th>ENSGR0000236871</th>\n",
       "      <th>ENSGR0000237531</th>\n",
       "      <th>ENSGR0000270726</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MSA_01</th>\n",
       "      <td>5.586990</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.466682</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.247240</td>\n",
       "      <td>0.436618</td>\n",
       "      <td>0.486599</td>\n",
       "      <td>0.246384</td>\n",
       "      <td>0.136780</td>\n",
       "      <td>0</td>\n",
       "      <td>0.054551</td>\n",
       "      <td>0.055922</td>\n",
       "      <td>0</td>\n",
       "      <td>0.098818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_02</th>\n",
       "      <td>0.187487</td>\n",
       "      <td>0</td>\n",
       "      <td>142.88720</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.153452</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.28841</td>\n",
       "      <td>...</td>\n",
       "      <td>1.875210</td>\n",
       "      <td>0.323546</td>\n",
       "      <td>0.228916</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.319740</td>\n",
       "      <td>0</td>\n",
       "      <td>0.171625</td>\n",
       "      <td>0.125074</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_03</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.578228</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.238153</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.035220</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_04</th>\n",
       "      <td>13.836200</td>\n",
       "      <td>0</td>\n",
       "      <td>67.18842</td>\n",
       "      <td>0.537852</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.371986</td>\n",
       "      <td>0.180851</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.284970</td>\n",
       "      <td>0.312847</td>\n",
       "      <td>3.358120</td>\n",
       "      <td>0.475807</td>\n",
       "      <td>0</td>\n",
       "      <td>0.028488</td>\n",
       "      <td>0.786396</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_05</th>\n",
       "      <td>24.617500</td>\n",
       "      <td>0</td>\n",
       "      <td>169.56110</td>\n",
       "      <td>0.301238</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.389548</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.32254</td>\n",
       "      <td>...</td>\n",
       "      <td>0.151971</td>\n",
       "      <td>3.342930</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.712829</td>\n",
       "      <td>0.062593</td>\n",
       "      <td>0</td>\n",
       "      <td>0.103309</td>\n",
       "      <td>0.143716</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34608 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ENSG00000000003  ENSG00000000005  ENSG00000000419  ENSG00000000457  \\\n",
       "MSA_01         5.586990                0          0.00000         0.466682   \n",
       "MSA_02         0.187487                0        142.88720         0.000000   \n",
       "MSA_03         0.000000                0          0.00000         0.578228   \n",
       "MSA_04        13.836200                0         67.18842         0.537852   \n",
       "MSA_05        24.617500                0        169.56110         0.301238   \n",
       "\n",
       "        ENSG00000000460  ENSG00000000938  ENSG00000000971  ENSG00000001036  \\\n",
       "MSA_01                0                0                0         0.000000   \n",
       "MSA_02                0                0                0         0.153452   \n",
       "MSA_03                0                0                0         0.000000   \n",
       "MSA_04                0                0                0         0.371986   \n",
       "MSA_05                0                0                0         0.389548   \n",
       "\n",
       "        ENSG00000001084  ENSG00000001167       ...         ENSGR0000214717  \\\n",
       "MSA_01         0.000000          0.00000       ...                0.247240   \n",
       "MSA_02         0.000000          0.28841       ...                1.875210   \n",
       "MSA_03         0.000000          0.00000       ...                0.000000   \n",
       "MSA_04         0.180851          0.00000       ...                0.000000   \n",
       "MSA_05         0.000000          2.32254       ...                0.151971   \n",
       "\n",
       "        ENSGR0000223511  ENSGR0000223571  ENSGR0000226179  ENSGR0000230542  \\\n",
       "MSA_01         0.436618         0.486599         0.246384         0.136780   \n",
       "MSA_02         0.323546         0.228916         0.000000         1.319740   \n",
       "MSA_03         0.238153         0.000000         7.035220         0.000000   \n",
       "MSA_04         2.284970         0.312847         3.358120         0.475807   \n",
       "MSA_05         3.342930         0.000000         0.712829         0.062593   \n",
       "\n",
       "        ENSGR0000234622  ENSGR0000236017  ENSGR0000236871  ENSGR0000237531  \\\n",
       "MSA_01                0         0.054551         0.055922                0   \n",
       "MSA_02                0         0.171625         0.125074                0   \n",
       "MSA_03                0         0.000000         0.000000                0   \n",
       "MSA_04                0         0.028488         0.786396                0   \n",
       "MSA_05                0         0.103309         0.143716                0   \n",
       "\n",
       "        ENSGR0000270726  \n",
       "MSA_01         0.098818  \n",
       "MSA_02         0.000000  \n",
       "MSA_03         0.000000  \n",
       "MSA_04         0.000000  \n",
       "MSA_05         0.000000  \n",
       "\n",
       "[5 rows x 34608 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_stressed = expression.ix[stressed_samples]\n",
    "expression_stressed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>event_name</th>\n",
       "      <th>chr10:101901555:101901876:-@chr10:101899839:101901554:-</th>\n",
       "      <th>chr10:101982148:101982972:-@chr10:101982044:101982147:-</th>\n",
       "      <th>chr10:102024660:102025207:-@chr10:102024445:102024659:-</th>\n",
       "      <th>chr10:102247855-102247784:-@chr10:102247552-102246403:-</th>\n",
       "      <th>chr10:102743062-102744114:+@chr10:102744247-102745373:+</th>\n",
       "      <th>chr10:102783825-102783685:-@chr10:102783367-102783193:-</th>\n",
       "      <th>chr10:103339574-103339430:-@chr10:103339310-103338639:-</th>\n",
       "      <th>chr10:103344676-103344635:-@chr10:103344477-103344359:-</th>\n",
       "      <th>chr10:103912016:103912355:+@chr10:103912356:103913611:+</th>\n",
       "      <th>chr10:104179571-104180056:+@chr10:104180809-104180939:+</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX:86082769-86082815:+@chrX:86087109-86087605:+</th>\n",
       "      <th>chrX:9874977:9876404:+@chrX:9876405:9877478:+</th>\n",
       "      <th>chrY:15026796-15026894:+@chrY:15026979-15027139:+</th>\n",
       "      <th>chrY:15448215-15448116:-@chrY:15447680-15447443:-</th>\n",
       "      <th>chrY:15467278-15467255:-@chrY:15466972-15466883:-</th>\n",
       "      <th>chrY:20326874:20327267:-@chrY:20326696:20326873:-</th>\n",
       "      <th>chrY:21163620:21163904:+@chrY:21163905:21164427:+</th>\n",
       "      <th>chrY:21868749-21868680:-@chrY:21868526-21868327:-</th>\n",
       "      <th>chrY:9236076-9236162:+@chrY:9236424-9236561:+</th>\n",
       "      <th>chrY:9365535-9365621:+@chrY:9365883-9366020:+</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MSA_01</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.38</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_02</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_03</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.71</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_05</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_06</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.63</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_07</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_08</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_09</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.52</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.73</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_12</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_13</th>\n",
       "      <td>0.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_14</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.66</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_16</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_18</th>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_19</th>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81</td>\n",
       "      <td>...</td>\n",
       "      <td>0.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_20</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_21</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_22</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.65</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_23</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.69</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_26</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_27</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_28</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.48</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_29</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_31</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_32</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_33</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_35</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.76</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_36</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_37</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.49</td>\n",
       "      <td>0.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36 rows × 5586 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "event_name  chr10:101901555:101901876:-@chr10:101899839:101901554:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                   0.89         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                   0.99         \n",
       "MSA_19                                                   0.95         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:101982148:101982972:-@chr10:101982044:101982147:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                   0.89         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                   0.94         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                   0.97         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                   0.92         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                   0.88         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:102024660:102025207:-@chr10:102024445:102024659:-  \\\n",
       "MSA_01                                                   0.38         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                   0.92         \n",
       "MSA_05                                                   0.42         \n",
       "MSA_06                                                   0.63         \n",
       "MSA_07                                                   0.05         \n",
       "MSA_08                                                   0.53         \n",
       "MSA_09                                                   0.52         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                   0.08         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                   0.89         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                   0.66         \n",
       "MSA_16                                                   0.01         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                   0.08         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                   0.01         \n",
       "MSA_21                                                   0.21         \n",
       "MSA_22                                                   0.65         \n",
       "MSA_23                                                   0.69         \n",
       "MSA_24                                                   0.01         \n",
       "MSA_26                                                   0.32         \n",
       "MSA_27                                                   0.00         \n",
       "MSA_28                                                   0.48         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                   0.81         \n",
       "MSA_31                                                   0.00         \n",
       "MSA_32                                                   0.01         \n",
       "MSA_33                                                   0.01         \n",
       "MSA_34                                                   0.00         \n",
       "MSA_35                                                   0.76         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                   0.49         \n",
       "\n",
       "event_name  chr10:102247855-102247784:-@chr10:102247552-102246403:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                   0.88         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                   0.09         \n",
       "\n",
       "event_name  chr10:102743062-102744114:+@chr10:102744247-102745373:+  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                   0.00         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                   0.01         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                   0.10         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:102783825-102783685:-@chr10:102783367-102783193:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                   0.99         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:103339574-103339430:-@chr10:103339310-103338639:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                   0.99         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                   1.00         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:103344676-103344635:-@chr10:103344477-103344359:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                   0.97         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                   0.96         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:103912016:103912355:+@chr10:103912356:103913611:+  \\\n",
       "MSA_01                                                   0.72         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                   0.71         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                   1.00         \n",
       "MSA_08                                                   1.00         \n",
       "MSA_09                                                   0.73         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                   0.99         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                   0.94         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                   1.00         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                   0.19         \n",
       "MSA_21                                                   0.47         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                   1.00         \n",
       "MSA_26                                                   1.00         \n",
       "MSA_27                                                   0.98         \n",
       "MSA_28                                                   0.96         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                   0.13         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                   1.00         \n",
       "MSA_35                                                   0.89         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:104179571-104180056:+@chr10:104180809-104180939:+  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                    NaN         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                   0.81         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name                      ...                        \\\n",
       "MSA_01                          ...                         \n",
       "MSA_02                          ...                         \n",
       "MSA_03                          ...                         \n",
       "MSA_04                          ...                         \n",
       "MSA_05                          ...                         \n",
       "MSA_06                          ...                         \n",
       "MSA_07                          ...                         \n",
       "MSA_08                          ...                         \n",
       "MSA_09                          ...                         \n",
       "MSA_10                          ...                         \n",
       "MSA_11                          ...                         \n",
       "MSA_12                          ...                         \n",
       "MSA_13                          ...                         \n",
       "MSA_14                          ...                         \n",
       "MSA_15                          ...                         \n",
       "MSA_16                          ...                         \n",
       "MSA_17                          ...                         \n",
       "MSA_18                          ...                         \n",
       "MSA_19                          ...                         \n",
       "MSA_20                          ...                         \n",
       "MSA_21                          ...                         \n",
       "MSA_22                          ...                         \n",
       "MSA_23                          ...                         \n",
       "MSA_24                          ...                         \n",
       "MSA_26                          ...                         \n",
       "MSA_27                          ...                         \n",
       "MSA_28                          ...                         \n",
       "MSA_29                          ...                         \n",
       "MSA_30                          ...                         \n",
       "MSA_31                          ...                         \n",
       "MSA_32                          ...                         \n",
       "MSA_33                          ...                         \n",
       "MSA_34                          ...                         \n",
       "MSA_35                          ...                         \n",
       "MSA_36                          ...                         \n",
       "MSA_37                          ...                         \n",
       "\n",
       "event_name  chrX:86082769-86082815:+@chrX:86087109-86087605:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                   1.00   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                   0.01   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                   0.94   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:9874977:9876404:+@chrX:9876405:9877478:+  \\\n",
       "MSA_01                                                NaN   \n",
       "MSA_02                                                NaN   \n",
       "MSA_03                                               1.00   \n",
       "MSA_04                                                NaN   \n",
       "MSA_05                                                NaN   \n",
       "MSA_06                                                NaN   \n",
       "MSA_07                                                NaN   \n",
       "MSA_08                                                NaN   \n",
       "MSA_09                                               0.01   \n",
       "MSA_10                                                NaN   \n",
       "MSA_11                                                NaN   \n",
       "MSA_12                                                NaN   \n",
       "MSA_13                                                NaN   \n",
       "MSA_14                                                NaN   \n",
       "MSA_15                                                NaN   \n",
       "MSA_16                                                NaN   \n",
       "MSA_17                                                NaN   \n",
       "MSA_18                                                NaN   \n",
       "MSA_19                                                NaN   \n",
       "MSA_20                                                NaN   \n",
       "MSA_21                                               1.00   \n",
       "MSA_22                                                NaN   \n",
       "MSA_23                                                NaN   \n",
       "MSA_24                                                NaN   \n",
       "MSA_26                                                NaN   \n",
       "MSA_27                                                NaN   \n",
       "MSA_28                                               0.88   \n",
       "MSA_29                                                NaN   \n",
       "MSA_30                                                NaN   \n",
       "MSA_31                                                NaN   \n",
       "MSA_32                                                NaN   \n",
       "MSA_33                                                NaN   \n",
       "MSA_34                                                NaN   \n",
       "MSA_35                                                NaN   \n",
       "MSA_36                                                NaN   \n",
       "MSA_37                                                NaN   \n",
       "\n",
       "event_name  chrY:15026796-15026894:+@chrY:15026979-15027139:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.01   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                   0.36   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                   0.99   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                   0.00   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.01   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                   0.00   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                   0.01   \n",
       "MSA_31                                                   0.03   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                   0.01   \n",
       "\n",
       "event_name  chrY:15448215-15448116:-@chrY:15447680-15447443:-  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                   0.98   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                   0.06   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrY:15467278-15467255:-@chrY:15466972-15466883:-  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   0.95   \n",
       "MSA_15                                                   0.99   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrY:20326874:20327267:-@chrY:20326696:20326873:-  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                   0.54   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                   0.99   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.98   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                   0.64   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                   0.91   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                   0.97   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                   0.99   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                   0.81   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrY:21163620:21163904:+@chrY:21163905:21164427:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                   0.02   \n",
       "MSA_06                                                   0.09   \n",
       "MSA_07                                                   0.06   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   0.01   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                   0.70   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                   0.22   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                   0.02   \n",
       "\n",
       "event_name  chrY:21868749-21868680:-@chrY:21868526-21868327:-  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                   1.00   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                   0.04   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrY:9236076-9236162:+@chrY:9236424-9236561:+  \\\n",
       "MSA_01                                                NaN   \n",
       "MSA_02                                                NaN   \n",
       "MSA_03                                                NaN   \n",
       "MSA_04                                                NaN   \n",
       "MSA_05                                                NaN   \n",
       "MSA_06                                                NaN   \n",
       "MSA_07                                                NaN   \n",
       "MSA_08                                                NaN   \n",
       "MSA_09                                                NaN   \n",
       "MSA_10                                                NaN   \n",
       "MSA_11                                                NaN   \n",
       "MSA_12                                                NaN   \n",
       "MSA_13                                                NaN   \n",
       "MSA_14                                                NaN   \n",
       "MSA_15                                                NaN   \n",
       "MSA_16                                                NaN   \n",
       "MSA_17                                                NaN   \n",
       "MSA_18                                                NaN   \n",
       "MSA_19                                               0.84   \n",
       "MSA_20                                                NaN   \n",
       "MSA_21                                                NaN   \n",
       "MSA_22                                                NaN   \n",
       "MSA_23                                                NaN   \n",
       "MSA_24                                                NaN   \n",
       "MSA_26                                                NaN   \n",
       "MSA_27                                                NaN   \n",
       "MSA_28                                                NaN   \n",
       "MSA_29                                                NaN   \n",
       "MSA_30                                                NaN   \n",
       "MSA_31                                                NaN   \n",
       "MSA_32                                                NaN   \n",
       "MSA_33                                                NaN   \n",
       "MSA_34                                                NaN   \n",
       "MSA_35                                                NaN   \n",
       "MSA_36                                                NaN   \n",
       "MSA_37                                                NaN   \n",
       "\n",
       "event_name  chrY:9365535-9365621:+@chrY:9365883-9366020:+  \n",
       "MSA_01                                                NaN  \n",
       "MSA_02                                                NaN  \n",
       "MSA_03                                                NaN  \n",
       "MSA_04                                                NaN  \n",
       "MSA_05                                                NaN  \n",
       "MSA_06                                                NaN  \n",
       "MSA_07                                                NaN  \n",
       "MSA_08                                                NaN  \n",
       "MSA_09                                                NaN  \n",
       "MSA_10                                                NaN  \n",
       "MSA_11                                                NaN  \n",
       "MSA_12                                                NaN  \n",
       "MSA_13                                                NaN  \n",
       "MSA_14                                                NaN  \n",
       "MSA_15                                                NaN  \n",
       "MSA_16                                                NaN  \n",
       "MSA_17                                                NaN  \n",
       "MSA_18                                                NaN  \n",
       "MSA_19                                               0.87  \n",
       "MSA_20                                                NaN  \n",
       "MSA_21                                                NaN  \n",
       "MSA_22                                                NaN  \n",
       "MSA_23                                                NaN  \n",
       "MSA_24                                                NaN  \n",
       "MSA_26                                                NaN  \n",
       "MSA_27                                                NaN  \n",
       "MSA_28                                                NaN  \n",
       "MSA_29                                                NaN  \n",
       "MSA_30                                                NaN  \n",
       "MSA_31                                                NaN  \n",
       "MSA_32                                                NaN  \n",
       "MSA_33                                                NaN  \n",
       "MSA_34                                                NaN  \n",
       "MSA_35                                                NaN  \n",
       "MSA_36                                                NaN  \n",
       "MSA_37                                                NaN  \n",
       "\n",
       "[36 rows x 5586 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "splicing_stressed = splicing.ix[stressed_samples].dropna(how='all', axis=1)\n",
    "splicing_stressed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Started job on</th>\n",
       "      <th>Started mapping on</th>\n",
       "      <th>Finished on</th>\n",
       "      <th>Mapping speed, Million of reads per hour</th>\n",
       "      <th>Number of input reads</th>\n",
       "      <th>Average input read length</th>\n",
       "      <th>UNIQUE READS</th>\n",
       "      <th>Uniquely mapped reads number</th>\n",
       "      <th>Uniquely mapped reads %</th>\n",
       "      <th>Average mapped length</th>\n",
       "      <th>...</th>\n",
       "      <th>% of reads mapped to too many loci</th>\n",
       "      <th>UNMAPPED READS</th>\n",
       "      <th>% of reads unmapped: too many mismatches</th>\n",
       "      <th>% of reads unmapped: too short</th>\n",
       "      <th>% of reads unmapped: other</th>\n",
       "      <th>% splices: Annotated (sjdb)</th>\n",
       "      <th>% splices: GT/AG</th>\n",
       "      <th>% splices: GC/AG</th>\n",
       "      <th>% splices: AT/AC</th>\n",
       "      <th>% splices: Non-canonical</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MSA_01</th>\n",
       "      <td>Feb 14 14:17:08</td>\n",
       "      <td>Feb 14 14:18:42</td>\n",
       "      <td>Feb 14 14:22:57</td>\n",
       "      <td>174.93</td>\n",
       "      <td>12391060</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11501110</td>\n",
       "      <td>92.82</td>\n",
       "      <td>195.64</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>5.25</td>\n",
       "      <td>0.03</td>\n",
       "      <td>97.499616</td>\n",
       "      <td>98.968184</td>\n",
       "      <td>0.824082</td>\n",
       "      <td>0.078761</td>\n",
       "      <td>0.128973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_02</th>\n",
       "      <td>Feb 14 17:19:46</td>\n",
       "      <td>Feb 14 17:22:55</td>\n",
       "      <td>Feb 14 17:27:34</td>\n",
       "      <td>180.24</td>\n",
       "      <td>13968818</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12682700</td>\n",
       "      <td>90.79</td>\n",
       "      <td>195.61</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>7.27</td>\n",
       "      <td>0.00</td>\n",
       "      <td>97.839862</td>\n",
       "      <td>99.064593</td>\n",
       "      <td>0.694267</td>\n",
       "      <td>0.099192</td>\n",
       "      <td>0.141948</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_03</th>\n",
       "      <td>Feb 14 17:14:45</td>\n",
       "      <td>Feb 14 17:16:50</td>\n",
       "      <td>Feb 14 17:19:14</td>\n",
       "      <td>159.53</td>\n",
       "      <td>6381088</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5447794</td>\n",
       "      <td>85.37</td>\n",
       "      <td>195.83</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>12.20</td>\n",
       "      <td>0.04</td>\n",
       "      <td>97.344815</td>\n",
       "      <td>98.967121</td>\n",
       "      <td>0.816992</td>\n",
       "      <td>0.116936</td>\n",
       "      <td>0.098951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_04</th>\n",
       "      <td>Feb 14 17:28:11</td>\n",
       "      <td>Feb 14 17:31:39</td>\n",
       "      <td>Feb 14 17:34:54</td>\n",
       "      <td>169.47</td>\n",
       "      <td>9179776</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8535754</td>\n",
       "      <td>92.98</td>\n",
       "      <td>195.20</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4.44</td>\n",
       "      <td>0.02</td>\n",
       "      <td>97.216173</td>\n",
       "      <td>99.062576</td>\n",
       "      <td>0.643185</td>\n",
       "      <td>0.116036</td>\n",
       "      <td>0.178202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_05</th>\n",
       "      <td>Feb 19 00:57:23</td>\n",
       "      <td>Feb 19 00:59:03</td>\n",
       "      <td>Feb 19 01:02:23</td>\n",
       "      <td>172.99</td>\n",
       "      <td>9610514</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9007508</td>\n",
       "      <td>93.73</td>\n",
       "      <td>195.29</td>\n",
       "      <td>...</td>\n",
       "      <td>0.05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>4.16</td>\n",
       "      <td>0.04</td>\n",
       "      <td>97.934275</td>\n",
       "      <td>98.912125</td>\n",
       "      <td>0.907125</td>\n",
       "      <td>0.094474</td>\n",
       "      <td>0.086276</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Started job on Started mapping on      Finished on  \\\n",
       "MSA_01  Feb 14 14:17:08    Feb 14 14:18:42  Feb 14 14:22:57   \n",
       "MSA_02  Feb 14 17:19:46    Feb 14 17:22:55  Feb 14 17:27:34   \n",
       "MSA_03  Feb 14 17:14:45    Feb 14 17:16:50  Feb 14 17:19:14   \n",
       "MSA_04  Feb 14 17:28:11    Feb 14 17:31:39  Feb 14 17:34:54   \n",
       "MSA_05  Feb 19 00:57:23    Feb 19 00:59:03  Feb 19 01:02:23   \n",
       "\n",
       "        Mapping speed, Million of reads per hour  Number of input reads  \\\n",
       "MSA_01                                    174.93               12391060   \n",
       "MSA_02                                    180.24               13968818   \n",
       "MSA_03                                    159.53                6381088   \n",
       "MSA_04                                    169.47                9179776   \n",
       "MSA_05                                    172.99                9610514   \n",
       "\n",
       "        Average input read length  UNIQUE READS  Uniquely mapped reads number  \\\n",
       "MSA_01                        197           NaN                      11501110   \n",
       "MSA_02                        197           NaN                      12682700   \n",
       "MSA_03                        197           NaN                       5447794   \n",
       "MSA_04                        197           NaN                       8535754   \n",
       "MSA_05                        197           NaN                       9007508   \n",
       "\n",
       "        Uniquely mapped reads %  Average mapped length  \\\n",
       "MSA_01                    92.82                 195.64   \n",
       "MSA_02                    90.79                 195.61   \n",
       "MSA_03                    85.37                 195.83   \n",
       "MSA_04                    92.98                 195.20   \n",
       "MSA_05                    93.73                 195.29   \n",
       "\n",
       "                  ...             % of reads mapped to too many loci  \\\n",
       "MSA_01            ...                                           0.04   \n",
       "MSA_02            ...                                           0.00   \n",
       "MSA_03            ...                                           0.04   \n",
       "MSA_04            ...                                           0.03   \n",
       "MSA_05            ...                                           0.05   \n",
       "\n",
       "        UNMAPPED READS  % of reads unmapped: too many mismatches  \\\n",
       "MSA_01             NaN                                         0   \n",
       "MSA_02             NaN                                         0   \n",
       "MSA_03             NaN                                         0   \n",
       "MSA_04             NaN                                         0   \n",
       "MSA_05             NaN                                         0   \n",
       "\n",
       "        % of reads unmapped: too short  % of reads unmapped: other  \\\n",
       "MSA_01                            5.25                        0.03   \n",
       "MSA_02                            7.27                        0.00   \n",
       "MSA_03                           12.20                        0.04   \n",
       "MSA_04                            4.44                        0.02   \n",
       "MSA_05                            4.16                        0.04   \n",
       "\n",
       "        % splices: Annotated (sjdb)  % splices: GT/AG  % splices: GC/AG  \\\n",
       "MSA_01                    97.499616         98.968184          0.824082   \n",
       "MSA_02                    97.839862         99.064593          0.694267   \n",
       "MSA_03                    97.344815         98.967121          0.816992   \n",
       "MSA_04                    97.216173         99.062576          0.643185   \n",
       "MSA_05                    97.934275         98.912125          0.907125   \n",
       "\n",
       "        % splices: AT/AC  % splices: Non-canonical  \n",
       "MSA_01          0.078761                  0.128973  \n",
       "MSA_02          0.099192                  0.141948  \n",
       "MSA_03          0.116936                  0.098951  \n",
       "MSA_04          0.116036                  0.178202  \n",
       "MSA_05          0.094474                  0.086276  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapping_stats_stressed = mapping_stats.ix[stressed_samples]\n",
    "mapping_stats_stressed.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After all these filters, how many cells are left?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phenotype\n",
       "MN      73\n",
       "NPC     76\n",
       "iPSC    65\n",
       "sMN     36\n",
       "dtype: int64"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.groupby('phenotype').size()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write the files to a safe place"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory `/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper': File exists\r\n"
     ]
    }
   ],
   "source": [
    "mkdir /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "folder = '/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write stressed samples separately"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "metadata_stressed.to_csv('{}/metadata_stressed.csv'.format(folder))\n",
    "expression_stressed.to_csv('{}/expression_stressed.csv'.format(folder))\n",
    "splicing_stressed.to_csv('{}/splicing_stressed.csv'.format(folder))\n",
    "mapping_stats_stressed.to_csv('{}/mapping_stats_stressed.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check retained intron events in stressed motor neurons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "retained_intron_ind = splicing_stressed.columns.map(lambda x: len(x.split('@')) == 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>event_name</th>\n",
       "      <th>chr10:103912016:103912355:+@chr10:103912356:103913611:+</th>\n",
       "      <th>chr10:104622630-104622689:+@chr10:104623487-104624718:+</th>\n",
       "      <th>chr10:118634438:118636054:-@chr10:118634296:118634437:-</th>\n",
       "      <th>chr10:12077440-12077264:-@chr10:12077173-12077058:-</th>\n",
       "      <th>chr10:120785248:120785539:-@chr10:120784543:120785247:-</th>\n",
       "      <th>chr10:3821782-3821539:-@chr10:3818595-3818188:-</th>\n",
       "      <th>chr10:51259236:51259392:+@chr10:51259393:51260740:+</th>\n",
       "      <th>chr10:61221520:61222679:-@chr10:61218525:61221519:-</th>\n",
       "      <th>chr10:69771817:69772034:+@chr10:69772035:69772956:+</th>\n",
       "      <th>chr10:69773751:69775529:-@chr10:69773279:69773750:-</th>\n",
       "      <th>...</th>\n",
       "      <th>chrX:48759510-48759661:+@chrX:48760009-48760072:+</th>\n",
       "      <th>chrX:51805760-51805462:-@chrX:51805347-51805318:-</th>\n",
       "      <th>chrX:51934527-51934825:+@chrX:51934940-51934957:+</th>\n",
       "      <th>chrX:51934527-51934825:+@chrX:51934940-51934969:+</th>\n",
       "      <th>chrX:53980875:53982314:-@chrX:53979836:53980874:-</th>\n",
       "      <th>chrX:54836155-54836241:+@chrX:54836356-54836646:+</th>\n",
       "      <th>chrX:54836155-54836250:+@chrX:54836305-54836646:+</th>\n",
       "      <th>chrX:54836155-54836251:+@chrX:54836314-54836646:+</th>\n",
       "      <th>chrX:54837254-54837337:+@chrX:54837479-54837562:+</th>\n",
       "      <th>chrX:70436651:70436844:+@chrX:70436845:70437743:+</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MSA_01</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.84</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_02</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.96</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_03</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.89</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_05</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>...</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_06</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.87</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_07</th>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_08</th>\n",
       "      <td>1.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.83</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_09</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>...</td>\n",
       "      <td>0.85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_10</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_12</th>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_13</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_14</th>\n",
       "      <td>0.94</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.92</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.84</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_15</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_16</th>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.82</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.97</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.82</td>\n",
       "      <td>0.87</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_17</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.87</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_18</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_19</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_20</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_21</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.92</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_22</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_23</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.89</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.97</td>\n",
       "      <td>...</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_24</th>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_26</th>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.94</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.89</td>\n",
       "      <td>...</td>\n",
       "      <td>0.92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_27</th>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>...</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_28</th>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_29</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.93</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_31</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.97</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_32</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.94</td>\n",
       "      <td>0.94</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_33</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_34</th>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.81</td>\n",
       "      <td>0.96</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_35</th>\n",
       "      <td>0.89</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.86</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>1.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_36</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.96</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.98</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MSA_37</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>0.95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.98</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.91</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36 rows × 417 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "event_name  chr10:103912016:103912355:+@chr10:103912356:103913611:+  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                   1.00         \n",
       "MSA_08                                                   1.00         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                   0.99         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                   0.94         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                   1.00         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                   1.00         \n",
       "MSA_26                                                   1.00         \n",
       "MSA_27                                                   0.98         \n",
       "MSA_28                                                   0.96         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                   1.00         \n",
       "MSA_35                                                   0.89         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:104622630-104622689:+@chr10:104623487-104624718:+  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                    NaN         \n",
       "MSA_04                                                   0.99         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                    NaN         \n",
       "MSA_07                                                   1.00         \n",
       "MSA_08                                                   0.99         \n",
       "MSA_09                                                   1.00         \n",
       "MSA_10                                                   0.94         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                   1.00         \n",
       "MSA_14                                                   1.00         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                   0.97         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                   1.00         \n",
       "MSA_24                                                   1.00         \n",
       "MSA_26                                                    NaN         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                   1.00         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                    NaN         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                   1.00         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:118634438:118636054:-@chr10:118634296:118634437:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                   0.99         \n",
       "MSA_04                                                    NaN         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                   0.97         \n",
       "MSA_07                                                   0.84         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                   0.96         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                    NaN         \n",
       "MSA_14                                                   0.99         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                   0.86         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                   0.99         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                    NaN         \n",
       "MSA_22                                                   0.98         \n",
       "MSA_23                                                   0.89         \n",
       "MSA_24                                                   0.98         \n",
       "MSA_26                                                   0.92         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                    NaN         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                   0.96         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                   0.99         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                   0.97         \n",
       "MSA_36                                                    NaN         \n",
       "MSA_37                                                   0.99         \n",
       "\n",
       "event_name  chr10:12077440-12077264:-@chr10:12077173-12077058:-  \\\n",
       "MSA_01                                                   0.84     \n",
       "MSA_02                                                   0.95     \n",
       "MSA_03                                                    NaN     \n",
       "MSA_04                                                    NaN     \n",
       "MSA_05                                                   0.91     \n",
       "MSA_06                                                   0.98     \n",
       "MSA_07                                                   0.97     \n",
       "MSA_08                                                    NaN     \n",
       "MSA_09                                                   0.99     \n",
       "MSA_10                                                    NaN     \n",
       "MSA_11                                                    NaN     \n",
       "MSA_12                                                    NaN     \n",
       "MSA_13                                                    NaN     \n",
       "MSA_14                                                   0.92     \n",
       "MSA_15                                                    NaN     \n",
       "MSA_16                                                   0.82     \n",
       "MSA_17                                                   0.99     \n",
       "MSA_18                                                   0.99     \n",
       "MSA_19                                                    NaN     \n",
       "MSA_20                                                   0.97     \n",
       "MSA_21                                                   0.97     \n",
       "MSA_22                                                   0.98     \n",
       "MSA_23                                                   0.95     \n",
       "MSA_24                                                   0.99     \n",
       "MSA_26                                                   0.94     \n",
       "MSA_27                                                    NaN     \n",
       "MSA_28                                                   0.88     \n",
       "MSA_29                                                    NaN     \n",
       "MSA_30                                                   0.99     \n",
       "MSA_31                                                    NaN     \n",
       "MSA_32                                                   0.90     \n",
       "MSA_33                                                    NaN     \n",
       "MSA_34                                                   0.96     \n",
       "MSA_35                                                   0.99     \n",
       "MSA_36                                                    NaN     \n",
       "MSA_37                                                   0.95     \n",
       "\n",
       "event_name  chr10:120785248:120785539:-@chr10:120784543:120785247:-  \\\n",
       "MSA_01                                                    NaN         \n",
       "MSA_02                                                    NaN         \n",
       "MSA_03                                                   0.99         \n",
       "MSA_04                                                   0.97         \n",
       "MSA_05                                                    NaN         \n",
       "MSA_06                                                   0.87         \n",
       "MSA_07                                                    NaN         \n",
       "MSA_08                                                    NaN         \n",
       "MSA_09                                                    NaN         \n",
       "MSA_10                                                    NaN         \n",
       "MSA_11                                                    NaN         \n",
       "MSA_12                                                    NaN         \n",
       "MSA_13                                                   0.99         \n",
       "MSA_14                                                   1.00         \n",
       "MSA_15                                                    NaN         \n",
       "MSA_16                                                    NaN         \n",
       "MSA_17                                                    NaN         \n",
       "MSA_18                                                    NaN         \n",
       "MSA_19                                                    NaN         \n",
       "MSA_20                                                    NaN         \n",
       "MSA_21                                                   0.92         \n",
       "MSA_22                                                    NaN         \n",
       "MSA_23                                                    NaN         \n",
       "MSA_24                                                    NaN         \n",
       "MSA_26                                                   1.00         \n",
       "MSA_27                                                    NaN         \n",
       "MSA_28                                                   0.98         \n",
       "MSA_29                                                    NaN         \n",
       "MSA_30                                                    NaN         \n",
       "MSA_31                                                    NaN         \n",
       "MSA_32                                                    NaN         \n",
       "MSA_33                                                   0.97         \n",
       "MSA_34                                                    NaN         \n",
       "MSA_35                                                    NaN         \n",
       "MSA_36                                                   0.96         \n",
       "MSA_37                                                    NaN         \n",
       "\n",
       "event_name  chr10:3821782-3821539:-@chr10:3818595-3818188:-  \\\n",
       "MSA_01                                                 0.99   \n",
       "MSA_02                                                 1.00   \n",
       "MSA_03                                                  NaN   \n",
       "MSA_04                                                 0.97   \n",
       "MSA_05                                                 0.99   \n",
       "MSA_06                                                 1.00   \n",
       "MSA_07                                                 1.00   \n",
       "MSA_08                                                  NaN   \n",
       "MSA_09                                                 0.82   \n",
       "MSA_10                                                  NaN   \n",
       "MSA_11                                                 0.90   \n",
       "MSA_12                                                  NaN   \n",
       "MSA_13                                                 0.92   \n",
       "MSA_14                                                 0.84   \n",
       "MSA_15                                                  NaN   \n",
       "MSA_16                                                 1.00   \n",
       "MSA_17                                                 0.96   \n",
       "MSA_18                                                 1.00   \n",
       "MSA_19                                                  NaN   \n",
       "MSA_20                                                 0.98   \n",
       "MSA_21                                                 1.00   \n",
       "MSA_22                                                  NaN   \n",
       "MSA_23                                                 0.91   \n",
       "MSA_24                                                 1.00   \n",
       "MSA_26                                                 1.00   \n",
       "MSA_27                                                 0.98   \n",
       "MSA_28                                                 1.00   \n",
       "MSA_29                                                  NaN   \n",
       "MSA_30                                                 1.00   \n",
       "MSA_31                                                 1.00   \n",
       "MSA_32                                                 0.94   \n",
       "MSA_33                                                  NaN   \n",
       "MSA_34                                                 1.00   \n",
       "MSA_35                                                 0.99   \n",
       "MSA_36                                                  NaN   \n",
       "MSA_37                                                 0.99   \n",
       "\n",
       "event_name  chr10:51259236:51259392:+@chr10:51259393:51260740:+  \\\n",
       "MSA_01                                                    NaN     \n",
       "MSA_02                                                   0.99     \n",
       "MSA_03                                                    NaN     \n",
       "MSA_04                                                   1.00     \n",
       "MSA_05                                                    NaN     \n",
       "MSA_06                                                    NaN     \n",
       "MSA_07                                                   0.98     \n",
       "MSA_08                                                   1.00     \n",
       "MSA_09                                                   0.82     \n",
       "MSA_10                                                   0.99     \n",
       "MSA_11                                                    NaN     \n",
       "MSA_12                                                    NaN     \n",
       "MSA_13                                                    NaN     \n",
       "MSA_14                                                    NaN     \n",
       "MSA_15                                                    NaN     \n",
       "MSA_16                                                   0.95     \n",
       "MSA_17                                                    NaN     \n",
       "MSA_18                                                    NaN     \n",
       "MSA_19                                                    NaN     \n",
       "MSA_20                                                    NaN     \n",
       "MSA_21                                                    NaN     \n",
       "MSA_22                                                    NaN     \n",
       "MSA_23                                                   0.96     \n",
       "MSA_24                                                    NaN     \n",
       "MSA_26                                                    NaN     \n",
       "MSA_27                                                    NaN     \n",
       "MSA_28                                                    NaN     \n",
       "MSA_29                                                    NaN     \n",
       "MSA_30                                                   1.00     \n",
       "MSA_31                                                   0.96     \n",
       "MSA_32                                                   0.94     \n",
       "MSA_33                                                    NaN     \n",
       "MSA_34                                                   0.96     \n",
       "MSA_35                                                   1.00     \n",
       "MSA_36                                                    NaN     \n",
       "MSA_37                                                    NaN     \n",
       "\n",
       "event_name  chr10:61221520:61222679:-@chr10:61218525:61221519:-  \\\n",
       "MSA_01                                                    NaN     \n",
       "MSA_02                                                   1.00     \n",
       "MSA_03                                                    NaN     \n",
       "MSA_04                                                    NaN     \n",
       "MSA_05                                                    NaN     \n",
       "MSA_06                                                    NaN     \n",
       "MSA_07                                                    NaN     \n",
       "MSA_08                                                    NaN     \n",
       "MSA_09                                                   0.99     \n",
       "MSA_10                                                    NaN     \n",
       "MSA_11                                                   1.00     \n",
       "MSA_12                                                   1.00     \n",
       "MSA_13                                                    NaN     \n",
       "MSA_14                                                    NaN     \n",
       "MSA_15                                                   1.00     \n",
       "MSA_16                                                    NaN     \n",
       "MSA_17                                                    NaN     \n",
       "MSA_18                                                    NaN     \n",
       "MSA_19                                                    NaN     \n",
       "MSA_20                                                    NaN     \n",
       "MSA_21                                                    NaN     \n",
       "MSA_22                                                    NaN     \n",
       "MSA_23                                                    NaN     \n",
       "MSA_24                                                    NaN     \n",
       "MSA_26                                                   1.00     \n",
       "MSA_27                                                   1.00     \n",
       "MSA_28                                                    NaN     \n",
       "MSA_29                                                    NaN     \n",
       "MSA_30                                                    NaN     \n",
       "MSA_31                                                   1.00     \n",
       "MSA_32                                                    NaN     \n",
       "MSA_33                                                   0.98     \n",
       "MSA_34                                                    NaN     \n",
       "MSA_35                                                   0.97     \n",
       "MSA_36                                                    NaN     \n",
       "MSA_37                                                    NaN     \n",
       "\n",
       "event_name  chr10:69771817:69772034:+@chr10:69772035:69772956:+  \\\n",
       "MSA_01                                                    NaN     \n",
       "MSA_02                                                   0.81     \n",
       "MSA_03                                                    NaN     \n",
       "MSA_04                                                    NaN     \n",
       "MSA_05                                                   1.00     \n",
       "MSA_06                                                   0.99     \n",
       "MSA_07                                                    NaN     \n",
       "MSA_08                                                   0.99     \n",
       "MSA_09                                                    NaN     \n",
       "MSA_10                                                   0.97     \n",
       "MSA_11                                                    NaN     \n",
       "MSA_12                                                    NaN     \n",
       "MSA_13                                                    NaN     \n",
       "MSA_14                                                   1.00     \n",
       "MSA_15                                                   0.93     \n",
       "MSA_16                                                   0.94     \n",
       "MSA_17                                                   1.00     \n",
       "MSA_18                                                    NaN     \n",
       "MSA_19                                                    NaN     \n",
       "MSA_20                                                    NaN     \n",
       "MSA_21                                                    NaN     \n",
       "MSA_22                                                   0.99     \n",
       "MSA_23                                                   0.99     \n",
       "MSA_24                                                   0.93     \n",
       "MSA_26                                                   1.00     \n",
       "MSA_27                                                    NaN     \n",
       "MSA_28                                                    NaN     \n",
       "MSA_29                                                    NaN     \n",
       "MSA_30                                                    NaN     \n",
       "MSA_31                                                   0.99     \n",
       "MSA_32                                                   0.97     \n",
       "MSA_33                                                    NaN     \n",
       "MSA_34                                                   0.81     \n",
       "MSA_35                                                   0.98     \n",
       "MSA_36                                                    NaN     \n",
       "MSA_37                                                   1.00     \n",
       "\n",
       "event_name  chr10:69773751:69775529:-@chr10:69773279:69773750:-  \\\n",
       "MSA_01                                                    NaN     \n",
       "MSA_02                                                   0.96     \n",
       "MSA_03                                                   0.97     \n",
       "MSA_04                                                    NaN     \n",
       "MSA_05                                                   0.98     \n",
       "MSA_06                                                   0.99     \n",
       "MSA_07                                                    NaN     \n",
       "MSA_08                                                    NaN     \n",
       "MSA_09                                                   0.97     \n",
       "MSA_10                                                    NaN     \n",
       "MSA_11                                                    NaN     \n",
       "MSA_12                                                    NaN     \n",
       "MSA_13                                                    NaN     \n",
       "MSA_14                                                    NaN     \n",
       "MSA_15                                                    NaN     \n",
       "MSA_16                                                   0.97     \n",
       "MSA_17                                                    NaN     \n",
       "MSA_18                                                   0.98     \n",
       "MSA_19                                                    NaN     \n",
       "MSA_20                                                   0.96     \n",
       "MSA_21                                                    NaN     \n",
       "MSA_22                                                    NaN     \n",
       "MSA_23                                                   0.97     \n",
       "MSA_24                                                    NaN     \n",
       "MSA_26                                                   0.89     \n",
       "MSA_27                                                   1.00     \n",
       "MSA_28                                                    NaN     \n",
       "MSA_29                                                    NaN     \n",
       "MSA_30                                                   0.92     \n",
       "MSA_31                                                    NaN     \n",
       "MSA_32                                                    NaN     \n",
       "MSA_33                                                    NaN     \n",
       "MSA_34                                                   0.96     \n",
       "MSA_35                                                    NaN     \n",
       "MSA_36                                                    NaN     \n",
       "MSA_37                                                   0.98     \n",
       "\n",
       "event_name                        ...                          \\\n",
       "MSA_01                            ...                           \n",
       "MSA_02                            ...                           \n",
       "MSA_03                            ...                           \n",
       "MSA_04                            ...                           \n",
       "MSA_05                            ...                           \n",
       "MSA_06                            ...                           \n",
       "MSA_07                            ...                           \n",
       "MSA_08                            ...                           \n",
       "MSA_09                            ...                           \n",
       "MSA_10                            ...                           \n",
       "MSA_11                            ...                           \n",
       "MSA_12                            ...                           \n",
       "MSA_13                            ...                           \n",
       "MSA_14                            ...                           \n",
       "MSA_15                            ...                           \n",
       "MSA_16                            ...                           \n",
       "MSA_17                            ...                           \n",
       "MSA_18                            ...                           \n",
       "MSA_19                            ...                           \n",
       "MSA_20                            ...                           \n",
       "MSA_21                            ...                           \n",
       "MSA_22                            ...                           \n",
       "MSA_23                            ...                           \n",
       "MSA_24                            ...                           \n",
       "MSA_26                            ...                           \n",
       "MSA_27                            ...                           \n",
       "MSA_28                            ...                           \n",
       "MSA_29                            ...                           \n",
       "MSA_30                            ...                           \n",
       "MSA_31                            ...                           \n",
       "MSA_32                            ...                           \n",
       "MSA_33                            ...                           \n",
       "MSA_34                            ...                           \n",
       "MSA_35                            ...                           \n",
       "MSA_36                            ...                           \n",
       "MSA_37                            ...                           \n",
       "\n",
       "event_name  chrX:48759510-48759661:+@chrX:48760009-48760072:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                   0.98   \n",
       "MSA_05                                                   0.97   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.97   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                   0.85   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                   0.99   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                   0.99   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.96   \n",
       "MSA_22                                                   0.99   \n",
       "MSA_23                                                   0.96   \n",
       "MSA_24                                                   0.99   \n",
       "MSA_26                                                   0.92   \n",
       "MSA_27                                                   0.93   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                   0.96   \n",
       "MSA_31                                                   0.97   \n",
       "MSA_32                                                   0.98   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                   0.86   \n",
       "MSA_36                                                   0.99   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:51805760-51805462:-@chrX:51805347-51805318:-  \\\n",
       "MSA_01                                                   1.00   \n",
       "MSA_02                                                   1.00   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.98   \n",
       "MSA_08                                                   0.83   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                   0.96   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                   0.96   \n",
       "MSA_14                                                   0.96   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.82   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                   0.97   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                   0.96   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.99   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                   0.96   \n",
       "MSA_33                                                   0.97   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:51934527-51934825:+@chrX:51934940-51934957:+  \\\n",
       "MSA_01                                                   1.00   \n",
       "MSA_02                                                   1.00   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                   0.89   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.97   \n",
       "MSA_08                                                   0.85   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                   0.95   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                   0.95   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.87   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                   0.96   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                   0.96   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.99   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                   0.96   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:51934527-51934825:+@chrX:51934940-51934969:+  \\\n",
       "MSA_01                                                   1.00   \n",
       "MSA_02                                                   1.00   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                   0.89   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.98   \n",
       "MSA_08                                                   0.82   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                   0.96   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                   0.96   \n",
       "MSA_14                                                   0.95   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.85   \n",
       "MSA_17                                                    NaN   \n",
       "MSA_18                                                   0.96   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                    NaN   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.99   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                    NaN   \n",
       "MSA_32                                                   0.95   \n",
       "MSA_33                                                   0.97   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:53980875:53982314:-@chrX:53979836:53980874:-  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                    NaN   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                   1.00   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                    NaN   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                    NaN   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.91   \n",
       "MSA_17                                                   0.87   \n",
       "MSA_18                                                   0.97   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.99   \n",
       "MSA_22                                                    NaN   \n",
       "MSA_23                                                   1.00   \n",
       "MSA_24                                                   1.00   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                   0.99   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                   0.99   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                   0.91   \n",
       "MSA_35                                                    NaN   \n",
       "MSA_36                                                    NaN   \n",
       "MSA_37                                                   0.91   \n",
       "\n",
       "event_name  chrX:54836155-54836241:+@chrX:54836356-54836646:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                   0.97   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   1.00   \n",
       "MSA_08                                                   1.00   \n",
       "MSA_09                                                   0.99   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   1.00   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.93   \n",
       "MSA_17                                                   1.00   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.99   \n",
       "MSA_22                                                   1.00   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.99   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                   0.91   \n",
       "MSA_31                                                   0.96   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                   1.00   \n",
       "MSA_35                                                   1.00   \n",
       "MSA_36                                                   0.98   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:54836155-54836250:+@chrX:54836305-54836646:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                   0.93   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   0.96   \n",
       "MSA_08                                                   0.95   \n",
       "MSA_09                                                   0.97   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   0.99   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                    NaN   \n",
       "MSA_17                                                   0.97   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.98   \n",
       "MSA_22                                                   0.97   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.98   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                   0.90   \n",
       "MSA_31                                                   0.97   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                   0.97   \n",
       "MSA_35                                                   0.98   \n",
       "MSA_36                                                   0.90   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:54836155-54836251:+@chrX:54836314-54836646:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                   0.96   \n",
       "MSA_04                                                    NaN   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                   1.00   \n",
       "MSA_08                                                   1.00   \n",
       "MSA_09                                                   0.99   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   1.00   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.92   \n",
       "MSA_17                                                   1.00   \n",
       "MSA_18                                                    NaN   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   0.99   \n",
       "MSA_22                                                   1.00   \n",
       "MSA_23                                                    NaN   \n",
       "MSA_24                                                    NaN   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                   0.99   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                   0.93   \n",
       "MSA_31                                                   0.97   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                    NaN   \n",
       "MSA_35                                                   1.00   \n",
       "MSA_36                                                   0.98   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:54837254-54837337:+@chrX:54837479-54837562:+  \\\n",
       "MSA_01                                                    NaN   \n",
       "MSA_02                                                    NaN   \n",
       "MSA_03                                                   0.99   \n",
       "MSA_04                                                   1.00   \n",
       "MSA_05                                                    NaN   \n",
       "MSA_06                                                    NaN   \n",
       "MSA_07                                                    NaN   \n",
       "MSA_08                                                   1.00   \n",
       "MSA_09                                                    NaN   \n",
       "MSA_10                                                    NaN   \n",
       "MSA_11                                                    NaN   \n",
       "MSA_12                                                    NaN   \n",
       "MSA_13                                                    NaN   \n",
       "MSA_14                                                   1.00   \n",
       "MSA_15                                                    NaN   \n",
       "MSA_16                                                   0.98   \n",
       "MSA_17                                                   1.00   \n",
       "MSA_18                                                   0.88   \n",
       "MSA_19                                                    NaN   \n",
       "MSA_20                                                    NaN   \n",
       "MSA_21                                                   1.00   \n",
       "MSA_22                                                   1.00   \n",
       "MSA_23                                                   1.00   \n",
       "MSA_24                                                   1.00   \n",
       "MSA_26                                                    NaN   \n",
       "MSA_27                                                    NaN   \n",
       "MSA_28                                                    NaN   \n",
       "MSA_29                                                    NaN   \n",
       "MSA_30                                                    NaN   \n",
       "MSA_31                                                   1.00   \n",
       "MSA_32                                                    NaN   \n",
       "MSA_33                                                    NaN   \n",
       "MSA_34                                                   1.00   \n",
       "MSA_35                                                   1.00   \n",
       "MSA_36                                                   0.99   \n",
       "MSA_37                                                    NaN   \n",
       "\n",
       "event_name  chrX:70436651:70436844:+@chrX:70436845:70437743:+  \n",
       "MSA_01                                                   1.00  \n",
       "MSA_02                                                    NaN  \n",
       "MSA_03                                                   1.00  \n",
       "MSA_04                                                   1.00  \n",
       "MSA_05                                                   1.00  \n",
       "MSA_06                                                   1.00  \n",
       "MSA_07                                                   0.95  \n",
       "MSA_08                                                   1.00  \n",
       "MSA_09                                                   1.00  \n",
       "MSA_10                                                    NaN  \n",
       "MSA_11                                                   0.90  \n",
       "MSA_12                                                   1.00  \n",
       "MSA_13                                                   1.00  \n",
       "MSA_14                                                   0.96  \n",
       "MSA_15                                                    NaN  \n",
       "MSA_16                                                   0.95  \n",
       "MSA_17                                                   1.00  \n",
       "MSA_18                                                   0.99  \n",
       "MSA_19                                                    NaN  \n",
       "MSA_20                                                    NaN  \n",
       "MSA_21                                                   1.00  \n",
       "MSA_22                                                   1.00  \n",
       "MSA_23                                                   1.00  \n",
       "MSA_24                                                   1.00  \n",
       "MSA_26                                                   0.97  \n",
       "MSA_27                                                    NaN  \n",
       "MSA_28                                                   1.00  \n",
       "MSA_29                                                   0.88  \n",
       "MSA_30                                                    NaN  \n",
       "MSA_31                                                   1.00  \n",
       "MSA_32                                                    NaN  \n",
       "MSA_33                                                   1.00  \n",
       "MSA_34                                                   1.00  \n",
       "MSA_35                                                    NaN  \n",
       "MSA_36                                                   1.00  \n",
       "MSA_37                                                   1.00  \n",
       "\n",
       "[36 rows x 417 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "retained_intron = splicing_stressed.loc[:, retained_intron_ind]\n",
    "retained_intron[retained_intron > 0.8].dropna(thresh=10, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "$\\Psi\\approx1$ events to check:\n",
    "   \n",
    "    chr10:12077440-12077264:-@chr10:12077173-12077058:-\n",
    "        chr10:12077173-12077264\n",
    "    chr10:104622630-104622689:+@chr10:104623487-104624718\n",
    "        chr10:104622630-104624718"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove stressed samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "samples_to_drop.extend(stressed_samples)\n",
    "to_drop = metadata.index[metadata.index.isin(samples_to_drop)]\n",
    "metadata = metadata.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = splicing.index[splicing.index.isin(samples_to_drop)]\n",
    "splicing = splicing.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = expression.index[expression.index.isin(samples_to_drop)]\n",
    "expression = expression.drop(to_drop).sort_index()\n",
    "\n",
    "to_drop = mapping_stats.index[mapping_stats.index.isin(samples_to_drop)]\n",
    "mapping_stats = mapping_stats.drop(to_drop).sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/kernel/__main__.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Back-add the split samples so we have all the split information\n",
    "splits = metadata.split.dropna()\n",
    "splits_reversed = pd.Series(splits.index, index=splits.values)\n",
    "metadata.split[splits_reversed.index] = splits_reversed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "P1_04    P7_07\n",
       "P1_05    P7_08\n",
       "P1_10    P8_11\n",
       "P2_10    P7_09\n",
       "P4_11    P8_10\n",
       "P7_07    P1_04\n",
       "P7_08    P1_05\n",
       "P7_09    P2_10\n",
       "P8_10    P4_11\n",
       "P8_11    P1_10\n",
       "Name: split, dtype: object"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.split.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "metadata.to_csv('{}/metadata.csv'.format(folder))\n",
    "expression.to_csv('{}/expression.csv'.format(folder))\n",
    "splicing.to_csv('{}/splicing.csv'.format(folder))\n",
    "mapping_stats.to_csv('{}/mapping_stats.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1.3G\r\n",
      "drwxr-xr-x 17 obotvinnik yeo-group  12K May 27 16:14 ..\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 669M Aug 13 08:23 splice_junctions_psi.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  28M Aug 13 09:41 splice_junction_metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  21M Aug 13 09:48 psi5_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  21M Aug 13 09:48 psi3_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  80M Aug 13 09:54 psi5.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  80M Aug 13 09:56 psi3.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 159M Aug 13 09:59 psi5_psi3_combined.csv\r\n",
      "drwxr-xr-x  2 obotvinnik yeo-group 4.0K Aug 13 10:01 .\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 162M Aug 13 10:17 psi5_psi3_combined_metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 1.9K Aug 13 10:45 metadata_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 615K Aug 13 10:45 splicing_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  12K Aug 13 10:45 mapping_stats_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 7.2M Aug 13 10:45 expression_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  11K Aug 13 10:45 metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 3.2M Aug 13 10:45 splicing.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  63K Aug 13 10:45 mapping_stats.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  41M Aug 13 10:45 expression.csv\r\n"
     ]
    }
   ],
   "source": [
    "! ls -lhaStr /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       215 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/expression.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/expression_stressed.csv\n",
      "       215 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/mapping_stats.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/mapping_stats_stressed.csv\n",
      "       215 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/metadata.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/metadata_stressed.csv\n",
      "       190 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi3.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi3_stressed.csv\n",
      "       190 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi5.csv\n",
      "       190 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi5_psi3_combined.csv\n",
      "    577643 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi5_psi3_combined_metadata.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/psi5_stressed.csv\n",
      "    288822 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/splice_junction_metadata.csv\n",
      "   9814465 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/splice_junctions_psi.csv\n",
      "       215 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/splicing.csv\n",
      "        37 /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/splicing_stressed.csv\n",
      "  10682582 total\n"
     ]
    }
   ],
   "source": [
    "! wc -l  /home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper/*.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Read SJ out files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 CVN_09\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/matplotlib/__init__.py:784: UserWarning: Found matplotlib configuration in ~/.matplotlib/. To conform with the XDG base directory standard, this configuration location has been deprecated on Linux, and the new location is now /home/obotvinnik/.config/matplotlib/. Please move your configuration there to ensure that matplotlib will continue to find it in the future.\n",
      "  _get_xdg_config_dir())\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'min_unique_reads' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-3ba25fc97d04>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mget_ipython\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mu'time'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34mu''\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34mu'import glob\\nimport os\\nimport pandas as pd\\nimport sj2psi\\n\\ndirs = {\\'pe\\': \\'/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_v6\\',\\n        \\'pe_redo\\': \\'/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_redo_v1\\',\\n        \\'se\\': \\'/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v6\\'}\\n\\n\\n\\nMIN_UNIQUE_JUNCTION_READS = 10\\nMIN_MULTIMAP_JUNCTION_READS = 20\\n\\n\\n# sj_raw = []\\ndfs = []\\nfor name, DIR in dirs.items():\\n    for i, filename in enumerate(glob.iglob(\"{}/*rmRep*SJ.out.tab\".format(DIR))):\\n        df = sj2psi.read_sj_out_tab(filename)\\n        basename = os.path.basename(filename)\\n        sample_id = \\'_\\'.join(basename.split(\\'_\\')[:2])\\n        print i, sample_id\\n#         min_unique_reads = df.unique_junction_reads >= MIN_UNIQUE_JUNCTION_READS\\n#         min_multimap_reads = df.multimap_junction_reads >= MIN_MULTIMAP_JUNCTION_READS\\n    #     print \\'\\\\t\\', df.shape\\n        df = df.loc[min_unique_reads]\\n        df[\\'sample_id\\'] = sample_id\\n    #     print \\'\\\\t\\', df.shape\\n        df[\\'run_name\\'] = name\\n        dfs.append(df)\\nsj_raw = pd.concat(dfs, ignore_index=True)'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[1;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[0;32m   2291\u001b[0m             \u001b[0mmagic_arg_s\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstack_depth\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2292\u001b[0m             \u001b[1;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2293\u001b[1;33m                 \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2294\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2295\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/core/magics/execution.pyc\u001b[0m in \u001b[0;36mtime\u001b[1;34m(self, line, cell, local_ns)\u001b[0m\n",
      "\u001b[1;32m/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/core/magic.pyc\u001b[0m in \u001b[0;36m<lambda>\u001b[1;34m(f, *a, **k)\u001b[0m\n\u001b[0;32m    191\u001b[0m     \u001b[1;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    192\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0marg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 193\u001b[1;33m         \u001b[0mcall\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m*\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    194\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    195\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0marg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/home/obotvinnik/anaconda/lib/python2.7/site-packages/IPython/core/magics/execution.pyc\u001b[0m in \u001b[0;36mtime\u001b[1;34m(self, line, cell, local_ns)\u001b[0m\n\u001b[0;32m   1165\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1166\u001b[0m             \u001b[0mst\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1167\u001b[1;33m             \u001b[1;32mexec\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mglob\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlocal_ns\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1168\u001b[0m             \u001b[0mend\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mclock2\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1169\u001b[0m             \u001b[0mout\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mNone\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<timed exec>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'min_unique_reads' is not defined"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "import sj2psi\n",
    "\n",
    "dirs = {'pe': '/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_v6',\n",
    "        'pe_redo': '/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_pe_redo_v1',\n",
    "        'se': '/home/obotvinnik/projects/singlecell_pnms/analysis/singlecell_pnms_se_v6'}\n",
    "\n",
    "\n",
    "\n",
    "MIN_UNIQUE_JUNCTION_READS = 10\n",
    "MIN_MULTIMAP_JUNCTION_READS = 20\n",
    "\n",
    "\n",
    "# sj_raw = []\n",
    "dfs = []\n",
    "for name, DIR in dirs.items():\n",
    "    for i, filename in enumerate(glob.iglob(\"{}/*rmRep*SJ.out.tab\".format(DIR))):\n",
    "        df = sj2psi.read_sj_out_tab(filename)\n",
    "        basename = os.path.basename(filename)\n",
    "        sample_id = '_'.join(basename.split('_')[:2])\n",
    "        print i, sample_id\n",
    "#         min_unique_reads = df.unique_junction_reads >= MIN_UNIQUE_JUNCTION_READS\n",
    "#         min_multimap_reads = df.multimap_junction_reads >= MIN_MULTIMAP_JUNCTION_READS\n",
    "    #     print '\\t', df.shape\n",
    "#         df = df.loc[min_unique_reads]\n",
    "        df['sample_id'] = sample_id\n",
    "    #     print '\\t', df.shape\n",
    "        df['run_name'] = name\n",
    "        dfs.append(df)\n",
    "sj_raw = pd.concat(dfs, ignore_index=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id run_name  \n",
       "0  chr1:1309283-1309379:-    CVN_09       se  \n",
       "1  chr1:1309826-1310084:-    CVN_09       se  \n",
       "2  chr1:1310171-1310533:-    CVN_09       se  \n",
       "3  chr1:1337637-1341188:-    CVN_09       se  \n",
       "4  chr1:1341267-1342288:-    CVN_09       se  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7083200, 12)\n",
      "(4581458, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id run_name  \n",
       "0  chr1:1309283-1309379:-    CVN_09       se  \n",
       "1  chr1:1309826-1310084:-    CVN_09       se  \n",
       "2  chr1:1310171-1310533:-    CVN_09       se  \n",
       "3  chr1:1337637-1341188:-    CVN_09       se  \n",
       "4  chr1:1341267-1342288:-    CVN_09       se  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print sj_raw.shape\n",
    "sj_raw = sj_raw.drop_duplicates(['intron_location', 'sample_id'])\n",
    "print sj_raw.shape\n",
    "sj_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remove the samples in `pe_redo` from `pe`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4581458, 12)\n",
      "(4581458, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>run_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "      <td>se</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id run_name  \n",
       "0  chr1:1309283-1309379:-    CVN_09       se  \n",
       "1  chr1:1309826-1310084:-    CVN_09       se  \n",
       "2  chr1:1310171-1310533:-    CVN_09       se  \n",
       "3  chr1:1337637-1341188:-    CVN_09       se  \n",
       "4  chr1:1341267-1342288:-    CVN_09       se  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "redo_ids = sj_raw.loc[sj_raw.run_name == 'pe_redo'].sample_id.unique()\n",
    "to_drop = sj_raw.index[(sj_raw.run_name == 'pe') & (sj_raw.sample_id.isin(redo_ids))]\n",
    "print sj_raw.shape\n",
    "sj = sj_raw.drop(to_drop, errors='ignore')\n",
    "print sj.shape\n",
    "sj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id  \n",
       "0  chr1:1309283-1309379:-    CVN_09  \n",
       "1  chr1:1309826-1310084:-    CVN_09  \n",
       "2  chr1:1310171-1310533:-    CVN_09  \n",
       "3  chr1:1337637-1341188:-    CVN_09  \n",
       "4  chr1:1341267-1342288:-    CVN_09  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj = sj.drop(['run_name'], axis=1, errors='ignore')\n",
    "sj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1128056</th>\n",
       "      <td>chr1</td>\n",
       "      <td>891342</td>\n",
       "      <td>894379</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>chr1:891342-894379:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128059</th>\n",
       "      <td>chr1</td>\n",
       "      <td>995774</td>\n",
       "      <td>1001209</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:995774-1001209:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128060</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1197141</td>\n",
       "      <td>1197351</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>15</td>\n",
       "      <td>3</td>\n",
       "      <td>42</td>\n",
       "      <td>chr1:1197141-1197351:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128069</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1247447</td>\n",
       "      <td>1254819</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>42</td>\n",
       "      <td>chr1:1247447-1254819:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128084</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1463103</td>\n",
       "      <td>1469750</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>49</td>\n",
       "      <td>2</td>\n",
       "      <td>30</td>\n",
       "      <td>chr1:1463103-1469750:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128095</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2323398</td>\n",
       "      <td>2324708</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:2323398-2324708:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128096</th>\n",
       "      <td>chr1</td>\n",
       "      <td>2324795</td>\n",
       "      <td>2327222</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>chr1:2324795-2327222:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128137</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8031024</td>\n",
       "      <td>8044953</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>240</td>\n",
       "      <td>2</td>\n",
       "      <td>44</td>\n",
       "      <td>chr1:8031024-8044953:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128140</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8460443</td>\n",
       "      <td>8460958</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>43</td>\n",
       "      <td>chr1:8460443-8460958:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128151</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8934974</td>\n",
       "      <td>8938638</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>178</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:8934974-8938638:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128178</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11129662</td>\n",
       "      <td>11129718</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>chr1:11129662-11129718:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128201</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11734868</td>\n",
       "      <td>11735138</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>83</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:11734868-11735138:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128210</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11740671</td>\n",
       "      <td>11751808</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>55</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:11740671-11751808:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128212</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11805892</td>\n",
       "      <td>11810758</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>34</td>\n",
       "      <td>chr1:11805892-11810758:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128276</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19585947</td>\n",
       "      <td>19586506</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>45</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:19585947-19586506:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128316</th>\n",
       "      <td>chr1</td>\n",
       "      <td>22030881</td>\n",
       "      <td>22032219</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>53</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:22030881-22032219:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128343</th>\n",
       "      <td>chr1</td>\n",
       "      <td>23651439</td>\n",
       "      <td>23656349</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>13</td>\n",
       "      <td>4</td>\n",
       "      <td>41</td>\n",
       "      <td>chr1:23651439-23656349:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128352</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24018320</td>\n",
       "      <td>24019101</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>82</td>\n",
       "      <td>0</td>\n",
       "      <td>27</td>\n",
       "      <td>chr1:24018320-24019101:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128370</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24301561</td>\n",
       "      <td>24304400</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:24301561-24304400:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128388</th>\n",
       "      <td>chr1</td>\n",
       "      <td>24981522</td>\n",
       "      <td>24996705</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:24981522-24996705:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128414</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26658110</td>\n",
       "      <td>26662828</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:26658110-26662828:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128431</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27678047</td>\n",
       "      <td>27680203</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>74</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>chr1:27678047-27680203:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128453</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28719028</td>\n",
       "      <td>28720330</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>132</td>\n",
       "      <td>4</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:28719028-28720330:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128456</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28832613</td>\n",
       "      <td>28834639</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>63</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>chr1:28832613-28834639:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128468</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29269972</td>\n",
       "      <td>29271407</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>30</td>\n",
       "      <td>chr1:29269972-29271407:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128502</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32688232</td>\n",
       "      <td>32694099</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>166</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:32688232-32694099:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128528</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33087550</td>\n",
       "      <td>33093377</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>41</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:33087550-33093377:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128530</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33093467</td>\n",
       "      <td>33116033</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:33093467-33116033:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128534</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33100394</td>\n",
       "      <td>33116002</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>40</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>chr1:33100394-33116002:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128538</th>\n",
       "      <td>chr1</td>\n",
       "      <td>33119387</td>\n",
       "      <td>33119870</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chr1:33119387-33119870:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154275</th>\n",
       "      <td>chrY</td>\n",
       "      <td>4868647</td>\n",
       "      <td>4870728</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>chrY:4868647-4870728:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154277</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14774638</td>\n",
       "      <td>14798441</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>168</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrY:14774638-14798441:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154279</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14798536</td>\n",
       "      <td>14811345</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>215</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>chrY:14798536-14811345:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154280</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14798536</td>\n",
       "      <td>14813938</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>347</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrY:14798536-14813938:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154281</th>\n",
       "      <td>chrY</td>\n",
       "      <td>14811570</td>\n",
       "      <td>14813938</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>115</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrY:14811570-14813938:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154297</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21094728</td>\n",
       "      <td>21095242</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>91</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chrY:21094728-21095242:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154298</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21095362</td>\n",
       "      <td>21237826</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>92</td>\n",
       "      <td>0</td>\n",
       "      <td>43</td>\n",
       "      <td>chrY:21095362-21237826:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154300</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21205233</td>\n",
       "      <td>21206427</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrY:21205233-21206427:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154301</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21206582</td>\n",
       "      <td>21230609</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrY:21206582-21230609:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154303</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21717441</td>\n",
       "      <td>21719115</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>60</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>chrY:21717441-21719115:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154304</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21719176</td>\n",
       "      <td>21719909</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>71</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chrY:21719176-21719909:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154305</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21720027</td>\n",
       "      <td>21728867</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>26</td>\n",
       "      <td>chrY:21720027-21728867:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154314</th>\n",
       "      <td>chrM</td>\n",
       "      <td>795</td>\n",
       "      <td>8830</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>46</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chrM:795-8830:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154315</th>\n",
       "      <td>chrM</td>\n",
       "      <td>2581</td>\n",
       "      <td>2763</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>41</td>\n",
       "      <td>chrM:2581-2763:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154316</th>\n",
       "      <td>chrM</td>\n",
       "      <td>2764</td>\n",
       "      <td>2854</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>chrM:2764-2854:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154317</th>\n",
       "      <td>chrM</td>\n",
       "      <td>2796</td>\n",
       "      <td>2841</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>86</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrM:2796-2841:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154318</th>\n",
       "      <td>chrM</td>\n",
       "      <td>2796</td>\n",
       "      <td>2854</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>106</td>\n",
       "      <td>0</td>\n",
       "      <td>46</td>\n",
       "      <td>chrM:2796-2854:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154319</th>\n",
       "      <td>chrM</td>\n",
       "      <td>3905</td>\n",
       "      <td>4156</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>110</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chrM:3905-4156:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154320</th>\n",
       "      <td>chrM</td>\n",
       "      <td>8636</td>\n",
       "      <td>9035</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>76</td>\n",
       "      <td>1</td>\n",
       "      <td>45</td>\n",
       "      <td>chrM:8636-9035:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1154321</th>\n",
       "      <td>chrM</td>\n",
       "      <td>14924</td>\n",
       "      <td>15007</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>33</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>chrM:14924-15007:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1892268</th>\n",
       "      <td>chr1</td>\n",
       "      <td>21299025</td>\n",
       "      <td>21317990</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>48</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr1:21299025-21317990:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1894854</th>\n",
       "      <td>chr2</td>\n",
       "      <td>20850006</td>\n",
       "      <td>20850791</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>45</td>\n",
       "      <td>chr2:20850006-20850791:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902487</th>\n",
       "      <td>chr7</td>\n",
       "      <td>5568274</td>\n",
       "      <td>5569257</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>37</td>\n",
       "      <td>chr7:5568274-5569257:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1905408</th>\n",
       "      <td>chr9</td>\n",
       "      <td>131402652</td>\n",
       "      <td>131446110</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>70</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>chr9:131402652-131446110:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1908478</th>\n",
       "      <td>chr12</td>\n",
       "      <td>49953588</td>\n",
       "      <td>49958359</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>chr12:49953588-49958359:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1911832</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89836295</td>\n",
       "      <td>89843030</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>39</td>\n",
       "      <td>chr15:89836295-89843030:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1912052</th>\n",
       "      <td>chr16</td>\n",
       "      <td>3977408</td>\n",
       "      <td>3978572</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>38</td>\n",
       "      <td>chr16:3977408-3978572:+</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1916559</th>\n",
       "      <td>chr21</td>\n",
       "      <td>15313300</td>\n",
       "      <td>15316064</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>chr21:15313300-15316064:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1917082</th>\n",
       "      <td>chr22</td>\n",
       "      <td>39714572</td>\n",
       "      <td>39715599</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>chr22:39714572-39715599:-</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1918124</th>\n",
       "      <td>chrX</td>\n",
       "      <td>153196248</td>\n",
       "      <td>153197796</td>\n",
       "      <td>0</td>\n",
       "      <td>non-canonical</td>\n",
       "      <td>False</td>\n",
       "      <td>40</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>chrX:153196248-153197796:0</td>\n",
       "      <td>P1_01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1316 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom  intron_start  intron_stop strand   intron_motif annotated  \\\n",
       "1128056   chr1        891342       894379      0  non-canonical     False   \n",
       "1128059   chr1        995774      1001209      +          GT/AG     False   \n",
       "1128060   chr1       1197141      1197351      -          GC/AG     False   \n",
       "1128069   chr1       1247447      1254819      0  non-canonical     False   \n",
       "1128084   chr1       1463103      1469750      0  non-canonical     False   \n",
       "1128095   chr1       2323398      2324708      +          GT/AG     False   \n",
       "1128096   chr1       2324795      2327222      +          GT/AG     False   \n",
       "1128137   chr1       8031024      8044953      +          GT/AG     False   \n",
       "1128140   chr1       8460443      8460958      +          GT/AG     False   \n",
       "1128151   chr1       8934974      8938638      -          GT/AG     False   \n",
       "1128178   chr1      11129662     11129718      0  non-canonical     False   \n",
       "1128201   chr1      11734868     11735138      -          GT/AG     False   \n",
       "1128210   chr1      11740671     11751808      -          GT/AG     False   \n",
       "1128212   chr1      11805892     11810758      -          GT/AG     False   \n",
       "1128276   chr1      19585947     19586506      -          GT/AG     False   \n",
       "1128316   chr1      22030881     22032219      -          GT/AG     False   \n",
       "1128343   chr1      23651439     23656349      -          GT/AG     False   \n",
       "1128352   chr1      24018320     24019101      +          GT/AG     False   \n",
       "1128370   chr1      24301561     24304400      -          GT/AG     False   \n",
       "1128388   chr1      24981522     24996705      -          GT/AG     False   \n",
       "1128414   chr1      26658110     26662828      -          GT/AG     False   \n",
       "1128431   chr1      27678047     27680203      +          GT/AG     False   \n",
       "1128453   chr1      28719028     28720330      -          GT/AG     False   \n",
       "1128456   chr1      28832613     28834639      +          GT/AG     False   \n",
       "1128468   chr1      29269972     29271407      -          GC/AG     False   \n",
       "1128502   chr1      32688232     32694099      +          GT/AG     False   \n",
       "1128528   chr1      33087550     33093377      -          GT/AG     False   \n",
       "1128530   chr1      33093467     33116033      -          GT/AG     False   \n",
       "1128534   chr1      33100394     33116002      -          GT/AG     False   \n",
       "1128538   chr1      33119387     33119870      +          GT/AG     False   \n",
       "...        ...           ...          ...    ...            ...       ...   \n",
       "1154275   chrY       4868647      4870728      +          GT/AG     False   \n",
       "1154277   chrY      14774638     14798441      +          GT/AG     False   \n",
       "1154279   chrY      14798536     14811345      +          GT/AG     False   \n",
       "1154280   chrY      14798536     14813938      +          GT/AG     False   \n",
       "1154281   chrY      14811570     14813938      +          GT/AG     False   \n",
       "1154297   chrY      21094728     21095242      -          GT/AG     False   \n",
       "1154298   chrY      21095362     21237826      -          GT/AG     False   \n",
       "1154300   chrY      21205233     21206427      -          GT/AG     False   \n",
       "1154301   chrY      21206582     21230609      -          GT/AG     False   \n",
       "1154303   chrY      21717441     21719115      -          GT/AG     False   \n",
       "1154304   chrY      21719176     21719909      -          GT/AG     False   \n",
       "1154305   chrY      21720027     21728867      -          GT/AG     False   \n",
       "1154314   chrM           795         8830      -          GT/AG     False   \n",
       "1154315   chrM          2581         2763      -          GT/AG     False   \n",
       "1154316   chrM          2764         2854      -          GT/AG     False   \n",
       "1154317   chrM          2796         2841      -          GT/AG     False   \n",
       "1154318   chrM          2796         2854      -          GT/AG     False   \n",
       "1154319   chrM          3905         4156      -          GT/AG     False   \n",
       "1154320   chrM          8636         9035      -          GT/AG     False   \n",
       "1154321   chrM         14924        15007      -          GC/AG     False   \n",
       "1892268   chr1      21299025     21317990      +          GC/AG     False   \n",
       "1894854   chr2      20850006     20850791      -          GT/AG     False   \n",
       "1902487   chr7       5568274      5569257      0  non-canonical     False   \n",
       "1905408   chr9     131402652    131446110      +          GC/AG     False   \n",
       "1908478  chr12      49953588     49958359      +          GT/AG     False   \n",
       "1911832  chr15      89836295     89843030      +          GT/AG     False   \n",
       "1912052  chr16       3977408      3978572      +          GT/AG     False   \n",
       "1916559  chr21      15313300     15316064      -          GT/AG     False   \n",
       "1917082  chr22      39714572     39715599      -          GT/AG     False   \n",
       "1918124   chrX     153196248    153197796      0  non-canonical     False   \n",
       "\n",
       "         unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "1128056                     12                        0            39   \n",
       "1128059                     66                        0            46   \n",
       "1128060                     15                        3            42   \n",
       "1128069                     24                        1            42   \n",
       "1128084                     49                        2            30   \n",
       "1128095                     15                        0            46   \n",
       "1128096                     16                        0            32   \n",
       "1128137                    240                        2            44   \n",
       "1128140                     10                        8            43   \n",
       "1128151                    178                        0            45   \n",
       "1128178                     11                        0            41   \n",
       "1128201                     83                        0            46   \n",
       "1128210                     55                        0            45   \n",
       "1128212                     18                        1            34   \n",
       "1128276                     45                        0            45   \n",
       "1128316                     53                        0            46   \n",
       "1128343                     13                        4            41   \n",
       "1128352                     82                        0            27   \n",
       "1128370                     90                        0            45   \n",
       "1128388                     16                        0            45   \n",
       "1128414                     90                        0            45   \n",
       "1128431                     74                        0            44   \n",
       "1128453                    132                        4            45   \n",
       "1128456                     63                        0            36   \n",
       "1128468                     14                        1            30   \n",
       "1128502                    166                        0            46   \n",
       "1128528                     41                        0            45   \n",
       "1128530                     36                        0            45   \n",
       "1128534                     40                        0            25   \n",
       "1128538                     22                        0            46   \n",
       "...                        ...                      ...           ...   \n",
       "1154275                     19                        0            44   \n",
       "1154277                    168                        0            46   \n",
       "1154279                    215                        0            44   \n",
       "1154280                    347                        0            46   \n",
       "1154281                    115                        0            46   \n",
       "1154297                     91                        0            45   \n",
       "1154298                     92                        0            43   \n",
       "1154300                     19                        0            46   \n",
       "1154301                     14                        0            46   \n",
       "1154303                     60                        0            17   \n",
       "1154304                     71                        0            45   \n",
       "1154305                     47                        0            26   \n",
       "1154314                     46                        0            45   \n",
       "1154315                     11                        0            41   \n",
       "1154316                     16                        0            42   \n",
       "1154317                     86                        0            46   \n",
       "1154318                    106                        0            46   \n",
       "1154319                    110                        0            45   \n",
       "1154320                     76                        1            45   \n",
       "1154321                     33                        1            38   \n",
       "1892268                     48                        0            45   \n",
       "1894854                     10                        0            45   \n",
       "1902487                     28                        0            37   \n",
       "1905408                     70                        0            12   \n",
       "1908478                     10                        1            20   \n",
       "1911832                     10                        0            39   \n",
       "1912052                     12                        1            38   \n",
       "1916559                     10                        0            22   \n",
       "1917082                     11                        0            32   \n",
       "1918124                     40                        1            32   \n",
       "\n",
       "                    intron_location sample_id  \n",
       "1128056        chr1:891342-894379:0     P1_01  \n",
       "1128059       chr1:995774-1001209:+     P1_01  \n",
       "1128060      chr1:1197141-1197351:-     P1_01  \n",
       "1128069      chr1:1247447-1254819:0     P1_01  \n",
       "1128084      chr1:1463103-1469750:0     P1_01  \n",
       "1128095      chr1:2323398-2324708:+     P1_01  \n",
       "1128096      chr1:2324795-2327222:+     P1_01  \n",
       "1128137      chr1:8031024-8044953:+     P1_01  \n",
       "1128140      chr1:8460443-8460958:+     P1_01  \n",
       "1128151      chr1:8934974-8938638:-     P1_01  \n",
       "1128178    chr1:11129662-11129718:0     P1_01  \n",
       "1128201    chr1:11734868-11735138:-     P1_01  \n",
       "1128210    chr1:11740671-11751808:-     P1_01  \n",
       "1128212    chr1:11805892-11810758:-     P1_01  \n",
       "1128276    chr1:19585947-19586506:-     P1_01  \n",
       "1128316    chr1:22030881-22032219:-     P1_01  \n",
       "1128343    chr1:23651439-23656349:-     P1_01  \n",
       "1128352    chr1:24018320-24019101:+     P1_01  \n",
       "1128370    chr1:24301561-24304400:-     P1_01  \n",
       "1128388    chr1:24981522-24996705:-     P1_01  \n",
       "1128414    chr1:26658110-26662828:-     P1_01  \n",
       "1128431    chr1:27678047-27680203:+     P1_01  \n",
       "1128453    chr1:28719028-28720330:-     P1_01  \n",
       "1128456    chr1:28832613-28834639:+     P1_01  \n",
       "1128468    chr1:29269972-29271407:-     P1_01  \n",
       "1128502    chr1:32688232-32694099:+     P1_01  \n",
       "1128528    chr1:33087550-33093377:-     P1_01  \n",
       "1128530    chr1:33093467-33116033:-     P1_01  \n",
       "1128534    chr1:33100394-33116002:-     P1_01  \n",
       "1128538    chr1:33119387-33119870:+     P1_01  \n",
       "...                             ...       ...  \n",
       "1154275      chrY:4868647-4870728:+     P1_01  \n",
       "1154277    chrY:14774638-14798441:+     P1_01  \n",
       "1154279    chrY:14798536-14811345:+     P1_01  \n",
       "1154280    chrY:14798536-14813938:+     P1_01  \n",
       "1154281    chrY:14811570-14813938:+     P1_01  \n",
       "1154297    chrY:21094728-21095242:-     P1_01  \n",
       "1154298    chrY:21095362-21237826:-     P1_01  \n",
       "1154300    chrY:21205233-21206427:-     P1_01  \n",
       "1154301    chrY:21206582-21230609:-     P1_01  \n",
       "1154303    chrY:21717441-21719115:-     P1_01  \n",
       "1154304    chrY:21719176-21719909:-     P1_01  \n",
       "1154305    chrY:21720027-21728867:-     P1_01  \n",
       "1154314             chrM:795-8830:-     P1_01  \n",
       "1154315            chrM:2581-2763:-     P1_01  \n",
       "1154316            chrM:2764-2854:-     P1_01  \n",
       "1154317            chrM:2796-2841:-     P1_01  \n",
       "1154318            chrM:2796-2854:-     P1_01  \n",
       "1154319            chrM:3905-4156:-     P1_01  \n",
       "1154320            chrM:8636-9035:-     P1_01  \n",
       "1154321          chrM:14924-15007:-     P1_01  \n",
       "1892268    chr1:21299025-21317990:+     P1_01  \n",
       "1894854    chr2:20850006-20850791:-     P1_01  \n",
       "1902487      chr7:5568274-5569257:0     P1_01  \n",
       "1905408  chr9:131402652-131446110:+     P1_01  \n",
       "1908478   chr12:49953588-49958359:+     P1_01  \n",
       "1911832   chr15:89836295-89843030:+     P1_01  \n",
       "1912052     chr16:3977408-3978572:+     P1_01  \n",
       "1916559   chr21:15313300-15316064:-     P1_01  \n",
       "1917082   chr22:39714572-39715599:-     P1_01  \n",
       "1918124  chrX:153196248-153197796:0     P1_01  \n",
       "\n",
       "[1316 rows x 11 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj.loc[(~sj.annotated) & (sj.sample_id == 'P1_01')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(233,)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj.sample_id.unique().shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "#### Add single unique index for each splice junction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj['intron_location'] = sj.chrom.astype(str) + ':' \\\n",
    "    + sj.intron_start.astype(int).astype(str) + '-' \\\n",
    "    + sj.intron_stop.astype(int).astype(str) + ':' \\\n",
    "    + sj.strand.astype(str)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Save to a safe place"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "folder = '/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper'\n",
    "sj.to_csv('{}/sj_raw.csv'.format(folder), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 50)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAY8AAAERCAYAAACD9ivUAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAF29JREFUeJzt3X+UnNV93/H3CrESWCslNluoG2yXJP1CGxMhAiJAJHMq\nfrqOW6ek/GhK1ET8iEzDSVLXVTkIE1EINoRwDoYG/5CocNKI4PziIKkmgFSZGjkIxdju16aST8nB\nbRZktCsZJMFO/3junp2spd29i8ysdt+vc+bszJ07d+7z1Wo+e59n5pmuVquFJEk1ZnR6ApKkI4/h\nIUmqZnhIkqoZHpKkaoaHJKma4SFJqjZztDsj4mjgc8B7gVnAKuBvgL8AvlW6fToz10XEMuBq4A1g\nVWY+GhHHAGuBXmAAuCozX46Is4C7S9+NmXlLeb6VwCWl/YbM3BoRxwFfAGYDLwFLM/O1w1YBSVK1\nsVYeVwJ9mbkIuAi4F1gA3JmZ55XLuog4AbgeOBu4ELgtIrqB64Dt5fEPAjeWce8HLs/Mc4GFETE/\nIhYAizJzIXBZeS6Am4C1ZYxtwDWHZ9MlSRM1Vniso3nxHup7ADgd+GBEPBURn4mIOcCZwJbMPJCZ\n/cALwKnAOcD68vj1wJKI6AG6M3Nnad8ALCl9NwJk5ovAzLLqaB/jsdJXktRBo4ZHZu7NzD3lBX8d\n8J+AZ4DfyszFwA5gJdAD7G576AAwD5gL9I/SNrL9UGMMte8pbZKkDhrzgHlEnAj8JfBgZv4h8MXM\n3Fbu/iJwGk0Y9LQ9rAd4dUT7wdqgCYeDtbf3nzuiTZLUQaOGR0QcT7Mr6WOZubo0r4+IM8r1JcBX\naVYjPxcRsyJiHnAK8DywheYAOMDFwKbMHAD2R8RJEdEFXABsKn0vjIiuiHgP0JWZrxxsjLE2qtWc\nsMuLFy9evIz/UmXUd1sBK2h2E90UEUPHPm4AfjciDgDfBa4uu7buATbTBNKKzNwXEfcBayJiM7AP\nuKKMcS3wEHAUsCEztwKUfk+XMZaXvqvKGMuAvrYxDqmrq4u+voGxt34a6O3tsRaFtRhmLYZZi0Zv\nb8/Yndp0TdGz6rb8ZWj4H2OYtRhmLYZZi0Zvb09XTf+xVh46gg0ODrJ79276+/2PAdDdPWgtCmsx\nzFo0alcehscUtmfPABv/54sMtvxnBpjzjl3s2buv09OYFKzFMGvR+PEf/7Gq/r6qTHHHHvsOBunu\n9DQmhXfMmc0gr3d6GpOCtRhmLSbGc1tJkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEh\nSapmeEiSqhkekqRqhockqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEh\nSapmeEiSqhkekqRqhockqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEh\nSapmeEiSqhkekqRqM0e7MyKOBj4HvBeYBawCvgmsBgaB54HlmdmKiGXA1cAbwKrMfDQijgHWAr3A\nAHBVZr4cEWcBd5e+GzPzlvJ8K4FLSvsNmbk1Io4DvgDMBl4Clmbma4exBpKkSmOtPK4E+jJzEXAR\ncC9wJ7CitHUBH46IE4DrgbOBC4HbIqIbuA7YXvo+CNxYxr0fuDwzzwUWRsT8iFgALMrMhcBl5bkA\nbgLWljG2Adccjg2XJE3cWOGxjubFe6jvAWBBZm4qbY8BS4AzgC2ZeSAz+4EXgFOBc4D1pe96YElE\n9ADdmbmztG8oY5wDbATIzBeBmWXV0T7G0PNJkjpo1PDIzL2Zuae84K+jWTm0P2YAmAfMBXYfor1/\nlLbxjjHUvqe0SZI6aNRjHgARcSLwCHBvZv5BRNzRdvdc4FWaMOhpa+85SPvB2trH2D/KGHOBvra2\nMfX29ozdaYrr7h6EHbvomTO701OZNKzFMGsxzFrUG+uA+fE0u5J+LTOfKM3bImJxZj4FXAw8DjwD\n3BoRs2gObJ9CczB9C80B8K2l76bMHIiI/RFxErATuAC4GXgTuCMiPgWcCHRl5isRMTTGmqExxrNh\nfX0D4yzB1NXf39RgYM/rHZ7J5NAzZ7a1KKzFMGsxMWOtPFbQ7Ca6KSKGjn38OnBPOSD+DeDh8m6r\ne4DNNLu1VmTmvoi4D1gTEZuBfcAVZYxrgYeAo4ANmbkVoPR7uoyxvPRdVcZYRrP6GBpDktQhXa1W\nq9Nz+GFoufKA/v7dPLdjF4N0d3oqk4J/YQ6zFsOsRePS80/uqunvhwQlSdUMD0lSNcNDklTN8JAk\nVTM8JEnVDA9JUjXDQ5JUzfCQJFUzPCRJ1QwPSVI1w0OSVM3wkCRVMzwkSdUMD0lSNcNDklTN8JAk\nVTM8JEnVDA9JUjXDQ5JUzfCQJFUzPCRJ1QwPSVI1w0OSVM3wkCRVMzwkSdUMD0lSNcNDklTN8JAk\nVTM8JEnVDA9JUjXDQ5JUzfCQJFUzPCRJ1QwPSVI1w0OSVM3wkCRVmzmeThGxELg9M8+LiNOAPwe+\nXe7+dGaui4hlwNXAG8CqzHw0Io4B1gK9wABwVWa+HBFnAXeXvhsz85byPCuBS0r7DZm5NSKOA74A\nzAZeApZm5muHZeslSRMy5sojIj4GPADMKk2nA3dl5nnlsi4iTgCuB84GLgRui4hu4Dpge2YuAh4E\nbixj3A9cnpnnAgsjYn5ELAAWZeZC4DLg3tL3JmBtGWMbcM1b32xJ0lsxnt1WLwAfAbrK7dOBD0bE\nUxHxmYiYA5wJbMnMA5nZXx5zKnAOsL48bj2wJCJ6gO7M3FnaNwBLSt+NAJn5IjCzrDrax3is9JUk\nddCY4ZGZj9DsRhryFeC3MnMxsANYCfQAu9v6DADzgLlA/yhtI9sPNcZQ+57SJknqoIkcMP9iZm4b\nug6cRhMGPW19eoBXR7QfrA2acDhYe3v/uSPaJEkdNK4D5iOsj4h/l5lbaXYhfRV4Brg1ImbRHNg+\nBXge2EJzAHwrcDGwKTMHImJ/RJwE7AQuAG4G3gTuiIhPAScCXZn5SkQMjbFmaIzxTLK3t2fsTlNc\nd/cg7NhFz5zZnZ7KpGEthlmLYdaiXk14tMrPa4F7I+IA8F3g6szcExH3AJtpVjMrMnNfRNwHrImI\nzcA+4Iq2MR4CjgI2lCCi9Hu6jLG89F1VxlgG9LWNMaq+voGKTZua+vubGgzseb3DM5kceubMthaF\ntRhmLSamq9Vqjd3ryNMyPKC/fzfP7djFIN2dnsqk4IvEMGsxzFo0Lj3/5K6xew3zQ4KSpGqGhySp\nmuEhSapmeEiSqhkekqRqhockqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySp\nmuEhSapmeEiSqhkekqRqhockqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySp\nmuEhSapmeEiSqhkekqRqhockqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGozx9Mp\nIhYCt2fmeRHxE8BqYBB4Hliema2IWAZcDbwBrMrMRyPiGGAt0AsMAFdl5ssRcRZwd+m7MTNvKc+z\nEriktN+QmVsj4jjgC8Bs4CVgaWa+dpi2X5I0AWOuPCLiY8ADwKzSdBewIjMXAV3AhyPiBOB64Gzg\nQuC2iOgGrgO2l74PAjeWMe4HLs/Mc4GFETE/IhYAizJzIXAZcG/pexOwtoyxDbjmrW60JOmtGc9u\nqxeAj9AEBcCCzNxUrj8GLAHOALZk5oHM7C+PORU4B1hf+q4HlkRED9CdmTtL+4YyxjnARoDMfBGY\nWVYd7WMMPZ8kqYPGDI/MfIRmN9KQrrbrA8A8YC6w+xDt/aO0jXeMofY9pU2S1EHjOuYxwmDb9bnA\nqzRh0NPW3nOQ9oO1tY+xf5Qx5gJ9bW1j6u3tGbvTFNfdPQg7dtEzZ3anpzJpWIth1mKYtag3kfDY\nFhGLM/Mp4GLgceAZ4NaImEVzYPsUmoPpW2gOgG8tfTdl5kBE7I+Ik4CdwAXAzcCbwB0R8SngRKAr\nM1+JiKEx1gyNMZ5J9vUNTGDTppb+/qYGA3te7/BMJoeeObOtRWEthlmLiakJj1b5+ZvAA+WA+DeA\nh8u7re4BNtPsCluRmfsi4j5gTURsBvYBV5QxrgUeAo4CNmTmVoDS7+kyxvLSd1UZYxnN6mNoDElS\nh3S1Wq2xex15Wq48oL9/N8/t2MUg3Z2eyqTgX5jDrMUwa9G49PyTu8buNcwPCUqSqhkekqRqhock\nqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEhSapmeEiSqhkekqRqhock\nqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEhSapmeEiSqhkekqRqhock\nqZrhIUmqZnhIkqoZHpKkaoaHJKma4SFJqmZ4SJKqGR6SpGqGhySpmuEhSapmeEiSqs2c6AMj4llg\nd7m5A7gNWA0MAs8DyzOzFRHLgKuBN4BVmfloRBwDrAV6gQHgqsx8OSLOAu4ufTdm5i3luVYCl5T2\nGzJz60TnLUl66ya08oiI2QCZeV65/ApwF7AiMxcBXcCHI+IE4HrgbOBC4LaI6AauA7aXvg8CN5ah\n7wcuz8xzgYURMT8iFgCLMnMhcBlw70Q3VpJ0eEx0t9VPA8dGxIaIeLysGBZk5qZy/2PAEuAMYEtm\nHsjMfuAF4FTgHGB96bseWBIRPUB3Zu4s7RvKGOcAGwEy80VgZkS8a4LzliQdBhMNj73AJzPzQuBa\n4KER9w8A84C5DO/aGtneP0rbeMaQJHXIRI95fItmFUFmfjsiXgFOa7t/LvAqTRj0tLX3HKT9YG3t\nY+w/xBij6u3tGavLlNfdPQg7dtEzZ3anpzJpWIth1mKYtag30fBYSrP7aXlEvJvmBX1jRCzOzKeA\ni4HHgWeAWyNiFjAbOIXmYPoWmgPgW0vfTZk5EBH7I+IkYCdwAXAz8CZwR0R8CjgRmJGZu8aaYF/f\nwAQ3bero729qMLDn9Q7PZHLomTPbWhTWYpi1mJiJhsdngc9HxNAxjqXAK8AD5YD4N4CHy7ut7gE2\n0+wiW5GZ+yLiPmBNRGwG9gFXlHGGdoEdBWwYeldV6fd0GePXJjhnSdJh0tVqtTo9hx+GlisP6O/f\nzXM7djFId6enMin4F+YwazHMWjQuPf/krpr+fkhQklTN8JAkVTM8JEnVDA9JUjXDQ5JUzfCQJFUz\nPCRJ1QwPSVI1w0OSVM3wkCRVMzwkSdUMD0lSNcNDklTN8JAkVTM8JEnVDA9JUjXDQ5JUzfCQJFUz\nPCRJ1QwPSVI1w0OSVM3wkCRVMzwkSdUMD0lSNcNDklTN8JAkVTM8JEnVDA9JUjXDQ5JUzfCQJFUz\nPCRJ1QwPSVI1w0OSVM3wkCRVMzwkSdUMD0lStZmdnsB4RMQM4NPAqcA+4Fcz8393dlaSNH0dKSuP\nfw50Z+bZwMeBOzs8H0ma1o6U8DgHWA+QmV8Bfqaz05Gk6e1ICY+5QH/b7TfLrixJUgccEcc8aIKj\np+32jMwcPFTn3bt3098/8MOf1SQ3MNDP97+/l8HWvk5PZVKYwX6+v9dagLVoZy0m5kgJjy3Ah4B1\nEXEW8NejdZ43b17XvHnz3paJTXbz53d6BpKmoiMlPL4InB8RW8rtpZ2cjCRNd12tVqvTc5AkHWE8\n6CxJqmZ4SJKqGR6SpGqGhySp2pHybqsxef6rRkQsBG7PzPMi4ieA1cAg8DywPDOn/DskIuJo4HPA\ne4FZwCrgm0zPWhwFPAD8I6AFXEvz/2M106wWQyLi7wF/BfxTmhqsZhrWIiKeBXaXmzuA26ioxVRa\neUz7819FxMdoXihmlaa7gBWZuQjoAj7cqbm9za4E+sp2XwTcS/P7MB1r8c+Awcw8F7gR+M9M31oM\n/WHxX4C9NNs+Lf+PRMRsgMw8r1x+hcpaTKXw8PxX8ALwEZp/eIAFmbmpXH8MWNKRWb391gE3lesz\ngANM01pk5p8C15Sb7wO+B5w+HWtRfBK4D/huuT0tfy+AnwaOjYgNEfF4+fB1VS2mUnhM+/NfZeYj\nwBttTV1t1/cA0+Jj95m5NzP3REQPTZDcyN/9XZ82tQDIzDcjYjXwe8BDTNPfi4j4ZZoV6cbS1MU0\nrQXNyuuTmXkhza7Mh0bcP2YtptKLa9X5r6aJ9u3vAV7t1ETebhFxIvCXwIOZ+QdM41oAZOYvAwF8\nBpjddtd0qsVSmjNVPAHMB9YAvW33T6dafIsSGJn5beAV4Pi2+8esxVQKjy3AJQDjOf/VNLEtIhaX\n6xcDm0brPFVExPHARuBjmbm6NE/XWvxSRPzHcvM14E3gq9OxFpm5ODM/kJnnAc8B/wZYPx1rQROk\ndwJExLtpwmJjTS2mzLut8PxX7YbeIfGbwAMR0Q18A3i4c1N6W62gWXLfFBFDxz5+HbhnGtbiYWB1\nRDwFHE1Th//F9Py9GKnF9P0/8lng8xExFBBLaVYf466F57aSJFWbSrutJElvE8NDklTN8JAkVTM8\nJEnVDA9JUjXDQ5JUzfDQES8ith2pzxURV0fEZeX6JyLiQ4dz/MMpIo6LiJ2dnocmh6n0IUFNU5l5\n2hH8XGcDT5SxVx7msaUfGsNDk05EfABYWU4jQTmp3xPADcDXgNOA/wdcmpnfi4jBzJwREe8CHgTe\nQ/P9He8DLgXOAxZn5tIy3pNl/Kci4uOlz1HAhsz8D2PMbei5bgZamfmJ0v4dYHF5rouAHwVOAjZm\n5vKI6AJup/nqgDdoTgv+deBDwAci4rvAFcATmbkmIpYCv0HzKei/Aj6amXtLv3XAuWWcX8zM74wy\n3ydpPjn8T4B/Bfx94BM0nzbfCSzLzF0RcWl5vmPK5Vczc3NEzKf5NHIX8NW2ca8A/j3N6U52Av86\nM/eNVjtNLe620pFg6DQIpwJ3Zub7aU7aduWIfr8NbC/3/x5wennsyNMotAAi4iJgAXBG+fljETFy\nzLHm1H57qO1naU6NfyrwoYj4KeBf0qwyfgo4k+Z0EM8DfwbcVM702gJaEfF+mlOsLMrMU2nOgDq0\nKjke+FJmLqA599BHxzHP7Zl5MvASzRf+XFAevxH4nRJs1wAfzMz5wO/QBAPAfwU+Xvpn2zb+NnB+\nZv4MzelOTh5jHppiXHnoSPK3mbm9XH+e5q/7douAywAyc0tEfJ0fPO12uyXAQpq/7KE52+x3xjmX\nLn4wQIae58uZuRcgInYA7yxz+2+ZeYDm+0VOK/cfbNxFwJ9l5vdK2+8Dn2/rs778fL70HctXys+F\nNKuyJ8vzHgW8kpmtiPgXwM9Hc8di4I2yknt3Zv738vjPMhxWfw58OSL+BPjjtn8XTROuPDQZtfi7\nL/hHl5+vj9IHmrPGtv9O7x9jvBnA3Zl5WjmWcTbNX+ZvZY6HmueB9v4R8b6IOPYQY88YMfYM2v7Q\ny8z9bfcdKhjbvdY2zv9o294zgV+MiDk0u6TeCzwJ3FP6jtzGN9vmcAPwC8AuYG3Fik1ThOGhyehl\n4KSImBUR7wR+7iB9Drai2Aj8W4Cy6+f9NC+AfcAppf0f0uxOatF838cvRcQ7ImIm8AjN7qbx6AP+\ncRnzTJpjCaOdZXQT8JGImFlCYz3wD2iOWxw9ou+TNKuAoZXVsjLXt+oZ4Gcj4ifL7RuBO4CfpAmG\n28pzXwIclZm7gJ0R8fOl/xUAETEjIhJ4OTNvpznONP8wzE9HEMNDk05mfh14lOaA8h8x/L0C7S/O\n7ccYhn7eCrwzIrYDq4C/Le1fAl4sL3h3A5vL8/wF8Mc0u3W+BmzLzAfHOc0/BN5Vdo19FHiW4V1Z\nP3A8JDP/hOY7Z56leRH/3fIlPF8CVkTEL7T1/RrNC/lTEfFNmm/JvHGMGowpM/8vTbj+UUT8Nc2u\ns98AttN8v8U3gadovgvnPeVhV5b5PUtz0L1VvmRtJfCliNhKE+53jXcemho8JbumrPLCe2Fm/p/D\nNN6PAH+TmXMOx3jSkcwD5lKbiDgG+PJB7poH/AjwwNs7o7FFxFqaVcFIf5qZN7/N09E04cpDklTN\nYx6SpGqGhySpmuEhSapmeEiSqhkekqRqhockqdr/B3oo7yw7eu5pAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2ae719a91510>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "reads = sj.unique_junction_reads[sj.unique_junction_reads > 0]\n",
    "# bins = np.linspace(0, 6, 1000)\n",
    "\n",
    "fig, ax = plt.subplots()\n",
    "\n",
    "sns.distplot(reads, bins=[-1, 0, 10, 100, 1000, 1e4, 1e5, 1e6, 1e7], kde=False)\n",
    "ax.set_xlim(0, 50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## restart here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import poshsplice\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "folder = '/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper'\n",
    "sj = pd.read_csv('{}/sj_raw.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Save metadata about splice junctions separately\n",
    "\n",
    "Drop duplicates since lots of junctions have been observed in multiple samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4581458, 7)\n",
      "(280294, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "          intron_location  \n",
       "0  chr1:1309283-1309379:-  \n",
       "1  chr1:1309826-1310084:-  \n",
       "2  chr1:1310171-1310533:-  \n",
       "3  chr1:1337637-1341188:-  \n",
       "4  chr1:1341267-1342288:-  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata = sj[['chrom', 'intron_start', 'intron_stop', 'strand', 'intron_motif', 'annotated', 'intron_location']]\n",
    "print sj_metadata.shape\n",
    "sj_metadata = sj_metadata.drop_duplicates()\n",
    "print sj_metadata.shape\n",
    "sj_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata.set_index('intron_location')\n",
    "sj_metadata.to_csv('{}/sj_metadata.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(280294, 7)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(sj_metadata.strand == 0).sum()/float(sj_metadata.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "3.26% of the splice junctions have an unannotated/incorrectly annotated strand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chrom, intron_start, intron_stop, strand, intron_motif, annotated, intron_location]\n",
       "Index: []"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.loc[sj_metadata.strand == 0].head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate Percent spliced-in/$\\Psi$/Psi scores\n",
    "\n",
    "If the number of reads is less than the threshold, treat it as zero reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id  \n",
       "0  chr1:1309283-1309379:-    CVN_09  \n",
       "1  chr1:1309826-1310084:-    CVN_09  \n",
       "2  chr1:1310171-1310533:-    CVN_09  \n",
       "3  chr1:1337637-1341188:-    CVN_09  \n",
       "4  chr1:1341267-1342288:-    CVN_09  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MIN_UNIQUE_JUNCTION_READS = 10\n",
    "\n",
    "sj.loc[sj.unique_junction_reads < MIN_UNIQUE_JUNCTION_READS, 'unique_junction_reads'] = 0\n",
    "# sj.loc[sj.multimap_junction_reads < MIN_MULTIMAP_JUNCTION_READS, 'multimap_junction_reads'] = 0\n",
    "sj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                                 1                                 2                               \n",
       "exon:chr10:100190328-100190427:-  exon:chr10:100189548-100189646:-  exon:chr10:100188908-100189399:-    ('chr10:100189647-100190327:-', 'chr10:1001894...\n",
       "                                                                    exon:chr10:100188913-100189399:-    ('chr10:100189647-100190327:-', 'chr10:1001894...\n",
       "                                                                    exon:chr10:100189330-100189399:-    ('chr10:100189647-100190327:-', 'chr10:1001894...\n",
       "exon:chr10:100193697-100193848:-  exon:chr10:100190888-100191048:-  exon:chr10:100190328-100190427:-    ('chr10:100191049-100193696:-', 'chr10:1001904...\n",
       "exon:chr10:100195392-100195529:-  exon:chr10:100195029-100195171:-  exon:chr10:100193697-100193848:-    ('chr10:100195172-100195391:-', 'chr10:1001938...\n",
       "Name: 3, dtype: object"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_folder = '/projects/ps-yeolab/obotvinnik/singlecell_pnms'\n",
    "exons_to_junctions = pd.read_csv('{}/exons_to_junctions_se.csv'.format(metadata_folder), \n",
    "                                 index_col=[0, 1, 2], squeeze=True, header=None)\n",
    "exons_to_junctions.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ \"('chr10:100189647-100190327:-', 'chr10:100189400-100189547:-', 'chr10:100189400-100190327:-')\",\n",
       "       \"('chr10:100189647-100190327:-', 'chr10:100189400-100189547:-', 'chr10:100189400-100190327:-')\",\n",
       "       \"('chr10:100189647-100190327:-', 'chr10:100189400-100189547:-', 'chr10:100189400-100190327:-')\",\n",
       "       ...,\n",
       "       \"('chrY:59356944-59357702:+', 'chrY:59357772-59357911:+', 'chrY:59356944-59357911:+')\",\n",
       "       \"('chrY:7194211-7209155:+', 'chrY:7209276-7224175:+', 'chrY:7194211-7224175:+')\",\n",
       "       \"('chrY:7194211-7209155:+', 'chrY:7209276-7224175:+', 'chrY:7194211-7224175:+')\"], dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exons_to_junctions.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7082563, 11)\n",
      "(4580821, 11)\n"
     ]
    }
   ],
   "source": [
    "print sj.shape\n",
    "sj = sj.drop_duplicates(['sample_id', 'intron_location'])\n",
    "print sj.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_09</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_09</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_09</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_09</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_09</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "sample_id                                                                  \n",
       "CVN_09     chr1       1309283      1309379      -        GT/AG      True   \n",
       "CVN_09     chr1       1309826      1310084      -        GT/AG      True   \n",
       "CVN_09     chr1       1310171      1310533      -        GT/AG      True   \n",
       "CVN_09     chr1       1337637      1341188      -        GT/AG      True   \n",
       "CVN_09     chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "           unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "sample_id                                                                 \n",
       "CVN_09                        37                        0            49   \n",
       "CVN_09                        43                        0            49   \n",
       "CVN_09                        27                        0            47   \n",
       "CVN_09                       409                        0            50   \n",
       "CVN_09                       337                        0            50   \n",
       "\n",
       "                  intron_location  \n",
       "sample_id                          \n",
       "CVN_09     chr1:1309283-1309379:-  \n",
       "CVN_09     chr1:1309826-1310084:-  \n",
       "CVN_09     chr1:1310171-1310533:-  \n",
       "CVN_09     chr1:1337637-1341188:-  \n",
       "CVN_09     chr1:1341267-1342288:-  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_samples = sj.set_index('sample_id')\n",
    "sj_samples_grouped = sj_samples.groupby('intron_location')\n",
    "sj_samples.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import poshsplice\n",
    "from poshsplice.region import Region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Automatic pdb calling has been turned ON\n"
     ]
    }
   ],
   "source": [
    "%pdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_samples_test = sj_samples.loc[sj_samples.intron_location.str.startswith('chr10:10018')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_35</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189400</td>\n",
       "      <td>100189547</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>chr10:100189400-100189547:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_35</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189647</td>\n",
       "      <td>100190327</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr10:100189647-100190327:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_17</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189647</td>\n",
       "      <td>100190327</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>chr10:100189647-100190327:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P1_10</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189400</td>\n",
       "      <td>100190327</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>44</td>\n",
       "      <td>chr10:100189400-100190327:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P4_06</th>\n",
       "      <td>chr10</td>\n",
       "      <td>100189400</td>\n",
       "      <td>100190327</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>chr10:100189400-100190327:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "sample_id                                                                   \n",
       "CVN_35     chr10     100189400    100189547      -        GT/AG      True   \n",
       "CVN_35     chr10     100189647    100190327      -        GT/AG      True   \n",
       "CVN_17     chr10     100189647    100190327      -        GT/AG      True   \n",
       "P1_10      chr10     100189400    100190327      -        GT/AG      True   \n",
       "P4_06      chr10     100189400    100190327      -        GT/AG      True   \n",
       "\n",
       "           unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "sample_id                                                                 \n",
       "CVN_35                        13                        0            48   \n",
       "CVN_35                        15                        0            47   \n",
       "CVN_17                        12                        0            44   \n",
       "P1_10                         43                        0            44   \n",
       "P4_06                         18                        0            48   \n",
       "\n",
       "                       intron_location  \n",
       "sample_id                               \n",
       "CVN_35     chr10:100189400-100189547:-  \n",
       "CVN_35     chr10:100189647-100190327:-  \n",
       "CVN_17     chr10:100189647-100190327:-  \n",
       "P1_10      chr10:100189400-100190327:-  \n",
       "P4_06      chr10:100189400-100190327:-  "
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_samples_test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "pd.concat?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0, 1, 2]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "range(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%time\n",
    "\n",
    "reads_col = 'unique_junction_reads'\n",
    "\n",
    "se_dfs = []\n",
    "\n",
    "def get_junction_reads(junction, sj_samples_grouped, reads_col):\n",
    "    try:\n",
    "        return sj_samples_grouped.get_group(junction)[reads_col].astype(float)\n",
    "    except KeyError:\n",
    "        return pd.Series()\n",
    "\n",
    "for junctions, s in exons_to_junctions.groupby(exons_to_junctions):\n",
    "    junctions = eval(junctions)\n",
    "    \n",
    "    # \"1\", \"2\", and \"3\" in this case are in genomic order, not in transcript order\n",
    "    exon12 = get_junction_reads(junctions[0], sj_samples_grouped, reads_col)\n",
    "    exon23 = get_junction_reads(junctions[1], sj_samples_grouped, reads_col)\n",
    "    exon13 = get_junction_reads(junctions[2], sj_samples_grouped, reads_col)\n",
    "    \n",
    "    df = pd.concat([exon12, exon23, exon13], axis=1, keys=['exon12', 'exon23', 'exon13'])\n",
    "#     df.columns = \n",
    "    df = df.fillna(0)\n",
    "    \n",
    "    df['$\\Psi$'] = (df.exon12 + df.exon23).divide(df.exon12 + df.exon23 + 2*df.exon13)\n",
    "    df['junction1'], df['junction2'], df['junction3'] = junctions\n",
    "    df = df.reset_index()\n",
    "    df = df.rename(columns={'index': 'sample_id'})\n",
    "#     print df.head()\n",
    "    se_dfs.append(df)\n",
    "se_df = pd.concat(se_dfs, ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     1.000000\n",
       "1     1.000000\n",
       "2     0.000000\n",
       "3     0.000000\n",
       "4     1.000000\n",
       "5     1.000000\n",
       "6     1.000000\n",
       "7     0.000000\n",
       "8     0.000000\n",
       "9     1.000000\n",
       "10    0.000000\n",
       "11    0.000000\n",
       "12    0.000000\n",
       "13    1.000000\n",
       "14    0.000000\n",
       "15    0.000000\n",
       "16    0.000000\n",
       "17    0.000000\n",
       "18    0.000000\n",
       "19    0.000000\n",
       "20    0.000000\n",
       "21    1.000000\n",
       "22    0.000000\n",
       "23    0.000000\n",
       "24    0.000000\n",
       "25    1.000000\n",
       "26    0.000000\n",
       "27    1.000000\n",
       "28    1.000000\n",
       "29    1.000000\n",
       "30    1.000000\n",
       "31    0.000000\n",
       "32    1.000000\n",
       "33    1.000000\n",
       "34    1.000000\n",
       "35    1.000000\n",
       "36    1.000000\n",
       "37    0.000000\n",
       "38    1.000000\n",
       "39    1.000000\n",
       "40    1.000000\n",
       "41    1.000000\n",
       "42    1.000000\n",
       "43    1.000000\n",
       "44    1.000000\n",
       "45    0.518072\n",
       "46    1.000000\n",
       "47    1.000000\n",
       "48    1.000000\n",
       "49    1.000000\n",
       "Name: $\\Psi$, dtype: float64"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "se_df['$\\Psi$'][:50]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For samples that share a donor (acceptor), add the acceptor (donor) to other samples, with zero reads. This ***explicitly*** adds all possible `donor, acceptor` pairs to each sample, so the other samples will properly get $\\Psi$ scores of 1 for the `donor, acceptor` pairs they have not seen.\n",
    "\n",
    "\n",
    "**From:**\n",
    "\n",
    "    Sample 1:  \n",
    "    chr1:100-200: 25 reads\n",
    "    [   ]-------------[   ]\n",
    "    Sample 2:  \n",
    "    chr1:100:250: 40 reads\n",
    "    [   ]----------------------[    ]\n",
    "    \n",
    "\n",
    "**To:**\n",
    "\n",
    "    Sample 1:  \n",
    "    chr1:100-200: 25 reads\n",
    "    [   ]-------------[   ]\n",
    "    chr1:100:250: 0 reads\n",
    "    [   ]----------------------[    ]\n",
    "    Sample 2:  \n",
    "    chr1:100-200: 0 reads\n",
    "    [   ]-------------[   ]\n",
    "    chr1:100:250: 40 reads\n",
    "    [   ]----------------------[    ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2h 33s, sys: 28.4 s, total: 2h 1min 2s\n",
      "Wall time: 2h 1min 3s\n"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "dfs = []\n",
    "\n",
    "sample_id = 'sample_id'\n",
    "full_index = ['sample_id', 'chrom', 'intron_start', 'intron_stop']\n",
    "sample_chrom_intron_start = ['sample_id', 'chrom', 'intron_start', 'strand']\n",
    "sample_chrom_intron_stop = ['sample_id', 'chrom', 'intron_stop', 'strand']\n",
    "sample_id = 'sample_id'\n",
    "full_index = ['sample_id', 'chrom', 'intron_start', 'intron_stop', 'strand']\n",
    "reads_col = 'unique_junction_reads'\n",
    "\n",
    "intron_start = 'intron_start'\n",
    "intron_stop = 'intron_stop'\n",
    "\n",
    "for intron in ('intron_start', 'intron_stop'):\n",
    "    for name, df in sj.groupby(['chrom', intron, 'strand']):\n",
    "        df = df.set_index(full_index)\n",
    "        df = df[reads_col].unstack(['sample_id', 'chrom', intron, 'strand'])\n",
    "        df = df.fillna(0)\n",
    "        df = df.stack(['sample_id', 'chrom', intron, 'strand']).reset_index().sort('sample_id')\n",
    "        df = df.rename(columns={0: reads_col})\n",
    "    #     df = df.reindex(columns=['sample_id', 'chrom', 'first_bp_intron', 'last_bp_intron', 'total_filtered_reads'])\n",
    "        dfs.append(df)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(11360949, 6)\n",
      "(7034954, 6)\n",
      "CPU times: user 4min 46s, sys: 20 s, total: 5min 6s\n",
      "Wall time: 5min 7s\n"
     ]
    }
   ],
   "source": [
    "%%time \n",
    "sj_appended = pd.concat(dfs, ignore_index=True)\n",
    "print sj_appended.shape\n",
    "sj_appended = sj_appended.drop_duplicates()\n",
    "print sj_appended.shape\n",
    "sj_appended = sj_appended.reindex(\n",
    "    columns=full_index + [reads_col])\n",
    "sj_appended.head()\n",
    "sj_appended = sj_appended.sort(full_index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5686310</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21094728</td>\n",
       "      <td>21237826</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358564</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21095362</td>\n",
       "      <td>21205048</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5686384</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21146519</td>\n",
       "      <td>21192634</td>\n",
       "      <td>-</td>\n",
       "      <td>683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5686385</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21152692</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358639</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21152844</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358640</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21152980</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5686390</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21154115</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358642</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21154260</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358643</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21154400</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358644</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21176408</td>\n",
       "      <td>21218028</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11358565</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>21203473</td>\n",
       "      <td>21205048</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5687719</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22737774</td>\n",
       "      <td>22741493</td>\n",
       "      <td>+</td>\n",
       "      <td>2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5687837</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22741578</td>\n",
       "      <td>22744475</td>\n",
       "      <td>+</td>\n",
       "      <td>2116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11360468</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22744487</td>\n",
       "      <td>22746359</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688059</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22744580</td>\n",
       "      <td>22746359</td>\n",
       "      <td>+</td>\n",
       "      <td>2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688060</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22744580</td>\n",
       "      <td>22749909</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688287</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22746411</td>\n",
       "      <td>22749909</td>\n",
       "      <td>+</td>\n",
       "      <td>2506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688288</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22746411</td>\n",
       "      <td>22749936</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688402</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22749992</td>\n",
       "      <td>22751369</td>\n",
       "      <td>+</td>\n",
       "      <td>2130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5688517</th>\n",
       "      <td>P9_04</td>\n",
       "      <td>chrY</td>\n",
       "      <td>22751462</td>\n",
       "      <td>22754226</td>\n",
       "      <td>+</td>\n",
       "      <td>1872</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_id chrom  intron_start  intron_stop strand  \\\n",
       "5686310      P9_04  chrY      21094728     21237826      -   \n",
       "11358564     P9_04  chrY      21095362     21205048      -   \n",
       "5686384      P9_04  chrY      21146519     21192634      -   \n",
       "5686385      P9_04  chrY      21152692     21218028      -   \n",
       "11358639     P9_04  chrY      21152844     21218028      -   \n",
       "11358640     P9_04  chrY      21152980     21218028      -   \n",
       "5686390      P9_04  chrY      21154115     21218028      -   \n",
       "11358642     P9_04  chrY      21154260     21218028      -   \n",
       "11358643     P9_04  chrY      21154400     21218028      -   \n",
       "11358644     P9_04  chrY      21176408     21218028      -   \n",
       "11358565     P9_04  chrY      21203473     21205048      -   \n",
       "5687719      P9_04  chrY      22737774     22741493      +   \n",
       "5687837      P9_04  chrY      22741578     22744475      +   \n",
       "11360468     P9_04  chrY      22744487     22746359      +   \n",
       "5688059      P9_04  chrY      22744580     22746359      +   \n",
       "5688060      P9_04  chrY      22744580     22749909      +   \n",
       "5688287      P9_04  chrY      22746411     22749909      +   \n",
       "5688288      P9_04  chrY      22746411     22749936      +   \n",
       "5688402      P9_04  chrY      22749992     22751369      +   \n",
       "5688517      P9_04  chrY      22751462     22754226      +   \n",
       "\n",
       "          unique_junction_reads  \n",
       "5686310                       0  \n",
       "11358564                      0  \n",
       "5686384                     683  \n",
       "5686385                      11  \n",
       "11358639                      0  \n",
       "11358640                      0  \n",
       "5686390                      10  \n",
       "11358642                      0  \n",
       "11358643                      0  \n",
       "11358644                      0  \n",
       "11358565                      0  \n",
       "5687719                    2014  \n",
       "5687837                    2116  \n",
       "11360468                      0  \n",
       "5688059                    2394  \n",
       "5688060                       0  \n",
       "5688287                    2506  \n",
       "5688288                       0  \n",
       "5688402                    2130  \n",
       "5688517                    1872  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_appended.tail(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate Percent spliced-in/Psi/$\\Psi$ from splice junctions\n",
    "\n",
    "This method of getting psi scores is 5x faster than using `lambda x: x/x.sum()`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 36s, sys: 13.2 s, total: 1min 49s\n",
      "Wall time: 1min 49s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "splice_sites = {r'$\\Psi_5$': 'intron_start', r'$\\Psi_3$': 'intron_stop'}\n",
    "\n",
    "for psi, ss in splice_sites.items():\n",
    "    sj_appended = sj_appended.sort(full_index)\n",
    "    groupby = ['sample_id', 'chrom', ss, 'strand']\n",
    "    reads = sj_appended.groupby(groupby)[reads_col].sum()\n",
    "    reads.name = reads_col\n",
    "    \n",
    "    # sort_index() is necessary for the rows of sj_appended\n",
    "    # to be exactly the right order and match with psi_scores\n",
    "    sj_appended = sj_appended.set_index(groupby).sort_index()\n",
    "    psi_scores = sj_appended[reads_col].divide(reads).fillna(0)\n",
    "    sj_appended[psi] = psi_scores.values\n",
    "    sj_appended = sj_appended.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>$\\Psi_5$</th>\n",
       "      <th>$\\Psi_3$</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CVN_01</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1328775</td>\n",
       "      <td>-</td>\n",
       "      <td>1326246</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CVN_01</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1328775</td>\n",
       "      <td>-</td>\n",
       "      <td>1328184</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CVN_01</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1337738</td>\n",
       "      <td>-</td>\n",
       "      <td>1337637</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CVN_01</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1337841</td>\n",
       "      <td>-</td>\n",
       "      <td>1337637</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CVN_01</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1341182</td>\n",
       "      <td>-</td>\n",
       "      <td>1337637</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  sample_id chrom  intron_stop strand  intron_start  unique_junction_reads  \\\n",
       "0    CVN_01  chr1      1328775      -       1326246                      0   \n",
       "1    CVN_01  chr1      1328775      -       1328184                     20   \n",
       "2    CVN_01  chr1      1337738      -       1337637                      0   \n",
       "3    CVN_01  chr1      1337841      -       1337637                      0   \n",
       "4    CVN_01  chr1      1341182      -       1337637                      0   \n",
       "\n",
       "   $\\Psi_5$  $\\Psi_3$  \n",
       "0         0         0  \n",
       "1         1         1  \n",
       "2         0         0  \n",
       "3         0         0  \n",
       "4         0         0  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_appended.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sj_appended['intron_location'] = sj_appended.chrom.astype(str) + ':' \\\n",
    "    + sj_appended.intron_start.astype(str) + '-' \\\n",
    "    + sj_appended.intron_stop.astype(str) + ':' \\\n",
    "    + sj_appended.strand.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ubc_middle_event = 'chr12:125396955-125397182:-|3p'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ubc_middle_intron_location = 'chr12:125396955-125397182:-'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>37</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>409</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>337</td>\n",
       "      <td>0</td>\n",
       "      <td>50</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "      <td>CVN_09</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "   unique_junction_reads  multimap_junction_reads  max_overhang  \\\n",
       "0                     37                        0            49   \n",
       "1                     43                        0            49   \n",
       "2                     27                        0            47   \n",
       "3                    409                        0            50   \n",
       "4                    337                        0            50   \n",
       "\n",
       "          intron_location sample_id  \n",
       "0  chr1:1309283-1309379:-    CVN_09  \n",
       "1  chr1:1309826-1310084:-    CVN_09  \n",
       "2  chr1:1310171-1310533:-    CVN_09  \n",
       "3  chr1:1337637-1341188:-    CVN_09  \n",
       "4  chr1:1341267-1342288:-    CVN_09  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309283</td>\n",
       "      <td>1309379</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1309283-1309379:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1309826</td>\n",
       "      <td>1310084</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1309826-1310084:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1310171</td>\n",
       "      <td>1310533</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1310171-1310533:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1337637</td>\n",
       "      <td>1341188</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1337637-1341188:-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1341267</td>\n",
       "      <td>1342288</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1341267-1342288:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1       1309283      1309379      -        GT/AG      True   \n",
       "1  chr1       1309826      1310084      -        GT/AG      True   \n",
       "2  chr1       1310171      1310533      -        GT/AG      True   \n",
       "3  chr1       1337637      1341188      -        GT/AG      True   \n",
       "4  chr1       1341267      1342288      -        GT/AG      True   \n",
       "\n",
       "          intron_location  \n",
       "0  chr1:1309283-1309379:-  \n",
       "1  chr1:1309826-1310084:-  \n",
       "2  chr1:1310171-1310533:-  \n",
       "3  chr1:1337637-1341188:-  \n",
       "4  chr1:1341267-1342288:-  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>multimap_junction_reads</th>\n",
       "      <th>max_overhang</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>sample_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [chrom, intron_start, intron_stop, strand, intron_motif, annotated, unique_junction_reads, multimap_junction_reads, max_overhang, intron_location, sample_id]\n",
       "Index: []"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj.loc[sj.intron_location == ubc_middle_intron_location]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ind = sj2psi.chr_start_stop_to_sj_ind('chr12:125394192-125401587', sj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ubc_introns = sj.loc[ind].intron_location.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>unique_junction_reads</th>\n",
       "      <th>$\\Psi_5$</th>\n",
       "      <th>$\\Psi_3$</th>\n",
       "      <th>intron_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sample_id, chrom, intron_stop, strand, intron_start, unique_junction_reads, $\\Psi_5$, $\\Psi_3$, intron_location]\n",
       "Index: []"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_appended.loc[sj_appended.intron_location == ubc_middle_intron_location]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 19.2 s, sys: 706 ms, total: 19.9 s\n",
      "Wall time: 20 s\n"
     ]
    }
   ],
   "source": [
    "folder = '/home/obotvinnik/projects/singlecell_pnms/analysis/csvs_for_paper'\n",
    "%time sj_appended.to_csv('{}/splice_junctions_psi.csv'.format(folder), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 11.3 s, sys: 1.2 s, total: 12.5 s\n",
      "Wall time: 12.5 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "psi5 = sj_appended.pivot(index='sample_id', columns='intron_location', values='$\\Psi_5$')\n",
    "psi3 = sj_appended.pivot(index='sample_id', columns='intron_location', values='$\\Psi_3$')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(189, 247334)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi5.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>intron_location</th>\n",
       "      <th>chr10:100008749-100010821:-</th>\n",
       "      <th>chr10:100010934-100011322:-</th>\n",
       "      <th>chr10:100143626-100144703:-</th>\n",
       "      <th>chr10:100144825-100146957:-</th>\n",
       "      <th>chr10:100147065-100155147:-</th>\n",
       "      <th>chr10:100150512-100150766:-</th>\n",
       "      <th>chr10:100155210-100157101:-</th>\n",
       "      <th>chr10:100157256-100159858:-</th>\n",
       "      <th>chr10:100160015-100167338:-</th>\n",
       "      <th>chr10:100167413-100167660:-</th>\n",
       "      <th>...</th>\n",
       "      <th>chrY:7194211-7209155:+</th>\n",
       "      <th>chrY:7194211-7224175:+</th>\n",
       "      <th>chrY:7198575-7201095:+</th>\n",
       "      <th>chrY:7208064-7209155:+</th>\n",
       "      <th>chrY:7209276-7224175:+</th>\n",
       "      <th>chrY:7223339-7223437:+</th>\n",
       "      <th>chrY:7224272-7235396:+</th>\n",
       "      <th>chrY:7225609-7226951:+</th>\n",
       "      <th>chrY:7235475-7239783:+</th>\n",
       "      <th>chrY:7239931-7243752:+</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CVN_01</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_02</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_03</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_04</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CVN_05</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 247334 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "intron_location  chr10:100008749-100010821:-  chr10:100010934-100011322:-  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                   NaN                          NaN   \n",
       "CVN_02                                   NaN                          NaN   \n",
       "CVN_03                                   NaN                          NaN   \n",
       "CVN_04                                   NaN                          NaN   \n",
       "CVN_05                                   NaN                          NaN   \n",
       "\n",
       "intron_location  chr10:100143626-100144703:-  chr10:100144825-100146957:-  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                   NaN                          NaN   \n",
       "CVN_02                                   NaN                          NaN   \n",
       "CVN_03                                   NaN                          NaN   \n",
       "CVN_04                                   NaN                          NaN   \n",
       "CVN_05                                   NaN                          NaN   \n",
       "\n",
       "intron_location  chr10:100147065-100155147:-  chr10:100150512-100150766:-  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                   NaN                          NaN   \n",
       "CVN_02                                   NaN                          NaN   \n",
       "CVN_03                                   NaN                          NaN   \n",
       "CVN_04                                   NaN                          NaN   \n",
       "CVN_05                                   NaN                          NaN   \n",
       "\n",
       "intron_location  chr10:100155210-100157101:-  chr10:100157256-100159858:-  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                   NaN                          NaN   \n",
       "CVN_02                                   NaN                          NaN   \n",
       "CVN_03                                   NaN                          NaN   \n",
       "CVN_04                                   NaN                          NaN   \n",
       "CVN_05                                   NaN                          NaN   \n",
       "\n",
       "intron_location  chr10:100160015-100167338:-  chr10:100167413-100167660:-  \\\n",
       "sample_id                                                                   \n",
       "CVN_01                                   NaN                          NaN   \n",
       "CVN_02                                   NaN                          NaN   \n",
       "CVN_03                                   NaN                          NaN   \n",
       "CVN_04                                   NaN                          NaN   \n",
       "CVN_05                                   NaN                          NaN   \n",
       "\n",
       "intron_location           ...            chrY:7194211-7209155:+  \\\n",
       "sample_id                 ...                                     \n",
       "CVN_01                    ...                               NaN   \n",
       "CVN_02                    ...                                 0   \n",
       "CVN_03                    ...                               NaN   \n",
       "CVN_04                    ...                               NaN   \n",
       "CVN_05                    ...                               NaN   \n",
       "\n",
       "intron_location  chrY:7194211-7224175:+  chrY:7198575-7201095:+  \\\n",
       "sample_id                                                         \n",
       "CVN_01                              NaN                     NaN   \n",
       "CVN_02                                0                     NaN   \n",
       "CVN_03                              NaN                     NaN   \n",
       "CVN_04                              NaN                     NaN   \n",
       "CVN_05                              NaN                     NaN   \n",
       "\n",
       "intron_location  chrY:7208064-7209155:+  chrY:7209276-7224175:+  \\\n",
       "sample_id                                                         \n",
       "CVN_01                              NaN                     NaN   \n",
       "CVN_02                                1                       1   \n",
       "CVN_03                              NaN                     NaN   \n",
       "CVN_04                              NaN                     NaN   \n",
       "CVN_05                              NaN                     NaN   \n",
       "\n",
       "intron_location  chrY:7223339-7223437:+  chrY:7224272-7235396:+  \\\n",
       "sample_id                                                         \n",
       "CVN_01                              NaN                     NaN   \n",
       "CVN_02                              NaN                       1   \n",
       "CVN_03                              NaN                     NaN   \n",
       "CVN_04                              NaN                     NaN   \n",
       "CVN_05                              NaN                     NaN   \n",
       "\n",
       "intron_location  chrY:7225609-7226951:+  chrY:7235475-7239783:+  \\\n",
       "sample_id                                                         \n",
       "CVN_01                              NaN                     NaN   \n",
       "CVN_02                              NaN                       1   \n",
       "CVN_03                              NaN                     NaN   \n",
       "CVN_04                              NaN                     NaN   \n",
       "CVN_05                              NaN                     NaN   \n",
       "\n",
       "intron_location  chrY:7239931-7243752:+  \n",
       "sample_id                                \n",
       "CVN_01                              NaN  \n",
       "CVN_02                                1  \n",
       "CVN_03                              NaN  \n",
       "CVN_04                              NaN  \n",
       "CVN_05                              NaN  \n",
       "\n",
       "[5 rows x 247334 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi5.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Combine `psi5` and `psi3`\n",
    "Add indicator to the ID whther this is the 5' or 3' splice site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "psi5.columns = psi5.columns.map(lambda x: x + '|5p')\n",
    "psi3.columns = psi3.columns.map(lambda x: x + '|3p')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Concatenate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(189, 494668)"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi_combined = pd.concat([psi5, psi3], axis=1)\n",
    "psi_combined.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Write stressed samples separately\n",
    "\n",
    "do this later - stressed samples SJ.out.files got purged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# psi5_stressed = psi5.loc[stressed_samples, :]\n",
    "# psi3_stressed = psi3.loc[stressed_samples, :]\n",
    "# psi_combined_stressed = psi_combined.loc[stressed_samples, :]\n",
    "# %%time\n",
    "# psi5_stressed.to_csv('{}/psi5_stressed.csv'.format(folder))\n",
    "# psi3_stressed.to_csv('{}/psi3_stressed.csv'.format(folder))\n",
    "# psi_combined_stressed.to_csv('{}/psi5_psi3_combined.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psi5_dropped = psi5.drop(samples_to_drop, errors='ignore').sort_index().dropna(how='all', axis=1)\n",
    "psi3_dropped = psi3.drop(samples_to_drop, errors='ignore').sort_index().dropna(how='all', axis=1)\n",
    "psi_combined_dropped = psi_combined.drop(samples_to_drop, errors='ignore').sort_index().dropna(how='all', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(189, 247334)\n",
      "(189, 247334)\n",
      "(189, 494668)\n"
     ]
    }
   ],
   "source": [
    "print psi5_dropped.shape\n",
    "print psi3_dropped.shape\n",
    "print psi_combined_dropped.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 5min 5s, sys: 15.3 s, total: 5min 20s\n",
      "Wall time: 5min 20s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "psi5_dropped.to_csv('{}/psi5.csv'.format(folder))\n",
    "psi3_dropped.to_csv('{}/psi3.csv'.format(folder))\n",
    "psi_combined_dropped.to_csv('{}/psi5_psi3_combined.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "psi_combined_dropped.to_hdf('{}/psi5_psi3_combined.hdf'.format(folder), 'psi', format='fixed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1.3G\r\n",
      "drwxr-xr-x  2 obotvinnik yeo-group 4.0K Aug 13 10:01 \u001b[0m\u001b[01;34m.\u001b[0m/\r\n",
      "drwxr-xr-x 17 obotvinnik yeo-group  12K May 27 16:14 \u001b[01;34m..\u001b[0m/\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  41M Aug 13 10:45 expression.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 7.2M Aug 13 10:45 expression_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  63K Aug 13 10:45 mapping_stats.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  12K Aug 13 10:45 mapping_stats_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  11K Aug 13 10:45 metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 1.9K Aug 13 10:45 metadata_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  98M Aug 13 14:42 psi3.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  23M Aug 13 14:38 psi3_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  98M Aug 13 14:41 psi5.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 195M Aug 13 14:46 psi5_psi3_combined.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 187M Aug 14 09:00 psi5_psi3_combined_metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  23M Aug 13 14:38 psi5_stressed.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group  67M Aug 14 09:00 splice_junction_metadata.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 578M Aug 13 14:33 splice_junctions_psi.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 3.2M Aug 13 10:45 splicing.csv\r\n",
      "-rw-r--r--  1 obotvinnik yeo-group 615K Aug 13 10:45 splicing_stressed.csv\r\n",
      "\u001b[m"
     ]
    }
   ],
   "source": [
    "ls -lha $folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata.to_csv('{}/splice_junction_metadata.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gffutils\n",
    "import pybedtools\n",
    "import re\n",
    "import numpy as np\n",
    "\n",
    "v19db_filename = '/projects/ps-yeolab/genomes/hg19/gencode/v19/gencode.v19.annotation.gtf.db'\n",
    "v19db = gffutils.FeatureDB(v19db_filename)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "562680"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum(1 for _ in v19db.features_of_type('exon'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232.4\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237613.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000268020.2\"; transcript_id \"EN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0       1     2      3      4  5  6  7  \\\n",
       "0  chr1  HAVANA  gene  11869  14412  .  +  .   \n",
       "1  chr1  HAVANA  gene  14363  29806  .  -  .   \n",
       "2  chr1  HAVANA  gene  29554  31109  .  +  .   \n",
       "3  chr1  HAVANA  gene  34554  36081  .  -  .   \n",
       "4  chr1  HAVANA  gene  52473  54936  .  +  .   \n",
       "\n",
       "                                                   8  \n",
       "0  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...  \n",
       "1  gene_id \"ENSG00000227232.4\"; transcript_id \"EN...  \n",
       "2  gene_id \"ENSG00000243485.2\"; transcript_id \"EN...  \n",
       "3  gene_id \"ENSG00000237613.2\"; transcript_id \"EN...  \n",
       "4  gene_id \"ENSG00000268020.2\"; transcript_id \"EN...  "
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = pd.read_table('/projects/ps-yeolab/genomes/hg19/gencode/v19/gencode.v19.annotation.gene.gtf', header=None)\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def extract_attributes(attribute_str):\n",
    "    split = attribute_str.rstrip(';').split(';')\n",
    "    individual_attributes = map(lambda x: map(lambda y: y.strip(' \"'), x.split()), split)\n",
    "#     print individual_attributes\n",
    "    return dict(individual_attributes)\n",
    "    \n",
    "attributes_df = pd.DataFrame.from_records(genes[8].map(extract_attributes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(57820, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_id   gene_name gene_status   gene_type  \\\n",
       "0  ENSG00000223972.4     DDX11L1       KNOWN  pseudogene   \n",
       "1  ENSG00000227232.4      WASH7P       KNOWN  pseudogene   \n",
       "2  ENSG00000243485.2  MIR1302-11       NOVEL     lincRNA   \n",
       "3  ENSG00000237613.2     FAM138A       KNOWN     lincRNA   \n",
       "4  ENSG00000268020.2      OR4G4P       KNOWN  pseudogene   \n",
       "\n",
       "            havana_gene level  tag      transcript_id transcript_name  \\\n",
       "0  OTTHUMG00000000961.2     2  NaN  ENSG00000223972.4         DDX11L1   \n",
       "1  OTTHUMG00000000958.1     2  NaN  ENSG00000227232.4          WASH7P   \n",
       "2  OTTHUMG00000000959.2     2  NaN  ENSG00000243485.2      MIR1302-11   \n",
       "3  OTTHUMG00000000960.1     2  NaN  ENSG00000237613.2         FAM138A   \n",
       "4  OTTHUMG00000185779.1     2  NaN  ENSG00000268020.2          OR4G4P   \n",
       "\n",
       "  transcript_status transcript_type  \n",
       "0             KNOWN      pseudogene  \n",
       "1             KNOWN      pseudogene  \n",
       "2             NOVEL         lincRNA  \n",
       "3             KNOWN         lincRNA  \n",
       "4             KNOWN      pseudogene  "
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print attributes_df.shape\n",
    "attributes_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000223972.4\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000227232.4\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000243485.2\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000237613.2\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSG00000268020.2\"; transcript_id \"EN...</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0       1     2      3      4  5  6  7  \\\n",
       "0  chr1  HAVANA  gene  11869  14412  .  +  .   \n",
       "1  chr1  HAVANA  gene  14363  29806  .  -  .   \n",
       "2  chr1  HAVANA  gene  29554  31109  .  +  .   \n",
       "3  chr1  HAVANA  gene  34554  36081  .  -  .   \n",
       "4  chr1  HAVANA  gene  52473  54936  .  +  .   \n",
       "\n",
       "                                                   8            gene_id  \\\n",
       "0  gene_id \"ENSG00000223972.4\"; transcript_id \"EN...  ENSG00000223972.4   \n",
       "1  gene_id \"ENSG00000227232.4\"; transcript_id \"EN...  ENSG00000227232.4   \n",
       "2  gene_id \"ENSG00000243485.2\"; transcript_id \"EN...  ENSG00000243485.2   \n",
       "3  gene_id \"ENSG00000237613.2\"; transcript_id \"EN...  ENSG00000237613.2   \n",
       "4  gene_id \"ENSG00000268020.2\"; transcript_id \"EN...  ENSG00000268020.2   \n",
       "\n",
       "    gene_name gene_status   gene_type           havana_gene level  tag  \\\n",
       "0     DDX11L1       KNOWN  pseudogene  OTTHUMG00000000961.2     2  NaN   \n",
       "1      WASH7P       KNOWN  pseudogene  OTTHUMG00000000958.1     2  NaN   \n",
       "2  MIR1302-11       NOVEL     lincRNA  OTTHUMG00000000959.2     2  NaN   \n",
       "3     FAM138A       KNOWN     lincRNA  OTTHUMG00000000960.1     2  NaN   \n",
       "4      OR4G4P       KNOWN  pseudogene  OTTHUMG00000185779.1     2  NaN   \n",
       "\n",
       "       transcript_id transcript_name transcript_status transcript_type  \n",
       "0  ENSG00000223972.4         DDX11L1             KNOWN      pseudogene  \n",
       "1  ENSG00000227232.4          WASH7P             KNOWN      pseudogene  \n",
       "2  ENSG00000243485.2      MIR1302-11             NOVEL         lincRNA  \n",
       "3  ENSG00000237613.2         FAM138A             KNOWN         lincRNA  \n",
       "4  ENSG00000268020.2          OR4G4P             KNOWN      pseudogene  "
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = pd.concat([genes, attributes_df], axis=1)\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0       1     2      3      4  5  6  7            gene_id   gene_name  \\\n",
       "0  chr1  HAVANA  gene  11869  14412  .  +  .  ENSG00000223972.4     DDX11L1   \n",
       "1  chr1  HAVANA  gene  14363  29806  .  -  .  ENSG00000227232.4      WASH7P   \n",
       "2  chr1  HAVANA  gene  29554  31109  .  +  .  ENSG00000243485.2  MIR1302-11   \n",
       "3  chr1  HAVANA  gene  34554  36081  .  -  .  ENSG00000237613.2     FAM138A   \n",
       "4  chr1  HAVANA  gene  52473  54936  .  +  .  ENSG00000268020.2      OR4G4P   \n",
       "\n",
       "  gene_status   gene_type           havana_gene level  tag      transcript_id  \\\n",
       "0       KNOWN  pseudogene  OTTHUMG00000000961.2     2  NaN  ENSG00000223972.4   \n",
       "1       KNOWN  pseudogene  OTTHUMG00000000958.1     2  NaN  ENSG00000227232.4   \n",
       "2       NOVEL     lincRNA  OTTHUMG00000000959.2     2  NaN  ENSG00000243485.2   \n",
       "3       KNOWN     lincRNA  OTTHUMG00000000960.1     2  NaN  ENSG00000237613.2   \n",
       "4       KNOWN  pseudogene  OTTHUMG00000185779.1     2  NaN  ENSG00000268020.2   \n",
       "\n",
       "  transcript_name transcript_status transcript_type  \n",
       "0         DDX11L1             KNOWN      pseudogene  \n",
       "1          WASH7P             KNOWN      pseudogene  \n",
       "2      MIR1302-11             NOVEL         lincRNA  \n",
       "3         FAM138A             KNOWN         lincRNA  \n",
       "4          OR4G4P             KNOWN      pseudogene  "
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = genes.drop(8, axis=1)\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>source</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>5</th>\n",
       "      <th>strand</th>\n",
       "      <th>7</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14412</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000961.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000223972.4</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000959.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000243485.2</td>\n",
       "      <td>MIR1302-11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>34554</td>\n",
       "      <td>36081</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000000960.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237613.2</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>52473</td>\n",
       "      <td>54936</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000185779.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000268020.2</td>\n",
       "      <td>OR4G4P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  source feature_type  start   stop  5 strand  7            gene_id  \\\n",
       "0  chr1  HAVANA         gene  11869  14412  .      +  .  ENSG00000223972.4   \n",
       "1  chr1  HAVANA         gene  14363  29806  .      -  .  ENSG00000227232.4   \n",
       "2  chr1  HAVANA         gene  29554  31109  .      +  .  ENSG00000243485.2   \n",
       "3  chr1  HAVANA         gene  34554  36081  .      -  .  ENSG00000237613.2   \n",
       "4  chr1  HAVANA         gene  52473  54936  .      +  .  ENSG00000268020.2   \n",
       "\n",
       "    gene_name gene_status   gene_type           havana_gene level  tag  \\\n",
       "0     DDX11L1       KNOWN  pseudogene  OTTHUMG00000000961.2     2  NaN   \n",
       "1      WASH7P       KNOWN  pseudogene  OTTHUMG00000000958.1     2  NaN   \n",
       "2  MIR1302-11       NOVEL     lincRNA  OTTHUMG00000000959.2     2  NaN   \n",
       "3     FAM138A       KNOWN     lincRNA  OTTHUMG00000000960.1     2  NaN   \n",
       "4      OR4G4P       KNOWN  pseudogene  OTTHUMG00000185779.1     2  NaN   \n",
       "\n",
       "       transcript_id transcript_name transcript_status transcript_type  \n",
       "0  ENSG00000223972.4         DDX11L1             KNOWN      pseudogene  \n",
       "1  ENSG00000227232.4          WASH7P             KNOWN      pseudogene  \n",
       "2  ENSG00000243485.2      MIR1302-11             NOVEL         lincRNA  \n",
       "3  ENSG00000237613.2         FAM138A             KNOWN         lincRNA  \n",
       "4  ENSG00000268020.2          OR4G4P             KNOWN      pseudogene  "
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes = genes.rename(columns={0: 'chrom', 1:'source', 2: 'feature_type', 3: 'start', 4: 'stop', 6: 'strand'})\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\t0\n",
      "\t"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "sj_metadata['gencode_id'] = ''\n",
    "sj_metadata['ensembl_id'] = ''\n",
    "for col in attributes_df.columns:\n",
    "    sj_metadata[col] = ''\n",
    "\n",
    "for i, row in genes.iterrows():\n",
    "    if i % 1000 == 0:\n",
    "        print '\\t', i\n",
    "    chrom = sj_metadata.chrom == row.chrom\n",
    "    start = sj_metadata.intron_start > row.start\n",
    "    stop = sj_metadata.intron_stop < row.stop\n",
    "    strand = sj_metadata.strand == row.strand\n",
    "    \n",
    "    rows = chrom & start & stop & strand\n",
    "    if rows.sum() == 0:\n",
    "        continue\n",
    "#     print gene\n",
    "    gencode_id = row.gene_id\n",
    "    ensembl_id = gencode_id.split('.')[:1][0]\n",
    "    \n",
    "    sj_metadata.loc[rows, 'gencode_id'] = sj_metadata.loc[rows, 'gencode_id'] + ',' + gencode_id\n",
    "    sj_metadata.loc[rows, 'ensembl_id'] = sj_metadata.loc[rows, 'ensembl_id'] + ',' + ensembl_id\n",
    "    for col, value in row[attributes_df.columns].dropna().iteritems():\n",
    "        sj_metadata.loc[rows, col] = sj_metadata.loc[rows, col] + ',' + value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6416034</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147733653</td>\n",
       "      <td>147743428</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chrX:147733653-147743428:+</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,ENSG00000155966</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "      <td>,OTTHUMG00000022613.2</td>\n",
       "      <td>,2</td>\n",
       "      <td></td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6416038</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147924940</td>\n",
       "      <td>147964960</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX:147924940-147964960:+</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,ENSG00000155966</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "      <td>,OTTHUMG00000022613.2</td>\n",
       "      <td>,2</td>\n",
       "      <td></td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6416039</th>\n",
       "      <td>chrX</td>\n",
       "      <td>147965051</td>\n",
       "      <td>147967418</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX:147965051-147967418:+</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,ENSG00000155966</td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "      <td>,OTTHUMG00000022613.2</td>\n",
       "      <td>,2</td>\n",
       "      <td></td>\n",
       "      <td>,ENSG00000155966.9</td>\n",
       "      <td>,AFF2</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6416043</th>\n",
       "      <td>chrX</td>\n",
       "      <td>152420132</td>\n",
       "      <td>152423800</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX:152420132-152423800:+</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6416101</th>\n",
       "      <td>chrY</td>\n",
       "      <td>15511932</td>\n",
       "      <td>15512429</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chrY:15511932-15512429:-</td>\n",
       "      <td>,ENSG00000183878.11</td>\n",
       "      <td>,ENSG00000183878</td>\n",
       "      <td>,ENSG00000183878.11</td>\n",
       "      <td>,UTY</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "      <td>,OTTHUMG00000036319.1</td>\n",
       "      <td>,2</td>\n",
       "      <td></td>\n",
       "      <td>,ENSG00000183878.11</td>\n",
       "      <td>,UTY</td>\n",
       "      <td>,KNOWN</td>\n",
       "      <td>,protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "6416034  chrX     147733653    147743428      +        GT/AG      True   \n",
       "6416038  chrX     147924940    147964960      +        GT/AG     False   \n",
       "6416039  chrX     147965051    147967418      +        GT/AG     False   \n",
       "6416043  chrX     152420132    152423800      +        GT/AG     False   \n",
       "6416101  chrY      15511932     15512429      -        GT/AG     False   \n",
       "\n",
       "                    intron_location           gencode_id        ensembl_id  \\\n",
       "6416034  chrX:147733653-147743428:+   ,ENSG00000155966.9  ,ENSG00000155966   \n",
       "6416038  chrX:147924940-147964960:+   ,ENSG00000155966.9  ,ENSG00000155966   \n",
       "6416039  chrX:147965051-147967418:+   ,ENSG00000155966.9  ,ENSG00000155966   \n",
       "6416043  chrX:152420132-152423800:+                                          \n",
       "6416101    chrY:15511932-15512429:-  ,ENSG00000183878.11  ,ENSG00000183878   \n",
       "\n",
       "                     gene_id gene_name gene_status        gene_type  \\\n",
       "6416034   ,ENSG00000155966.9     ,AFF2      ,KNOWN  ,protein_coding   \n",
       "6416038   ,ENSG00000155966.9     ,AFF2      ,KNOWN  ,protein_coding   \n",
       "6416039   ,ENSG00000155966.9     ,AFF2      ,KNOWN  ,protein_coding   \n",
       "6416043                                                               \n",
       "6416101  ,ENSG00000183878.11      ,UTY      ,KNOWN  ,protein_coding   \n",
       "\n",
       "                   havana_gene level tag        transcript_id transcript_name  \\\n",
       "6416034  ,OTTHUMG00000022613.2    ,2       ,ENSG00000155966.9           ,AFF2   \n",
       "6416038  ,OTTHUMG00000022613.2    ,2       ,ENSG00000155966.9           ,AFF2   \n",
       "6416039  ,OTTHUMG00000022613.2    ,2       ,ENSG00000155966.9           ,AFF2   \n",
       "6416043                                                                         \n",
       "6416101  ,OTTHUMG00000036319.1    ,2      ,ENSG00000183878.11            ,UTY   \n",
       "\n",
       "        transcript_status  transcript_type  \n",
       "6416034            ,KNOWN  ,protein_coding  \n",
       "6416038            ,KNOWN  ,protein_coding  \n",
       "6416039            ,KNOWN  ,protein_coding  \n",
       "6416043                                     \n",
       "6416101            ,KNOWN  ,protein_coding  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000001096.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "      <td>OTTHUMG00000156968.5,OTTHUMG00000002861.2</td>\n",
       "      <td>2,1</td>\n",
       "      <td>pseudo_consens</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>RP4-669L17.10</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000156968.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>RP4-669L17.10</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>RP5-857K21.4</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002331.4</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>RP5-857K21.4</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            gencode_id                       ensembl_id  \\\n",
       "0                    ENSG00000238009.2                  ENSG00000238009   \n",
       "1                    ENSG00000237683.5                  ENSG00000237683   \n",
       "2  ENSG00000237094.7,ENSG00000250575.1  ENSG00000237094,ENSG00000250575   \n",
       "3                    ENSG00000237094.7                  ENSG00000237094   \n",
       "4                    ENSG00000230021.3                  ENSG00000230021   \n",
       "\n",
       "                               gene_id                   gene_name  \\\n",
       "0                    ENSG00000238009.2                RP11-34P13.7   \n",
       "1                    ENSG00000237683.5                  AL627309.1   \n",
       "2  ENSG00000237094.7,ENSG00000250575.1  RP4-669L17.10,RP4-669L17.8   \n",
       "3                    ENSG00000237094.7               RP4-669L17.10   \n",
       "4                    ENSG00000230021.3                RP5-857K21.4   \n",
       "\n",
       "   gene_status           gene_type                                havana_gene  \\\n",
       "0        NOVEL             lincRNA                       OTTHUMG00000001096.2   \n",
       "1        KNOWN      protein_coding                                        NaN   \n",
       "2  NOVEL,KNOWN  lincRNA,pseudogene  OTTHUMG00000156968.5,OTTHUMG00000002861.2   \n",
       "3        NOVEL             lincRNA                       OTTHUMG00000156968.5   \n",
       "4        NOVEL             lincRNA                       OTTHUMG00000002331.4   \n",
       "\n",
       "  level             tag                        transcript_id  \\\n",
       "0     2             NaN                    ENSG00000238009.2   \n",
       "1     3             NaN                    ENSG00000237683.5   \n",
       "2   2,1  pseudo_consens  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3     2             NaN                    ENSG00000237094.7   \n",
       "4     2             NaN                    ENSG00000230021.3   \n",
       "\n",
       "              transcript_name transcript_status     transcript_type  \n",
       "0                RP11-34P13.7             NOVEL             lincRNA  \n",
       "1                  AL627309.1             KNOWN      protein_coding  \n",
       "2  RP4-669L17.10,RP4-669L17.8       NOVEL,KNOWN  lincRNA,pseudogene  \n",
       "3               RP4-669L17.10             NOVEL             lincRNA  \n",
       "4                RP5-857K21.4             NOVEL             lincRNA  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "cols = ['gencode_id', 'ensembl_id'] + attributes_df.columns.tolist()\n",
    "\n",
    "sj_metadata[cols] = sj_metadata[cols].applymap(lambda x: x.lstrip(',') if isinstance(x, str) else x)\n",
    "sj_metadata[cols] = sj_metadata[cols].replace('', np.nan)\n",
    "sj_metadata[cols].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:326417-327587:+</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "      <td>OTTHUMG00000156968.5,OTTHUMG00000002861.2</td>\n",
       "      <td>2,1</td>\n",
       "      <td>pseudo_consens</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1634439</td>\n",
       "      <td>1634518</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1634439-1634518:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1643840</td>\n",
       "      <td>1647784</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1643840-1647784:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1647918</td>\n",
       "      <td>1650766</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1647918-1650766:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1650895</td>\n",
       "      <td>1653034</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1650895-1653034:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1653151</td>\n",
       "      <td>1654146</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1653151-1654146:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1654271</td>\n",
       "      <td>1655696</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:1654271-1655696:-</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000000703.14,OTTHUMG00000183552.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10490626</td>\n",
       "      <td>10493898</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:10490626-10493898:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10490626</td>\n",
       "      <td>10500403</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:10490626-10500403:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10491459</td>\n",
       "      <td>10493898</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:10491459-10493898:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10494023</td>\n",
       "      <td>10494713</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:10494023-10494713:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10494748</td>\n",
       "      <td>10500403</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:10494748-10500403:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10500471</td>\n",
       "      <td>10502321</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:10500471-10502321:+</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059085.8,OTTHUMG00000162436.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19923604</td>\n",
       "      <td>19948593</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:19923604-19948593:+</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000183956.1,OTTHUMG00000002712.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19948642</td>\n",
       "      <td>19949967</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:19948642-19949967:+</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000183956.1,OTTHUMG00000002712.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19950078</td>\n",
       "      <td>19952880</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:19950078-19952880:+</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000183956.1,OTTHUMG00000002712.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28832597</td>\n",
       "      <td>28834639</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:28832597-28834639:+</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000003645.10,OTTHUMG00000003657.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28832613</td>\n",
       "      <td>28834639</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:28832613-28834639:+</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000003645.10,OTTHUMG00000003657.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28834673</td>\n",
       "      <td>28835341</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:28834673-28835341:+</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000003645.10,OTTHUMG00000003657.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32374562</td>\n",
       "      <td>32375638</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:32374562-32375638:-</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000184139.1,OTTHUMG00000003801.9</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>505</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32375714</td>\n",
       "      <td>32377296</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:32375714-32377296:-</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000184139.1,OTTHUMG00000003801.9</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>506</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32377428</td>\n",
       "      <td>32381495</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:32377428-32381495:-</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000184139.1,OTTHUMG00000003801.9</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>507</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32381589</td>\n",
       "      <td>32384570</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:32381589-32384570:-</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000184139.1,OTTHUMG00000003801.9</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35447721</td>\n",
       "      <td>35449367</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35447721-35449367:-</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>ENSG00000241014,ENSG00000271741,ENSG00000243749</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB</td>\n",
       "      <td>NOVEL,NOVEL,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "      <td>OTTHUMG00000004046.3,OTTHUMG00000185379.1,OTTH...</td>\n",
       "      <td>2,2,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB</td>\n",
       "      <td>NOVEL,NOVEL,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>564</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35449549</td>\n",
       "      <td>35450798</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35449549-35450798:-</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>ENSG00000241014,ENSG00000271741,ENSG0000024374...</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6</td>\n",
       "      <td>NOVEL,NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "      <td>OTTHUMG00000004046.3,OTTHUMG00000185379.1,OTTH...</td>\n",
       "      <td>2,2,1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6</td>\n",
       "      <td>NOVEL,NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35454537</td>\n",
       "      <td>35457834</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35454537-35457834:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>566</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35457989</td>\n",
       "      <td>35469568</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35457989-35469568:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>567</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35469607</td>\n",
       "      <td>35470732</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35469607-35470732:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>568</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35470874</td>\n",
       "      <td>35472551</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35470874-35472551:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35472700</td>\n",
       "      <td>35474352</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:35472700-35474352:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6313120</th>\n",
       "      <td>chr12</td>\n",
       "      <td>110837706</td>\n",
       "      <td>110837773</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr12:110837706-110837773:-</td>\n",
       "      <td>ENSG00000196510.8,ENSG00000258210.1</td>\n",
       "      <td>ENSG00000196510,ENSG00000258210</td>\n",
       "      <td>ENSG00000196510.8,ENSG00000258210.1</td>\n",
       "      <td>ANAPC7,RP11-478C19.2</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "      <td>OTTHUMG00000157009.5,OTTHUMG00000169522.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000196510.8,ENSG00000258210.1</td>\n",
       "      <td>ANAPC7,RP11-478C19.2</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6313340</th>\n",
       "      <td>chr12</td>\n",
       "      <td>122234149</td>\n",
       "      <td>122234818</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr12:122234149-122234818:-</td>\n",
       "      <td>ENSG00000139725.3,ENSG00000212694.4</td>\n",
       "      <td>ENSG00000139725,ENSG00000212694</td>\n",
       "      <td>ENSG00000139725.3,ENSG00000212694.4</td>\n",
       "      <td>RHOF,AC084018.1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,lincRNA</td>\n",
       "      <td>OTTHUMG00000169077.3,OTTHUMG00000128905.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000139725.3,ENSG00000212694.4</td>\n",
       "      <td>RHOF,AC084018.1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6313578</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951348</td>\n",
       "      <td>21951639</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr13:21951348-21951639:-</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000017410.1,OTTHUMG00000046192.1</td>\n",
       "      <td>2,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6313579</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951348</td>\n",
       "      <td>21951805</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr13:21951348-21951805:-</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000017410.1,OTTHUMG00000046192.1</td>\n",
       "      <td>2,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6313585</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951682</td>\n",
       "      <td>21951763</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr13:21951682-21951763:-</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000017410.1,OTTHUMG00000046192.1</td>\n",
       "      <td>2,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6314556</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24633344</td>\n",
       "      <td>24633795</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr14:24633344-24633795:+</td>\n",
       "      <td>ENSG00000259529.1,ENSG00000213928.4</td>\n",
       "      <td>ENSG00000259529,ENSG00000213928</td>\n",
       "      <td>ENSG00000259529.1,ENSG00000213928.4</td>\n",
       "      <td>RP11-468E2.4,IRF9</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000171951.1,OTTHUMG00000028799.3</td>\n",
       "      <td>2,1</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000259529.1,ENSG00000213928.4</td>\n",
       "      <td>RP11-468E2.4,IRF9</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6314571</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24683358</td>\n",
       "      <td>24683453</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr14:24683358-24683453:-</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>ENSG00000260669,ENSG00000213920,ENSG00000255526</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "      <td>OTTHUMG00000029324.4,OTTHUMG00000133477.4,OTTH...</td>\n",
       "      <td>2,2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6314576</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24683806</td>\n",
       "      <td>24685124</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr14:24683806-24685124:-</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>ENSG00000260669,ENSG00000213920,ENSG00000255526</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "      <td>OTTHUMG00000029324.4,OTTHUMG00000133477.4,OTTH...</td>\n",
       "      <td>2,2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>NOVEL,KNOWN,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding,protein_co...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6315008</th>\n",
       "      <td>chr14</td>\n",
       "      <td>62229307</td>\n",
       "      <td>62242812</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr14:62229307-62242812:+</td>\n",
       "      <td>ENSG00000258964.1,ENSG00000023608.4</td>\n",
       "      <td>ENSG00000258964,ENSG00000023608</td>\n",
       "      <td>ENSG00000258964.1,ENSG00000023608.4</td>\n",
       "      <td>RP11-618G20.1,SNAPC1</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding</td>\n",
       "      <td>OTTHUMG00000171115.1,OTTHUMG00000140343.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000258964.1,ENSG00000023608.4</td>\n",
       "      <td>RP11-618G20.1,SNAPC1</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>processed_transcript,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6315072</th>\n",
       "      <td>chr14</td>\n",
       "      <td>65393042</td>\n",
       "      <td>65395245</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr14:65393042-65395245:+</td>\n",
       "      <td>ENSG00000258289.3,ENSG00000257365.3,ENSG000001...</td>\n",
       "      <td>ENSG00000258289,ENSG00000257365,ENSG00000125954</td>\n",
       "      <td>ENSG00000258289.3,ENSG00000257365.3,ENSG000001...</td>\n",
       "      <td>CHURC1,FNTB,CHURC1-FNTB</td>\n",
       "      <td>KNOWN,KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000170218.5,OTTHUMG00000142811.4,OTTH...</td>\n",
       "      <td>1,1,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000258289.3,ENSG00000257365.3,ENSG000001...</td>\n",
       "      <td>CHURC1,FNTB,CHURC1-FNTB</td>\n",
       "      <td>KNOWN,KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6315722</th>\n",
       "      <td>chr15</td>\n",
       "      <td>25265872</td>\n",
       "      <td>25282858</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr15:25265872-25282858:+</td>\n",
       "      <td>ENSG00000224078.8,ENSG00000272846.1</td>\n",
       "      <td>ENSG00000224078,ENSG00000272846</td>\n",
       "      <td>ENSG00000224078.8,ENSG00000272846.1</td>\n",
       "      <td>SNHG14,RP11-701H24.10</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>processed_transcript,processed_transcript</td>\n",
       "      <td>OTTHUMG00000056661.6,OTTHUMG00000186505.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000224078.8,ENSG00000272846.1</td>\n",
       "      <td>SNHG14,RP11-701H24.10</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>processed_transcript,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6316003</th>\n",
       "      <td>chr15</td>\n",
       "      <td>43942099</td>\n",
       "      <td>43955851</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr15:43942099-43955851:-</td>\n",
       "      <td>ENSG00000242866.5,ENSG00000166762.12</td>\n",
       "      <td>ENSG00000242866,ENSG00000166762</td>\n",
       "      <td>ENSG00000242866.5,ENSG00000166762.12</td>\n",
       "      <td>STRC,CATSPER2</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000059899.5,OTTHUMG00000059902.10</td>\n",
       "      <td>2,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000242866.5,ENSG00000166762.12</td>\n",
       "      <td>STRC,CATSPER2</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6316239</th>\n",
       "      <td>chr15</td>\n",
       "      <td>59561890</td>\n",
       "      <td>59562581</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr15:59561890-59562581:-</td>\n",
       "      <td>ENSG00000157483.4,ENSG00000259771.1</td>\n",
       "      <td>ENSG00000157483,ENSG00000259771</td>\n",
       "      <td>ENSG00000157483.4,ENSG00000259771.1</td>\n",
       "      <td>MYO1E,RP11-429D19.1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000171944.1,OTTHUMG00000171945.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000157483.4,ENSG00000259771.1</td>\n",
       "      <td>MYO1E,RP11-429D19.1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6319193</th>\n",
       "      <td>chr17</td>\n",
       "      <td>41108343</td>\n",
       "      <td>41108420</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr17:41108343-41108420:-</td>\n",
       "      <td>ENSG00000108825.13,ENSG00000266967.2</td>\n",
       "      <td>ENSG00000108825,ENSG00000266967</td>\n",
       "      <td>ENSG00000108825.13,ENSG00000266967.2</td>\n",
       "      <td>PTGES3L-AARSD1,AARSD1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000180905.1,OTTHUMG00000153515.7</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000108825.13,ENSG00000266967.2</td>\n",
       "      <td>PTGES3L-AARSD1,AARSD1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6319615</th>\n",
       "      <td>chr17</td>\n",
       "      <td>57279061</td>\n",
       "      <td>57280856</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr17:57279061-57280856:+</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>ENSG00000068489,ENSG00000265303</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000179362.1,OTTHUMG00000179375.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6319616</th>\n",
       "      <td>chr17</td>\n",
       "      <td>57279095</td>\n",
       "      <td>57280796</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr17:57279095-57280796:+</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>ENSG00000068489,ENSG00000265303</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000179362.1,OTTHUMG00000179375.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6321434</th>\n",
       "      <td>chr19</td>\n",
       "      <td>23455185</td>\n",
       "      <td>23456877</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr19:23455185-23456877:-</td>\n",
       "      <td>ENSG00000269107.1,ENSG00000213971.3</td>\n",
       "      <td>ENSG00000269107,ENSG00000213971</td>\n",
       "      <td>ENSG00000269107.1,ENSG00000213971.3</td>\n",
       "      <td>RP11-15H20.7,RP11-15H20.6</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>lincRNA,processed_transcript</td>\n",
       "      <td>OTTHUMG00000183234.1,OTTHUMG00000183235.2</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000269107.1,ENSG00000213971.3</td>\n",
       "      <td>RP11-15H20.7,RP11-15H20.6</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>lincRNA,processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6321996</th>\n",
       "      <td>chr19</td>\n",
       "      <td>50925803</td>\n",
       "      <td>50926861</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr19:50925803-50926861:+</td>\n",
       "      <td>ENSG00000142539.9,ENSG00000269404.2</td>\n",
       "      <td>ENSG00000142539,ENSG00000269404</td>\n",
       "      <td>ENSG00000142539.9,ENSG00000269404.2</td>\n",
       "      <td>CTD-2545M3.6,SPIB</td>\n",
       "      <td>PUTATIVE,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000183038.1,OTTHUMG00000183039.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000142539.9,ENSG00000269404.2</td>\n",
       "      <td>CTD-2545M3.6,SPIB</td>\n",
       "      <td>PUTATIVE,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6322633</th>\n",
       "      <td>chr20</td>\n",
       "      <td>34246937</td>\n",
       "      <td>34252155</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr20:34246937-34252155:-</td>\n",
       "      <td>ENSG00000214078.7,ENSG00000272897.1,ENSG000002...</td>\n",
       "      <td>ENSG00000214078,ENSG00000272897,ENSG00000244462</td>\n",
       "      <td>ENSG00000214078.7,ENSG00000272897.1,ENSG000002...</td>\n",
       "      <td>CPNE1,RP1-309K20.6,RBM12</td>\n",
       "      <td>KNOWN,NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000032356.31,OTTHUMG00000186060.1,OTT...</td>\n",
       "      <td>1,2,2</td>\n",
       "      <td>ncRNA_host,ncRNA_host</td>\n",
       "      <td>ENSG00000214078.7,ENSG00000272897.1,ENSG000002...</td>\n",
       "      <td>CPNE1,RP1-309K20.6,RBM12</td>\n",
       "      <td>KNOWN,NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6323440</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47882677</td>\n",
       "      <td>47882978</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr21:47882677-47882978:+</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>ENSG00000160305,ENSG00000223692</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000090717.3,OTTHUMG00000090716.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6323441</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47883142</td>\n",
       "      <td>47889058</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr21:47883142-47889058:+</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>ENSG00000160305,ENSG00000223692</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "      <td>OTTHUMG00000090717.3,OTTHUMG00000090716.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,sense_intronic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6323825</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38747005</td>\n",
       "      <td>38757403</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr22:38747005-38757403:-</td>\n",
       "      <td>ENSG00000213923.6,ENSG00000244627.1</td>\n",
       "      <td>ENSG00000213923,ENSG00000244627</td>\n",
       "      <td>ENSG00000213923.6,ENSG00000244627.1</td>\n",
       "      <td>CSNK1E,RP3-449O17.1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,pseudogene</td>\n",
       "      <td>OTTHUMG00000151135.2,OTTHUMG00000151245.2</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000213923.6,ENSG00000244627.1</td>\n",
       "      <td>CSNK1E,RP3-449O17.1</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6324222</th>\n",
       "      <td>chrX</td>\n",
       "      <td>37214152</td>\n",
       "      <td>37214592</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chrX:37214152-37214592:+</td>\n",
       "      <td>ENSG00000130962.13,ENSG00000250349.3</td>\n",
       "      <td>ENSG00000130962,ENSG00000250349</td>\n",
       "      <td>ENSG00000130962.13,ENSG00000250349.3</td>\n",
       "      <td>PRRG1,TM4SF2</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000021360.2,OTTHUMG00000161009.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000130962.13,ENSG00000250349.3</td>\n",
       "      <td>PRRG1,TM4SF2</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6378907</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238895074</td>\n",
       "      <td>238900785</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr2:238895074-238900785:+</td>\n",
       "      <td>ENSG00000184182.14,ENSG00000258984.1</td>\n",
       "      <td>ENSG00000184182,ENSG00000258984</td>\n",
       "      <td>ENSG00000184182.14,ENSG00000258984.1</td>\n",
       "      <td>UBE2F,UBE2F-SCLY</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000133341.3,OTTHUMG00000170602.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000184182.14,ENSG00000258984.1</td>\n",
       "      <td>UBE2F,UBE2F-SCLY</td>\n",
       "      <td>KNOWN,NOVEL</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6382763</th>\n",
       "      <td>chr10</td>\n",
       "      <td>75570662</td>\n",
       "      <td>75571231</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr10:75570662-75571231:-</td>\n",
       "      <td>ENSG00000272916.1,ENSG00000166507.12</td>\n",
       "      <td>ENSG00000272916,ENSG00000166507</td>\n",
       "      <td>ENSG00000272916.1,ENSG00000166507.12</td>\n",
       "      <td>RP11-574K11.31,NDST2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185992.1,OTTHUMG00000018489.3</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000272916.1,ENSG00000166507.12</td>\n",
       "      <td>RP11-574K11.31,NDST2</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6383939</th>\n",
       "      <td>chr12</td>\n",
       "      <td>54479630</td>\n",
       "      <td>54495997</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr12:54479630-54495997:+</td>\n",
       "      <td>ENSG00000248265.1,ENSG00000249388.1</td>\n",
       "      <td>ENSG00000248265,ENSG00000249388</td>\n",
       "      <td>ENSG00000248265.1,ENSG00000249388.1</td>\n",
       "      <td>RP11-834C11.3,RP11-834C11.6</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>lincRNA,lincRNA</td>\n",
       "      <td>OTTHUMG00000160034.1,OTTHUMG00000160029.1</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000248265.1,ENSG00000249388.1</td>\n",
       "      <td>RP11-834C11.3,RP11-834C11.6</td>\n",
       "      <td>NOVEL,NOVEL</td>\n",
       "      <td>lincRNA,lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6387685</th>\n",
       "      <td>chr22</td>\n",
       "      <td>18593343</td>\n",
       "      <td>18604245</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr22:18593343-18604245:+</td>\n",
       "      <td>ENSG00000215193.8,ENSG00000183785.10</td>\n",
       "      <td>ENSG00000215193,ENSG00000183785</td>\n",
       "      <td>ENSG00000215193.8,ENSG00000183785.10</td>\n",
       "      <td>PEX26,TUBA8</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000149970.5,OTTHUMG00000150097.4</td>\n",
       "      <td>1,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000215193.8,ENSG00000183785.10</td>\n",
       "      <td>PEX26,TUBA8</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6406822</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35480764</td>\n",
       "      <td>35485152</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:35480764-35485152:-</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000185379.1,OTTHUMG00000004163.4</td>\n",
       "      <td>2,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6414267</th>\n",
       "      <td>chr16</td>\n",
       "      <td>18800784</td>\n",
       "      <td>18801565</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr16:18800784-18801565:-</td>\n",
       "      <td>ENSG00000134419.11,ENSG00000260342.2</td>\n",
       "      <td>ENSG00000134419,ENSG00000260342</td>\n",
       "      <td>ENSG00000134419.11,ENSG00000260342.2</td>\n",
       "      <td>RPS15A,RP11-1035H13.3</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000131365.6,OTTHUMG00000177194.3</td>\n",
       "      <td>1,1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000134419.11,ENSG00000260342.2</td>\n",
       "      <td>RPS15A,RP11-1035H13.3</td>\n",
       "      <td>KNOWN,PUTATIVE</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6414602</th>\n",
       "      <td>chr17</td>\n",
       "      <td>26883285</td>\n",
       "      <td>26890448</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr17:26883285-26890448:-</td>\n",
       "      <td>ENSG00000258472.4,ENSG00000087111.16</td>\n",
       "      <td>ENSG00000258472,ENSG00000087111</td>\n",
       "      <td>ENSG00000258472.4,ENSG00000087111.16</td>\n",
       "      <td>RP11-192H23.4,PIGS</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "      <td>OTTHUMG00000171061.28,OTTHUMG00000132604.6</td>\n",
       "      <td>1,2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000258472.4,ENSG00000087111.16</td>\n",
       "      <td>RP11-192H23.4,PIGS</td>\n",
       "      <td>KNOWN,KNOWN</td>\n",
       "      <td>protein_coding,protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12736 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "2         chr1        326417       327587      +        GC/AG     False   \n",
       "42        chr1       1634439      1634518      -        GT/AG      True   \n",
       "43        chr1       1643840      1647784      -        GT/AG      True   \n",
       "44        chr1       1647918      1650766      -        GT/AG      True   \n",
       "45        chr1       1650895      1653034      -        GT/AG      True   \n",
       "46        chr1       1653151      1654146      -        GT/AG      True   \n",
       "47        chr1       1654271      1655696      -        GT/AG      True   \n",
       "128       chr1      10490626     10493898      +        GT/AG      True   \n",
       "129       chr1      10490626     10500403      +        GT/AG     False   \n",
       "130       chr1      10491459     10493898      +        GT/AG      True   \n",
       "131       chr1      10494023     10494713      +        GT/AG      True   \n",
       "132       chr1      10494748     10500403      +        GT/AG      True   \n",
       "133       chr1      10500471     10502321      +        GT/AG      True   \n",
       "237       chr1      19923604     19948593      +        GT/AG      True   \n",
       "238       chr1      19948642     19949967      +        GT/AG      True   \n",
       "239       chr1      19950078     19952880      +        GT/AG      True   \n",
       "430       chr1      28832597     28834639      +        GT/AG      True   \n",
       "431       chr1      28832613     28834639      +        GT/AG     False   \n",
       "432       chr1      28834673     28835341      +        GT/AG      True   \n",
       "504       chr1      32374562     32375638      -        GT/AG      True   \n",
       "505       chr1      32375714     32377296      -        GT/AG      True   \n",
       "506       chr1      32377428     32381495      -        GT/AG      True   \n",
       "507       chr1      32381589     32384570      -        GT/AG      True   \n",
       "563       chr1      35447721     35449367      -        GT/AG      True   \n",
       "564       chr1      35449549     35450798      -        GT/AG      True   \n",
       "565       chr1      35454537     35457834      -        GT/AG      True   \n",
       "566       chr1      35457989     35469568      -        GT/AG      True   \n",
       "567       chr1      35469607     35470732      -        GT/AG      True   \n",
       "568       chr1      35470874     35472551      -        GT/AG      True   \n",
       "569       chr1      35472700     35474352      -        GT/AG      True   \n",
       "...        ...           ...          ...    ...          ...       ...   \n",
       "6313120  chr12     110837706    110837773      -        GC/AG     False   \n",
       "6313340  chr12     122234149    122234818      -        GC/AG     False   \n",
       "6313578  chr13      21951348     21951639      -        GT/AG     False   \n",
       "6313579  chr13      21951348     21951805      -        GT/AG      True   \n",
       "6313585  chr13      21951682     21951763      -        GT/AG     False   \n",
       "6314556  chr14      24633344     24633795      +        GT/AG      True   \n",
       "6314571  chr14      24683358     24683453      -        GT/AG     False   \n",
       "6314576  chr14      24683806     24685124      -        GT/AG     False   \n",
       "6315008  chr14      62229307     62242812      +        GT/AG     False   \n",
       "6315072  chr14      65393042     65395245      +        GC/AG     False   \n",
       "6315722  chr15      25265872     25282858      +        GT/AG     False   \n",
       "6316003  chr15      43942099     43955851      -        GT/AG     False   \n",
       "6316239  chr15      59561890     59562581      -        GT/AG     False   \n",
       "6319193  chr17      41108343     41108420      -        GT/AG     False   \n",
       "6319615  chr17      57279061     57280856      +        GT/AG     False   \n",
       "6319616  chr17      57279095     57280796      +        GC/AG     False   \n",
       "6321434  chr19      23455185     23456877      -        GT/AG     False   \n",
       "6321996  chr19      50925803     50926861      +        GT/AG      True   \n",
       "6322633  chr20      34246937     34252155      -        GT/AG     False   \n",
       "6323440  chr21      47882677     47882978      +        GT/AG      True   \n",
       "6323441  chr21      47883142     47889058      +        GT/AG      True   \n",
       "6323825  chr22      38747005     38757403      -        GT/AG      True   \n",
       "6324222   chrX      37214152     37214592      +        GT/AG     False   \n",
       "6378907   chr2     238895074    238900785      +        GC/AG     False   \n",
       "6382763  chr10      75570662     75571231      -        GT/AG      True   \n",
       "6383939  chr12      54479630     54495997      +        GT/AG     False   \n",
       "6387685  chr22      18593343     18604245      +        GT/AG      True   \n",
       "6406822   chr1      35480764     35485152      -        GT/AG     False   \n",
       "6414267  chr16      18800784     18801565      -        GT/AG     False   \n",
       "6414602  chr17      26883285     26890448      -        GT/AG     False   \n",
       "\n",
       "                     intron_location  \\\n",
       "2               chr1:326417-327587:+   \n",
       "42            chr1:1634439-1634518:-   \n",
       "43            chr1:1643840-1647784:-   \n",
       "44            chr1:1647918-1650766:-   \n",
       "45            chr1:1650895-1653034:-   \n",
       "46            chr1:1653151-1654146:-   \n",
       "47            chr1:1654271-1655696:-   \n",
       "128         chr1:10490626-10493898:+   \n",
       "129         chr1:10490626-10500403:+   \n",
       "130         chr1:10491459-10493898:+   \n",
       "131         chr1:10494023-10494713:+   \n",
       "132         chr1:10494748-10500403:+   \n",
       "133         chr1:10500471-10502321:+   \n",
       "237         chr1:19923604-19948593:+   \n",
       "238         chr1:19948642-19949967:+   \n",
       "239         chr1:19950078-19952880:+   \n",
       "430         chr1:28832597-28834639:+   \n",
       "431         chr1:28832613-28834639:+   \n",
       "432         chr1:28834673-28835341:+   \n",
       "504         chr1:32374562-32375638:-   \n",
       "505         chr1:32375714-32377296:-   \n",
       "506         chr1:32377428-32381495:-   \n",
       "507         chr1:32381589-32384570:-   \n",
       "563         chr1:35447721-35449367:-   \n",
       "564         chr1:35449549-35450798:-   \n",
       "565         chr1:35454537-35457834:-   \n",
       "566         chr1:35457989-35469568:-   \n",
       "567         chr1:35469607-35470732:-   \n",
       "568         chr1:35470874-35472551:-   \n",
       "569         chr1:35472700-35474352:-   \n",
       "...                              ...   \n",
       "6313120  chr12:110837706-110837773:-   \n",
       "6313340  chr12:122234149-122234818:-   \n",
       "6313578    chr13:21951348-21951639:-   \n",
       "6313579    chr13:21951348-21951805:-   \n",
       "6313585    chr13:21951682-21951763:-   \n",
       "6314556    chr14:24633344-24633795:+   \n",
       "6314571    chr14:24683358-24683453:-   \n",
       "6314576    chr14:24683806-24685124:-   \n",
       "6315008    chr14:62229307-62242812:+   \n",
       "6315072    chr14:65393042-65395245:+   \n",
       "6315722    chr15:25265872-25282858:+   \n",
       "6316003    chr15:43942099-43955851:-   \n",
       "6316239    chr15:59561890-59562581:-   \n",
       "6319193    chr17:41108343-41108420:-   \n",
       "6319615    chr17:57279061-57280856:+   \n",
       "6319616    chr17:57279095-57280796:+   \n",
       "6321434    chr19:23455185-23456877:-   \n",
       "6321996    chr19:50925803-50926861:+   \n",
       "6322633    chr20:34246937-34252155:-   \n",
       "6323440    chr21:47882677-47882978:+   \n",
       "6323441    chr21:47883142-47889058:+   \n",
       "6323825    chr22:38747005-38757403:-   \n",
       "6324222     chrX:37214152-37214592:+   \n",
       "6378907   chr2:238895074-238900785:+   \n",
       "6382763    chr10:75570662-75571231:-   \n",
       "6383939    chr12:54479630-54495997:+   \n",
       "6387685    chr22:18593343-18604245:+   \n",
       "6406822     chr1:35480764-35485152:-   \n",
       "6414267    chr16:18800784-18801565:-   \n",
       "6414602    chr17:26883285-26890448:-   \n",
       "\n",
       "                                                gencode_id  \\\n",
       "2                      ENSG00000237094.7,ENSG00000250575.1   \n",
       "42                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "43                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "44                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "45                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "46                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "47                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "128                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "129                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "130                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "131                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "132                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "133                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "237                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "238                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "239                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "430                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "431                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "432                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "504                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "505                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "506                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "507                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "563      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "564      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "565                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "566                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "567                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "568                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "569                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "...                                                    ...   \n",
       "6313120                ENSG00000196510.8,ENSG00000258210.1   \n",
       "6313340                ENSG00000139725.3,ENSG00000212694.4   \n",
       "6313578               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313579               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313585               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6314556                ENSG00000259529.1,ENSG00000213928.4   \n",
       "6314571  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6314576  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6315008                ENSG00000258964.1,ENSG00000023608.4   \n",
       "6315072  ENSG00000258289.3,ENSG00000257365.3,ENSG000001...   \n",
       "6315722                ENSG00000224078.8,ENSG00000272846.1   \n",
       "6316003               ENSG00000242866.5,ENSG00000166762.12   \n",
       "6316239                ENSG00000157483.4,ENSG00000259771.1   \n",
       "6319193               ENSG00000108825.13,ENSG00000266967.2   \n",
       "6319615                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6319616                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6321434                ENSG00000269107.1,ENSG00000213971.3   \n",
       "6321996                ENSG00000142539.9,ENSG00000269404.2   \n",
       "6322633  ENSG00000214078.7,ENSG00000272897.1,ENSG000002...   \n",
       "6323440               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323441               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323825                ENSG00000213923.6,ENSG00000244627.1   \n",
       "6324222               ENSG00000130962.13,ENSG00000250349.3   \n",
       "6378907               ENSG00000184182.14,ENSG00000258984.1   \n",
       "6382763               ENSG00000272916.1,ENSG00000166507.12   \n",
       "6383939                ENSG00000248265.1,ENSG00000249388.1   \n",
       "6387685               ENSG00000215193.8,ENSG00000183785.10   \n",
       "6406822               ENSG00000271741.1,ENSG00000163867.12   \n",
       "6414267               ENSG00000134419.11,ENSG00000260342.2   \n",
       "6414602               ENSG00000258472.4,ENSG00000087111.16   \n",
       "\n",
       "                                                ensembl_id  \\\n",
       "2                          ENSG00000237094,ENSG00000250575   \n",
       "42                         ENSG00000008128,ENSG00000268575   \n",
       "43                         ENSG00000008128,ENSG00000268575   \n",
       "44                         ENSG00000008128,ENSG00000268575   \n",
       "45                         ENSG00000008128,ENSG00000268575   \n",
       "46                         ENSG00000008128,ENSG00000268575   \n",
       "47                         ENSG00000008128,ENSG00000268575   \n",
       "128                        ENSG00000175279,ENSG00000251503   \n",
       "129                        ENSG00000175279,ENSG00000251503   \n",
       "130                        ENSG00000175279,ENSG00000251503   \n",
       "131                        ENSG00000175279,ENSG00000251503   \n",
       "132                        ENSG00000175279,ENSG00000251503   \n",
       "133                        ENSG00000175279,ENSG00000251503   \n",
       "237                        ENSG00000270136,ENSG00000173436   \n",
       "238                        ENSG00000270136,ENSG00000173436   \n",
       "239                        ENSG00000270136,ENSG00000173436   \n",
       "430                        ENSG00000180198,ENSG00000242125   \n",
       "431                        ENSG00000180198,ENSG00000242125   \n",
       "432                        ENSG00000180198,ENSG00000242125   \n",
       "504                        ENSG00000269967,ENSG00000184007   \n",
       "505                        ENSG00000269967,ENSG00000184007   \n",
       "506                        ENSG00000269967,ENSG00000184007   \n",
       "507                        ENSG00000269967,ENSG00000184007   \n",
       "563        ENSG00000241014,ENSG00000271741,ENSG00000243749   \n",
       "564      ENSG00000241014,ENSG00000271741,ENSG0000024374...   \n",
       "565                        ENSG00000271741,ENSG00000163867   \n",
       "566                        ENSG00000271741,ENSG00000163867   \n",
       "567                        ENSG00000271741,ENSG00000163867   \n",
       "568                        ENSG00000271741,ENSG00000163867   \n",
       "569                        ENSG00000271741,ENSG00000163867   \n",
       "...                                                    ...   \n",
       "6313120                    ENSG00000196510,ENSG00000258210   \n",
       "6313340                    ENSG00000139725,ENSG00000212694   \n",
       "6313578                    ENSG00000180776,ENSG00000236953   \n",
       "6313579                    ENSG00000180776,ENSG00000236953   \n",
       "6313585                    ENSG00000180776,ENSG00000236953   \n",
       "6314556                    ENSG00000259529,ENSG00000213928   \n",
       "6314571    ENSG00000260669,ENSG00000213920,ENSG00000255526   \n",
       "6314576    ENSG00000260669,ENSG00000213920,ENSG00000255526   \n",
       "6315008                    ENSG00000258964,ENSG00000023608   \n",
       "6315072    ENSG00000258289,ENSG00000257365,ENSG00000125954   \n",
       "6315722                    ENSG00000224078,ENSG00000272846   \n",
       "6316003                    ENSG00000242866,ENSG00000166762   \n",
       "6316239                    ENSG00000157483,ENSG00000259771   \n",
       "6319193                    ENSG00000108825,ENSG00000266967   \n",
       "6319615                    ENSG00000068489,ENSG00000265303   \n",
       "6319616                    ENSG00000068489,ENSG00000265303   \n",
       "6321434                    ENSG00000269107,ENSG00000213971   \n",
       "6321996                    ENSG00000142539,ENSG00000269404   \n",
       "6322633    ENSG00000214078,ENSG00000272897,ENSG00000244462   \n",
       "6323440                    ENSG00000160305,ENSG00000223692   \n",
       "6323441                    ENSG00000160305,ENSG00000223692   \n",
       "6323825                    ENSG00000213923,ENSG00000244627   \n",
       "6324222                    ENSG00000130962,ENSG00000250349   \n",
       "6378907                    ENSG00000184182,ENSG00000258984   \n",
       "6382763                    ENSG00000272916,ENSG00000166507   \n",
       "6383939                    ENSG00000248265,ENSG00000249388   \n",
       "6387685                    ENSG00000215193,ENSG00000183785   \n",
       "6406822                    ENSG00000271741,ENSG00000163867   \n",
       "6414267                    ENSG00000134419,ENSG00000260342   \n",
       "6414602                    ENSG00000258472,ENSG00000087111   \n",
       "\n",
       "                                                   gene_id  \\\n",
       "2                      ENSG00000237094.7,ENSG00000250575.1   \n",
       "42                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "43                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "44                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "45                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "46                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "47                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "128                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "129                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "130                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "131                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "132                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "133                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "237                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "238                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "239                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "430                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "431                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "432                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "504                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "505                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "506                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "507                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "563      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "564      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "565                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "566                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "567                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "568                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "569                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "...                                                    ...   \n",
       "6313120                ENSG00000196510.8,ENSG00000258210.1   \n",
       "6313340                ENSG00000139725.3,ENSG00000212694.4   \n",
       "6313578               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313579               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313585               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6314556                ENSG00000259529.1,ENSG00000213928.4   \n",
       "6314571  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6314576  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6315008                ENSG00000258964.1,ENSG00000023608.4   \n",
       "6315072  ENSG00000258289.3,ENSG00000257365.3,ENSG000001...   \n",
       "6315722                ENSG00000224078.8,ENSG00000272846.1   \n",
       "6316003               ENSG00000242866.5,ENSG00000166762.12   \n",
       "6316239                ENSG00000157483.4,ENSG00000259771.1   \n",
       "6319193               ENSG00000108825.13,ENSG00000266967.2   \n",
       "6319615                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6319616                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6321434                ENSG00000269107.1,ENSG00000213971.3   \n",
       "6321996                ENSG00000142539.9,ENSG00000269404.2   \n",
       "6322633  ENSG00000214078.7,ENSG00000272897.1,ENSG000002...   \n",
       "6323440               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323441               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323825                ENSG00000213923.6,ENSG00000244627.1   \n",
       "6324222               ENSG00000130962.13,ENSG00000250349.3   \n",
       "6378907               ENSG00000184182.14,ENSG00000258984.1   \n",
       "6382763               ENSG00000272916.1,ENSG00000166507.12   \n",
       "6383939                ENSG00000248265.1,ENSG00000249388.1   \n",
       "6387685               ENSG00000215193.8,ENSG00000183785.10   \n",
       "6406822               ENSG00000271741.1,ENSG00000163867.12   \n",
       "6414267               ENSG00000134419.11,ENSG00000260342.2   \n",
       "6414602               ENSG00000258472.4,ENSG00000087111.16   \n",
       "\n",
       "                                gene_name              gene_status  \\\n",
       "2              RP4-669L17.10,RP4-669L17.8              NOVEL,KNOWN   \n",
       "42                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "43                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "44                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "45                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "46                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "47                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "128                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "129                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "130                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "131                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "132                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "133                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "237                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "238                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "239                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "430                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "431                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "432                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "504                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "505                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "506                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "507                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "563            RP11-244H3.1,ZMYM6,ZMYM6NB        NOVEL,NOVEL,KNOWN   \n",
       "564      RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6  NOVEL,NOVEL,KNOWN,KNOWN   \n",
       "565                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "566                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "567                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "568                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "569                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "...                                   ...                      ...   \n",
       "6313120              ANAPC7,RP11-478C19.2              KNOWN,NOVEL   \n",
       "6313340                   RHOF,AC084018.1              KNOWN,NOVEL   \n",
       "6313578               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6313579               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6313585               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6314556                 RP11-468E2.4,IRF9              NOVEL,KNOWN   \n",
       "6314571        AL136419.6,MDP1,NEDD8-MDP1        NOVEL,KNOWN,KNOWN   \n",
       "6314576        AL136419.6,MDP1,NEDD8-MDP1        NOVEL,KNOWN,KNOWN   \n",
       "6315008              RP11-618G20.1,SNAPC1              NOVEL,KNOWN   \n",
       "6315072           CHURC1,FNTB,CHURC1-FNTB        KNOWN,KNOWN,KNOWN   \n",
       "6315722             SNHG14,RP11-701H24.10              NOVEL,NOVEL   \n",
       "6316003                     STRC,CATSPER2              KNOWN,KNOWN   \n",
       "6316239               MYO1E,RP11-429D19.1              KNOWN,NOVEL   \n",
       "6319193             PTGES3L-AARSD1,AARSD1              KNOWN,KNOWN   \n",
       "6319615                PRR11,CTD-2510F5.6           KNOWN,PUTATIVE   \n",
       "6319616                PRR11,CTD-2510F5.6           KNOWN,PUTATIVE   \n",
       "6321434         RP11-15H20.7,RP11-15H20.6              NOVEL,NOVEL   \n",
       "6321996                 CTD-2545M3.6,SPIB           PUTATIVE,KNOWN   \n",
       "6322633          CPNE1,RP1-309K20.6,RBM12        KNOWN,NOVEL,KNOWN   \n",
       "6323440                   DIP2A,DIP2A-IT1              KNOWN,NOVEL   \n",
       "6323441                   DIP2A,DIP2A-IT1              KNOWN,NOVEL   \n",
       "6323825               CSNK1E,RP3-449O17.1              KNOWN,KNOWN   \n",
       "6324222                      PRRG1,TM4SF2              KNOWN,KNOWN   \n",
       "6378907                  UBE2F,UBE2F-SCLY              KNOWN,NOVEL   \n",
       "6382763              RP11-574K11.31,NDST2              NOVEL,KNOWN   \n",
       "6383939       RP11-834C11.3,RP11-834C11.6              NOVEL,NOVEL   \n",
       "6387685                       PEX26,TUBA8              KNOWN,KNOWN   \n",
       "6406822                       ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "6414267             RPS15A,RP11-1035H13.3           KNOWN,PUTATIVE   \n",
       "6414602                RP11-192H23.4,PIGS              KNOWN,KNOWN   \n",
       "\n",
       "                                                 gene_type  \\\n",
       "2                                       lincRNA,pseudogene   \n",
       "42                     protein_coding,processed_transcript   \n",
       "43                     protein_coding,processed_transcript   \n",
       "44                     protein_coding,processed_transcript   \n",
       "45                     protein_coding,processed_transcript   \n",
       "46                     protein_coding,processed_transcript   \n",
       "47                     protein_coding,processed_transcript   \n",
       "128                          protein_coding,protein_coding   \n",
       "129                          protein_coding,protein_coding   \n",
       "130                          protein_coding,protein_coding   \n",
       "131                          protein_coding,protein_coding   \n",
       "132                          protein_coding,protein_coding   \n",
       "133                          protein_coding,protein_coding   \n",
       "237                          protein_coding,protein_coding   \n",
       "238                          protein_coding,protein_coding   \n",
       "239                          protein_coding,protein_coding   \n",
       "430                          protein_coding,sense_intronic   \n",
       "431                          protein_coding,sense_intronic   \n",
       "432                          protein_coding,sense_intronic   \n",
       "504                                 lincRNA,protein_coding   \n",
       "505                                 lincRNA,protein_coding   \n",
       "506                                 lincRNA,protein_coding   \n",
       "507                                 lincRNA,protein_coding   \n",
       "563      processed_transcript,protein_coding,protein_co...   \n",
       "564      processed_transcript,protein_coding,protein_co...   \n",
       "565                          protein_coding,protein_coding   \n",
       "566                          protein_coding,protein_coding   \n",
       "567                          protein_coding,protein_coding   \n",
       "568                          protein_coding,protein_coding   \n",
       "569                          protein_coding,protein_coding   \n",
       "...                                                    ...   \n",
       "6313120                protein_coding,processed_transcript   \n",
       "6313340                             protein_coding,lincRNA   \n",
       "6313578                      protein_coding,sense_intronic   \n",
       "6313579                      protein_coding,sense_intronic   \n",
       "6313585                      protein_coding,sense_intronic   \n",
       "6314556                      protein_coding,protein_coding   \n",
       "6314571  processed_transcript,protein_coding,protein_co...   \n",
       "6314576  processed_transcript,protein_coding,protein_co...   \n",
       "6315008                processed_transcript,protein_coding   \n",
       "6315072       protein_coding,protein_coding,protein_coding   \n",
       "6315722          processed_transcript,processed_transcript   \n",
       "6316003                      protein_coding,protein_coding   \n",
       "6316239                      protein_coding,sense_intronic   \n",
       "6319193                      protein_coding,protein_coding   \n",
       "6319615                      protein_coding,protein_coding   \n",
       "6319616                      protein_coding,protein_coding   \n",
       "6321434                       lincRNA,processed_transcript   \n",
       "6321996                      protein_coding,protein_coding   \n",
       "6322633       protein_coding,protein_coding,protein_coding   \n",
       "6323440                      protein_coding,sense_intronic   \n",
       "6323441                      protein_coding,sense_intronic   \n",
       "6323825                          protein_coding,pseudogene   \n",
       "6324222                      protein_coding,protein_coding   \n",
       "6378907                      protein_coding,protein_coding   \n",
       "6382763                      protein_coding,protein_coding   \n",
       "6383939                                    lincRNA,lincRNA   \n",
       "6387685                      protein_coding,protein_coding   \n",
       "6406822                      protein_coding,protein_coding   \n",
       "6414267                      protein_coding,protein_coding   \n",
       "6414602                      protein_coding,protein_coding   \n",
       "\n",
       "                                               havana_gene    level  \\\n",
       "2                OTTHUMG00000156968.5,OTTHUMG00000002861.2      2,1   \n",
       "42              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "43              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "44              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "45              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "46              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "47              OTTHUMG00000000703.14,OTTHUMG00000183552.1      2,2   \n",
       "128              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "129              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "130              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "131              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "132              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "133              OTTHUMG00000059085.8,OTTHUMG00000162436.2      1,2   \n",
       "237              OTTHUMG00000183956.1,OTTHUMG00000002712.2      2,2   \n",
       "238              OTTHUMG00000183956.1,OTTHUMG00000002712.2      2,2   \n",
       "239              OTTHUMG00000183956.1,OTTHUMG00000002712.2      2,2   \n",
       "430             OTTHUMG00000003645.10,OTTHUMG00000003657.1      1,2   \n",
       "431             OTTHUMG00000003645.10,OTTHUMG00000003657.1      1,2   \n",
       "432             OTTHUMG00000003645.10,OTTHUMG00000003657.1      1,2   \n",
       "504              OTTHUMG00000184139.1,OTTHUMG00000003801.9      2,2   \n",
       "505              OTTHUMG00000184139.1,OTTHUMG00000003801.9      2,2   \n",
       "506              OTTHUMG00000184139.1,OTTHUMG00000003801.9      2,2   \n",
       "507              OTTHUMG00000184139.1,OTTHUMG00000003801.9      2,2   \n",
       "563      OTTHUMG00000004046.3,OTTHUMG00000185379.1,OTTH...    2,2,1   \n",
       "564      OTTHUMG00000004046.3,OTTHUMG00000185379.1,OTTH...  2,2,1,2   \n",
       "565              OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "566              OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "567              OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "568              OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "569              OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "...                                                    ...      ...   \n",
       "6313120          OTTHUMG00000157009.5,OTTHUMG00000169522.1      1,2   \n",
       "6313340          OTTHUMG00000169077.3,OTTHUMG00000128905.2      2,2   \n",
       "6313578          OTTHUMG00000017410.1,OTTHUMG00000046192.1      2,1   \n",
       "6313579          OTTHUMG00000017410.1,OTTHUMG00000046192.1      2,1   \n",
       "6313585          OTTHUMG00000017410.1,OTTHUMG00000046192.1      2,1   \n",
       "6314556          OTTHUMG00000171951.1,OTTHUMG00000028799.3      2,1   \n",
       "6314571  OTTHUMG00000029324.4,OTTHUMG00000133477.4,OTTH...    2,2,2   \n",
       "6314576  OTTHUMG00000029324.4,OTTHUMG00000133477.4,OTTH...    2,2,2   \n",
       "6315008          OTTHUMG00000171115.1,OTTHUMG00000140343.2      2,2   \n",
       "6315072  OTTHUMG00000170218.5,OTTHUMG00000142811.4,OTTH...    1,1,2   \n",
       "6315722          OTTHUMG00000056661.6,OTTHUMG00000186505.1      2,2   \n",
       "6316003         OTTHUMG00000059899.5,OTTHUMG00000059902.10      2,2   \n",
       "6316239          OTTHUMG00000171944.1,OTTHUMG00000171945.1      1,2   \n",
       "6319193          OTTHUMG00000180905.1,OTTHUMG00000153515.7      1,2   \n",
       "6319615          OTTHUMG00000179362.1,OTTHUMG00000179375.1      1,2   \n",
       "6319616          OTTHUMG00000179362.1,OTTHUMG00000179375.1      1,2   \n",
       "6321434          OTTHUMG00000183234.1,OTTHUMG00000183235.2      2,2   \n",
       "6321996          OTTHUMG00000183038.1,OTTHUMG00000183039.1      2,2   \n",
       "6322633  OTTHUMG00000032356.31,OTTHUMG00000186060.1,OTT...    1,2,2   \n",
       "6323440          OTTHUMG00000090717.3,OTTHUMG00000090716.1      2,2   \n",
       "6323441          OTTHUMG00000090717.3,OTTHUMG00000090716.1      2,2   \n",
       "6323825          OTTHUMG00000151135.2,OTTHUMG00000151245.2      1,2   \n",
       "6324222          OTTHUMG00000021360.2,OTTHUMG00000161009.1      1,2   \n",
       "6378907          OTTHUMG00000133341.3,OTTHUMG00000170602.1      2,2   \n",
       "6382763          OTTHUMG00000185992.1,OTTHUMG00000018489.3      2,2   \n",
       "6383939          OTTHUMG00000160034.1,OTTHUMG00000160029.1      2,2   \n",
       "6387685          OTTHUMG00000149970.5,OTTHUMG00000150097.4      1,1   \n",
       "6406822          OTTHUMG00000185379.1,OTTHUMG00000004163.4      2,2   \n",
       "6414267          OTTHUMG00000131365.6,OTTHUMG00000177194.3      1,1   \n",
       "6414602         OTTHUMG00000171061.28,OTTHUMG00000132604.6      1,2   \n",
       "\n",
       "                           tag  \\\n",
       "2               pseudo_consens   \n",
       "42                         NaN   \n",
       "43                         NaN   \n",
       "44                         NaN   \n",
       "45                         NaN   \n",
       "46                         NaN   \n",
       "47                         NaN   \n",
       "128                        NaN   \n",
       "129                        NaN   \n",
       "130                        NaN   \n",
       "131                        NaN   \n",
       "132                        NaN   \n",
       "133                        NaN   \n",
       "237                        NaN   \n",
       "238                        NaN   \n",
       "239                        NaN   \n",
       "430      ncRNA_host,ncRNA_host   \n",
       "431      ncRNA_host,ncRNA_host   \n",
       "432      ncRNA_host,ncRNA_host   \n",
       "504                        NaN   \n",
       "505                        NaN   \n",
       "506                        NaN   \n",
       "507                        NaN   \n",
       "563                        NaN   \n",
       "564                        NaN   \n",
       "565                        NaN   \n",
       "566                        NaN   \n",
       "567                        NaN   \n",
       "568                        NaN   \n",
       "569                        NaN   \n",
       "...                        ...   \n",
       "6313120                    NaN   \n",
       "6313340                    NaN   \n",
       "6313578                    NaN   \n",
       "6313579                    NaN   \n",
       "6313585                    NaN   \n",
       "6314556             ncRNA_host   \n",
       "6314571                    NaN   \n",
       "6314576                    NaN   \n",
       "6315008                    NaN   \n",
       "6315072  ncRNA_host,ncRNA_host   \n",
       "6315722  ncRNA_host,ncRNA_host   \n",
       "6316003  ncRNA_host,ncRNA_host   \n",
       "6316239             ncRNA_host   \n",
       "6319193                    NaN   \n",
       "6319615                    NaN   \n",
       "6319616                    NaN   \n",
       "6321434                    NaN   \n",
       "6321996                    NaN   \n",
       "6322633  ncRNA_host,ncRNA_host   \n",
       "6323440             ncRNA_host   \n",
       "6323441             ncRNA_host   \n",
       "6323825                    NaN   \n",
       "6324222             ncRNA_host   \n",
       "6378907                    NaN   \n",
       "6382763                    NaN   \n",
       "6383939                    NaN   \n",
       "6387685                    NaN   \n",
       "6406822                    NaN   \n",
       "6414267                    NaN   \n",
       "6414602                    NaN   \n",
       "\n",
       "                                             transcript_id  \\\n",
       "2                      ENSG00000237094.7,ENSG00000250575.1   \n",
       "42                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "43                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "44                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "45                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "46                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "47                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "128                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "129                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "130                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "131                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "132                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "133                   ENSG00000175279.17,ENSG00000251503.3   \n",
       "237                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "238                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "239                    ENSG00000270136.1,ENSG00000173436.9   \n",
       "430                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "431                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "432                   ENSG00000180198.11,ENSG00000242125.2   \n",
       "504                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "505                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "506                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "507                   ENSG00000269967.1,ENSG00000184007.13   \n",
       "563      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "564      ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "565                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "566                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "567                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "568                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "569                   ENSG00000271741.1,ENSG00000163867.12   \n",
       "...                                                    ...   \n",
       "6313120                ENSG00000196510.8,ENSG00000258210.1   \n",
       "6313340                ENSG00000139725.3,ENSG00000212694.4   \n",
       "6313578               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313579               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6313585               ENSG00000180776.11,ENSG00000236953.1   \n",
       "6314556                ENSG00000259529.1,ENSG00000213928.4   \n",
       "6314571  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6314576  ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "6315008                ENSG00000258964.1,ENSG00000023608.4   \n",
       "6315072  ENSG00000258289.3,ENSG00000257365.3,ENSG000001...   \n",
       "6315722                ENSG00000224078.8,ENSG00000272846.1   \n",
       "6316003               ENSG00000242866.5,ENSG00000166762.12   \n",
       "6316239                ENSG00000157483.4,ENSG00000259771.1   \n",
       "6319193               ENSG00000108825.13,ENSG00000266967.2   \n",
       "6319615                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6319616                ENSG00000068489.8,ENSG00000265303.1   \n",
       "6321434                ENSG00000269107.1,ENSG00000213971.3   \n",
       "6321996                ENSG00000142539.9,ENSG00000269404.2   \n",
       "6322633  ENSG00000214078.7,ENSG00000272897.1,ENSG000002...   \n",
       "6323440               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323441               ENSG00000160305.13,ENSG00000223692.1   \n",
       "6323825                ENSG00000213923.6,ENSG00000244627.1   \n",
       "6324222               ENSG00000130962.13,ENSG00000250349.3   \n",
       "6378907               ENSG00000184182.14,ENSG00000258984.1   \n",
       "6382763               ENSG00000272916.1,ENSG00000166507.12   \n",
       "6383939                ENSG00000248265.1,ENSG00000249388.1   \n",
       "6387685               ENSG00000215193.8,ENSG00000183785.10   \n",
       "6406822               ENSG00000271741.1,ENSG00000163867.12   \n",
       "6414267               ENSG00000134419.11,ENSG00000260342.2   \n",
       "6414602               ENSG00000258472.4,ENSG00000087111.16   \n",
       "\n",
       "                          transcript_name        transcript_status  \\\n",
       "2              RP4-669L17.10,RP4-669L17.8              NOVEL,KNOWN   \n",
       "42                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "43                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "44                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "45                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "46                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "47                     CDK11A,RP1-283E3.8              KNOWN,NOVEL   \n",
       "128                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "129                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "130                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "131                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "132                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "133                    APITD1,APITD1-CORT              KNOWN,KNOWN   \n",
       "237                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "238                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "239                    MINOS1-NBL1,MINOS1              KNOWN,KNOWN   \n",
       "430                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "431                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "432                            RCC1,SNHG3              KNOWN,KNOWN   \n",
       "504                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "505                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "506                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "507                   RP11-84A19.4,PTP4A2              NOVEL,KNOWN   \n",
       "563            RP11-244H3.1,ZMYM6,ZMYM6NB        NOVEL,NOVEL,KNOWN   \n",
       "564      RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6  NOVEL,NOVEL,KNOWN,KNOWN   \n",
       "565                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "566                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "567                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "568                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "569                           ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "...                                   ...                      ...   \n",
       "6313120              ANAPC7,RP11-478C19.2              KNOWN,NOVEL   \n",
       "6313340                   RHOF,AC084018.1              KNOWN,NOVEL   \n",
       "6313578               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6313579               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6313585               ZDHHC20,ZDHHC20-IT1              KNOWN,NOVEL   \n",
       "6314556                 RP11-468E2.4,IRF9              NOVEL,KNOWN   \n",
       "6314571        AL136419.6,MDP1,NEDD8-MDP1        NOVEL,KNOWN,KNOWN   \n",
       "6314576        AL136419.6,MDP1,NEDD8-MDP1        NOVEL,KNOWN,KNOWN   \n",
       "6315008              RP11-618G20.1,SNAPC1              NOVEL,KNOWN   \n",
       "6315072           CHURC1,FNTB,CHURC1-FNTB        KNOWN,KNOWN,KNOWN   \n",
       "6315722             SNHG14,RP11-701H24.10              NOVEL,NOVEL   \n",
       "6316003                     STRC,CATSPER2              KNOWN,KNOWN   \n",
       "6316239               MYO1E,RP11-429D19.1              KNOWN,NOVEL   \n",
       "6319193             PTGES3L-AARSD1,AARSD1              KNOWN,KNOWN   \n",
       "6319615                PRR11,CTD-2510F5.6           KNOWN,PUTATIVE   \n",
       "6319616                PRR11,CTD-2510F5.6           KNOWN,PUTATIVE   \n",
       "6321434         RP11-15H20.7,RP11-15H20.6              NOVEL,NOVEL   \n",
       "6321996                 CTD-2545M3.6,SPIB           PUTATIVE,KNOWN   \n",
       "6322633          CPNE1,RP1-309K20.6,RBM12        KNOWN,NOVEL,KNOWN   \n",
       "6323440                   DIP2A,DIP2A-IT1              KNOWN,NOVEL   \n",
       "6323441                   DIP2A,DIP2A-IT1              KNOWN,NOVEL   \n",
       "6323825               CSNK1E,RP3-449O17.1              KNOWN,KNOWN   \n",
       "6324222                      PRRG1,TM4SF2              KNOWN,KNOWN   \n",
       "6378907                  UBE2F,UBE2F-SCLY              KNOWN,NOVEL   \n",
       "6382763              RP11-574K11.31,NDST2              NOVEL,KNOWN   \n",
       "6383939       RP11-834C11.3,RP11-834C11.6              NOVEL,NOVEL   \n",
       "6387685                       PEX26,TUBA8              KNOWN,KNOWN   \n",
       "6406822                       ZMYM6,ZMYM6              NOVEL,KNOWN   \n",
       "6414267             RPS15A,RP11-1035H13.3           KNOWN,PUTATIVE   \n",
       "6414602                RP11-192H23.4,PIGS              KNOWN,KNOWN   \n",
       "\n",
       "                                           transcript_type  \n",
       "2                                       lincRNA,pseudogene  \n",
       "42                     protein_coding,processed_transcript  \n",
       "43                     protein_coding,processed_transcript  \n",
       "44                     protein_coding,processed_transcript  \n",
       "45                     protein_coding,processed_transcript  \n",
       "46                     protein_coding,processed_transcript  \n",
       "47                     protein_coding,processed_transcript  \n",
       "128                          protein_coding,protein_coding  \n",
       "129                          protein_coding,protein_coding  \n",
       "130                          protein_coding,protein_coding  \n",
       "131                          protein_coding,protein_coding  \n",
       "132                          protein_coding,protein_coding  \n",
       "133                          protein_coding,protein_coding  \n",
       "237                          protein_coding,protein_coding  \n",
       "238                          protein_coding,protein_coding  \n",
       "239                          protein_coding,protein_coding  \n",
       "430                          protein_coding,sense_intronic  \n",
       "431                          protein_coding,sense_intronic  \n",
       "432                          protein_coding,sense_intronic  \n",
       "504                                 lincRNA,protein_coding  \n",
       "505                                 lincRNA,protein_coding  \n",
       "506                                 lincRNA,protein_coding  \n",
       "507                                 lincRNA,protein_coding  \n",
       "563      processed_transcript,protein_coding,protein_co...  \n",
       "564      processed_transcript,protein_coding,protein_co...  \n",
       "565                          protein_coding,protein_coding  \n",
       "566                          protein_coding,protein_coding  \n",
       "567                          protein_coding,protein_coding  \n",
       "568                          protein_coding,protein_coding  \n",
       "569                          protein_coding,protein_coding  \n",
       "...                                                    ...  \n",
       "6313120                protein_coding,processed_transcript  \n",
       "6313340                             protein_coding,lincRNA  \n",
       "6313578                      protein_coding,sense_intronic  \n",
       "6313579                      protein_coding,sense_intronic  \n",
       "6313585                      protein_coding,sense_intronic  \n",
       "6314556                      protein_coding,protein_coding  \n",
       "6314571  processed_transcript,protein_coding,protein_co...  \n",
       "6314576  processed_transcript,protein_coding,protein_co...  \n",
       "6315008                processed_transcript,protein_coding  \n",
       "6315072       protein_coding,protein_coding,protein_coding  \n",
       "6315722          processed_transcript,processed_transcript  \n",
       "6316003                      protein_coding,protein_coding  \n",
       "6316239                      protein_coding,sense_intronic  \n",
       "6319193                      protein_coding,protein_coding  \n",
       "6319615                      protein_coding,protein_coding  \n",
       "6319616                      protein_coding,protein_coding  \n",
       "6321434                       lincRNA,processed_transcript  \n",
       "6321996                      protein_coding,protein_coding  \n",
       "6322633       protein_coding,protein_coding,protein_coding  \n",
       "6323440                      protein_coding,sense_intronic  \n",
       "6323441                      protein_coding,sense_intronic  \n",
       "6323825                          protein_coding,pseudogene  \n",
       "6324222                      protein_coding,protein_coding  \n",
       "6378907                      protein_coding,protein_coding  \n",
       "6382763                      protein_coding,protein_coding  \n",
       "6383939                                    lincRNA,lincRNA  \n",
       "6387685                      protein_coding,protein_coding  \n",
       "6406822                      protein_coding,protein_coding  \n",
       "6414267                      protein_coding,protein_coding  \n",
       "6414602                      protein_coding,protein_coding  \n",
       "\n",
       "[12736 rows x 20 columns]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.loc[sj_metadata['gencode_id'].astype(str).str.contains(',')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>one_ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120933</td>\n",
       "      <td>129054</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:120933-129054:-</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000001096.2</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135767</td>\n",
       "      <td>137718</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:135767-137718:-</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:326417-327587:+</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "      <td>OTTHUMG00000156968.5,OTTHUMG00000002861.2</td>\n",
       "      <td>2,1</td>\n",
       "      <td>pseudo_consens</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>NOVEL,KNOWN</td>\n",
       "      <td>lincRNA,pseudogene</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>334298</td>\n",
       "      <td>342391</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:334298-342391:+</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000156968.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>RP4-669L17.10</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>566538</td>\n",
       "      <td>566579</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:566538-566579:-</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>...</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002331.4</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>RP5-857K21.4</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1        120933       129054      -        GT/AG      True   \n",
       "1  chr1        135767       137718      -        GC/AG     False   \n",
       "2  chr1        326417       327587      +        GC/AG     False   \n",
       "3  chr1        334298       342391      +        GT/AG      True   \n",
       "4  chr1        566538       566579      -        GC/AG     False   \n",
       "\n",
       "        intron_location                           gencode_id  \\\n",
       "0  chr1:120933-129054:-                    ENSG00000238009.2   \n",
       "1  chr1:135767-137718:-                    ENSG00000237683.5   \n",
       "2  chr1:326417-327587:+  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3  chr1:334298-342391:+                    ENSG00000237094.7   \n",
       "4  chr1:566538-566579:-                    ENSG00000230021.3   \n",
       "\n",
       "                        ensembl_id                              gene_id  \\\n",
       "0                  ENSG00000238009                    ENSG00000238009.2   \n",
       "1                  ENSG00000237683                    ENSG00000237683.5   \n",
       "2  ENSG00000237094,ENSG00000250575  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3                  ENSG00000237094                    ENSG00000237094.7   \n",
       "4                  ENSG00000230021                    ENSG00000230021.3   \n",
       "\n",
       "        ...         gene_status           gene_type  \\\n",
       "0       ...               NOVEL             lincRNA   \n",
       "1       ...               KNOWN      protein_coding   \n",
       "2       ...         NOVEL,KNOWN  lincRNA,pseudogene   \n",
       "3       ...               NOVEL             lincRNA   \n",
       "4       ...               NOVEL             lincRNA   \n",
       "\n",
       "                                 havana_gene level             tag  \\\n",
       "0                       OTTHUMG00000001096.2     2             NaN   \n",
       "1                                        NaN     3             NaN   \n",
       "2  OTTHUMG00000156968.5,OTTHUMG00000002861.2   2,1  pseudo_consens   \n",
       "3                       OTTHUMG00000156968.5     2             NaN   \n",
       "4                       OTTHUMG00000002331.4     2             NaN   \n",
       "\n",
       "                         transcript_id             transcript_name  \\\n",
       "0                    ENSG00000238009.2                RP11-34P13.7   \n",
       "1                    ENSG00000237683.5                  AL627309.1   \n",
       "2  ENSG00000237094.7,ENSG00000250575.1  RP4-669L17.10,RP4-669L17.8   \n",
       "3                    ENSG00000237094.7               RP4-669L17.10   \n",
       "4                    ENSG00000230021.3                RP5-857K21.4   \n",
       "\n",
       "  transcript_status     transcript_type   one_ensembl_id  \n",
       "0             NOVEL             lincRNA  ENSG00000238009  \n",
       "1             KNOWN      protein_coding  ENSG00000237683  \n",
       "2       NOVEL,KNOWN  lincRNA,pseudogene  ENSG00000237094  \n",
       "3             NOVEL             lincRNA  ENSG00000237094  \n",
       "4             NOVEL             lincRNA  ENSG00000230021  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata['one_ensembl_id'] = sj_metadata.ensembl_id.str.split(',').str.get(0)\n",
    "sj_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "expression_feature_data = pd.read_csv('/home/obotvinnik/flotilla_projects/hg19/expression_feature_data.csv', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomark_neural_panel</th>\n",
       "      <th>confident_rbp</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_parologous_family_id</th>\n",
       "      <th>gerstberger2014_rbp_target_mrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_diverse</th>\n",
       "      <th>gerstberger2014_rbp_family_targets</th>\n",
       "      <th>cell_surface_markers</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>cell_cycle</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000000003.10</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>OTTHUMG00000022002.1</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000000005.5</td>\n",
       "      <td>TNMD</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>OTTHUMG00000022001.1</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000000419.8</td>\n",
       "      <td>DPM1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>OTTHUMG00000032742.2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000000457.9</td>\n",
       "      <td>SCYL3</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>OTTHUMG00000035941.4</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000000460.12</td>\n",
       "      <td>C1orf112</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>OTTHUMG00000035821.6</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                biomark_neural_panel confident_rbp          gencode_id  \\\n",
       "ENSG00000000003                False         False  ENSG00000000003.10   \n",
       "ENSG00000000005                False         False   ENSG00000000005.5   \n",
       "ENSG00000000419                False         False   ENSG00000000419.8   \n",
       "ENSG00000000457                False         False   ENSG00000000457.9   \n",
       "ENSG00000000460                False         False  ENSG00000000460.12   \n",
       "\n",
       "                gene_name gene_status       gene_type           havana_gene  \\\n",
       "ENSG00000000003    TSPAN6       KNOWN  protein_coding  OTTHUMG00000022002.1   \n",
       "ENSG00000000005      TNMD       KNOWN  protein_coding  OTTHUMG00000022001.1   \n",
       "ENSG00000000419      DPM1       KNOWN  protein_coding  OTTHUMG00000032742.2   \n",
       "ENSG00000000457     SCYL3       KNOWN  protein_coding  OTTHUMG00000035941.4   \n",
       "ENSG00000000460  C1orf112       KNOWN  protein_coding  OTTHUMG00000035821.6   \n",
       "\n",
       "                 level    rbp ribosomal    ...      \\\n",
       "ENSG00000000003      2  False     False    ...       \n",
       "ENSG00000000005      2  False     False    ...       \n",
       "ENSG00000000419      2  False     False    ...       \n",
       "ENSG00000000457      2  False     False    ...       \n",
       "ENSG00000000460      2  False     False    ...       \n",
       "\n",
       "                gerstberger2014_rbp_parologous_family_id  \\\n",
       "ENSG00000000003                                      NaN   \n",
       "ENSG00000000005                                      NaN   \n",
       "ENSG00000000419                                      NaN   \n",
       "ENSG00000000457                                      NaN   \n",
       "ENSG00000000460                                      NaN   \n",
       "\n",
       "                gerstberger2014_rbp_target_mrna  \\\n",
       "ENSG00000000003                           False   \n",
       "ENSG00000000005                           False   \n",
       "ENSG00000000419                           False   \n",
       "ENSG00000000457                           False   \n",
       "ENSG00000000460                           False   \n",
       "\n",
       "                gerstberger2014_rbp_target_ribosome  \\\n",
       "ENSG00000000003                               False   \n",
       "ENSG00000000005                               False   \n",
       "ENSG00000000419                               False   \n",
       "ENSG00000000457                               False   \n",
       "ENSG00000000460                               False   \n",
       "\n",
       "                gerstberger2014_rbp_target_trna  \\\n",
       "ENSG00000000003                           False   \n",
       "ENSG00000000005                           False   \n",
       "ENSG00000000419                           False   \n",
       "ENSG00000000457                           False   \n",
       "ENSG00000000460                           False   \n",
       "\n",
       "                gerstberger2014_rbp_target_ncrna  \\\n",
       "ENSG00000000003                            False   \n",
       "ENSG00000000005                            False   \n",
       "ENSG00000000419                            False   \n",
       "ENSG00000000457                            False   \n",
       "ENSG00000000460                            False   \n",
       "\n",
       "                gerstberger2014_rbp_target_diverse  \\\n",
       "ENSG00000000003                              False   \n",
       "ENSG00000000005                              False   \n",
       "ENSG00000000419                              False   \n",
       "ENSG00000000457                              False   \n",
       "ENSG00000000460                              False   \n",
       "\n",
       "                gerstberger2014_rbp_family_targets cell_surface_markers  \\\n",
       "ENSG00000000003                                NaN                False   \n",
       "ENSG00000000005                                NaN                False   \n",
       "ENSG00000000419                                NaN                False   \n",
       "ENSG00000000457                                NaN                False   \n",
       "ENSG00000000460                                NaN                False   \n",
       "\n",
       "                gerstberger2014_tf cell_cycle  \n",
       "ENSG00000000003              False      False  \n",
       "ENSG00000000005              False      False  \n",
       "ENSG00000000419              False      False  \n",
       "ENSG00000000457              False      False  \n",
       "ENSG00000000460              False       True  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression_feature_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(313, 303983)"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "psi3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(316291, 21)"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(316291, 41)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_mrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120933</td>\n",
       "      <td>129054</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:120933-129054:-</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135767</td>\n",
       "      <td>137718</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:135767-137718:-</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:326417-327587:+</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>334298</td>\n",
       "      <td>342391</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:334298-342391:+</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>566538</td>\n",
       "      <td>566579</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:566538-566579:-</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1        120933       129054      -        GT/AG      True   \n",
       "1  chr1        135767       137718      -        GC/AG     False   \n",
       "2  chr1        326417       327587      +        GC/AG     False   \n",
       "3  chr1        334298       342391      +        GT/AG      True   \n",
       "4  chr1        566538       566579      -        GC/AG     False   \n",
       "\n",
       "        intron_location                           gencode_id  \\\n",
       "0  chr1:120933-129054:-                    ENSG00000238009.2   \n",
       "1  chr1:135767-137718:-                    ENSG00000237683.5   \n",
       "2  chr1:326417-327587:+  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3  chr1:334298-342391:+                    ENSG00000237094.7   \n",
       "4  chr1:566538-566579:-                    ENSG00000230021.3   \n",
       "\n",
       "                        ensembl_id                              gene_id  \\\n",
       "0                  ENSG00000238009                    ENSG00000238009.2   \n",
       "1                  ENSG00000237683                    ENSG00000237683.5   \n",
       "2  ENSG00000237094,ENSG00000250575  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3                  ENSG00000237094                    ENSG00000237094.7   \n",
       "4                  ENSG00000230021                    ENSG00000230021.3   \n",
       "\n",
       "          ...          gerstberger2014_rbp_target_mrna  \\\n",
       "0         ...                                    False   \n",
       "1         ...                                    False   \n",
       "2         ...                                    False   \n",
       "3         ...                                    False   \n",
       "4         ...                                    False   \n",
       "\n",
       "  gerstberger2014_rbp_target_ncrna gerstberger2014_rbp_target_ribosome  \\\n",
       "0                            False                               False   \n",
       "1                            False                               False   \n",
       "2                            False                               False   \n",
       "3                            False                               False   \n",
       "4                            False                               False   \n",
       "\n",
       "  gerstberger2014_rbp_target_trna gerstberger2014_tf    rbp ribosomal  \\\n",
       "0                           False              False  False     False   \n",
       "1                           False              False  False     False   \n",
       "2                           False              False  False     False   \n",
       "3                           False              False  False     False   \n",
       "4                           False              False  False     False   \n",
       "\n",
       "  ribosomal_subunit synapse transcription_factor  \n",
       "0             False   False                False  \n",
       "1             False   False                False  \n",
       "2             False   False                False  \n",
       "3             False   False                False  \n",
       "4             False   False                False  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "join_cols = expression_feature_data.columns.difference(sj_metadata.columns)\n",
    "sj_metadata_annotated = sj_metadata.join(expression_feature_data[join_cols], how='left', on='one_ensembl_id')\n",
    "print sj_metadata_annotated.shape\n",
    "sj_metadata_annotated.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120933</td>\n",
       "      <td>129054</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:120933-129054:-</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135767</td>\n",
       "      <td>137718</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:135767-137718:-</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:326417-327587:+</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>334298</td>\n",
       "      <td>342391</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>chr1:334298-342391:+</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>566538</td>\n",
       "      <td>566579</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>chr1:566538-566579:-</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chrom  intron_start  intron_stop strand intron_motif annotated  \\\n",
       "0  chr1        120933       129054      -        GT/AG      True   \n",
       "1  chr1        135767       137718      -        GC/AG     False   \n",
       "2  chr1        326417       327587      +        GC/AG     False   \n",
       "3  chr1        334298       342391      +        GT/AG      True   \n",
       "4  chr1        566538       566579      -        GC/AG     False   \n",
       "\n",
       "        intron_location                           gencode_id  \\\n",
       "0  chr1:120933-129054:-                    ENSG00000238009.2   \n",
       "1  chr1:135767-137718:-                    ENSG00000237683.5   \n",
       "2  chr1:326417-327587:+  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3  chr1:334298-342391:+                    ENSG00000237094.7   \n",
       "4  chr1:566538-566579:-                    ENSG00000230021.3   \n",
       "\n",
       "                        ensembl_id                              gene_id  \\\n",
       "0                  ENSG00000238009                    ENSG00000238009.2   \n",
       "1                  ENSG00000237683                    ENSG00000237683.5   \n",
       "2  ENSG00000237094,ENSG00000250575  ENSG00000237094.7,ENSG00000250575.1   \n",
       "3                  ENSG00000237094                    ENSG00000237094.7   \n",
       "4                  ENSG00000230021                    ENSG00000230021.3   \n",
       "\n",
       "       ...      gerstberger2014_rbp_target_ncrna  \\\n",
       "0      ...                                 False   \n",
       "1      ...                                 False   \n",
       "2      ...                                 False   \n",
       "3      ...                                 False   \n",
       "4      ...                                 False   \n",
       "\n",
       "  gerstberger2014_rbp_target_ribosome gerstberger2014_rbp_target_trna  \\\n",
       "0                               False                           False   \n",
       "1                               False                           False   \n",
       "2                               False                           False   \n",
       "3                               False                           False   \n",
       "4                               False                           False   \n",
       "\n",
       "  gerstberger2014_tf    rbp ribosomal ribosomal_subunit synapse  \\\n",
       "0              False  False     False             False   False   \n",
       "1              False  False     False             False   False   \n",
       "2              False  False     False             False   False   \n",
       "3              False  False     False             False   False   \n",
       "4              False  False     False             False   False   \n",
       "\n",
       "  transcription_factor intron_length  \n",
       "0                False          8121  \n",
       "1                False          1951  \n",
       "2                False          1170  \n",
       "3                False          8093  \n",
       "4                False            41  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata_annotated['intron_length'] = sj_metadata_annotated.intron_stop - sj_metadata_annotated.intron_start\n",
    "sj_metadata_annotated.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata.to_csv('{}/splice_junction_metadata.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata_5p = sj_metadata_annotated.copy()\n",
    "sj_metadata_5p.intron_location = sj_metadata_5p.intron_location + '|5p'\n",
    "\n",
    "sj_metadata_3p = sj_metadata_annotated.copy()\n",
    "sj_metadata_3p.intron_location = sj_metadata_3p.intron_location + '|3p'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(632582, 41)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intron_location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:120933-129054:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>120933</td>\n",
       "      <td>129054</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENSG00000238009.2</td>\n",
       "      <td>RP11-34P13.7</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:135767-137718:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>135767</td>\n",
       "      <td>137718</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>ENSG00000237683</td>\n",
       "      <td>ENSG00000237683.5</td>\n",
       "      <td>AL627309.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:326417-327587:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:334298-342391:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>334298</td>\n",
       "      <td>342391</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>ENSG00000237094</td>\n",
       "      <td>ENSG00000237094.7</td>\n",
       "      <td>RP4-669L17.10</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>8093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:566538-566579:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>566538</td>\n",
       "      <td>566579</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>ENSG00000230021</td>\n",
       "      <td>ENSG00000230021.3</td>\n",
       "      <td>RP5-857K21.4</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        chrom  intron_start  intron_stop strand intron_motif  \\\n",
       "intron_location                                                                \n",
       "chr1:120933-129054:-|5p  chr1        120933       129054      -        GT/AG   \n",
       "chr1:135767-137718:-|5p  chr1        135767       137718      -        GC/AG   \n",
       "chr1:326417-327587:+|5p  chr1        326417       327587      +        GC/AG   \n",
       "chr1:334298-342391:+|5p  chr1        334298       342391      +        GT/AG   \n",
       "chr1:566538-566579:-|5p  chr1        566538       566579      -        GC/AG   \n",
       "\n",
       "                        annotated                           gencode_id  \\\n",
       "intron_location                                                          \n",
       "chr1:120933-129054:-|5p      True                    ENSG00000238009.2   \n",
       "chr1:135767-137718:-|5p     False                    ENSG00000237683.5   \n",
       "chr1:326417-327587:+|5p     False  ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:334298-342391:+|5p      True                    ENSG00000237094.7   \n",
       "chr1:566538-566579:-|5p     False                    ENSG00000230021.3   \n",
       "\n",
       "                                              ensembl_id  \\\n",
       "intron_location                                            \n",
       "chr1:120933-129054:-|5p                  ENSG00000238009   \n",
       "chr1:135767-137718:-|5p                  ENSG00000237683   \n",
       "chr1:326417-327587:+|5p  ENSG00000237094,ENSG00000250575   \n",
       "chr1:334298-342391:+|5p                  ENSG00000237094   \n",
       "chr1:566538-566579:-|5p                  ENSG00000230021   \n",
       "\n",
       "                                                     gene_id  \\\n",
       "intron_location                                                \n",
       "chr1:120933-129054:-|5p                    ENSG00000238009.2   \n",
       "chr1:135767-137718:-|5p                    ENSG00000237683.5   \n",
       "chr1:326417-327587:+|5p  ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:334298-342391:+|5p                    ENSG00000237094.7   \n",
       "chr1:566538-566579:-|5p                    ENSG00000230021.3   \n",
       "\n",
       "                                          gene_name      ...       \\\n",
       "intron_location                                          ...        \n",
       "chr1:120933-129054:-|5p                RP11-34P13.7      ...        \n",
       "chr1:135767-137718:-|5p                  AL627309.1      ...        \n",
       "chr1:326417-327587:+|5p  RP4-669L17.10,RP4-669L17.8      ...        \n",
       "chr1:334298-342391:+|5p               RP4-669L17.10      ...        \n",
       "chr1:566538-566579:-|5p                RP5-857K21.4      ...        \n",
       "\n",
       "                        gerstberger2014_rbp_target_ncrna  \\\n",
       "intron_location                                            \n",
       "chr1:120933-129054:-|5p                            False   \n",
       "chr1:135767-137718:-|5p                            False   \n",
       "chr1:326417-327587:+|5p                            False   \n",
       "chr1:334298-342391:+|5p                            False   \n",
       "chr1:566538-566579:-|5p                            False   \n",
       "\n",
       "                        gerstberger2014_rbp_target_ribosome  \\\n",
       "intron_location                                               \n",
       "chr1:120933-129054:-|5p                               False   \n",
       "chr1:135767-137718:-|5p                               False   \n",
       "chr1:326417-327587:+|5p                               False   \n",
       "chr1:334298-342391:+|5p                               False   \n",
       "chr1:566538-566579:-|5p                               False   \n",
       "\n",
       "                        gerstberger2014_rbp_target_trna gerstberger2014_tf  \\\n",
       "intron_location                                                              \n",
       "chr1:120933-129054:-|5p                           False              False   \n",
       "chr1:135767-137718:-|5p                           False              False   \n",
       "chr1:326417-327587:+|5p                           False              False   \n",
       "chr1:334298-342391:+|5p                           False              False   \n",
       "chr1:566538-566579:-|5p                           False              False   \n",
       "\n",
       "                           rbp ribosomal ribosomal_subunit synapse  \\\n",
       "intron_location                                                      \n",
       "chr1:120933-129054:-|5p  False     False             False   False   \n",
       "chr1:135767-137718:-|5p  False     False             False   False   \n",
       "chr1:326417-327587:+|5p  False     False             False   False   \n",
       "chr1:334298-342391:+|5p  False     False             False   False   \n",
       "chr1:566538-566579:-|5p  False     False             False   False   \n",
       "\n",
       "                        transcription_factor intron_length  \n",
       "intron_location                                             \n",
       "chr1:120933-129054:-|5p                False          8121  \n",
       "chr1:135767-137718:-|5p                False          1951  \n",
       "chr1:326417-327587:+|5p                False          1170  \n",
       "chr1:334298-342391:+|5p                False          8093  \n",
       "chr1:566538-566579:-|5p                False            41  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata_combined = pd.concat([sj_metadata_5p, sj_metadata_3p], ignore_index=True)\n",
    "sj_metadata_combined = sj_metadata_combined.set_index('intron_location')\n",
    "print sj_metadata_combined.shape\n",
    "sj_metadata_combined.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>intron_start</th>\n",
       "      <th>intron_stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>intron_motif</th>\n",
       "      <th>annotated</th>\n",
       "      <th>gencode_id</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>...</th>\n",
       "      <th>gerstberger2014_rbp_target_ncrna</th>\n",
       "      <th>gerstberger2014_rbp_target_ribosome</th>\n",
       "      <th>gerstberger2014_rbp_target_trna</th>\n",
       "      <th>gerstberger2014_tf</th>\n",
       "      <th>rbp</th>\n",
       "      <th>ribosomal</th>\n",
       "      <th>ribosomal_subunit</th>\n",
       "      <th>synapse</th>\n",
       "      <th>transcription_factor</th>\n",
       "      <th>intron_length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intron_location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1:326417-327587:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>326417</td>\n",
       "      <td>327587</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>ENSG00000237094,ENSG00000250575</td>\n",
       "      <td>ENSG00000237094.7,ENSG00000250575.1</td>\n",
       "      <td>RP4-669L17.10,RP4-669L17.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1634439-1634518:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1634439</td>\n",
       "      <td>1634518</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1643840-1647784:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1643840</td>\n",
       "      <td>1647784</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1647918-1650766:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1647918</td>\n",
       "      <td>1650766</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1650895-1653034:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1650895</td>\n",
       "      <td>1653034</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1653151-1654146:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1653151</td>\n",
       "      <td>1654146</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:1654271-1655696:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>1654271</td>\n",
       "      <td>1655696</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>ENSG00000008128,ENSG00000268575</td>\n",
       "      <td>ENSG00000008128.18,ENSG00000268575.1</td>\n",
       "      <td>CDK11A,RP1-283E3.8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10490626-10493898:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10490626</td>\n",
       "      <td>10493898</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10490626-10500403:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10490626</td>\n",
       "      <td>10500403</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>9777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10491459-10493898:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10491459</td>\n",
       "      <td>10493898</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10494023-10494713:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10494023</td>\n",
       "      <td>10494713</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10494748-10500403:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10494748</td>\n",
       "      <td>10500403</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:10500471-10502321:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10500471</td>\n",
       "      <td>10502321</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>ENSG00000175279,ENSG00000251503</td>\n",
       "      <td>ENSG00000175279.17,ENSG00000251503.3</td>\n",
       "      <td>APITD1,APITD1-CORT</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:19923604-19948593:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19923604</td>\n",
       "      <td>19948593</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>24989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:19948642-19949967:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19948642</td>\n",
       "      <td>19949967</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:19950078-19952880:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>19950078</td>\n",
       "      <td>19952880</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>ENSG00000270136,ENSG00000173436</td>\n",
       "      <td>ENSG00000270136.1,ENSG00000173436.9</td>\n",
       "      <td>MINOS1-NBL1,MINOS1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:28832597-28834639:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28832597</td>\n",
       "      <td>28834639</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:28832613-28834639:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28832613</td>\n",
       "      <td>28834639</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:28834673-28835341:+|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>28834673</td>\n",
       "      <td>28835341</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>ENSG00000180198,ENSG00000242125</td>\n",
       "      <td>ENSG00000180198.11,ENSG00000242125.2</td>\n",
       "      <td>RCC1,SNHG3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:32374562-32375638:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32374562</td>\n",
       "      <td>32375638</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:32375714-32377296:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32375714</td>\n",
       "      <td>32377296</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:32377428-32381495:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32377428</td>\n",
       "      <td>32381495</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>4067</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:32381589-32384570:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>32381589</td>\n",
       "      <td>32384570</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>ENSG00000269967,ENSG00000184007</td>\n",
       "      <td>ENSG00000269967.1,ENSG00000184007.13</td>\n",
       "      <td>RP11-84A19.4,PTP4A2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35447721-35449367:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35447721</td>\n",
       "      <td>35449367</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>ENSG00000241014,ENSG00000271741,ENSG00000243749</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35449549-35450798:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35449549</td>\n",
       "      <td>35450798</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>ENSG00000241014,ENSG00000271741,ENSG0000024374...</td>\n",
       "      <td>ENSG00000241014.1,ENSG00000271741.1,ENSG000002...</td>\n",
       "      <td>RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35454537-35457834:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35454537</td>\n",
       "      <td>35457834</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35457989-35469568:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35457989</td>\n",
       "      <td>35469568</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>11579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35469607-35470732:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35469607</td>\n",
       "      <td>35470732</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35470874-35472551:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35470874</td>\n",
       "      <td>35472551</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35472700-35474352:-|5p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35472700</td>\n",
       "      <td>35474352</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:110837706-110837773:-|3p</th>\n",
       "      <td>chr12</td>\n",
       "      <td>110837706</td>\n",
       "      <td>110837773</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000196510.8,ENSG00000258210.1</td>\n",
       "      <td>ENSG00000196510,ENSG00000258210</td>\n",
       "      <td>ENSG00000196510.8,ENSG00000258210.1</td>\n",
       "      <td>ANAPC7,RP11-478C19.2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:122234149-122234818:-|3p</th>\n",
       "      <td>chr12</td>\n",
       "      <td>122234149</td>\n",
       "      <td>122234818</td>\n",
       "      <td>-</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000139725.3,ENSG00000212694.4</td>\n",
       "      <td>ENSG00000139725,ENSG00000212694</td>\n",
       "      <td>ENSG00000139725.3,ENSG00000212694.4</td>\n",
       "      <td>RHOF,AC084018.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:21951348-21951639:-|3p</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951348</td>\n",
       "      <td>21951639</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:21951348-21951805:-|3p</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951348</td>\n",
       "      <td>21951805</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:21951682-21951763:-|3p</th>\n",
       "      <td>chr13</td>\n",
       "      <td>21951682</td>\n",
       "      <td>21951763</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ENSG00000180776,ENSG00000236953</td>\n",
       "      <td>ENSG00000180776.11,ENSG00000236953.1</td>\n",
       "      <td>ZDHHC20,ZDHHC20-IT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:24633344-24633795:+|3p</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24633344</td>\n",
       "      <td>24633795</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000259529.1,ENSG00000213928.4</td>\n",
       "      <td>ENSG00000259529,ENSG00000213928</td>\n",
       "      <td>ENSG00000259529.1,ENSG00000213928.4</td>\n",
       "      <td>RP11-468E2.4,IRF9</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:24683358-24683453:-|3p</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24683358</td>\n",
       "      <td>24683453</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>ENSG00000260669,ENSG00000213920,ENSG00000255526</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:24683806-24685124:-|3p</th>\n",
       "      <td>chr14</td>\n",
       "      <td>24683806</td>\n",
       "      <td>24685124</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>ENSG00000260669,ENSG00000213920,ENSG00000255526</td>\n",
       "      <td>ENSG00000260669.2,ENSG00000213920.4,ENSG000002...</td>\n",
       "      <td>AL136419.6,MDP1,NEDD8-MDP1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:62229307-62242812:+|3p</th>\n",
       "      <td>chr14</td>\n",
       "      <td>62229307</td>\n",
       "      <td>62242812</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000258964.1,ENSG00000023608.4</td>\n",
       "      <td>ENSG00000258964,ENSG00000023608</td>\n",
       "      <td>ENSG00000258964.1,ENSG00000023608.4</td>\n",
       "      <td>RP11-618G20.1,SNAPC1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>13505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:65393042-65395245:+|3p</th>\n",
       "      <td>chr14</td>\n",
       "      <td>65393042</td>\n",
       "      <td>65395245</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000258289.3,ENSG00000257365.3,ENSG000001...</td>\n",
       "      <td>ENSG00000258289,ENSG00000257365,ENSG00000125954</td>\n",
       "      <td>ENSG00000258289.3,ENSG00000257365.3,ENSG000001...</td>\n",
       "      <td>CHURC1,FNTB,CHURC1-FNTB</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:25265872-25282858:+|3p</th>\n",
       "      <td>chr15</td>\n",
       "      <td>25265872</td>\n",
       "      <td>25282858</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000224078.8,ENSG00000272846.1</td>\n",
       "      <td>ENSG00000224078,ENSG00000272846</td>\n",
       "      <td>ENSG00000224078.8,ENSG00000272846.1</td>\n",
       "      <td>SNHG14,RP11-701H24.10</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:43942099-43955851:-|3p</th>\n",
       "      <td>chr15</td>\n",
       "      <td>43942099</td>\n",
       "      <td>43955851</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000242866.5,ENSG00000166762.12</td>\n",
       "      <td>ENSG00000242866,ENSG00000166762</td>\n",
       "      <td>ENSG00000242866.5,ENSG00000166762.12</td>\n",
       "      <td>STRC,CATSPER2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>13752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:59561890-59562581:-|3p</th>\n",
       "      <td>chr15</td>\n",
       "      <td>59561890</td>\n",
       "      <td>59562581</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000157483.4,ENSG00000259771.1</td>\n",
       "      <td>ENSG00000157483,ENSG00000259771</td>\n",
       "      <td>ENSG00000157483.4,ENSG00000259771.1</td>\n",
       "      <td>MYO1E,RP11-429D19.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:41108343-41108420:-|3p</th>\n",
       "      <td>chr17</td>\n",
       "      <td>41108343</td>\n",
       "      <td>41108420</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000108825.13,ENSG00000266967.2</td>\n",
       "      <td>ENSG00000108825,ENSG00000266967</td>\n",
       "      <td>ENSG00000108825.13,ENSG00000266967.2</td>\n",
       "      <td>PTGES3L-AARSD1,AARSD1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:57279061-57280856:+|3p</th>\n",
       "      <td>chr17</td>\n",
       "      <td>57279061</td>\n",
       "      <td>57280856</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>ENSG00000068489,ENSG00000265303</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:57279095-57280796:+|3p</th>\n",
       "      <td>chr17</td>\n",
       "      <td>57279095</td>\n",
       "      <td>57280796</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>ENSG00000068489,ENSG00000265303</td>\n",
       "      <td>ENSG00000068489.8,ENSG00000265303.1</td>\n",
       "      <td>PRR11,CTD-2510F5.6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:23455185-23456877:-|3p</th>\n",
       "      <td>chr19</td>\n",
       "      <td>23455185</td>\n",
       "      <td>23456877</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000269107.1,ENSG00000213971.3</td>\n",
       "      <td>ENSG00000269107,ENSG00000213971</td>\n",
       "      <td>ENSG00000269107.1,ENSG00000213971.3</td>\n",
       "      <td>RP11-15H20.7,RP11-15H20.6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:50925803-50926861:+|3p</th>\n",
       "      <td>chr19</td>\n",
       "      <td>50925803</td>\n",
       "      <td>50926861</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000142539.9,ENSG00000269404.2</td>\n",
       "      <td>ENSG00000142539,ENSG00000269404</td>\n",
       "      <td>ENSG00000142539.9,ENSG00000269404.2</td>\n",
       "      <td>CTD-2545M3.6,SPIB</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>1058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr20:34246937-34252155:-|3p</th>\n",
       "      <td>chr20</td>\n",
       "      <td>34246937</td>\n",
       "      <td>34252155</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000214078.7,ENSG00000272897.1,ENSG000002...</td>\n",
       "      <td>ENSG00000214078,ENSG00000272897,ENSG00000244462</td>\n",
       "      <td>ENSG00000214078.7,ENSG00000272897.1,ENSG000002...</td>\n",
       "      <td>CPNE1,RP1-309K20.6,RBM12</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr21:47882677-47882978:+|3p</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47882677</td>\n",
       "      <td>47882978</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>ENSG00000160305,ENSG00000223692</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr21:47883142-47889058:+|3p</th>\n",
       "      <td>chr21</td>\n",
       "      <td>47883142</td>\n",
       "      <td>47889058</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>ENSG00000160305,ENSG00000223692</td>\n",
       "      <td>ENSG00000160305.13,ENSG00000223692.1</td>\n",
       "      <td>DIP2A,DIP2A-IT1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:38747005-38757403:-|3p</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38747005</td>\n",
       "      <td>38757403</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000213923.6,ENSG00000244627.1</td>\n",
       "      <td>ENSG00000213923,ENSG00000244627</td>\n",
       "      <td>ENSG00000213923.6,ENSG00000244627.1</td>\n",
       "      <td>CSNK1E,RP3-449O17.1</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>10398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:37214152-37214592:+|3p</th>\n",
       "      <td>chrX</td>\n",
       "      <td>37214152</td>\n",
       "      <td>37214592</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000130962.13,ENSG00000250349.3</td>\n",
       "      <td>ENSG00000130962,ENSG00000250349</td>\n",
       "      <td>ENSG00000130962.13,ENSG00000250349.3</td>\n",
       "      <td>PRRG1,TM4SF2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:238895074-238900785:+|3p</th>\n",
       "      <td>chr2</td>\n",
       "      <td>238895074</td>\n",
       "      <td>238900785</td>\n",
       "      <td>+</td>\n",
       "      <td>GC/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000184182.14,ENSG00000258984.1</td>\n",
       "      <td>ENSG00000184182,ENSG00000258984</td>\n",
       "      <td>ENSG00000184182.14,ENSG00000258984.1</td>\n",
       "      <td>UBE2F,UBE2F-SCLY</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:75570662-75571231:-|3p</th>\n",
       "      <td>chr10</td>\n",
       "      <td>75570662</td>\n",
       "      <td>75571231</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000272916.1,ENSG00000166507.12</td>\n",
       "      <td>ENSG00000272916,ENSG00000166507</td>\n",
       "      <td>ENSG00000272916.1,ENSG00000166507.12</td>\n",
       "      <td>RP11-574K11.31,NDST2</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:54479630-54495997:+|3p</th>\n",
       "      <td>chr12</td>\n",
       "      <td>54479630</td>\n",
       "      <td>54495997</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000248265.1,ENSG00000249388.1</td>\n",
       "      <td>ENSG00000248265,ENSG00000249388</td>\n",
       "      <td>ENSG00000248265.1,ENSG00000249388.1</td>\n",
       "      <td>RP11-834C11.3,RP11-834C11.6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>16367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr22:18593343-18604245:+|3p</th>\n",
       "      <td>chr22</td>\n",
       "      <td>18593343</td>\n",
       "      <td>18604245</td>\n",
       "      <td>+</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>True</td>\n",
       "      <td>ENSG00000215193.8,ENSG00000183785.10</td>\n",
       "      <td>ENSG00000215193,ENSG00000183785</td>\n",
       "      <td>ENSG00000215193.8,ENSG00000183785.10</td>\n",
       "      <td>PEX26,TUBA8</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>10902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:35480764-35485152:-|3p</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35480764</td>\n",
       "      <td>35485152</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ENSG00000271741,ENSG00000163867</td>\n",
       "      <td>ENSG00000271741.1,ENSG00000163867.12</td>\n",
       "      <td>ZMYM6,ZMYM6</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>4388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr16:18800784-18801565:-|3p</th>\n",
       "      <td>chr16</td>\n",
       "      <td>18800784</td>\n",
       "      <td>18801565</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000134419.11,ENSG00000260342.2</td>\n",
       "      <td>ENSG00000134419,ENSG00000260342</td>\n",
       "      <td>ENSG00000134419.11,ENSG00000260342.2</td>\n",
       "      <td>RPS15A,RP11-1035H13.3</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:26883285-26890448:-|3p</th>\n",
       "      <td>chr17</td>\n",
       "      <td>26883285</td>\n",
       "      <td>26890448</td>\n",
       "      <td>-</td>\n",
       "      <td>GT/AG</td>\n",
       "      <td>False</td>\n",
       "      <td>ENSG00000258472.4,ENSG00000087111.16</td>\n",
       "      <td>ENSG00000258472,ENSG00000087111</td>\n",
       "      <td>ENSG00000258472.4,ENSG00000087111.16</td>\n",
       "      <td>RP11-192H23.4,PIGS</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>7163</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25472 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                chrom  intron_start  intron_stop strand  \\\n",
       "intron_location                                                           \n",
       "chr1:326417-327587:+|5p          chr1        326417       327587      +   \n",
       "chr1:1634439-1634518:-|5p        chr1       1634439      1634518      -   \n",
       "chr1:1643840-1647784:-|5p        chr1       1643840      1647784      -   \n",
       "chr1:1647918-1650766:-|5p        chr1       1647918      1650766      -   \n",
       "chr1:1650895-1653034:-|5p        chr1       1650895      1653034      -   \n",
       "chr1:1653151-1654146:-|5p        chr1       1653151      1654146      -   \n",
       "chr1:1654271-1655696:-|5p        chr1       1654271      1655696      -   \n",
       "chr1:10490626-10493898:+|5p      chr1      10490626     10493898      +   \n",
       "chr1:10490626-10500403:+|5p      chr1      10490626     10500403      +   \n",
       "chr1:10491459-10493898:+|5p      chr1      10491459     10493898      +   \n",
       "chr1:10494023-10494713:+|5p      chr1      10494023     10494713      +   \n",
       "chr1:10494748-10500403:+|5p      chr1      10494748     10500403      +   \n",
       "chr1:10500471-10502321:+|5p      chr1      10500471     10502321      +   \n",
       "chr1:19923604-19948593:+|5p      chr1      19923604     19948593      +   \n",
       "chr1:19948642-19949967:+|5p      chr1      19948642     19949967      +   \n",
       "chr1:19950078-19952880:+|5p      chr1      19950078     19952880      +   \n",
       "chr1:28832597-28834639:+|5p      chr1      28832597     28834639      +   \n",
       "chr1:28832613-28834639:+|5p      chr1      28832613     28834639      +   \n",
       "chr1:28834673-28835341:+|5p      chr1      28834673     28835341      +   \n",
       "chr1:32374562-32375638:-|5p      chr1      32374562     32375638      -   \n",
       "chr1:32375714-32377296:-|5p      chr1      32375714     32377296      -   \n",
       "chr1:32377428-32381495:-|5p      chr1      32377428     32381495      -   \n",
       "chr1:32381589-32384570:-|5p      chr1      32381589     32384570      -   \n",
       "chr1:35447721-35449367:-|5p      chr1      35447721     35449367      -   \n",
       "chr1:35449549-35450798:-|5p      chr1      35449549     35450798      -   \n",
       "chr1:35454537-35457834:-|5p      chr1      35454537     35457834      -   \n",
       "chr1:35457989-35469568:-|5p      chr1      35457989     35469568      -   \n",
       "chr1:35469607-35470732:-|5p      chr1      35469607     35470732      -   \n",
       "chr1:35470874-35472551:-|5p      chr1      35470874     35472551      -   \n",
       "chr1:35472700-35474352:-|5p      chr1      35472700     35474352      -   \n",
       "...                               ...           ...          ...    ...   \n",
       "chr12:110837706-110837773:-|3p  chr12     110837706    110837773      -   \n",
       "chr12:122234149-122234818:-|3p  chr12     122234149    122234818      -   \n",
       "chr13:21951348-21951639:-|3p    chr13      21951348     21951639      -   \n",
       "chr13:21951348-21951805:-|3p    chr13      21951348     21951805      -   \n",
       "chr13:21951682-21951763:-|3p    chr13      21951682     21951763      -   \n",
       "chr14:24633344-24633795:+|3p    chr14      24633344     24633795      +   \n",
       "chr14:24683358-24683453:-|3p    chr14      24683358     24683453      -   \n",
       "chr14:24683806-24685124:-|3p    chr14      24683806     24685124      -   \n",
       "chr14:62229307-62242812:+|3p    chr14      62229307     62242812      +   \n",
       "chr14:65393042-65395245:+|3p    chr14      65393042     65395245      +   \n",
       "chr15:25265872-25282858:+|3p    chr15      25265872     25282858      +   \n",
       "chr15:43942099-43955851:-|3p    chr15      43942099     43955851      -   \n",
       "chr15:59561890-59562581:-|3p    chr15      59561890     59562581      -   \n",
       "chr17:41108343-41108420:-|3p    chr17      41108343     41108420      -   \n",
       "chr17:57279061-57280856:+|3p    chr17      57279061     57280856      +   \n",
       "chr17:57279095-57280796:+|3p    chr17      57279095     57280796      +   \n",
       "chr19:23455185-23456877:-|3p    chr19      23455185     23456877      -   \n",
       "chr19:50925803-50926861:+|3p    chr19      50925803     50926861      +   \n",
       "chr20:34246937-34252155:-|3p    chr20      34246937     34252155      -   \n",
       "chr21:47882677-47882978:+|3p    chr21      47882677     47882978      +   \n",
       "chr21:47883142-47889058:+|3p    chr21      47883142     47889058      +   \n",
       "chr22:38747005-38757403:-|3p    chr22      38747005     38757403      -   \n",
       "chrX:37214152-37214592:+|3p      chrX      37214152     37214592      +   \n",
       "chr2:238895074-238900785:+|3p    chr2     238895074    238900785      +   \n",
       "chr10:75570662-75571231:-|3p    chr10      75570662     75571231      -   \n",
       "chr12:54479630-54495997:+|3p    chr12      54479630     54495997      +   \n",
       "chr22:18593343-18604245:+|3p    chr22      18593343     18604245      +   \n",
       "chr1:35480764-35485152:-|3p      chr1      35480764     35485152      -   \n",
       "chr16:18800784-18801565:-|3p    chr16      18800784     18801565      -   \n",
       "chr17:26883285-26890448:-|3p    chr17      26883285     26890448      -   \n",
       "\n",
       "                               intron_motif annotated  \\\n",
       "intron_location                                         \n",
       "chr1:326417-327587:+|5p               GC/AG     False   \n",
       "chr1:1634439-1634518:-|5p             GT/AG      True   \n",
       "chr1:1643840-1647784:-|5p             GT/AG      True   \n",
       "chr1:1647918-1650766:-|5p             GT/AG      True   \n",
       "chr1:1650895-1653034:-|5p             GT/AG      True   \n",
       "chr1:1653151-1654146:-|5p             GT/AG      True   \n",
       "chr1:1654271-1655696:-|5p             GT/AG      True   \n",
       "chr1:10490626-10493898:+|5p           GT/AG      True   \n",
       "chr1:10490626-10500403:+|5p           GT/AG     False   \n",
       "chr1:10491459-10493898:+|5p           GT/AG      True   \n",
       "chr1:10494023-10494713:+|5p           GT/AG      True   \n",
       "chr1:10494748-10500403:+|5p           GT/AG      True   \n",
       "chr1:10500471-10502321:+|5p           GT/AG      True   \n",
       "chr1:19923604-19948593:+|5p           GT/AG      True   \n",
       "chr1:19948642-19949967:+|5p           GT/AG      True   \n",
       "chr1:19950078-19952880:+|5p           GT/AG      True   \n",
       "chr1:28832597-28834639:+|5p           GT/AG      True   \n",
       "chr1:28832613-28834639:+|5p           GT/AG     False   \n",
       "chr1:28834673-28835341:+|5p           GT/AG      True   \n",
       "chr1:32374562-32375638:-|5p           GT/AG      True   \n",
       "chr1:32375714-32377296:-|5p           GT/AG      True   \n",
       "chr1:32377428-32381495:-|5p           GT/AG      True   \n",
       "chr1:32381589-32384570:-|5p           GT/AG      True   \n",
       "chr1:35447721-35449367:-|5p           GT/AG      True   \n",
       "chr1:35449549-35450798:-|5p           GT/AG      True   \n",
       "chr1:35454537-35457834:-|5p           GT/AG      True   \n",
       "chr1:35457989-35469568:-|5p           GT/AG      True   \n",
       "chr1:35469607-35470732:-|5p           GT/AG      True   \n",
       "chr1:35470874-35472551:-|5p           GT/AG      True   \n",
       "chr1:35472700-35474352:-|5p           GT/AG      True   \n",
       "...                                     ...       ...   \n",
       "chr12:110837706-110837773:-|3p        GC/AG     False   \n",
       "chr12:122234149-122234818:-|3p        GC/AG     False   \n",
       "chr13:21951348-21951639:-|3p          GT/AG     False   \n",
       "chr13:21951348-21951805:-|3p          GT/AG      True   \n",
       "chr13:21951682-21951763:-|3p          GT/AG     False   \n",
       "chr14:24633344-24633795:+|3p          GT/AG      True   \n",
       "chr14:24683358-24683453:-|3p          GT/AG     False   \n",
       "chr14:24683806-24685124:-|3p          GT/AG     False   \n",
       "chr14:62229307-62242812:+|3p          GT/AG     False   \n",
       "chr14:65393042-65395245:+|3p          GC/AG     False   \n",
       "chr15:25265872-25282858:+|3p          GT/AG     False   \n",
       "chr15:43942099-43955851:-|3p          GT/AG     False   \n",
       "chr15:59561890-59562581:-|3p          GT/AG     False   \n",
       "chr17:41108343-41108420:-|3p          GT/AG     False   \n",
       "chr17:57279061-57280856:+|3p          GT/AG     False   \n",
       "chr17:57279095-57280796:+|3p          GC/AG     False   \n",
       "chr19:23455185-23456877:-|3p          GT/AG     False   \n",
       "chr19:50925803-50926861:+|3p          GT/AG      True   \n",
       "chr20:34246937-34252155:-|3p          GT/AG     False   \n",
       "chr21:47882677-47882978:+|3p          GT/AG      True   \n",
       "chr21:47883142-47889058:+|3p          GT/AG      True   \n",
       "chr22:38747005-38757403:-|3p          GT/AG      True   \n",
       "chrX:37214152-37214592:+|3p           GT/AG     False   \n",
       "chr2:238895074-238900785:+|3p         GC/AG     False   \n",
       "chr10:75570662-75571231:-|3p          GT/AG      True   \n",
       "chr12:54479630-54495997:+|3p          GT/AG     False   \n",
       "chr22:18593343-18604245:+|3p          GT/AG      True   \n",
       "chr1:35480764-35485152:-|3p           GT/AG     False   \n",
       "chr16:18800784-18801565:-|3p          GT/AG     False   \n",
       "chr17:26883285-26890448:-|3p          GT/AG     False   \n",
       "\n",
       "                                                                       gencode_id  \\\n",
       "intron_location                                                                     \n",
       "chr1:326417-327587:+|5p                       ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:1634439-1634518:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1643840-1647784:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1647918-1650766:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1650895-1653034:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1653151-1654146:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1654271-1655696:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:10490626-10493898:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10490626-10500403:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10491459-10493898:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10494023-10494713:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10494748-10500403:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10500471-10502321:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:19923604-19948593:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:19948642-19949967:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:19950078-19952880:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:28832597-28834639:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:28832613-28834639:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:28834673-28835341:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:32374562-32375638:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32375714-32377296:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32377428-32381495:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32381589-32384570:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:35447721-35449367:-|5p     ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "chr1:35449549-35450798:-|5p     ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "chr1:35454537-35457834:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35457989-35469568:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35469607-35470732:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35470874-35472551:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35472700-35474352:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "...                                                                           ...   \n",
       "chr12:110837706-110837773:-|3p                ENSG00000196510.8,ENSG00000258210.1   \n",
       "chr12:122234149-122234818:-|3p                ENSG00000139725.3,ENSG00000212694.4   \n",
       "chr13:21951348-21951639:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr13:21951348-21951805:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr13:21951682-21951763:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr14:24633344-24633795:+|3p                  ENSG00000259529.1,ENSG00000213928.4   \n",
       "chr14:24683358-24683453:-|3p    ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "chr14:24683806-24685124:-|3p    ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "chr14:62229307-62242812:+|3p                  ENSG00000258964.1,ENSG00000023608.4   \n",
       "chr14:65393042-65395245:+|3p    ENSG00000258289.3,ENSG00000257365.3,ENSG000001...   \n",
       "chr15:25265872-25282858:+|3p                  ENSG00000224078.8,ENSG00000272846.1   \n",
       "chr15:43942099-43955851:-|3p                 ENSG00000242866.5,ENSG00000166762.12   \n",
       "chr15:59561890-59562581:-|3p                  ENSG00000157483.4,ENSG00000259771.1   \n",
       "chr17:41108343-41108420:-|3p                 ENSG00000108825.13,ENSG00000266967.2   \n",
       "chr17:57279061-57280856:+|3p                  ENSG00000068489.8,ENSG00000265303.1   \n",
       "chr17:57279095-57280796:+|3p                  ENSG00000068489.8,ENSG00000265303.1   \n",
       "chr19:23455185-23456877:-|3p                  ENSG00000269107.1,ENSG00000213971.3   \n",
       "chr19:50925803-50926861:+|3p                  ENSG00000142539.9,ENSG00000269404.2   \n",
       "chr20:34246937-34252155:-|3p    ENSG00000214078.7,ENSG00000272897.1,ENSG000002...   \n",
       "chr21:47882677-47882978:+|3p                 ENSG00000160305.13,ENSG00000223692.1   \n",
       "chr21:47883142-47889058:+|3p                 ENSG00000160305.13,ENSG00000223692.1   \n",
       "chr22:38747005-38757403:-|3p                  ENSG00000213923.6,ENSG00000244627.1   \n",
       "chrX:37214152-37214592:+|3p                  ENSG00000130962.13,ENSG00000250349.3   \n",
       "chr2:238895074-238900785:+|3p                ENSG00000184182.14,ENSG00000258984.1   \n",
       "chr10:75570662-75571231:-|3p                 ENSG00000272916.1,ENSG00000166507.12   \n",
       "chr12:54479630-54495997:+|3p                  ENSG00000248265.1,ENSG00000249388.1   \n",
       "chr22:18593343-18604245:+|3p                 ENSG00000215193.8,ENSG00000183785.10   \n",
       "chr1:35480764-35485152:-|3p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr16:18800784-18801565:-|3p                 ENSG00000134419.11,ENSG00000260342.2   \n",
       "chr17:26883285-26890448:-|3p                 ENSG00000258472.4,ENSG00000087111.16   \n",
       "\n",
       "                                                                       ensembl_id  \\\n",
       "intron_location                                                                     \n",
       "chr1:326417-327587:+|5p                           ENSG00000237094,ENSG00000250575   \n",
       "chr1:1634439-1634518:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:1643840-1647784:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:1647918-1650766:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:1650895-1653034:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:1653151-1654146:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:1654271-1655696:-|5p                         ENSG00000008128,ENSG00000268575   \n",
       "chr1:10490626-10493898:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:10490626-10500403:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:10491459-10493898:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:10494023-10494713:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:10494748-10500403:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:10500471-10502321:+|5p                       ENSG00000175279,ENSG00000251503   \n",
       "chr1:19923604-19948593:+|5p                       ENSG00000270136,ENSG00000173436   \n",
       "chr1:19948642-19949967:+|5p                       ENSG00000270136,ENSG00000173436   \n",
       "chr1:19950078-19952880:+|5p                       ENSG00000270136,ENSG00000173436   \n",
       "chr1:28832597-28834639:+|5p                       ENSG00000180198,ENSG00000242125   \n",
       "chr1:28832613-28834639:+|5p                       ENSG00000180198,ENSG00000242125   \n",
       "chr1:28834673-28835341:+|5p                       ENSG00000180198,ENSG00000242125   \n",
       "chr1:32374562-32375638:-|5p                       ENSG00000269967,ENSG00000184007   \n",
       "chr1:32375714-32377296:-|5p                       ENSG00000269967,ENSG00000184007   \n",
       "chr1:32377428-32381495:-|5p                       ENSG00000269967,ENSG00000184007   \n",
       "chr1:32381589-32384570:-|5p                       ENSG00000269967,ENSG00000184007   \n",
       "chr1:35447721-35449367:-|5p       ENSG00000241014,ENSG00000271741,ENSG00000243749   \n",
       "chr1:35449549-35450798:-|5p     ENSG00000241014,ENSG00000271741,ENSG0000024374...   \n",
       "chr1:35454537-35457834:-|5p                       ENSG00000271741,ENSG00000163867   \n",
       "chr1:35457989-35469568:-|5p                       ENSG00000271741,ENSG00000163867   \n",
       "chr1:35469607-35470732:-|5p                       ENSG00000271741,ENSG00000163867   \n",
       "chr1:35470874-35472551:-|5p                       ENSG00000271741,ENSG00000163867   \n",
       "chr1:35472700-35474352:-|5p                       ENSG00000271741,ENSG00000163867   \n",
       "...                                                                           ...   \n",
       "chr12:110837706-110837773:-|3p                    ENSG00000196510,ENSG00000258210   \n",
       "chr12:122234149-122234818:-|3p                    ENSG00000139725,ENSG00000212694   \n",
       "chr13:21951348-21951639:-|3p                      ENSG00000180776,ENSG00000236953   \n",
       "chr13:21951348-21951805:-|3p                      ENSG00000180776,ENSG00000236953   \n",
       "chr13:21951682-21951763:-|3p                      ENSG00000180776,ENSG00000236953   \n",
       "chr14:24633344-24633795:+|3p                      ENSG00000259529,ENSG00000213928   \n",
       "chr14:24683358-24683453:-|3p      ENSG00000260669,ENSG00000213920,ENSG00000255526   \n",
       "chr14:24683806-24685124:-|3p      ENSG00000260669,ENSG00000213920,ENSG00000255526   \n",
       "chr14:62229307-62242812:+|3p                      ENSG00000258964,ENSG00000023608   \n",
       "chr14:65393042-65395245:+|3p      ENSG00000258289,ENSG00000257365,ENSG00000125954   \n",
       "chr15:25265872-25282858:+|3p                      ENSG00000224078,ENSG00000272846   \n",
       "chr15:43942099-43955851:-|3p                      ENSG00000242866,ENSG00000166762   \n",
       "chr15:59561890-59562581:-|3p                      ENSG00000157483,ENSG00000259771   \n",
       "chr17:41108343-41108420:-|3p                      ENSG00000108825,ENSG00000266967   \n",
       "chr17:57279061-57280856:+|3p                      ENSG00000068489,ENSG00000265303   \n",
       "chr17:57279095-57280796:+|3p                      ENSG00000068489,ENSG00000265303   \n",
       "chr19:23455185-23456877:-|3p                      ENSG00000269107,ENSG00000213971   \n",
       "chr19:50925803-50926861:+|3p                      ENSG00000142539,ENSG00000269404   \n",
       "chr20:34246937-34252155:-|3p      ENSG00000214078,ENSG00000272897,ENSG00000244462   \n",
       "chr21:47882677-47882978:+|3p                      ENSG00000160305,ENSG00000223692   \n",
       "chr21:47883142-47889058:+|3p                      ENSG00000160305,ENSG00000223692   \n",
       "chr22:38747005-38757403:-|3p                      ENSG00000213923,ENSG00000244627   \n",
       "chrX:37214152-37214592:+|3p                       ENSG00000130962,ENSG00000250349   \n",
       "chr2:238895074-238900785:+|3p                     ENSG00000184182,ENSG00000258984   \n",
       "chr10:75570662-75571231:-|3p                      ENSG00000272916,ENSG00000166507   \n",
       "chr12:54479630-54495997:+|3p                      ENSG00000248265,ENSG00000249388   \n",
       "chr22:18593343-18604245:+|3p                      ENSG00000215193,ENSG00000183785   \n",
       "chr1:35480764-35485152:-|3p                       ENSG00000271741,ENSG00000163867   \n",
       "chr16:18800784-18801565:-|3p                      ENSG00000134419,ENSG00000260342   \n",
       "chr17:26883285-26890448:-|3p                      ENSG00000258472,ENSG00000087111   \n",
       "\n",
       "                                                                          gene_id  \\\n",
       "intron_location                                                                     \n",
       "chr1:326417-327587:+|5p                       ENSG00000237094.7,ENSG00000250575.1   \n",
       "chr1:1634439-1634518:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1643840-1647784:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1647918-1650766:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1650895-1653034:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1653151-1654146:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:1654271-1655696:-|5p                    ENSG00000008128.18,ENSG00000268575.1   \n",
       "chr1:10490626-10493898:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10490626-10500403:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10491459-10493898:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10494023-10494713:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10494748-10500403:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:10500471-10502321:+|5p                  ENSG00000175279.17,ENSG00000251503.3   \n",
       "chr1:19923604-19948593:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:19948642-19949967:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:19950078-19952880:+|5p                   ENSG00000270136.1,ENSG00000173436.9   \n",
       "chr1:28832597-28834639:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:28832613-28834639:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:28834673-28835341:+|5p                  ENSG00000180198.11,ENSG00000242125.2   \n",
       "chr1:32374562-32375638:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32375714-32377296:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32377428-32381495:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:32381589-32384570:-|5p                  ENSG00000269967.1,ENSG00000184007.13   \n",
       "chr1:35447721-35449367:-|5p     ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "chr1:35449549-35450798:-|5p     ENSG00000241014.1,ENSG00000271741.1,ENSG000002...   \n",
       "chr1:35454537-35457834:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35457989-35469568:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35469607-35470732:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35470874-35472551:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr1:35472700-35474352:-|5p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "...                                                                           ...   \n",
       "chr12:110837706-110837773:-|3p                ENSG00000196510.8,ENSG00000258210.1   \n",
       "chr12:122234149-122234818:-|3p                ENSG00000139725.3,ENSG00000212694.4   \n",
       "chr13:21951348-21951639:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr13:21951348-21951805:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr13:21951682-21951763:-|3p                 ENSG00000180776.11,ENSG00000236953.1   \n",
       "chr14:24633344-24633795:+|3p                  ENSG00000259529.1,ENSG00000213928.4   \n",
       "chr14:24683358-24683453:-|3p    ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "chr14:24683806-24685124:-|3p    ENSG00000260669.2,ENSG00000213920.4,ENSG000002...   \n",
       "chr14:62229307-62242812:+|3p                  ENSG00000258964.1,ENSG00000023608.4   \n",
       "chr14:65393042-65395245:+|3p    ENSG00000258289.3,ENSG00000257365.3,ENSG000001...   \n",
       "chr15:25265872-25282858:+|3p                  ENSG00000224078.8,ENSG00000272846.1   \n",
       "chr15:43942099-43955851:-|3p                 ENSG00000242866.5,ENSG00000166762.12   \n",
       "chr15:59561890-59562581:-|3p                  ENSG00000157483.4,ENSG00000259771.1   \n",
       "chr17:41108343-41108420:-|3p                 ENSG00000108825.13,ENSG00000266967.2   \n",
       "chr17:57279061-57280856:+|3p                  ENSG00000068489.8,ENSG00000265303.1   \n",
       "chr17:57279095-57280796:+|3p                  ENSG00000068489.8,ENSG00000265303.1   \n",
       "chr19:23455185-23456877:-|3p                  ENSG00000269107.1,ENSG00000213971.3   \n",
       "chr19:50925803-50926861:+|3p                  ENSG00000142539.9,ENSG00000269404.2   \n",
       "chr20:34246937-34252155:-|3p    ENSG00000214078.7,ENSG00000272897.1,ENSG000002...   \n",
       "chr21:47882677-47882978:+|3p                 ENSG00000160305.13,ENSG00000223692.1   \n",
       "chr21:47883142-47889058:+|3p                 ENSG00000160305.13,ENSG00000223692.1   \n",
       "chr22:38747005-38757403:-|3p                  ENSG00000213923.6,ENSG00000244627.1   \n",
       "chrX:37214152-37214592:+|3p                  ENSG00000130962.13,ENSG00000250349.3   \n",
       "chr2:238895074-238900785:+|3p                ENSG00000184182.14,ENSG00000258984.1   \n",
       "chr10:75570662-75571231:-|3p                 ENSG00000272916.1,ENSG00000166507.12   \n",
       "chr12:54479630-54495997:+|3p                  ENSG00000248265.1,ENSG00000249388.1   \n",
       "chr22:18593343-18604245:+|3p                 ENSG00000215193.8,ENSG00000183785.10   \n",
       "chr1:35480764-35485152:-|3p                  ENSG00000271741.1,ENSG00000163867.12   \n",
       "chr16:18800784-18801565:-|3p                 ENSG00000134419.11,ENSG00000260342.2   \n",
       "chr17:26883285-26890448:-|3p                 ENSG00000258472.4,ENSG00000087111.16   \n",
       "\n",
       "                                                       gene_name  \\\n",
       "intron_location                                                    \n",
       "chr1:326417-327587:+|5p               RP4-669L17.10,RP4-669L17.8   \n",
       "chr1:1634439-1634518:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:1643840-1647784:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:1647918-1650766:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:1650895-1653034:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:1653151-1654146:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:1654271-1655696:-|5p                     CDK11A,RP1-283E3.8   \n",
       "chr1:10490626-10493898:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:10490626-10500403:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:10491459-10493898:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:10494023-10494713:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:10494748-10500403:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:10500471-10502321:+|5p                   APITD1,APITD1-CORT   \n",
       "chr1:19923604-19948593:+|5p                   MINOS1-NBL1,MINOS1   \n",
       "chr1:19948642-19949967:+|5p                   MINOS1-NBL1,MINOS1   \n",
       "chr1:19950078-19952880:+|5p                   MINOS1-NBL1,MINOS1   \n",
       "chr1:28832597-28834639:+|5p                           RCC1,SNHG3   \n",
       "chr1:28832613-28834639:+|5p                           RCC1,SNHG3   \n",
       "chr1:28834673-28835341:+|5p                           RCC1,SNHG3   \n",
       "chr1:32374562-32375638:-|5p                  RP11-84A19.4,PTP4A2   \n",
       "chr1:32375714-32377296:-|5p                  RP11-84A19.4,PTP4A2   \n",
       "chr1:32377428-32381495:-|5p                  RP11-84A19.4,PTP4A2   \n",
       "chr1:32381589-32384570:-|5p                  RP11-84A19.4,PTP4A2   \n",
       "chr1:35447721-35449367:-|5p           RP11-244H3.1,ZMYM6,ZMYM6NB   \n",
       "chr1:35449549-35450798:-|5p     RP11-244H3.1,ZMYM6,ZMYM6NB,ZMYM6   \n",
       "chr1:35454537-35457834:-|5p                          ZMYM6,ZMYM6   \n",
       "chr1:35457989-35469568:-|5p                          ZMYM6,ZMYM6   \n",
       "chr1:35469607-35470732:-|5p                          ZMYM6,ZMYM6   \n",
       "chr1:35470874-35472551:-|5p                          ZMYM6,ZMYM6   \n",
       "chr1:35472700-35474352:-|5p                          ZMYM6,ZMYM6   \n",
       "...                                                          ...   \n",
       "chr12:110837706-110837773:-|3p              ANAPC7,RP11-478C19.2   \n",
       "chr12:122234149-122234818:-|3p                   RHOF,AC084018.1   \n",
       "chr13:21951348-21951639:-|3p                 ZDHHC20,ZDHHC20-IT1   \n",
       "chr13:21951348-21951805:-|3p                 ZDHHC20,ZDHHC20-IT1   \n",
       "chr13:21951682-21951763:-|3p                 ZDHHC20,ZDHHC20-IT1   \n",
       "chr14:24633344-24633795:+|3p                   RP11-468E2.4,IRF9   \n",
       "chr14:24683358-24683453:-|3p          AL136419.6,MDP1,NEDD8-MDP1   \n",
       "chr14:24683806-24685124:-|3p          AL136419.6,MDP1,NEDD8-MDP1   \n",
       "chr14:62229307-62242812:+|3p                RP11-618G20.1,SNAPC1   \n",
       "chr14:65393042-65395245:+|3p             CHURC1,FNTB,CHURC1-FNTB   \n",
       "chr15:25265872-25282858:+|3p               SNHG14,RP11-701H24.10   \n",
       "chr15:43942099-43955851:-|3p                       STRC,CATSPER2   \n",
       "chr15:59561890-59562581:-|3p                 MYO1E,RP11-429D19.1   \n",
       "chr17:41108343-41108420:-|3p               PTGES3L-AARSD1,AARSD1   \n",
       "chr17:57279061-57280856:+|3p                  PRR11,CTD-2510F5.6   \n",
       "chr17:57279095-57280796:+|3p                  PRR11,CTD-2510F5.6   \n",
       "chr19:23455185-23456877:-|3p           RP11-15H20.7,RP11-15H20.6   \n",
       "chr19:50925803-50926861:+|3p                   CTD-2545M3.6,SPIB   \n",
       "chr20:34246937-34252155:-|3p            CPNE1,RP1-309K20.6,RBM12   \n",
       "chr21:47882677-47882978:+|3p                     DIP2A,DIP2A-IT1   \n",
       "chr21:47883142-47889058:+|3p                     DIP2A,DIP2A-IT1   \n",
       "chr22:38747005-38757403:-|3p                 CSNK1E,RP3-449O17.1   \n",
       "chrX:37214152-37214592:+|3p                         PRRG1,TM4SF2   \n",
       "chr2:238895074-238900785:+|3p                   UBE2F,UBE2F-SCLY   \n",
       "chr10:75570662-75571231:-|3p                RP11-574K11.31,NDST2   \n",
       "chr12:54479630-54495997:+|3p         RP11-834C11.3,RP11-834C11.6   \n",
       "chr22:18593343-18604245:+|3p                         PEX26,TUBA8   \n",
       "chr1:35480764-35485152:-|3p                          ZMYM6,ZMYM6   \n",
       "chr16:18800784-18801565:-|3p               RPS15A,RP11-1035H13.3   \n",
       "chr17:26883285-26890448:-|3p                  RP11-192H23.4,PIGS   \n",
       "\n",
       "                                    ...      gerstberger2014_rbp_target_ncrna  \\\n",
       "intron_location                     ...                                         \n",
       "chr1:326417-327587:+|5p             ...                                 False   \n",
       "chr1:1634439-1634518:-|5p           ...                                 False   \n",
       "chr1:1643840-1647784:-|5p           ...                                 False   \n",
       "chr1:1647918-1650766:-|5p           ...                                 False   \n",
       "chr1:1650895-1653034:-|5p           ...                                 False   \n",
       "chr1:1653151-1654146:-|5p           ...                                 False   \n",
       "chr1:1654271-1655696:-|5p           ...                                 False   \n",
       "chr1:10490626-10493898:+|5p         ...                                 False   \n",
       "chr1:10490626-10500403:+|5p         ...                                 False   \n",
       "chr1:10491459-10493898:+|5p         ...                                 False   \n",
       "chr1:10494023-10494713:+|5p         ...                                 False   \n",
       "chr1:10494748-10500403:+|5p         ...                                 False   \n",
       "chr1:10500471-10502321:+|5p         ...                                 False   \n",
       "chr1:19923604-19948593:+|5p         ...                                 False   \n",
       "chr1:19948642-19949967:+|5p         ...                                 False   \n",
       "chr1:19950078-19952880:+|5p         ...                                 False   \n",
       "chr1:28832597-28834639:+|5p         ...                                 False   \n",
       "chr1:28832613-28834639:+|5p         ...                                 False   \n",
       "chr1:28834673-28835341:+|5p         ...                                 False   \n",
       "chr1:32374562-32375638:-|5p         ...                                 False   \n",
       "chr1:32375714-32377296:-|5p         ...                                 False   \n",
       "chr1:32377428-32381495:-|5p         ...                                 False   \n",
       "chr1:32381589-32384570:-|5p         ...                                 False   \n",
       "chr1:35447721-35449367:-|5p         ...                                 False   \n",
       "chr1:35449549-35450798:-|5p         ...                                 False   \n",
       "chr1:35454537-35457834:-|5p         ...                                 False   \n",
       "chr1:35457989-35469568:-|5p         ...                                 False   \n",
       "chr1:35469607-35470732:-|5p         ...                                 False   \n",
       "chr1:35470874-35472551:-|5p         ...                                 False   \n",
       "chr1:35472700-35474352:-|5p         ...                                 False   \n",
       "...                                 ...                                   ...   \n",
       "chr12:110837706-110837773:-|3p      ...                                 False   \n",
       "chr12:122234149-122234818:-|3p      ...                                 False   \n",
       "chr13:21951348-21951639:-|3p        ...                                 False   \n",
       "chr13:21951348-21951805:-|3p        ...                                 False   \n",
       "chr13:21951682-21951763:-|3p        ...                                 False   \n",
       "chr14:24633344-24633795:+|3p        ...                                 False   \n",
       "chr14:24683358-24683453:-|3p        ...                                 False   \n",
       "chr14:24683806-24685124:-|3p        ...                                 False   \n",
       "chr14:62229307-62242812:+|3p        ...                                 False   \n",
       "chr14:65393042-65395245:+|3p        ...                                 False   \n",
       "chr15:25265872-25282858:+|3p        ...                                 False   \n",
       "chr15:43942099-43955851:-|3p        ...                                 False   \n",
       "chr15:59561890-59562581:-|3p        ...                                 False   \n",
       "chr17:41108343-41108420:-|3p        ...                                 False   \n",
       "chr17:57279061-57280856:+|3p        ...                                 False   \n",
       "chr17:57279095-57280796:+|3p        ...                                 False   \n",
       "chr19:23455185-23456877:-|3p        ...                                 False   \n",
       "chr19:50925803-50926861:+|3p        ...                                 False   \n",
       "chr20:34246937-34252155:-|3p        ...                                 False   \n",
       "chr21:47882677-47882978:+|3p        ...                                 False   \n",
       "chr21:47883142-47889058:+|3p        ...                                 False   \n",
       "chr22:38747005-38757403:-|3p        ...                                 False   \n",
       "chrX:37214152-37214592:+|3p         ...                                 False   \n",
       "chr2:238895074-238900785:+|3p       ...                                 False   \n",
       "chr10:75570662-75571231:-|3p        ...                                 False   \n",
       "chr12:54479630-54495997:+|3p        ...                                 False   \n",
       "chr22:18593343-18604245:+|3p        ...                                 False   \n",
       "chr1:35480764-35485152:-|3p         ...                                 False   \n",
       "chr16:18800784-18801565:-|3p        ...                                 False   \n",
       "chr17:26883285-26890448:-|3p        ...                                 False   \n",
       "\n",
       "                               gerstberger2014_rbp_target_ribosome  \\\n",
       "intron_location                                                      \n",
       "chr1:326417-327587:+|5p                                      False   \n",
       "chr1:1634439-1634518:-|5p                                    False   \n",
       "chr1:1643840-1647784:-|5p                                    False   \n",
       "chr1:1647918-1650766:-|5p                                    False   \n",
       "chr1:1650895-1653034:-|5p                                    False   \n",
       "chr1:1653151-1654146:-|5p                                    False   \n",
       "chr1:1654271-1655696:-|5p                                    False   \n",
       "chr1:10490626-10493898:+|5p                                  False   \n",
       "chr1:10490626-10500403:+|5p                                  False   \n",
       "chr1:10491459-10493898:+|5p                                  False   \n",
       "chr1:10494023-10494713:+|5p                                  False   \n",
       "chr1:10494748-10500403:+|5p                                  False   \n",
       "chr1:10500471-10502321:+|5p                                  False   \n",
       "chr1:19923604-19948593:+|5p                                  False   \n",
       "chr1:19948642-19949967:+|5p                                  False   \n",
       "chr1:19950078-19952880:+|5p                                  False   \n",
       "chr1:28832597-28834639:+|5p                                  False   \n",
       "chr1:28832613-28834639:+|5p                                  False   \n",
       "chr1:28834673-28835341:+|5p                                  False   \n",
       "chr1:32374562-32375638:-|5p                                  False   \n",
       "chr1:32375714-32377296:-|5p                                  False   \n",
       "chr1:32377428-32381495:-|5p                                  False   \n",
       "chr1:32381589-32384570:-|5p                                  False   \n",
       "chr1:35447721-35449367:-|5p                                  False   \n",
       "chr1:35449549-35450798:-|5p                                  False   \n",
       "chr1:35454537-35457834:-|5p                                  False   \n",
       "chr1:35457989-35469568:-|5p                                  False   \n",
       "chr1:35469607-35470732:-|5p                                  False   \n",
       "chr1:35470874-35472551:-|5p                                  False   \n",
       "chr1:35472700-35474352:-|5p                                  False   \n",
       "...                                                            ...   \n",
       "chr12:110837706-110837773:-|3p                               False   \n",
       "chr12:122234149-122234818:-|3p                               False   \n",
       "chr13:21951348-21951639:-|3p                                 False   \n",
       "chr13:21951348-21951805:-|3p                                 False   \n",
       "chr13:21951682-21951763:-|3p                                 False   \n",
       "chr14:24633344-24633795:+|3p                                 False   \n",
       "chr14:24683358-24683453:-|3p                                 False   \n",
       "chr14:24683806-24685124:-|3p                                 False   \n",
       "chr14:62229307-62242812:+|3p                                 False   \n",
       "chr14:65393042-65395245:+|3p                                 False   \n",
       "chr15:25265872-25282858:+|3p                                 False   \n",
       "chr15:43942099-43955851:-|3p                                 False   \n",
       "chr15:59561890-59562581:-|3p                                 False   \n",
       "chr17:41108343-41108420:-|3p                                 False   \n",
       "chr17:57279061-57280856:+|3p                                 False   \n",
       "chr17:57279095-57280796:+|3p                                 False   \n",
       "chr19:23455185-23456877:-|3p                                 False   \n",
       "chr19:50925803-50926861:+|3p                                 False   \n",
       "chr20:34246937-34252155:-|3p                                 False   \n",
       "chr21:47882677-47882978:+|3p                                 False   \n",
       "chr21:47883142-47889058:+|3p                                 False   \n",
       "chr22:38747005-38757403:-|3p                                 False   \n",
       "chrX:37214152-37214592:+|3p                                  False   \n",
       "chr2:238895074-238900785:+|3p                                False   \n",
       "chr10:75570662-75571231:-|3p                                 False   \n",
       "chr12:54479630-54495997:+|3p                                 False   \n",
       "chr22:18593343-18604245:+|3p                                 False   \n",
       "chr1:35480764-35485152:-|3p                                  False   \n",
       "chr16:18800784-18801565:-|3p                                  True   \n",
       "chr17:26883285-26890448:-|3p                                 False   \n",
       "\n",
       "                               gerstberger2014_rbp_target_trna  \\\n",
       "intron_location                                                  \n",
       "chr1:326417-327587:+|5p                                  False   \n",
       "chr1:1634439-1634518:-|5p                                False   \n",
       "chr1:1643840-1647784:-|5p                                False   \n",
       "chr1:1647918-1650766:-|5p                                False   \n",
       "chr1:1650895-1653034:-|5p                                False   \n",
       "chr1:1653151-1654146:-|5p                                False   \n",
       "chr1:1654271-1655696:-|5p                                False   \n",
       "chr1:10490626-10493898:+|5p                              False   \n",
       "chr1:10490626-10500403:+|5p                              False   \n",
       "chr1:10491459-10493898:+|5p                              False   \n",
       "chr1:10494023-10494713:+|5p                              False   \n",
       "chr1:10494748-10500403:+|5p                              False   \n",
       "chr1:10500471-10502321:+|5p                              False   \n",
       "chr1:19923604-19948593:+|5p                              False   \n",
       "chr1:19948642-19949967:+|5p                              False   \n",
       "chr1:19950078-19952880:+|5p                              False   \n",
       "chr1:28832597-28834639:+|5p                              False   \n",
       "chr1:28832613-28834639:+|5p                              False   \n",
       "chr1:28834673-28835341:+|5p                              False   \n",
       "chr1:32374562-32375638:-|5p                              False   \n",
       "chr1:32375714-32377296:-|5p                              False   \n",
       "chr1:32377428-32381495:-|5p                              False   \n",
       "chr1:32381589-32384570:-|5p                              False   \n",
       "chr1:35447721-35449367:-|5p                              False   \n",
       "chr1:35449549-35450798:-|5p                              False   \n",
       "chr1:35454537-35457834:-|5p                              False   \n",
       "chr1:35457989-35469568:-|5p                              False   \n",
       "chr1:35469607-35470732:-|5p                              False   \n",
       "chr1:35470874-35472551:-|5p                              False   \n",
       "chr1:35472700-35474352:-|5p                              False   \n",
       "...                                                        ...   \n",
       "chr12:110837706-110837773:-|3p                           False   \n",
       "chr12:122234149-122234818:-|3p                           False   \n",
       "chr13:21951348-21951639:-|3p                             False   \n",
       "chr13:21951348-21951805:-|3p                             False   \n",
       "chr13:21951682-21951763:-|3p                             False   \n",
       "chr14:24633344-24633795:+|3p                             False   \n",
       "chr14:24683358-24683453:-|3p                             False   \n",
       "chr14:24683806-24685124:-|3p                             False   \n",
       "chr14:62229307-62242812:+|3p                             False   \n",
       "chr14:65393042-65395245:+|3p                             False   \n",
       "chr15:25265872-25282858:+|3p                             False   \n",
       "chr15:43942099-43955851:-|3p                             False   \n",
       "chr15:59561890-59562581:-|3p                             False   \n",
       "chr17:41108343-41108420:-|3p                              True   \n",
       "chr17:57279061-57280856:+|3p                             False   \n",
       "chr17:57279095-57280796:+|3p                             False   \n",
       "chr19:23455185-23456877:-|3p                             False   \n",
       "chr19:50925803-50926861:+|3p                             False   \n",
       "chr20:34246937-34252155:-|3p                             False   \n",
       "chr21:47882677-47882978:+|3p                             False   \n",
       "chr21:47883142-47889058:+|3p                             False   \n",
       "chr22:38747005-38757403:-|3p                             False   \n",
       "chrX:37214152-37214592:+|3p                              False   \n",
       "chr2:238895074-238900785:+|3p                            False   \n",
       "chr10:75570662-75571231:-|3p                             False   \n",
       "chr12:54479630-54495997:+|3p                             False   \n",
       "chr22:18593343-18604245:+|3p                             False   \n",
       "chr1:35480764-35485152:-|3p                              False   \n",
       "chr16:18800784-18801565:-|3p                             False   \n",
       "chr17:26883285-26890448:-|3p                             False   \n",
       "\n",
       "                               gerstberger2014_tf    rbp ribosomal  \\\n",
       "intron_location                                                      \n",
       "chr1:326417-327587:+|5p                     False  False     False   \n",
       "chr1:1634439-1634518:-|5p                   False  False     False   \n",
       "chr1:1643840-1647784:-|5p                   False  False     False   \n",
       "chr1:1647918-1650766:-|5p                   False  False     False   \n",
       "chr1:1650895-1653034:-|5p                   False  False     False   \n",
       "chr1:1653151-1654146:-|5p                   False  False     False   \n",
       "chr1:1654271-1655696:-|5p                   False  False     False   \n",
       "chr1:10490626-10493898:+|5p                 False  False     False   \n",
       "chr1:10490626-10500403:+|5p                 False  False     False   \n",
       "chr1:10491459-10493898:+|5p                 False  False     False   \n",
       "chr1:10494023-10494713:+|5p                 False  False     False   \n",
       "chr1:10494748-10500403:+|5p                 False  False     False   \n",
       "chr1:10500471-10502321:+|5p                 False  False     False   \n",
       "chr1:19923604-19948593:+|5p                 False  False     False   \n",
       "chr1:19948642-19949967:+|5p                 False  False     False   \n",
       "chr1:19950078-19952880:+|5p                 False  False     False   \n",
       "chr1:28832597-28834639:+|5p                 False  False     False   \n",
       "chr1:28832613-28834639:+|5p                 False  False     False   \n",
       "chr1:28834673-28835341:+|5p                 False  False     False   \n",
       "chr1:32374562-32375638:-|5p                 False  False     False   \n",
       "chr1:32375714-32377296:-|5p                 False  False     False   \n",
       "chr1:32377428-32381495:-|5p                 False  False     False   \n",
       "chr1:32381589-32384570:-|5p                 False  False     False   \n",
       "chr1:35447721-35449367:-|5p                 False  False     False   \n",
       "chr1:35449549-35450798:-|5p                 False  False     False   \n",
       "chr1:35454537-35457834:-|5p                 False  False     False   \n",
       "chr1:35457989-35469568:-|5p                 False  False     False   \n",
       "chr1:35469607-35470732:-|5p                 False  False     False   \n",
       "chr1:35470874-35472551:-|5p                 False  False     False   \n",
       "chr1:35472700-35474352:-|5p                 False  False     False   \n",
       "...                                           ...    ...       ...   \n",
       "chr12:110837706-110837773:-|3p              False  False     False   \n",
       "chr12:122234149-122234818:-|3p              False  False     False   \n",
       "chr13:21951348-21951639:-|3p                False  False     False   \n",
       "chr13:21951348-21951805:-|3p                False  False     False   \n",
       "chr13:21951682-21951763:-|3p                False  False     False   \n",
       "chr14:24633344-24633795:+|3p                False  False     False   \n",
       "chr14:24683358-24683453:-|3p                False  False     False   \n",
       "chr14:24683806-24685124:-|3p                False  False     False   \n",
       "chr14:62229307-62242812:+|3p                False  False     False   \n",
       "chr14:65393042-65395245:+|3p                False  False     False   \n",
       "chr15:25265872-25282858:+|3p                False  False     False   \n",
       "chr15:43942099-43955851:-|3p                False  False     False   \n",
       "chr15:59561890-59562581:-|3p                False  False     False   \n",
       "chr17:41108343-41108420:-|3p                False  False     False   \n",
       "chr17:57279061-57280856:+|3p                False  False     False   \n",
       "chr17:57279095-57280796:+|3p                False  False     False   \n",
       "chr19:23455185-23456877:-|3p                False  False     False   \n",
       "chr19:50925803-50926861:+|3p                 True  False     False   \n",
       "chr20:34246937-34252155:-|3p                False  False     False   \n",
       "chr21:47882677-47882978:+|3p                False  False     False   \n",
       "chr21:47883142-47889058:+|3p                False  False     False   \n",
       "chr22:38747005-38757403:-|3p                False  False     False   \n",
       "chrX:37214152-37214592:+|3p                 False  False     False   \n",
       "chr2:238895074-238900785:+|3p               False  False     False   \n",
       "chr10:75570662-75571231:-|3p                False  False     False   \n",
       "chr12:54479630-54495997:+|3p                False  False     False   \n",
       "chr22:18593343-18604245:+|3p                False  False     False   \n",
       "chr1:35480764-35485152:-|3p                 False  False     False   \n",
       "chr16:18800784-18801565:-|3p                False  False      True   \n",
       "chr17:26883285-26890448:-|3p                False  False     False   \n",
       "\n",
       "                               ribosomal_subunit synapse transcription_factor  \\\n",
       "intron_location                                                                 \n",
       "chr1:326417-327587:+|5p                    False   False                False   \n",
       "chr1:1634439-1634518:-|5p                  False   False                False   \n",
       "chr1:1643840-1647784:-|5p                  False   False                False   \n",
       "chr1:1647918-1650766:-|5p                  False   False                False   \n",
       "chr1:1650895-1653034:-|5p                  False   False                False   \n",
       "chr1:1653151-1654146:-|5p                  False   False                False   \n",
       "chr1:1654271-1655696:-|5p                  False   False                False   \n",
       "chr1:10490626-10493898:+|5p                False   False                False   \n",
       "chr1:10490626-10500403:+|5p                False   False                False   \n",
       "chr1:10491459-10493898:+|5p                False   False                False   \n",
       "chr1:10494023-10494713:+|5p                False   False                False   \n",
       "chr1:10494748-10500403:+|5p                False   False                False   \n",
       "chr1:10500471-10502321:+|5p                False   False                False   \n",
       "chr1:19923604-19948593:+|5p                False   False                False   \n",
       "chr1:19948642-19949967:+|5p                False   False                False   \n",
       "chr1:19950078-19952880:+|5p                False   False                False   \n",
       "chr1:28832597-28834639:+|5p                False   False                False   \n",
       "chr1:28832613-28834639:+|5p                False   False                False   \n",
       "chr1:28834673-28835341:+|5p                False   False                False   \n",
       "chr1:32374562-32375638:-|5p                False   False                False   \n",
       "chr1:32375714-32377296:-|5p                False   False                False   \n",
       "chr1:32377428-32381495:-|5p                False   False                False   \n",
       "chr1:32381589-32384570:-|5p                False   False                False   \n",
       "chr1:35447721-35449367:-|5p                False   False                False   \n",
       "chr1:35449549-35450798:-|5p                False   False                False   \n",
       "chr1:35454537-35457834:-|5p                False   False                False   \n",
       "chr1:35457989-35469568:-|5p                False   False                False   \n",
       "chr1:35469607-35470732:-|5p                False   False                False   \n",
       "chr1:35470874-35472551:-|5p                False   False                False   \n",
       "chr1:35472700-35474352:-|5p                False   False                False   \n",
       "...                                          ...     ...                  ...   \n",
       "chr12:110837706-110837773:-|3p             False   False                False   \n",
       "chr12:122234149-122234818:-|3p             False   False                False   \n",
       "chr13:21951348-21951639:-|3p               False   False                False   \n",
       "chr13:21951348-21951805:-|3p               False   False                False   \n",
       "chr13:21951682-21951763:-|3p               False   False                False   \n",
       "chr14:24633344-24633795:+|3p               False   False                False   \n",
       "chr14:24683358-24683453:-|3p               False   False                False   \n",
       "chr14:24683806-24685124:-|3p               False   False                False   \n",
       "chr14:62229307-62242812:+|3p               False   False                False   \n",
       "chr14:65393042-65395245:+|3p               False   False                False   \n",
       "chr15:25265872-25282858:+|3p               False   False                False   \n",
       "chr15:43942099-43955851:-|3p               False   False                False   \n",
       "chr15:59561890-59562581:-|3p               False   False                False   \n",
       "chr17:41108343-41108420:-|3p               False   False                False   \n",
       "chr17:57279061-57280856:+|3p               False   False                False   \n",
       "chr17:57279095-57280796:+|3p               False   False                False   \n",
       "chr19:23455185-23456877:-|3p               False   False                False   \n",
       "chr19:50925803-50926861:+|3p               False   False                 True   \n",
       "chr20:34246937-34252155:-|3p               False   False                False   \n",
       "chr21:47882677-47882978:+|3p               False   False                False   \n",
       "chr21:47883142-47889058:+|3p               False   False                False   \n",
       "chr22:38747005-38757403:-|3p               False   False                False   \n",
       "chrX:37214152-37214592:+|3p                False   False                False   \n",
       "chr2:238895074-238900785:+|3p              False   False                False   \n",
       "chr10:75570662-75571231:-|3p               False   False                False   \n",
       "chr12:54479630-54495997:+|3p               False   False                False   \n",
       "chr22:18593343-18604245:+|3p               False   False                False   \n",
       "chr1:35480764-35485152:-|3p                False   False                False   \n",
       "chr16:18800784-18801565:-|3p                True   False                False   \n",
       "chr17:26883285-26890448:-|3p               False   False                False   \n",
       "\n",
       "                               intron_length  \n",
       "intron_location                               \n",
       "chr1:326417-327587:+|5p                 1170  \n",
       "chr1:1634439-1634518:-|5p                 79  \n",
       "chr1:1643840-1647784:-|5p               3944  \n",
       "chr1:1647918-1650766:-|5p               2848  \n",
       "chr1:1650895-1653034:-|5p               2139  \n",
       "chr1:1653151-1654146:-|5p                995  \n",
       "chr1:1654271-1655696:-|5p               1425  \n",
       "chr1:10490626-10493898:+|5p             3272  \n",
       "chr1:10490626-10500403:+|5p             9777  \n",
       "chr1:10491459-10493898:+|5p             2439  \n",
       "chr1:10494023-10494713:+|5p              690  \n",
       "chr1:10494748-10500403:+|5p             5655  \n",
       "chr1:10500471-10502321:+|5p             1850  \n",
       "chr1:19923604-19948593:+|5p            24989  \n",
       "chr1:19948642-19949967:+|5p             1325  \n",
       "chr1:19950078-19952880:+|5p             2802  \n",
       "chr1:28832597-28834639:+|5p             2042  \n",
       "chr1:28832613-28834639:+|5p             2026  \n",
       "chr1:28834673-28835341:+|5p              668  \n",
       "chr1:32374562-32375638:-|5p             1076  \n",
       "chr1:32375714-32377296:-|5p             1582  \n",
       "chr1:32377428-32381495:-|5p             4067  \n",
       "chr1:32381589-32384570:-|5p             2981  \n",
       "chr1:35447721-35449367:-|5p             1646  \n",
       "chr1:35449549-35450798:-|5p             1249  \n",
       "chr1:35454537-35457834:-|5p             3297  \n",
       "chr1:35457989-35469568:-|5p            11579  \n",
       "chr1:35469607-35470732:-|5p             1125  \n",
       "chr1:35470874-35472551:-|5p             1677  \n",
       "chr1:35472700-35474352:-|5p             1652  \n",
       "...                                      ...  \n",
       "chr12:110837706-110837773:-|3p            67  \n",
       "chr12:122234149-122234818:-|3p           669  \n",
       "chr13:21951348-21951639:-|3p             291  \n",
       "chr13:21951348-21951805:-|3p             457  \n",
       "chr13:21951682-21951763:-|3p              81  \n",
       "chr14:24633344-24633795:+|3p             451  \n",
       "chr14:24683358-24683453:-|3p              95  \n",
       "chr14:24683806-24685124:-|3p            1318  \n",
       "chr14:62229307-62242812:+|3p           13505  \n",
       "chr14:65393042-65395245:+|3p            2203  \n",
       "chr15:25265872-25282858:+|3p           16986  \n",
       "chr15:43942099-43955851:-|3p           13752  \n",
       "chr15:59561890-59562581:-|3p             691  \n",
       "chr17:41108343-41108420:-|3p              77  \n",
       "chr17:57279061-57280856:+|3p            1795  \n",
       "chr17:57279095-57280796:+|3p            1701  \n",
       "chr19:23455185-23456877:-|3p            1692  \n",
       "chr19:50925803-50926861:+|3p            1058  \n",
       "chr20:34246937-34252155:-|3p            5218  \n",
       "chr21:47882677-47882978:+|3p             301  \n",
       "chr21:47883142-47889058:+|3p            5916  \n",
       "chr22:38747005-38757403:-|3p           10398  \n",
       "chrX:37214152-37214592:+|3p              440  \n",
       "chr2:238895074-238900785:+|3p           5711  \n",
       "chr10:75570662-75571231:-|3p             569  \n",
       "chr12:54479630-54495997:+|3p           16367  \n",
       "chr22:18593343-18604245:+|3p           10902  \n",
       "chr1:35480764-35485152:-|3p             4388  \n",
       "chr16:18800784-18801565:-|3p             781  \n",
       "chr17:26883285-26890448:-|3p            7163  \n",
       "\n",
       "[25472 rows x 41 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata_combined.loc[sj_metadata_combined.ensembl_id.str.contains(',').fillna(False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata_combined.to_csv('{}/psi5_psi3_combined_metadata.csv'.format(folder))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chrom                                       category\n",
       "intron_start                                   int64\n",
       "intron_stop                                    int64\n",
       "strand                                      category\n",
       "intron_motif                                category\n",
       "annotated                                       bool\n",
       "gencode_id                                    object\n",
       "ensembl_id                                    object\n",
       "gene_id                                       object\n",
       "gene_name                                     object\n",
       "gene_status                                   object\n",
       "gene_type                                     object\n",
       "havana_gene                                   object\n",
       "level                                         object\n",
       "tag                                           object\n",
       "transcript_id                                 object\n",
       "transcript_name                               object\n",
       "transcript_status                             object\n",
       "transcript_type                               object\n",
       "one_ensembl_id                                object\n",
       "biomark_neural_panel                          object\n",
       "cell_cycle                                    object\n",
       "cell_surface_markers                          object\n",
       "confident_rbp                                 object\n",
       "gerstberger2014_rbp                           object\n",
       "gerstberger2014_rbp_consensus_rna_target      object\n",
       "gerstberger2014_rbp_domains[count]            object\n",
       "gerstberger2014_rbp_family_targets            object\n",
       "gerstberger2014_rbp_parologous_family_id      object\n",
       "gerstberger2014_rbp_target_diverse            object\n",
       "gerstberger2014_rbp_target_mrna               object\n",
       "gerstberger2014_rbp_target_ncrna              object\n",
       "gerstberger2014_rbp_target_ribosome           object\n",
       "gerstberger2014_rbp_target_trna               object\n",
       "gerstberger2014_tf                            object\n",
       "rbp                                           object\n",
       "ribosomal                                     object\n",
       "ribosomal_subunit                             object\n",
       "synapse                                       object\n",
       "transcription_factor                          object\n",
       "intron_length                                  int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata_combined.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sj_metadata_combined.loc[:, 'chrom'] = sj_metadata_combined.chrom.astype(str)\n",
    "sj_metadata_combined.loc[:, 'strand'] = sj_metadata_combined.strand.astype(str)\n",
    "sj_metadata_combined.loc[:, 'intron_motif'] = sj_metadata_combined.intron_motif.astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chrom                                       object\n",
       "intron_start                                 int64\n",
       "intron_stop                                  int64\n",
       "strand                                      object\n",
       "intron_motif                                object\n",
       "annotated                                     bool\n",
       "gencode_id                                  object\n",
       "ensembl_id                                  object\n",
       "gene_id                                     object\n",
       "gene_name                                   object\n",
       "gene_status                                 object\n",
       "gene_type                                   object\n",
       "havana_gene                                 object\n",
       "level                                       object\n",
       "tag                                         object\n",
       "transcript_id                               object\n",
       "transcript_name                             object\n",
       "transcript_status                           object\n",
       "transcript_type                             object\n",
       "one_ensembl_id                              object\n",
       "biomark_neural_panel                        object\n",
       "cell_cycle                                  object\n",
       "cell_surface_markers                        object\n",
       "confident_rbp                               object\n",
       "gerstberger2014_rbp                         object\n",
       "gerstberger2014_rbp_consensus_rna_target    object\n",
       "gerstberger2014_rbp_domains[count]          object\n",
       "gerstberger2014_rbp_family_targets          object\n",
       "gerstberger2014_rbp_parologous_family_id    object\n",
       "gerstberger2014_rbp_target_diverse          object\n",
       "gerstberger2014_rbp_target_mrna             object\n",
       "gerstberger2014_rbp_target_ncrna            object\n",
       "gerstberger2014_rbp_target_ribosome         object\n",
       "gerstberger2014_rbp_target_trna             object\n",
       "gerstberger2014_tf                          object\n",
       "rbp                                         object\n",
       "ribosomal                                   object\n",
       "ribosomal_subunit                           object\n",
       "synapse                                     object\n",
       "transcription_factor                        object\n",
       "intron_length                                int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sj_metadata_combined.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/obotvinnik/anaconda/lib/python2.7/site-packages/pandas/io/pytables.py:2577: PerformanceWarning: \n",
      "your performance may suffer as PyTables will pickle object types that it cannot\n",
      "map directly to c-types [inferred_type->mixed,key->block2_values] [items->['chrom', 'strand', 'intron_motif', 'gencode_id', 'ensembl_id', 'gene_id', 'gene_name', 'gene_status', 'gene_type', 'havana_gene', 'level', 'tag', 'transcript_id', 'transcript_name', 'transcript_status', 'transcript_type', 'one_ensembl_id', 'biomark_neural_panel', 'cell_cycle', 'cell_surface_markers', 'confident_rbp', 'gerstberger2014_rbp', 'gerstberger2014_rbp_consensus_rna_target', 'gerstberger2014_rbp_domains[count]', 'gerstberger2014_rbp_family_targets', 'gerstberger2014_rbp_parologous_family_id', 'gerstberger2014_rbp_target_diverse', 'gerstberger2014_rbp_target_mrna', 'gerstberger2014_rbp_target_ncrna', 'gerstberger2014_rbp_target_ribosome', 'gerstberger2014_rbp_target_trna', 'gerstberger2014_tf', 'rbp', 'ribosomal', 'ribosomal_subunit', 'synapse', 'transcription_factor']]\n",
      "\n",
      "  warnings.warn(ws, PerformanceWarning)\n"
     ]
    }
   ],
   "source": [
    "sj_metadata_combined.to_hdf('{}/psi5_psi3_combined_metadata.hdf'.format(folder), 'metadata', format='fixed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
