{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction12</th>\n",
       "      <th>junction13</th>\n",
       "      <th>junction24</th>\n",
       "      <th>junction34</th>\n",
       "      <th>incompatible_junctions</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>junction:chr17:75212018-75212131:+</td>\n",
       "      <td>junction:chr17:75212018-75212222:+</td>\n",
       "      <td>junction:chr17:75212145-75212326:+</td>\n",
       "      <td>junction:chr17:75212301-75212326:+</td>\n",
       "      <td>junction:chr17:75212018-75212326:+|junction:ch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>junction:chr17:75212018-75212079:+</td>\n",
       "      <td>junction:chr17:75212018-75212222:+</td>\n",
       "      <td>junction:chr17:75212145-75212326:+</td>\n",
       "      <td>junction:chr17:75212288-75212326:+</td>\n",
       "      <td>junction:chr17:75212018-75212326:+|junction:ch...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            junction12  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212131:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212079:+   \n",
       "\n",
       "                                                                            junction13  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212222:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212222:+   \n",
       "\n",
       "                                                                            junction24  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212145-75212326:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212145-75212326:+   \n",
       "\n",
       "                                                                            junction34  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212301-75212326:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212288-75212326:+   \n",
       "\n",
       "                                                                               incompatible_junctions  \n",
       "event_id                                                                                               \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212326:+|junction:ch...  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212326:+|junction:ch...  "
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "mxe_events = pd.read_csv('/home/obotvinnik/projects/singlecell_pnms/analysis/outrigger_v2/index/mxe/events.csv', index_col=0)\n",
    "print(mxe_events.shape)\n",
    "mxe_events.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# mxe_events_not_duplicated = mxe_events.drop_duplicates()\n",
    "# print(mxe_events_not_duplicated.shape)\n",
    "# mxe_events_not_duplicated.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# ! head -n 5 /home/obotvinnik/projects/singlecell_pnms/analysis/outrigger_v2/index/mxe/exon*.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from outrigger.region import Region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211927-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212132-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212300:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212352:+)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211940-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212080-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212287:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212417:+)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>outrigger.Region (exon:chr17:75212005-75212079:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212171-75212209:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212210-75212300:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212379-75212391:+)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211979-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212132-75212157:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212158-75212222:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212275-75212300:+)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>outrigger.Region (exon:chr5:180670692-18067087...</td>\n",
       "      <td>outrigger.Region (exon:chr5:180669170-18066923...</td>\n",
       "      <td>outrigger.Region (exon:chr5:180668492-18066863...</td>\n",
       "      <td>outrigger.Region (exon:chr5:180666517-18066658...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               exon1  \\\n",
       "0  outrigger.Region (exon:chr17:75211927-75212017:+)   \n",
       "1  outrigger.Region (exon:chr17:75211940-75212017:+)   \n",
       "2  outrigger.Region (exon:chr17:75212005-75212079:+)   \n",
       "3  outrigger.Region (exon:chr17:75211979-75212017:+)   \n",
       "4  outrigger.Region (exon:chr5:180670692-18067087...   \n",
       "\n",
       "                                               exon2  \\\n",
       "0  outrigger.Region (exon:chr17:75212132-75212144:+)   \n",
       "1  outrigger.Region (exon:chr17:75212080-75212144:+)   \n",
       "2  outrigger.Region (exon:chr17:75212171-75212209:+)   \n",
       "3  outrigger.Region (exon:chr17:75212132-75212157:+)   \n",
       "4  outrigger.Region (exon:chr5:180669170-18066923...   \n",
       "\n",
       "                                               exon3  \\\n",
       "0  outrigger.Region (exon:chr17:75212223-75212300:+)   \n",
       "1  outrigger.Region (exon:chr17:75212223-75212287:+)   \n",
       "2  outrigger.Region (exon:chr17:75212210-75212300:+)   \n",
       "3  outrigger.Region (exon:chr17:75212158-75212222:+)   \n",
       "4  outrigger.Region (exon:chr5:180668492-18066863...   \n",
       "\n",
       "                                               exon4  \n",
       "0  outrigger.Region (exon:chr17:75212327-75212352:+)  \n",
       "1  outrigger.Region (exon:chr17:75212327-75212417:+)  \n",
       "2  outrigger.Region (exon:chr17:75212379-75212391:+)  \n",
       "3  outrigger.Region (exon:chr17:75212275-75212300:+)  \n",
       "4  outrigger.Region (exon:chr5:180666517-18066658...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_exon_bed_template = '/home/obotvinnik/projects/singlecell_pnms/analysis/outrigger_v2/index/mxe/exon{}.bed'\n",
    "\n",
    "exons = {}\n",
    "\n",
    "for i in range(1, 5):\n",
    "    exon = 'exon{}'.format(i)\n",
    "    bed = mxe_exon_bed_template.format(i)\n",
    "    exon_df = pd.read_table(bed, names=['chrom', 'start', 'stop', 'name', 'score', 'strand'])\n",
    "    exon_regions = exon_df.apply(lambda row: Region('exon:{chrom}:{start}-{stop}:{strand}'.format(\n",
    "                chrom=row.chrom, start=row.start+1, stop=row.stop, strand=row.strand)), axis=1)\n",
    "    exons[exon] = exon_regions\n",
    "exon_regions_df = pd.DataFrame(exons)\n",
    "exon_regions_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(325347, 4)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_regions_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211927-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212132-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212300:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212352:+)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211940-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212080-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212287:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212417:+)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                exon1  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75211927-75212017:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75211940-75212017:+)   \n",
       "\n",
       "                                                                                                exon2  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212132-75212144:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212080-75212144:+)   \n",
       "\n",
       "                                                                                                exon3  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212223-75212300:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212223-75212287:+)   \n",
       "\n",
       "                                                                                                exon4  \n",
       "event_id                                                                                               \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212327-75212352:+)  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212327-75212417:+)  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_regions_df.index = mxe_events.index\n",
    "print(exon_regions_df.shape)\n",
    "exon_regions_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>event_location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211927-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212132-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212300:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212352:+)</td>\n",
       "      <td>outrigger.Region (intron:chr17:75212017-752120...</td>\n",
       "      <td>outrigger.Region (event:chr17:75211927-7521201...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>outrigger.Region (exon:chr17:75211940-75212017:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212080-75212144:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212223-75212287:+)</td>\n",
       "      <td>outrigger.Region (exon:chr17:75212327-75212417:+)</td>\n",
       "      <td>outrigger.Region (intron:chr17:75212017-752120...</td>\n",
       "      <td>outrigger.Region (event:chr17:75211940-7521201...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                exon1  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75211927-75212017:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75211940-75212017:+)   \n",
       "\n",
       "                                                                                                exon2  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212132-75212144:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212080-75212144:+)   \n",
       "\n",
       "                                                                                                exon3  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212223-75212300:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212223-75212287:+)   \n",
       "\n",
       "                                                                                                exon4  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212327-75212352:+)   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (exon:chr17:75212327-75212417:+)   \n",
       "\n",
       "                                                                                      intron_location  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (intron:chr17:75212017-752120...   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (intron:chr17:75212017-752120...   \n",
       "\n",
       "                                                                                       event_location  \n",
       "event_id                                                                                               \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (event:chr17:75211927-7521201...  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  outrigger.Region (event:chr17:75211940-7521201...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def make_introns(row):\n",
    "    exon1 = row.iloc[0]\n",
    "    exonN = row.iloc[-1]\n",
    "    \n",
    "    if exon1.strand == \"+\":\n",
    "        start = exon1.stop\n",
    "        stop = exonN.start\n",
    "    elif exon1.strand == '-':\n",
    "        start = exonN.stop\n",
    "        stop = exon1.start\n",
    "    return Region('intron:{chrom}:{start}-{stop}:{strand}'.format(\n",
    "                chrom=exon1.chrom, start=start, stop=stop, strand=exon1.strand))\n",
    "\n",
    "def event_location(row):\n",
    "    exon1 = row.iloc[0]\n",
    "    exonN = row.iloc[-1]\n",
    "    \n",
    "    if exon1.strand == \"+\":\n",
    "        start = exon1.start\n",
    "        stop = exonN.stop\n",
    "    elif exon1.strand == '-':\n",
    "        start = exonN.start\n",
    "        stop = exon1.stop\n",
    "    return Region('event:{chrom}:{start}-{stop}:{strand}'.format(\n",
    "                chrom=exon1.chrom, start=start, stop=stop, strand=exon1.strand))\n",
    "\n",
    "\n",
    "exon_regions_df['intron_location'] = exon_regions_df.apply(make_introns, axis=1)\n",
    "exon_regions_df['event_location'] = exon_regions_df.apply(event_location, axis=1)\n",
    "exon_regions_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1_length</th>\n",
       "      <th>exon2_length</th>\n",
       "      <th>exon3_length</th>\n",
       "      <th>exon4_length</th>\n",
       "      <th>intron_length</th>\n",
       "      <th>event_length</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>91</td>\n",
       "      <td>13</td>\n",
       "      <td>78</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>78</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>91</td>\n",
       "      <td>1</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212080-75212209:+@novel_exon:chr17:75212210-75212300:+@junction:chr17:75212301-75212378:+|isoform2=junction:chr17:75212080-75212170:+@novel_exon:chr17:75212171-75212209:+@junction:chr17:75212210-75212378:+</th>\n",
       "      <td>75</td>\n",
       "      <td>39</td>\n",
       "      <td>91</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212157:+@novel_exon:chr17:75212158-75212222:+@junction:chr17:75212223-75212274:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212157:+@junction:chr17:75212158-75212274:+</th>\n",
       "      <td>39</td>\n",
       "      <td>26</td>\n",
       "      <td>65</td>\n",
       "      <td>26</td>\n",
       "      <td>1</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr5:180668640-180670691:-@exon:chr5:180668492-180668639:-@junction:chr5:180666583-180668491:-|isoform2=junction:chr5:180669235-180670691:-@novel_exon:chr5:180669170-180669234:-@junction:chr5:180666583-180669169:-</th>\n",
       "      <td>181</td>\n",
       "      <td>65</td>\n",
       "      <td>148</td>\n",
       "      <td>66</td>\n",
       "      <td>1</td>\n",
       "      <td>181</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    exon1_length  \\\n",
       "event_id                                                           \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            91   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            78   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...            75   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...            39   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...           181   \n",
       "\n",
       "                                                    exon2_length  \\\n",
       "event_id                                                           \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            13   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            65   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...            39   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...            26   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...            65   \n",
       "\n",
       "                                                    exon3_length  \\\n",
       "event_id                                                           \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            78   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            65   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...            91   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...            65   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...           148   \n",
       "\n",
       "                                                    exon4_length  \\\n",
       "event_id                                                           \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            26   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            91   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...            13   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...            26   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...            66   \n",
       "\n",
       "                                                    intron_length  \\\n",
       "event_id                                                            \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...              1   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...              1   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...              1   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...              1   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...              1   \n",
       "\n",
       "                                                    event_length  \n",
       "event_id                                                          \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            91  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...            78  \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...            75  \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...            39  \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...           181  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lengths = exon_regions_df.applymap(len)\n",
    "lengths.columns = [x.split('_')[0] + '_length' for x in lengths.columns]\n",
    "print(lengths.shape)\n",
    "lengths.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>event_location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>exon:chr17:75211927-75212017:+</td>\n",
       "      <td>exon:chr17:75212132-75212144:+</td>\n",
       "      <td>exon:chr17:75212223-75212300:+</td>\n",
       "      <td>exon:chr17:75212327-75212352:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211927-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>exon:chr17:75211940-75212017:+</td>\n",
       "      <td>exon:chr17:75212080-75212144:+</td>\n",
       "      <td>exon:chr17:75212223-75212287:+</td>\n",
       "      <td>exon:chr17:75212327-75212417:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211940-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212080-75212209:+@novel_exon:chr17:75212210-75212300:+@junction:chr17:75212301-75212378:+|isoform2=junction:chr17:75212080-75212170:+@novel_exon:chr17:75212171-75212209:+@junction:chr17:75212210-75212378:+</th>\n",
       "      <td>exon:chr17:75212005-75212079:+</td>\n",
       "      <td>exon:chr17:75212171-75212209:+</td>\n",
       "      <td>exon:chr17:75212210-75212300:+</td>\n",
       "      <td>exon:chr17:75212379-75212391:+</td>\n",
       "      <td>intron:chr17:75212079-75212079:+</td>\n",
       "      <td>event:chr17:75212005-75212079:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212157:+@novel_exon:chr17:75212158-75212222:+@junction:chr17:75212223-75212274:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212157:+@junction:chr17:75212158-75212274:+</th>\n",
       "      <td>exon:chr17:75211979-75212017:+</td>\n",
       "      <td>exon:chr17:75212132-75212157:+</td>\n",
       "      <td>exon:chr17:75212158-75212222:+</td>\n",
       "      <td>exon:chr17:75212275-75212300:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211979-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr5:180668640-180670691:-@exon:chr5:180668492-180668639:-@junction:chr5:180666583-180668491:-|isoform2=junction:chr5:180669235-180670691:-@novel_exon:chr5:180669170-180669234:-@junction:chr5:180666583-180669169:-</th>\n",
       "      <td>exon:chr5:180670692-180670872:-</td>\n",
       "      <td>exon:chr5:180669170-180669234:-</td>\n",
       "      <td>exon:chr5:180668492-180668639:-</td>\n",
       "      <td>exon:chr5:180666517-180666582:-</td>\n",
       "      <td>intron:chr5:180670692-180670692:-</td>\n",
       "      <td>event:chr5:180670692-180670872:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                              exon1  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75211927-75212017:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75211940-75212017:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212005-75212079:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75211979-75212017:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180670692-180670872:-   \n",
       "\n",
       "                                                                              exon2  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212132-75212144:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212080-75212144:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212171-75212209:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212132-75212157:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180669170-180669234:-   \n",
       "\n",
       "                                                                              exon3  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212223-75212300:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212223-75212287:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212210-75212300:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212158-75212222:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180668492-180668639:-   \n",
       "\n",
       "                                                                              exon4  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212327-75212352:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212327-75212417:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212379-75212391:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212275-75212300:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180666517-180666582:-   \n",
       "\n",
       "                                                                      intron_location  \\\n",
       "event_id                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   intron:chr17:75212079-75212079:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  intron:chr5:180670692-180670692:-   \n",
       "\n",
       "                                                                      event_location  \n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   event:chr17:75211927-75212017:+  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   event:chr17:75211940-75212017:+  \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   event:chr17:75212005-75212079:+  \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   event:chr17:75211979-75212017:+  \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  event:chr5:180670692-180670872:-  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_str = exon_regions_df.applymap(lambda x: x.name)\n",
    "exon_str.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 6)\n"
     ]
    }
   ],
   "source": [
    "print(exon_str.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(386635,)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "event_location.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction12</th>\n",
       "      <th>junction13</th>\n",
       "      <th>junction24</th>\n",
       "      <th>junction34</th>\n",
       "      <th>incompatible_junctions</th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>event_location</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212300:+@junction:chr17:75212301-75212326:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>junction:chr17:75212018-75212131:+</td>\n",
       "      <td>junction:chr17:75212018-75212222:+</td>\n",
       "      <td>junction:chr17:75212145-75212326:+</td>\n",
       "      <td>junction:chr17:75212301-75212326:+</td>\n",
       "      <td>junction:chr17:75212018-75212326:+|junction:ch...</td>\n",
       "      <td>exon:chr17:75211927-75212017:+</td>\n",
       "      <td>exon:chr17:75212132-75212144:+</td>\n",
       "      <td>exon:chr17:75212223-75212300:+</td>\n",
       "      <td>exon:chr17:75212327-75212352:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211927-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212222:+@novel_exon:chr17:75212223-75212287:+@junction:chr17:75212288-75212326:+|isoform2=junction:chr17:75212018-75212079:+@novel_exon:chr17:75212080-75212144:+@junction:chr17:75212145-75212326:+</th>\n",
       "      <td>junction:chr17:75212018-75212079:+</td>\n",
       "      <td>junction:chr17:75212018-75212222:+</td>\n",
       "      <td>junction:chr17:75212145-75212326:+</td>\n",
       "      <td>junction:chr17:75212288-75212326:+</td>\n",
       "      <td>junction:chr17:75212018-75212326:+|junction:ch...</td>\n",
       "      <td>exon:chr17:75211940-75212017:+</td>\n",
       "      <td>exon:chr17:75212080-75212144:+</td>\n",
       "      <td>exon:chr17:75212223-75212287:+</td>\n",
       "      <td>exon:chr17:75212327-75212417:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211940-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212080-75212209:+@novel_exon:chr17:75212210-75212300:+@junction:chr17:75212301-75212378:+|isoform2=junction:chr17:75212080-75212170:+@novel_exon:chr17:75212171-75212209:+@junction:chr17:75212210-75212378:+</th>\n",
       "      <td>junction:chr17:75212080-75212170:+</td>\n",
       "      <td>junction:chr17:75212080-75212209:+</td>\n",
       "      <td>junction:chr17:75212210-75212378:+</td>\n",
       "      <td>junction:chr17:75212301-75212378:+</td>\n",
       "      <td>junction:chr17:75212080-75212378:+|junction:ch...</td>\n",
       "      <td>exon:chr17:75212005-75212079:+</td>\n",
       "      <td>exon:chr17:75212171-75212209:+</td>\n",
       "      <td>exon:chr17:75212210-75212300:+</td>\n",
       "      <td>exon:chr17:75212379-75212391:+</td>\n",
       "      <td>intron:chr17:75212079-75212079:+</td>\n",
       "      <td>event:chr17:75212005-75212079:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr17:75212018-75212157:+@novel_exon:chr17:75212158-75212222:+@junction:chr17:75212223-75212274:+|isoform2=junction:chr17:75212018-75212131:+@novel_exon:chr17:75212132-75212157:+@junction:chr17:75212158-75212274:+</th>\n",
       "      <td>junction:chr17:75212018-75212131:+</td>\n",
       "      <td>junction:chr17:75212018-75212157:+</td>\n",
       "      <td>junction:chr17:75212158-75212274:+</td>\n",
       "      <td>junction:chr17:75212223-75212274:+</td>\n",
       "      <td>junction:chr17:75212018-75212274:+|junction:ch...</td>\n",
       "      <td>exon:chr17:75211979-75212017:+</td>\n",
       "      <td>exon:chr17:75212132-75212157:+</td>\n",
       "      <td>exon:chr17:75212158-75212222:+</td>\n",
       "      <td>exon:chr17:75212275-75212300:+</td>\n",
       "      <td>intron:chr17:75212017-75212017:+</td>\n",
       "      <td>event:chr17:75211979-75212017:+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr5:180668640-180670691:-@exon:chr5:180668492-180668639:-@junction:chr5:180666583-180668491:-|isoform2=junction:chr5:180669235-180670691:-@novel_exon:chr5:180669170-180669234:-@junction:chr5:180666583-180669169:-</th>\n",
       "      <td>junction:chr5:180669235-180670691:-</td>\n",
       "      <td>junction:chr5:180668640-180670691:-</td>\n",
       "      <td>junction:chr5:180666583-180669169:-</td>\n",
       "      <td>junction:chr5:180666583-180668491:-</td>\n",
       "      <td>junction:chr5:180666583-180670691:-|junction:c...</td>\n",
       "      <td>exon:chr5:180670692-180670872:-</td>\n",
       "      <td>exon:chr5:180669170-180669234:-</td>\n",
       "      <td>exon:chr5:180668492-180668639:-</td>\n",
       "      <td>exon:chr5:180666517-180666582:-</td>\n",
       "      <td>intron:chr5:180670692-180670692:-</td>\n",
       "      <td>event:chr5:180670692-180670872:-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                             junction12  \\\n",
       "event_id                                                                                  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212018-75212131:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212018-75212079:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   junction:chr17:75212080-75212170:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   junction:chr17:75212018-75212131:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  junction:chr5:180669235-180670691:-   \n",
       "\n",
       "                                                                             junction13  \\\n",
       "event_id                                                                                  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212018-75212222:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212018-75212222:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   junction:chr17:75212080-75212209:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   junction:chr17:75212018-75212157:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  junction:chr5:180668640-180670691:-   \n",
       "\n",
       "                                                                             junction24  \\\n",
       "event_id                                                                                  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212145-75212326:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212145-75212326:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   junction:chr17:75212210-75212378:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   junction:chr17:75212158-75212274:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  junction:chr5:180666583-180669169:-   \n",
       "\n",
       "                                                                             junction34  \\\n",
       "event_id                                                                                  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212301-75212326:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   junction:chr17:75212288-75212326:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   junction:chr17:75212301-75212378:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   junction:chr17:75212223-75212274:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  junction:chr5:180666583-180668491:-   \n",
       "\n",
       "                                                                               incompatible_junctions  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212326:+|junction:ch...   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...  junction:chr17:75212018-75212326:+|junction:ch...   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...  junction:chr17:75212080-75212378:+|junction:ch...   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...  junction:chr17:75212018-75212274:+|junction:ch...   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  junction:chr5:180666583-180670691:-|junction:c...   \n",
       "\n",
       "                                                                              exon1  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75211927-75212017:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75211940-75212017:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212005-75212079:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75211979-75212017:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180670692-180670872:-   \n",
       "\n",
       "                                                                              exon2  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212132-75212144:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212080-75212144:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212171-75212209:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212132-75212157:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180669170-180669234:-   \n",
       "\n",
       "                                                                              exon3  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212223-75212300:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212223-75212287:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212210-75212300:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212158-75212222:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180668492-180668639:-   \n",
       "\n",
       "                                                                              exon4  \\\n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212327-75212352:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   exon:chr17:75212327-75212417:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   exon:chr17:75212379-75212391:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   exon:chr17:75212275-75212300:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  exon:chr5:180666517-180666582:-   \n",
       "\n",
       "                                                                      intron_location  \\\n",
       "event_id                                                                                \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   intron:chr17:75212079-75212079:+   \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   intron:chr17:75212017-75212017:+   \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  intron:chr5:180670692-180670692:-   \n",
       "\n",
       "                                                                      event_location  \n",
       "event_id                                                                              \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   event:chr17:75211927-75212017:+  \n",
       "isoform1=junction:chr17:75212018-75212222:+@nov...   event:chr17:75211940-75212017:+  \n",
       "isoform1=junction:chr17:75212080-75212209:+@nov...   event:chr17:75212005-75212079:+  \n",
       "isoform1=junction:chr17:75212018-75212157:+@nov...   event:chr17:75211979-75212017:+  \n",
       "isoform1=junction:chr5:180668640-180670691:-@ex...  event:chr5:180670692-180670872:-  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_events_exons = pd.concat([mxe_events, exon_str], axis=1)\n",
    "print(mxe_events_exons.shape)\n",
    "mxe_events_exons.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine with gencode information so we have the genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17914</td>\n",
       "      <td>29370</td>\n",
       "      <td>isoform1=junction:chr1:18367-29320:-@exon:chr1...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14363</td>\n",
       "      <td>29806</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_status \"KNOWN\"; havana_gene \"OTTHUMG00000...</td>\n",
       "      <td>11456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>677192</td>\n",
       "      <td>685396</td>\n",
       "      <td>isoform1=junction:chr1:679354-682074:-@novel_e...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>677193</td>\n",
       "      <td>685396</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...</td>\n",
       "      <td>8204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>678665</td>\n",
       "      <td>685396</td>\n",
       "      <td>isoform1=junction:chr1:679354-682074:-@novel_e...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>677193</td>\n",
       "      <td>685396</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...</td>\n",
       "      <td>6731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>762987</td>\n",
       "      <td>787490</td>\n",
       "      <td>isoform1=junction:chr1:763156-783033:+@exon:ch...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>762988</td>\n",
       "      <td>794826</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...</td>\n",
       "      <td>24503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>762987</td>\n",
       "      <td>784982</td>\n",
       "      <td>isoform1=junction:chr1:763156-783033:+@exon:ch...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>762988</td>\n",
       "      <td>794826</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...</td>\n",
       "      <td>21995</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     0       1       2                                                  3  4   \\\n",
       "0  chr1   17914   29370  isoform1=junction:chr1:18367-29320:-@exon:chr1...  .   \n",
       "1  chr1  677192  685396  isoform1=junction:chr1:679354-682074:-@novel_e...  .   \n",
       "2  chr1  678665  685396  isoform1=junction:chr1:679354-682074:-@novel_e...  .   \n",
       "3  chr1  762987  787490  isoform1=junction:chr1:763156-783033:+@exon:ch...  .   \n",
       "4  chr1  762987  784982  isoform1=junction:chr1:763156-783033:+@exon:ch...  .   \n",
       "\n",
       "  5     6       7     8       9       10 11 12 13  \\\n",
       "0  -  chr1  HAVANA  gene   14363   29806  .  -  .   \n",
       "1  -  chr1  HAVANA  gene  677193  685396  .  -  .   \n",
       "2  -  chr1  HAVANA  gene  677193  685396  .  -  .   \n",
       "3  +  chr1  HAVANA  gene  762988  794826  .  +  .   \n",
       "4  +  chr1  HAVANA  gene  762988  794826  .  +  .   \n",
       "\n",
       "                                                  14     15  \n",
       "0  gene_status \"KNOWN\"; havana_gene \"OTTHUMG00000...  11456  \n",
       "1  gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...   8204  \n",
       "2  gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...   6731  \n",
       "3  gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...  24503  \n",
       "4  gene_status \"NOVEL\"; havana_gene \"OTTHUMG00000...  21995  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_gencode_filename = '/home/obotvinnik/projects/singlecell_pnms/analysis/outrigger_v2/index/mxe/event.sorted.gencode.v19.bed'\n",
    "mxe_gencode = pd.read_table(mxe_gencode_filename, header=None)\n",
    "mxe_gencode.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def split_gtf_attributes(attributes):\n",
    "    split = attributes.split('; ')\n",
    "    pairs = [x.split(' ') for x in split]\n",
    "    no_quotes = [map(lambda x: x.strip('\";'), pair) for pair in pairs]\n",
    "    mapping = dict(no_quotes)\n",
    "    return mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(386635, 11)\n",
      "CPU times: user 3min 8s, sys: 3.24 s, total: 3min 11s\n",
      "Wall time: 3min 11s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "attributes = mxe_gencode[14].map(split_gtf_attributes).apply(pd.Series)\n",
    "print(attributes.shape)\n",
    "attributes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:18367-29320:-@exon:chr1:18268-18366:-@junction:chr1:18062-18267:-|isoform2=junction:chr1:24892-29320:-@exon:chr1:24738-24891:-@junction:chr1:18062-24737:-</th>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:679354-682074:-@novel_exon:chr1:679297-679353:-@junction:chr1:678731-679296:-|isoform2=junction:chr1:680206-682074:-@novel_exon:chr1:680108-680205:-@junction:chr1:678731-680107:-</th>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002407.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:679354-682074:-@novel_exon:chr1:679297-679353:-@junction:chr1:678731-679296:-|isoform2=junction:chr1:680206-682074:-@novel_exon:chr1:680108-680205:-@junction:chr1:678731-680107:-</th>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002407.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:763156-783033:+@exon:chr1:783034-783186:+@junction:chr1:783187-787306:+|isoform2=junction:chr1:763156-764382:+@exon:chr1:764383-764484:+@junction:chr1:764485-787306:+</th>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>OTTHUMG00000002468.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:763156-783033:+@exon:chr1:783034-783186:+@junction:chr1:783187-784863:+|isoform2=junction:chr1:763156-764382:+@exon:chr1:764383-764484:+@junction:chr1:764485-784863:+</th>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>OTTHUMG00000002468.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                              gene_id  \\\n",
       "event_id                                                                \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  ENSG00000227232.4   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "\n",
       "                                                         gene_name  \\\n",
       "event_id                                                             \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...          WASH7P   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "\n",
       "                                                   gene_status  \\\n",
       "event_id                                                         \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...       KNOWN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...       NOVEL   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...       NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...       NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...       NOVEL   \n",
       "\n",
       "                                                               gene_type  \\\n",
       "event_id                                                                   \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...            pseudogene   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "\n",
       "                                                             havana_gene  \\\n",
       "event_id                                                                   \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  OTTHUMG00000000958.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  OTTHUMG00000002407.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  OTTHUMG00000002407.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  OTTHUMG00000002468.5   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  OTTHUMG00000002468.5   \n",
       "\n",
       "                                                   level  tag  \\\n",
       "event_id                                                        \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...     2  NaN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...     2  NaN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...     2  NaN   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...     2  NaN   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...     2  NaN   \n",
       "\n",
       "                                                        transcript_id  \\\n",
       "event_id                                                                \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  ENSG00000227232.4   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "\n",
       "                                                   transcript_name  \\\n",
       "event_id                                                             \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...          WASH7P   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "\n",
       "                                                   transcript_status  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...             KNOWN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...             NOVEL   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...             NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...             NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...             NOVEL   \n",
       "\n",
       "                                                         transcript_type  \n",
       "event_id                                                                  \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...            pseudogene  \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA  \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA  \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript  \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "attributes.index = mxe_gencode[3]\n",
    "attributes.index.name = 'event_id'\n",
    "attributes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:18367-29320:-@exon:chr1:18268-18366:-@junction:chr1:18062-18267:-|isoform2=junction:chr1:24892-29320:-@exon:chr1:24738-24891:-@junction:chr1:18062-24737:-</th>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>OTTHUMG00000000958.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000227232.4</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:679354-682074:-@novel_exon:chr1:679297-679353:-@junction:chr1:678731-679296:-|isoform2=junction:chr1:680206-682074:-@novel_exon:chr1:680108-680205:-@junction:chr1:678731-680107:-</th>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002407.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>ENSG00000235373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:679354-682074:-@novel_exon:chr1:679297-679353:-@junction:chr1:678731-679296:-|isoform2=junction:chr1:680206-682074:-@novel_exon:chr1:680108-680205:-@junction:chr1:678731-680107:-</th>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>OTTHUMG00000002407.1</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000235373.1</td>\n",
       "      <td>RP11-206L10.3</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>lincRNA</td>\n",
       "      <td>ENSG00000235373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:763156-783033:+@exon:chr1:783034-783186:+@junction:chr1:783187-787306:+|isoform2=junction:chr1:763156-764382:+@exon:chr1:764383-764484:+@junction:chr1:764485-787306:+</th>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>OTTHUMG00000002468.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>ENSG00000228794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr1:763156-783033:+@exon:chr1:783034-783186:+@junction:chr1:783187-784863:+|isoform2=junction:chr1:763156-764382:+@exon:chr1:764383-764484:+@junction:chr1:764485-784863:+</th>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>OTTHUMG00000002468.5</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000228794.4</td>\n",
       "      <td>RP11-206L10.11</td>\n",
       "      <td>NOVEL</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>ENSG00000228794</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                              gene_id  \\\n",
       "event_id                                                                \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  ENSG00000227232.4   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "\n",
       "                                                         gene_name  \\\n",
       "event_id                                                             \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...          WASH7P   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "\n",
       "                                                   gene_status  \\\n",
       "event_id                                                         \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...       KNOWN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...       NOVEL   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...       NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...       NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...       NOVEL   \n",
       "\n",
       "                                                               gene_type  \\\n",
       "event_id                                                                   \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...            pseudogene   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "\n",
       "                                                             havana_gene  \\\n",
       "event_id                                                                   \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  OTTHUMG00000000958.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  OTTHUMG00000002407.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  OTTHUMG00000002407.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  OTTHUMG00000002468.5   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  OTTHUMG00000002468.5   \n",
       "\n",
       "                                                   level  tag  \\\n",
       "event_id                                                        \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...     2  NaN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...     2  NaN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...     2  NaN   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...     2  NaN   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...     2  NaN   \n",
       "\n",
       "                                                        transcript_id  \\\n",
       "event_id                                                                \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  ENSG00000227232.4   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373.1   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794.4   \n",
       "\n",
       "                                                   transcript_name  \\\n",
       "event_id                                                             \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...          WASH7P   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...   RP11-206L10.3   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  RP11-206L10.11   \n",
       "\n",
       "                                                   transcript_status  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...             KNOWN   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...             NOVEL   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...             NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...             NOVEL   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...             NOVEL   \n",
       "\n",
       "                                                         transcript_type  \\\n",
       "event_id                                                                   \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...            pseudogene   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...               lincRNA   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  processed_transcript   \n",
       "\n",
       "                                                         ensembl_id  \n",
       "event_id                                                             \n",
       "isoform1=junction:chr1:18367-29320:-@exon:chr1:...  ENSG00000227232  \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373  \n",
       "isoform1=junction:chr1:679354-682074:-@novel_ex...  ENSG00000235373  \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794  \n",
       "isoform1=junction:chr1:763156-783033:+@exon:chr...  ENSG00000228794  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "attributes['ensembl_id'] = attributes['gene_id'].str.split('.').str[0]\n",
    "attributes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(18225, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:101996727-102003454:-@exon:chr10:101996607-101996726:-@junction:chr10:101995522-101996606:-|isoform2=junction:chr10:101997989-102003454:-@exon:chr10:101997779-101997988:-@junction:chr10:101995522-101997778:-</th>\n",
       "      <td>ENSG00000212464.1,ENSG00000095485.12</td>\n",
       "      <td>CWF19L1,SNORA12</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5</td>\n",
       "      <td>3,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000212464.1,ENSG00000095485.12</td>\n",
       "      <td>CWF19L1,SNORA12</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>ENSG00000212464,ENSG00000095485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:102003535-102013177:-@exon:chr10:102003455-102003534:-@junction:chr10:101997989-102003454:-|isoform2=junction:chr10:102010071-102013177:-@novel_exon:chr10:102010005-102010070:-@junction:chr10:101997989-102010004:-</th>\n",
       "      <td>ENSG00000095485.12,ENSG00000230224.1</td>\n",
       "      <td>CWF19L1,PHBP9</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5,OTTHUMG00000018902.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>pseudo_consens,ncRNA_host</td>\n",
       "      <td>ENSG00000095485.12,ENSG00000230224.1</td>\n",
       "      <td>CWF19L1,PHBP9</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>ENSG00000230224,ENSG00000095485</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                 gene_id  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464.1,ENSG00000095485.12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000095485.12,ENSG00000230224.1   \n",
       "\n",
       "                                                          gene_name  \\\n",
       "event_id                                                              \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  CWF19L1,SNORA12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...    CWF19L1,PHBP9   \n",
       "\n",
       "                                                   gene_status  \\\n",
       "event_id                                                         \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...       KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...       KNOWN   \n",
       "\n",
       "                                                                    gene_type  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "\n",
       "                                                                                  havana_gene  \\\n",
       "event_id                                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                       OTTHUMG00000018901.5   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  OTTHUMG00000018901.5,OTTHUMG00000018902.1   \n",
       "\n",
       "                                                   level  \\\n",
       "event_id                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...   3,2   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...   1,2   \n",
       "\n",
       "                                                                          tag  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                 ncRNA_host   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudo_consens,ncRNA_host   \n",
       "\n",
       "                                                                           transcript_id  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464.1,ENSG00000095485.12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000095485.12,ENSG00000230224.1   \n",
       "\n",
       "                                                    transcript_name  \\\n",
       "event_id                                                              \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  CWF19L1,SNORA12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...    CWF19L1,PHBP9   \n",
       "\n",
       "                                                   transcript_status  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...             KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...             KNOWN   \n",
       "\n",
       "                                                              transcript_type  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "\n",
       "                                                                         ensembl_id  \n",
       "event_id                                                                             \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464,ENSG00000095485  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000230224,ENSG00000095485  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "attributes_grouped = attributes.groupby(level=0, axis=0).apply(lambda df: df.apply(\n",
    "        lambda x: ','.join(map(str, set(x.dropna().values)))))\n",
    "print(attributes_grouped.shape)\n",
    "attributes_grouped.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(325347, 23)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction12</th>\n",
       "      <th>junction13</th>\n",
       "      <th>junction24</th>\n",
       "      <th>junction34</th>\n",
       "      <th>incompatible_junctions</th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:101996727-102003454:-@exon:chr10:101996607-101996726:-@junction:chr10:101995522-101996606:-|isoform2=junction:chr10:101997989-102003454:-@exon:chr10:101997779-101997988:-@junction:chr10:101995522-101997778:-</th>\n",
       "      <td>junction:chr10:101997989-102003454:-</td>\n",
       "      <td>junction:chr10:101996727-102003454:-</td>\n",
       "      <td>junction:chr10:101995522-101997778:-</td>\n",
       "      <td>junction:chr10:101995522-101996606:-</td>\n",
       "      <td>junction:chr10:101995522-102003454:-|junction:...</td>\n",
       "      <td>exon:chr10:102003455-102003534:-</td>\n",
       "      <td>exon:chr10:101997779-101997988:-</td>\n",
       "      <td>exon:chr10:101996607-101996726:-</td>\n",
       "      <td>exon:chr10:101995424-101995521:-</td>\n",
       "      <td>intron:chr10:102003455-102003455:-</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5</td>\n",
       "      <td>3,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000212464.1,ENSG00000095485.12</td>\n",
       "      <td>CWF19L1,SNORA12</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>ENSG00000212464,ENSG00000095485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:101996727-102003454:-@exon:chr10:101996607-101996726:-@junction:chr10:101995522-101996606:-|isoform2=junction:chr10:101997989-102003454:-@exon:chr10:101997779-101997988:-@junction:chr10:101995522-101997778:-</th>\n",
       "      <td>junction:chr10:101997989-102003454:-</td>\n",
       "      <td>junction:chr10:101996727-102003454:-</td>\n",
       "      <td>junction:chr10:101995522-101997778:-</td>\n",
       "      <td>junction:chr10:101995522-101996606:-</td>\n",
       "      <td>junction:chr10:101995522-102003454:-|junction:...</td>\n",
       "      <td>exon:chr10:102003455-102003529:-</td>\n",
       "      <td>exon:chr10:101997779-101997988:-</td>\n",
       "      <td>exon:chr10:101996607-101996726:-</td>\n",
       "      <td>exon:chr10:101995424-101995521:-</td>\n",
       "      <td>intron:chr10:102003455-102003455:-</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5</td>\n",
       "      <td>3,2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000212464.1,ENSG00000095485.12</td>\n",
       "      <td>CWF19L1,SNORA12</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>snoRNA,protein_coding</td>\n",
       "      <td>ENSG00000212464,ENSG00000095485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:102003535-102013177:-@exon:chr10:102003455-102003534:-@junction:chr10:101997989-102003454:-|isoform2=junction:chr10:102010071-102013177:-@novel_exon:chr10:102010005-102010070:-@junction:chr10:101997989-102010004:-</th>\n",
       "      <td>junction:chr10:102010071-102013177:-</td>\n",
       "      <td>junction:chr10:102003535-102013177:-</td>\n",
       "      <td>junction:chr10:101997989-102010004:-</td>\n",
       "      <td>junction:chr10:101997989-102003454:-</td>\n",
       "      <td>junction:chr10:101997989-102013177:-|junction:...</td>\n",
       "      <td>exon:chr10:102013178-102013296:-</td>\n",
       "      <td>exon:chr10:102010005-102010070:-</td>\n",
       "      <td>exon:chr10:102003455-102003534:-</td>\n",
       "      <td>exon:chr10:101997779-101997988:-</td>\n",
       "      <td>intron:chr10:102013178-102013178:-</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5,OTTHUMG00000018902.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>pseudo_consens,ncRNA_host</td>\n",
       "      <td>ENSG00000095485.12,ENSG00000230224.1</td>\n",
       "      <td>CWF19L1,PHBP9</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>ENSG00000230224,ENSG00000095485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:102003535-102013177:-@exon:chr10:102003455-102003534:-@junction:chr10:101997989-102003454:-|isoform2=junction:chr10:102010071-102013177:-@novel_exon:chr10:102010005-102010070:-@junction:chr10:101997989-102010004:-</th>\n",
       "      <td>junction:chr10:102010071-102013177:-</td>\n",
       "      <td>junction:chr10:102003535-102013177:-</td>\n",
       "      <td>junction:chr10:101997989-102010004:-</td>\n",
       "      <td>junction:chr10:101997989-102003454:-</td>\n",
       "      <td>junction:chr10:101997989-102013177:-|junction:...</td>\n",
       "      <td>exon:chr10:102013178-102014376:-</td>\n",
       "      <td>exon:chr10:102010005-102010070:-</td>\n",
       "      <td>exon:chr10:102003455-102003534:-</td>\n",
       "      <td>exon:chr10:101997779-101997988:-</td>\n",
       "      <td>intron:chr10:102013178-102013178:-</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5,OTTHUMG00000018902.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>pseudo_consens,ncRNA_host</td>\n",
       "      <td>ENSG00000095485.12,ENSG00000230224.1</td>\n",
       "      <td>CWF19L1,PHBP9</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>ENSG00000230224,ENSG00000095485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr10:102003535-102013177:-@exon:chr10:102003455-102003534:-@junction:chr10:101997989-102003454:-|isoform2=junction:chr10:102010071-102013177:-@novel_exon:chr10:102010005-102010070:-@junction:chr10:101997989-102010004:-</th>\n",
       "      <td>junction:chr10:102010071-102013177:-</td>\n",
       "      <td>junction:chr10:102003535-102013177:-</td>\n",
       "      <td>junction:chr10:101997989-102010004:-</td>\n",
       "      <td>junction:chr10:101997989-102003454:-</td>\n",
       "      <td>junction:chr10:101997989-102013177:-|junction:...</td>\n",
       "      <td>exon:chr10:102013178-102013296:-</td>\n",
       "      <td>exon:chr10:102010005-102010070:-</td>\n",
       "      <td>exon:chr10:102003455-102003534:-</td>\n",
       "      <td>exon:chr10:101997940-101997988:-</td>\n",
       "      <td>intron:chr10:102013178-102013178:-</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>OTTHUMG00000018901.5,OTTHUMG00000018902.1</td>\n",
       "      <td>1,2</td>\n",
       "      <td>pseudo_consens,ncRNA_host</td>\n",
       "      <td>ENSG00000095485.12,ENSG00000230224.1</td>\n",
       "      <td>CWF19L1,PHBP9</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>pseudogene,protein_coding</td>\n",
       "      <td>ENSG00000230224,ENSG00000095485</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                              junction12  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101997989-102003454:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101997989-102003454:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102010071-102013177:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102010071-102013177:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102010071-102013177:-   \n",
       "\n",
       "                                                                              junction13  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101996727-102003454:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101996727-102003454:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102003535-102013177:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102003535-102013177:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:102003535-102013177:-   \n",
       "\n",
       "                                                                              junction24  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-101997778:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-101997778:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102010004:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102010004:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102010004:-   \n",
       "\n",
       "                                                                              junction34  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-101996606:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-101996606:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102003454:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102003454:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102003454:-   \n",
       "\n",
       "                                                                               incompatible_junctions  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-102003454:-|junction:...   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  junction:chr10:101995522-102003454:-|junction:...   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102013177:-|junction:...   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102013177:-|junction:...   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  junction:chr10:101997989-102013177:-|junction:...   \n",
       "\n",
       "                                                                               exon1  \\\n",
       "event_id                                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:102003455-102003534:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:102003455-102003529:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102013178-102013296:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102013178-102014376:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102013178-102013296:-   \n",
       "\n",
       "                                                                               exon2  \\\n",
       "event_id                                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101997779-101997988:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101997779-101997988:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102010005-102010070:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102010005-102010070:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102010005-102010070:-   \n",
       "\n",
       "                                                                               exon3  \\\n",
       "event_id                                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101996607-101996726:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101996607-101996726:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102003455-102003534:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102003455-102003534:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:102003455-102003534:-   \n",
       "\n",
       "                                                                               exon4  \\\n",
       "event_id                                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101995424-101995521:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  exon:chr10:101995424-101995521:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:101997779-101997988:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:101997779-101997988:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  exon:chr10:101997940-101997988:-   \n",
       "\n",
       "                                                                       intron_location  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  intron:chr10:102003455-102003455:-   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  intron:chr10:102003455-102003455:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  intron:chr10:102013178-102013178:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  intron:chr10:102013178-102013178:-   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  intron:chr10:102013178-102013178:-   \n",
       "\n",
       "                                                                 ...                 \\\n",
       "event_id                                                         ...                  \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...               ...                  \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...               ...                  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...               ...                  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...               ...                  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...               ...                  \n",
       "\n",
       "                                                   gene_status  \\\n",
       "event_id                                                         \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...       KNOWN   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...       KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...       KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...       KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...       KNOWN   \n",
       "\n",
       "                                                                    gene_type  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "\n",
       "                                                                                  havana_gene  \\\n",
       "event_id                                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                       OTTHUMG00000018901.5   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                       OTTHUMG00000018901.5   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  OTTHUMG00000018901.5,OTTHUMG00000018902.1   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  OTTHUMG00000018901.5,OTTHUMG00000018902.1   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  OTTHUMG00000018901.5,OTTHUMG00000018902.1   \n",
       "\n",
       "                                                   level  \\\n",
       "event_id                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...   3,2   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...   3,2   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...   1,2   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...   1,2   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...   1,2   \n",
       "\n",
       "                                                                          tag  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                 ncRNA_host   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...                 ncRNA_host   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudo_consens,ncRNA_host   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudo_consens,ncRNA_host   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudo_consens,ncRNA_host   \n",
       "\n",
       "                                                                           transcript_id  \\\n",
       "event_id                                                                                   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464.1,ENSG00000095485.12   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464.1,ENSG00000095485.12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000095485.12,ENSG00000230224.1   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000095485.12,ENSG00000230224.1   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000095485.12,ENSG00000230224.1   \n",
       "\n",
       "                                                    transcript_name  \\\n",
       "event_id                                                              \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  CWF19L1,SNORA12   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  CWF19L1,SNORA12   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...    CWF19L1,PHBP9   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...    CWF19L1,PHBP9   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...    CWF19L1,PHBP9   \n",
       "\n",
       "                                                   transcript_status  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...             KNOWN   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...             KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...             KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...             KNOWN   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...             KNOWN   \n",
       "\n",
       "                                                              transcript_type  \\\n",
       "event_id                                                                        \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...      snoRNA,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  pseudogene,protein_coding   \n",
       "\n",
       "                                                                         ensembl_id  \n",
       "event_id                                                                             \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464,ENSG00000095485  \n",
       "isoform1=junction:chr10:101996727-102003454:-@e...  ENSG00000212464,ENSG00000095485  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000230224,ENSG00000095485  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000230224,ENSG00000095485  \n",
       "isoform1=junction:chr10:102003535-102013177:-@e...  ENSG00000230224,ENSG00000095485  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_metadata = mxe_events_exons.join(attributes_grouped)\n",
    "print(mxe_metadata.shape)\n",
    "mxe_metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'isoform1=junction:chr15:72494962-72499068:-@exon:chr15:72494795-72494961:-@junction:chr15:72492997-72494794:-|isoform2=junction:chr15:72495530-72499068:-@exon:chr15:72495363-72495529:-@junction:chr15:72492997-72495362:-'], dtype='object', name=u'event_id')"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_metadata.query('gene_name == \"PKM\"').index.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>junction12</th>\n",
       "      <th>junction13</th>\n",
       "      <th>junction24</th>\n",
       "      <th>junction34</th>\n",
       "      <th>incompatible_junctions</th>\n",
       "      <th>exon1</th>\n",
       "      <th>exon2</th>\n",
       "      <th>exon3</th>\n",
       "      <th>exon4</th>\n",
       "      <th>intron_location</th>\n",
       "      <th>...</th>\n",
       "      <th>gene_status</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>havana_gene</th>\n",
       "      <th>level</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_status</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>ensembl_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>event_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10224622:+@novel_exon:chr20:10224623-10224682:+@junction:chr20:10224683-10256076:+|isoform2=junction:chr20:10199626-10207989:+@novel_exon:chr20:10207990-10208086:+@junction:chr20:10208087-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10207989:+</td>\n",
       "      <td>junction:chr20:10199626-10224622:+</td>\n",
       "      <td>junction:chr20:10208087-10256076:+</td>\n",
       "      <td>junction:chr20:10224683-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199481-10199625:+</td>\n",
       "      <td>exon:chr20:10207990-10208086:+</td>\n",
       "      <td>exon:chr20:10224623-10224682:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10224622:+@novel_exon:chr20:10224623-10224682:+@junction:chr20:10224683-10256076:+|isoform2=junction:chr20:10199626-10207989:+@novel_exon:chr20:10207990-10208086:+@junction:chr20:10208087-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10207989:+</td>\n",
       "      <td>junction:chr20:10199626-10224622:+</td>\n",
       "      <td>junction:chr20:10208087-10256076:+</td>\n",
       "      <td>junction:chr20:10224683-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199478-10199625:+</td>\n",
       "      <td>exon:chr20:10207990-10208086:+</td>\n",
       "      <td>exon:chr20:10224623-10224682:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10241878:+@novel_exon:chr20:10241879-10241963:+@junction:chr20:10241964-10256076:+|isoform2=junction:chr20:10199626-10207989:+@novel_exon:chr20:10207990-10208086:+@junction:chr20:10208087-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10207989:+</td>\n",
       "      <td>junction:chr20:10199626-10241878:+</td>\n",
       "      <td>junction:chr20:10208087-10256076:+</td>\n",
       "      <td>junction:chr20:10241964-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199478-10199625:+</td>\n",
       "      <td>exon:chr20:10207990-10208086:+</td>\n",
       "      <td>exon:chr20:10241879-10241963:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10241878:+@novel_exon:chr20:10241879-10241963:+@junction:chr20:10241964-10256076:+|isoform2=junction:chr20:10199626-10207989:+@novel_exon:chr20:10207990-10208086:+@junction:chr20:10208087-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10207989:+</td>\n",
       "      <td>junction:chr20:10199626-10241878:+</td>\n",
       "      <td>junction:chr20:10208087-10256076:+</td>\n",
       "      <td>junction:chr20:10241964-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199481-10199625:+</td>\n",
       "      <td>exon:chr20:10207990-10208086:+</td>\n",
       "      <td>exon:chr20:10241879-10241963:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10241878:+@novel_exon:chr20:10241879-10241963:+@junction:chr20:10241964-10256076:+|isoform2=junction:chr20:10199626-10224622:+@novel_exon:chr20:10224623-10224682:+@junction:chr20:10224683-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10224622:+</td>\n",
       "      <td>junction:chr20:10199626-10241878:+</td>\n",
       "      <td>junction:chr20:10224683-10256076:+</td>\n",
       "      <td>junction:chr20:10241964-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199481-10199625:+</td>\n",
       "      <td>exon:chr20:10224623-10224682:+</td>\n",
       "      <td>exon:chr20:10241879-10241963:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10199626-10241878:+@novel_exon:chr20:10241879-10241963:+@junction:chr20:10241964-10256076:+|isoform2=junction:chr20:10199626-10224622:+@novel_exon:chr20:10224623-10224682:+@junction:chr20:10224683-10256076:+</th>\n",
       "      <td>junction:chr20:10199626-10224622:+</td>\n",
       "      <td>junction:chr20:10199626-10241878:+</td>\n",
       "      <td>junction:chr20:10224683-10256076:+</td>\n",
       "      <td>junction:chr20:10241964-10256076:+</td>\n",
       "      <td>junction:chr20:10199626-10256076:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10199478-10199625:+</td>\n",
       "      <td>exon:chr20:10224623-10224682:+</td>\n",
       "      <td>exon:chr20:10241879-10241963:+</td>\n",
       "      <td>exon:chr20:10256077-10256211:+</td>\n",
       "      <td>intron:chr20:10199625-10199625:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>isoform1=junction:chr20:10265421-10273808:+@exon:chr20:10273809-10273926:+@junction:chr20:10273927-10277572:+|isoform2=junction:chr20:10265421-10273529:+@exon:chr20:10273530-10273647:+@junction:chr20:10273648-10277572:+</th>\n",
       "      <td>junction:chr20:10265421-10273529:+</td>\n",
       "      <td>junction:chr20:10265421-10273808:+</td>\n",
       "      <td>junction:chr20:10273648-10277572:+</td>\n",
       "      <td>junction:chr20:10273927-10277572:+</td>\n",
       "      <td>junction:chr20:10265421-10277572:+|junction:ch...</td>\n",
       "      <td>exon:chr20:10265372-10265420:+</td>\n",
       "      <td>exon:chr20:10273530-10273647:+</td>\n",
       "      <td>exon:chr20:10273809-10273926:+</td>\n",
       "      <td>exon:chr20:10277573-10277698:+</td>\n",
       "      <td>intron:chr20:10265420-10265420:+</td>\n",
       "      <td>...</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>OTTHUMG00000031867.3,OTTHUMG00000031863.3</td>\n",
       "      <td>2</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>ENSG00000132639.8,ENSG00000227906.3</td>\n",
       "      <td>SNAP25,SNAP25-AS1</td>\n",
       "      <td>KNOWN</td>\n",
       "      <td>antisense,protein_coding</td>\n",
       "      <td>ENSG00000227906,ENSG00000132639</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows  23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                            junction12  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10207989:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10207989:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10207989:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10207989:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10224622:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10224622:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  junction:chr20:10265421-10273529:+   \n",
       "\n",
       "                                                                            junction13  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10224622:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10224622:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10241878:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10241878:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10241878:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10241878:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  junction:chr20:10265421-10273808:+   \n",
       "\n",
       "                                                                            junction24  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10208087-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10208087-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10208087-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10208087-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10224683-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10224683-10256076:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  junction:chr20:10273648-10277572:+   \n",
       "\n",
       "                                                                            junction34  \\\n",
       "event_id                                                                                 \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10224683-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10224683-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10241964-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10241964-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10241964-10256076:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10241964-10256076:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  junction:chr20:10273927-10277572:+   \n",
       "\n",
       "                                                                               incompatible_junctions  \\\n",
       "event_id                                                                                                \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  junction:chr20:10199626-10256076:+|junction:ch...   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  junction:chr20:10265421-10277572:+|junction:ch...   \n",
       "\n",
       "                                                                             exon1  \\\n",
       "event_id                                                                             \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10199481-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10199478-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10199478-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10199481-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10199481-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10199478-10199625:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  exon:chr20:10265372-10265420:+   \n",
       "\n",
       "                                                                             exon2  \\\n",
       "event_id                                                                             \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10207990-10208086:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10207990-10208086:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10207990-10208086:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10207990-10208086:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10224623-10224682:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10224623-10224682:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  exon:chr20:10273530-10273647:+   \n",
       "\n",
       "                                                                             exon3  \\\n",
       "event_id                                                                             \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10224623-10224682:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10224623-10224682:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10241879-10241963:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10241879-10241963:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10241879-10241963:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10241879-10241963:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  exon:chr20:10273809-10273926:+   \n",
       "\n",
       "                                                                             exon4  \\\n",
       "event_id                                                                             \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  exon:chr20:10256077-10256211:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  exon:chr20:10277573-10277698:+   \n",
       "\n",
       "                                                                     intron_location  \\\n",
       "event_id                                                                               \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  intron:chr20:10199625-10199625:+   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  intron:chr20:10265420-10265420:+   \n",
       "\n",
       "                                                                 ...                 \\\n",
       "event_id                                                         ...                  \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...               ...                  \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...               ...                  \n",
       "\n",
       "                                                   gene_status  \\\n",
       "event_id                                                         \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...       KNOWN   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...       KNOWN   \n",
       "\n",
       "                                                                   gene_type  \\\n",
       "event_id                                                                       \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  antisense,protein_coding   \n",
       "\n",
       "                                                                                  havana_gene  \\\n",
       "event_id                                                                                        \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  OTTHUMG00000031867.3,OTTHUMG00000031863.3   \n",
       "\n",
       "                                                   level         tag  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...     2  ncRNA_host   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...     2  ncRNA_host   \n",
       "\n",
       "                                                                          transcript_id  \\\n",
       "event_id                                                                                  \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  ENSG00000132639.8,ENSG00000227906.3   \n",
       "\n",
       "                                                      transcript_name  \\\n",
       "event_id                                                                \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  SNAP25,SNAP25-AS1   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  SNAP25,SNAP25-AS1   \n",
       "\n",
       "                                                   transcript_status  \\\n",
       "event_id                                                               \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...             KNOWN   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...             KNOWN   \n",
       "\n",
       "                                                             transcript_type  \\\n",
       "event_id                                                                       \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  antisense,protein_coding   \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  antisense,protein_coding   \n",
       "\n",
       "                                                                         ensembl_id  \n",
       "event_id                                                                             \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10199626-10224622:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10199626-10241878:+@nov...  ENSG00000227906,ENSG00000132639  \n",
       "isoform1=junction:chr20:10265421-10273808:+@exo...  ENSG00000227906,ENSG00000132639  \n",
       "\n",
       "[7 rows x 23 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s = mxe_metadata['gene_name'].str.contains('SNAP25').dropna()\n",
    "\n",
    "mxe_metadata.loc[s[s].index.unique()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mxe_metadata['gene_name'].isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mxe_metadata.to_csv('/projects/ps-yeolab/obotvinnik/singlecell_pnms/outrigger_v2/index/mxe/events_with_metadata.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
